// @motiadev/core/dist/src/adapters/defaults/stream/file-stream-adapter.mjs
import { StreamAdapter } from "../../interfaces/stream-adapter.interface.mjs";
import fs from "fs";
import * as path$1 from "path";

//#region src/adapters/defaults/stream/file-stream-adapter.ts
var FileStreamAdapter = class extends StreamAdapter {
	constructor(filePath, streamName, motiaFileStoragePath = ".motia") {
		super(streamName);
		this.streamsDir = path$1.join(filePath, motiaFileStoragePath, "streams");
		this.filePath = path$1.join(this.streamsDir, `${streamName}.stream.json`);
	}
	init() {
		try {
			fs.realpathSync(this.streamsDir);
		} catch {
			fs.mkdirSync(this.streamsDir, { recursive: true });
		}
		try {
			fs.readFileSync(this.filePath, "utf-8");
		} catch {
			fs.writeFileSync(this.filePath, JSON.stringify({}), "utf-8");
		}
	}
	async getGroup(groupId) {
		const data = this._readFile();
		const prefix = this._makeKey(groupId, "");
		return Object.entries(data).filter(([key]) => key.startsWith(prefix)).map(([, value]) => JSON.parse(value));
	}
	async get(groupId, key) {
		const data = this._readFile();
		const fullKey = this._makeKey(groupId, key);
		return data[fullKey] ? JSON.parse(data[fullKey]) : null;
	}
	async set(groupId, id, value) {
		const data = this._readFile();
		const key = this._makeKey(groupId, id);
		data[key] = JSON.stringify(value);
		this._writeFile(data);
		return {
			...value,
			id
		};
	}
	async delete(groupId, id) {
		const data = this._readFile();
		const key = this._makeKey(groupId, id);
		const value = await this.get(groupId, id);
		if (value) {
			delete data[key];
			this._writeFile(data);
		}
		return value;
	}
	async clear(groupId) {
		const data = this._readFile();
		const pattern = this._makeKey(groupId, "");
		for (const key in data) if (key.startsWith(pattern)) delete data[key];
		this._writeFile(data);
	}
	_makeKey(groupId, id) {
		return `${groupId}:${id}`;
	}
	_readFile() {
		try {
			const content = fs.readFileSync(this.filePath, "utf-8");
			return JSON.parse(content);
		} catch (error) {
			this.init();
			return {};
		}
	}
	_writeFile(data) {
		try {
			fs.writeFileSync(this.filePath, JSON.stringify(data, null, 2), "utf-8");
		} catch (error) {
			this.init();
			fs.writeFileSync(this.filePath, JSON.stringify(data, null, 2), "utf-8");
		}
	}
};

//#endregion
export { FileStreamAdapter };
