// @motiadev/core/dist/src/adapters/defaults/stream/memory-stream-adapter.mjs
import { StreamAdapter } from "../../interfaces/stream-adapter.interface.mjs";

//#region src/adapters/defaults/stream/memory-stream-adapter.ts
var MemoryStreamAdapter = class extends StreamAdapter {
	constructor(streamName) {
		super(streamName);
		this.state = {};
	}
	async init() {
		this.state = {};
	}
	async getGroup(groupId) {
		const prefix = this._makeKeyPrefix(groupId);
		return Object.entries(this.state).filter(([key]) => key.startsWith(prefix)).map(([key, value]) => {
			const id = key.split(":").pop();
			return {
				...value,
				id
			};
		});
	}
	async get(groupId, id) {
		const key = this._makeKey(groupId, id);
		const value = this.state[key];
		return value ? {
			...value,
			id
		} : null;
	}
	async set(groupId, id, value) {
		const key = this._makeKey(groupId, id);
		this.state[key] = value;
		return {
			...value,
			id
		};
	}
	async delete(groupId, id) {
		const key = this._makeKey(groupId, id);
		const value = await this.get(groupId, id);
		if (value) delete this.state[key];
		return value;
	}
	async clear(groupId) {
		const prefix = this._makeKeyPrefix(groupId);
		for (const key in this.state) if (key.startsWith(prefix)) delete this.state[key];
	}
	_makeKeyPrefix(groupId) {
		return `${this.streamName}:${groupId}`;
	}
	_makeKey(groupId, id) {
		return `${this.streamName}:${groupId}:${id}`;
	}
};

//#endregion
export { MemoryStreamAdapter };
