// @motiadev/core/dist/src/observability/tracer.mjs
import { createTrace } from "./create-trace.mjs";
import { RedisTraceStreamAdapter } from "./redis-trace-stream-adapter.mjs";
import { StreamTracer } from "./stream-tracer.mjs";
import { TraceManager } from "./trace-manager.mjs";

//#region src/observability/tracer.ts
var BaseTracerFactory = class {
	constructor(traceStream, traceGroupStream) {
		this.traceStream = traceStream;
		this.traceGroupStream = traceGroupStream;
	}
	async getAllGroups() {
		return await this.traceGroupStream.getGroup("default");
	}
	async deleteGroup(group) {
		const traces = await this.traceStream.getGroup(group.id);
		for (const trace of traces) await this.traceStream.delete(group.id, trace.id);
		await this.traceGroupStream.delete("default", group.id);
	}
	async clear() {
		const groups = await this.getAllGroups();
		for (const group of groups) await this.deleteGroup(group);
	}
	async createTracer(traceId, step, logger) {
		const traceGroup = {
			id: traceId,
			name: step.config.name,
			startTime: Date.now()
		};
		const trace = createTrace(traceGroup, step);
		const manager = new TraceManager(this.traceStream, this.traceGroupStream, traceGroup, trace);
		await manager.updateTraceGroup();
		return new StreamTracer(manager, traceGroup, trace, logger);
	}
	async attachToTrace(traceId, step, logger) {
		const existingGroup = await this.traceGroupStream.get("default", traceId);
		if (!existingGroup) return this.createTracer(traceId, step, logger);
		const trace = createTrace(existingGroup, step);
		return new StreamTracer(new TraceManager(this.traceStream, this.traceGroupStream, existingGroup, trace), existingGroup, trace, logger);
	}
};
const createTracerFactory = (lockedData) => {
	if (!lockedData.redisClient) throw new Error("Redis client is required for tracer factory. Please provide a redisClient when creating LockedData.");
	const traceStreamName = "motia-trace";
	const traceStreamAdapter = new RedisTraceStreamAdapter(traceStreamName, lockedData.redisClient);
	const traceStream = lockedData.createStream({
		filePath: traceStreamName,
		hidden: true,
		config: {
			name: traceStreamName,
			baseConfig: {
				storageType: "custom",
				factory: () => traceStreamAdapter
			},
			schema: null
		}
	})();
	const traceGroupName = "motia-trace-group";
	const traceGroupStreamAdapter = new RedisTraceStreamAdapter(traceGroupName, lockedData.redisClient);
	return new BaseTracerFactory(traceStream, lockedData.createStream({
		filePath: traceGroupName,
		hidden: true,
		config: {
			name: traceGroupName,
			baseConfig: {
				storageType: "custom",
				factory: () => traceGroupStreamAdapter
			},
			schema: null
		}
	})());
};

//#endregion
export { createTracerFactory };
