// @motiadev/core/dist/src/node/rpc.mjs
import crypto from "crypto";

//#region src/node/rpc.ts
var RpcSender = class {
	constructor(process) {
		this.process = process;
		this.pendingRequests = {};
	}
	async close() {
		if (Object.values(this.pendingRequests).length > 0) {
			console.error("Process ended while there are some promises outstanding");
			this.process.exit(1);
		}
	}
	send(method, args) {
		return new Promise((resolve, reject) => {
			const id = crypto.randomUUID();
			this.pendingRequests[id] = {
				resolve,
				reject,
				method,
				args
			};
			this.process.send?.({
				type: "rpc_request",
				id,
				method,
				args
			});
		});
	}
	sendNoWait(method, args) {
		this.process.send?.({
			type: "rpc_request",
			method,
			args
		});
	}
	init() {
		this.process.on("message", (msg) => {
			if (msg.type === "rpc_response") {
				const { id, result, error } = msg;
				const callbacks = this.pendingRequests[id];
				if (!callbacks) return;
				else if (error) callbacks.reject(error);
				else callbacks.resolve(result);
				delete this.pendingRequests[id];
			}
		});
	}
};

//#endregion
export { RpcSender };
