// @motiadev/adapter-bullmq-events/dist/bullmq-event-adapter.mjs
import { buildConfig } from "./config-builder.mjs";
import { ConnectionManager } from "./connection-manager.mjs";
import { DLQManager } from "./dlq-manager.mjs";
import { QueueManager } from "./queue-manager.mjs";
import { WorkerManager } from "./worker-manager.mjs";

//#region src/bullmq-event-adapter.ts
var BullMQEventAdapter = class {
	constructor(config) {
		this._config = buildConfig(config);
		this.connectionManager = new ConnectionManager(config.connection);
		this._queueManager = new QueueManager(this.connectionManager.connection, this._config);
		this._dlqManager = new DLQManager(this.connectionManager.connection, this._config);
		this._workerManager = new WorkerManager(this.connectionManager.connection, this._config, (topic, stepName) => this._queueManager.getQueueName(topic, stepName), this._dlqManager);
	}
	get connection() {
		return this.connectionManager.connection;
	}
	get prefix() {
		return this._config.prefix;
	}
	get dlqSuffix() {
		return this._config.dlq.suffix;
	}
	get queueManager() {
		return this._queueManager;
	}
	get workerManager() {
		return this._workerManager;
	}
	get dlqManager() {
		return this._dlqManager;
	}
	async emit(event) {
		const subscribers = this._workerManager.getSubscribers(event.topic);
		if (subscribers.length === 0) return;
		await this._queueManager.enqueueToAll(event, subscribers);
	}
	async subscribe(topic, stepName, handler, options) {
		const queueName = this._queueManager.getQueueName(topic, stepName);
		this._queueManager.getQueue(queueName);
		return this._workerManager.createWorker(topic, stepName, handler, options);
	}
	async unsubscribe(handle) {
		const workerInfo = this._workerManager.getWorkerInfo(handle.id);
		if (workerInfo) {
			const queueName = this._queueManager.getQueueName(workerInfo.topic, workerInfo.stepName);
			await this._queueManager.closeQueue(queueName);
			await this._workerManager.removeWorker(handle.id);
		}
	}
	async shutdown() {
		await Promise.allSettled([
			this._workerManager.closeAll(),
			this._queueManager.closeAll(),
			this._dlqManager.closeAll()
		]);
		try {
			await this.connectionManager.close();
		} catch (err) {
			console.error("[BullMQ] Error closing connection:", err);
		}
	}
	async getSubscriptionCount(topic) {
		return this._workerManager.getSubscriptionCount(topic);
	}
	async listTopics() {
		return this._workerManager.listTopics();
	}
};

//#endregion
export { BullMQEventAdapter };
