// @motiadev/adapter-bullmq-events/dist/dlq-manager.mjs
import { DLQ_JOB_PREFIX, LOG_PREFIX, MILLISECONDS_PER_SECOND } from "./constants.mjs";
import { QueueCreationError } from "./errors.mjs";
import { Queue } from "bullmq";

//#region src/dlq-manager.ts
var DLQManager = class {
	constructor(connection, config) {
		this.dlqQueues = /* @__PURE__ */ new Map();
		this.connection = connection;
		this.config = config;
	}
	getDLQQueueName(topic, stepName) {
		return `${`${topic}.${stepName}`}${this.config.dlq.suffix}`;
	}
	getOrCreateDLQQueue(queueName) {
		if (!this.dlqQueues.has(queueName)) {
			const ttlMs = this.config.dlq.ttl * MILLISECONDS_PER_SECOND;
			const queue$1 = new Queue(queueName, {
				connection: this.connection,
				prefix: this.config.prefix,
				defaultJobOptions: {
					removeOnComplete: { age: ttlMs },
					removeOnFail: { age: ttlMs }
				}
			});
			queue$1.on("error", (err) => {
				console.error(`${LOG_PREFIX} DLQ error for ${queueName}:`, err);
			});
			this.dlqQueues.set(queueName, queue$1);
		}
		const queue = this.dlqQueues.get(queueName);
		if (!queue) throw new QueueCreationError(queueName);
		return queue;
	}
	async moveToDLQ(topic, stepName, event, error, attemptsMade, originalJobId) {
		if (!this.config.dlq.enabled) {
			console.warn(`${LOG_PREFIX} DLQ is disabled, skipping move to DLQ for topic ${topic}, step ${stepName}`);
			return;
		}
		try {
			const dlqQueueName = this.getDLQQueueName(topic, stepName);
			const dlqQueue = this.getOrCreateDLQQueue(dlqQueueName);
			const dlqJobData = {
				originalEvent: {
					topic: event.topic,
					data: event.data,
					traceId: event.traceId || "unknown",
					...event.flows && { flows: event.flows },
					...event.messageGroupId && { messageGroupId: event.messageGroupId }
				},
				failureReason: error.message || "Unknown error",
				failureTimestamp: Date.now(),
				attemptsMade,
				...originalJobId && { originalJobId }
			};
			const jobOptions = originalJobId ? { jobId: `${DLQ_JOB_PREFIX}${originalJobId}` } : {};
			await dlqQueue.add(`${topic}.dlq`, dlqJobData, jobOptions);
			console.warn(`${LOG_PREFIX} Moved failed job to DLQ: ${dlqQueueName}`, {
				topic,
				stepName,
				attemptsMade,
				error: error.message
			});
		} catch (err) {
			const dlqError = err instanceof Error ? err : new Error(String(err));
			console.error(`${LOG_PREFIX} Failed to move job to DLQ for topic ${topic}, step ${stepName}:`, dlqError);
		}
	}
	async closeDLQQueue(queueName) {
		const queue = this.dlqQueues.get(queueName);
		if (queue) {
			await queue.close();
			this.dlqQueues.delete(queueName);
		}
	}
	async closeAll() {
		const promises = Array.from(this.dlqQueues.values()).map((queue) => queue.close().catch((err) => {
			console.error(`${LOG_PREFIX} Error closing DLQ queue:`, err);
		}));
		await Promise.allSettled(promises);
		this.dlqQueues.clear();
	}
	getDLQQueue(queueName) {
		return this.dlqQueues.get(queueName);
	}
	getOrCreateDLQ(queueName) {
		return this.getOrCreateDLQQueue(queueName);
	}
	listDLQQueueNames() {
		return Array.from(this.dlqQueues.keys());
	}
	getDLQSuffix() {
		return this.config.dlq.suffix;
	}
	getPrefix() {
		return this.config.prefix;
	}
	getConnection() {
		return this.connection;
	}
};

//#endregion
export { DLQManager };
