// @motiadev/stream-client-react/dist/index.js
import { Stream, Stream as Stream$1, StreamGroupSubscription, StreamItemSubscription } from "@motiadev/stream-client-browser";
import { c } from "react/compiler-runtime";
import React, { useEffect, useRef, useState } from "react";
import { jsx } from "react/jsx-runtime";

//#region src/motia-stream-context.ts
const MotiaStreamContext = React.createContext({ stream: null });

//#endregion
//#region src/motia-stream-provider.tsx
const MotiaStreamProvider = (t0) => {
	const $ = c(9);
	const { children, address, protocols } = t0;
	const [stream, setStream] = useState(null);
	let t1;
	let t2;
	if ($[0] !== address || $[1] !== protocols) {
		t1 = () => {
			const streamInstance = new Stream$1(address, { protocols });
			setStream(streamInstance);
			return () => streamInstance.close();
		};
		t2 = [address, protocols];
		$[0] = address;
		$[1] = protocols;
		$[2] = t1;
		$[3] = t2;
	} else {
		t1 = $[2];
		t2 = $[3];
	}
	useEffect(t1, t2);
	let t3;
	if ($[4] !== stream) {
		t3 = { stream };
		$[4] = stream;
		$[5] = t3;
	} else t3 = $[5];
	const contextValue = t3;
	let t4;
	if ($[6] !== children || $[7] !== contextValue) {
		t4 = /* @__PURE__ */ jsx(MotiaStreamContext.Provider, {
			value: contextValue,
			children
		});
		$[6] = children;
		$[7] = contextValue;
		$[8] = t4;
	} else t4 = $[8];
	return t4;
};

//#endregion
//#region src/use-motia-stream.ts
/**
* A hook to get the stream context.
*
* @example
* ```tsx
* const { stream } = useMotiaStream()
* ```
*/
const useMotiaStream = () => {
	const context = React.useContext(MotiaStreamContext);
	if (!context) throw new Error("useMotiaStream must be used within a MotiaStreamProvider");
	return context;
};

//#endregion
//#region src/use-stream-event-handler.ts
/**
* A hook to handle custom stream events.
*
* @example
* ```tsx
* const { event } = useStreamItem({ streamName: 'my-stream', id: '123' })
*
* const onEventHandled = (event: any) => {
*   // this is going to be called whenever 'on-custom-event' is sent from the server
*   console.log(event)
* }
*
* useStreamEventHandler({ event, type: 'on-custom-event', listener: onEventHandled }, [])
* ```
*/
const useStreamEventHandler = (t0, dependencies) => {
	const $ = c(8);
	const { event, type, listener } = t0;
	let t1;
	if ($[0] !== event || $[1] !== listener || $[2] !== type) {
		t1 = () => {
			if (event) {
				event.onEvent(type, listener);
				return () => event.offEvent(type, listener);
			}
		};
		$[0] = event;
		$[1] = listener;
		$[2] = type;
		$[3] = t1;
	} else t1 = $[3];
	let t2;
	if ($[4] !== dependencies || $[5] !== event || $[6] !== type) {
		t2 = [
			event,
			type,
			...dependencies
		];
		$[4] = dependencies;
		$[5] = event;
		$[6] = type;
		$[7] = t2;
	} else t2 = $[7];
	useEffect(t1, t2);
};

//#endregion
//#region src/use-stream-group.ts
/**
* A hook to get a group of items from a stream.
*
* @example
* ```tsx
* const { data } = useStreamGroup<{ id:string; name: string }>({
*   streamName: 'my-stream',
*   groupId: '123',
* })
*
* return (
*   <div>
*     {data.map((item) => (
*       <div key={item.id}>{item.name}</div>
*     ))}
*   </div>
* )
* ```
*/
const useStreamGroup = (args) => {
	const { stream } = useMotiaStream();
	const [data, setData] = useState([]);
	const subscriptionRef = useRef(null);
	const { streamName, groupId, sortKey, setData: setDataCallback } = args || {};
	useEffect(() => {
		if (!streamName || !groupId || !stream) {
			console.error("useStreamGroup: streamName, groupId or stream is not defined", {
				streamName,
				groupId,
				stream
			});
			return;
		}
		subscriptionRef.current = stream.subscribeGroup(streamName, groupId, sortKey);
		subscriptionRef.current.addChangeListener((data_0) => {
			const typedData = data_0;
			setData(typedData);
			setDataCallback?.(typedData);
		});
		return () => {
			subscriptionRef.current?.close();
			subscriptionRef.current = null;
			setData([]);
		};
	}, [
		stream,
		streamName,
		groupId,
		sortKey,
		setDataCallback
	]);
	return {
		data,
		event: subscriptionRef.current
	};
};

//#endregion
//#region src/use-stream-item.ts
/**
* A hook to get a single item from a stream.
*
* @example
* ```tsx
* const { data } = useStreamItem<{ id:string; name: string }>({
*   streamName: 'my-stream',
*   groupId: '123',
*   id: '123',
* })
*
* return (
*   <div>{data?.name}</div>
* )
* ```
*/
const useStreamItem = (args) => {
	const $ = c(12);
	const { stream } = useMotiaStream();
	const [data, setData] = useState();
	const [event, setEvent] = useState(null);
	let t0;
	if ($[0] !== args) {
		t0 = args || {};
		$[0] = args;
		$[1] = t0;
	} else t0 = $[1];
	const { streamName, groupId, id } = t0;
	let t1;
	if ($[2] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = (data_0) => {
			setData(data_0);
		};
		$[2] = t1;
	} else t1 = $[2];
	const handleChange = t1;
	let t2;
	let t3;
	if ($[3] !== groupId || $[4] !== id || $[5] !== stream || $[6] !== streamName) {
		t2 = () => {
			if (!streamName || !groupId || !id || !stream) return;
			const subscription = stream.subscribeItem(streamName, groupId, id);
			subscription.addChangeListener(handleChange);
			setEvent(subscription);
			return () => {
				setData(void 0);
				setEvent(null);
				subscription.close();
			};
		};
		t3 = [
			stream,
			streamName,
			groupId,
			id,
			handleChange
		];
		$[3] = groupId;
		$[4] = id;
		$[5] = stream;
		$[6] = streamName;
		$[7] = t2;
		$[8] = t3;
	} else {
		t2 = $[7];
		t3 = $[8];
	}
	useEffect(t2, t3);
	let t4;
	if ($[9] !== data || $[10] !== event) {
		t4 = {
			data,
			event
		};
		$[9] = data;
		$[10] = event;
		$[11] = t4;
	} else t4 = $[11];
	return t4;
};

//#endregion
export { MotiaStreamProvider, Stream, StreamGroupSubscription, StreamItemSubscription, useMotiaStream, useStreamEventHandler, useStreamGroup, useStreamItem };
