// motia/dist/openapi/generate.mjs
import { processSchema } from "./process-schema.mjs";
import { isHttpMethod } from "./utils.mjs";
import * as fs$1 from "fs";
import * as path$1 from "path";

//#region src/openapi/generate.ts
function generateOpenApi(projectDir, apiSteps, title, version, outputFile = "openapi.json") {
	if (!title || !version) try {
		const packageJsonPath = path$1.join(projectDir, "package.json");
		const packageJson = JSON.parse(fs$1.readFileSync(packageJsonPath, "utf8"));
		if (!title) title = packageJson.name;
		if (!version) version = packageJson.version;
	} catch (error) {
		console.warn(`Could not read package.json in ${projectDir} to determine project name. Using default.`);
	}
	const openApi = {
		openapi: "3.0.0",
		info: {
			title: title ?? "Motia Project API",
			version: version ?? "1.0.0"
		},
		paths: {},
		components: { schemas: {} }
	};
	for (const step of apiSteps) {
		const pathItem = openApi.paths[step.config.path] || {};
		const method = step.config.method.toLowerCase();
		const operation = {
			summary: step.config.name,
			description: step.config.description,
			requestBody: void 0,
			responses: {}
		};
		if (step.config.queryParams) {
			operation.parameters = operation.parameters || [];
			for (const param of step.config.queryParams) operation.parameters.push({
				in: "query",
				name: param.name,
				description: param.description,
				required: false,
				schema: { type: "string" }
			});
		}
		if (step.config.bodySchema) {
			const bodySchema = step.config.bodySchema;
			delete bodySchema.$schema;
			const processedSchema = processSchema(bodySchema, openApi);
			operation.requestBody = { content: { "application/json": { schema: processedSchema } } };
		}
		if (step.config.responseSchema) for (const [statusCode, responseSchema] of Object.entries(step.config.responseSchema)) {
			const resSchema = responseSchema;
			delete resSchema.$schema;
			const processedSchema = processSchema(resSchema, openApi);
			operation.responses[statusCode] = {
				description: `Response for status code ${statusCode}`,
				content: { "application/json": { schema: processedSchema } }
			};
		}
		if (isHttpMethod(method)) pathItem[method] = operation;
		openApi.paths[step.config.path] = pathItem;
	}
	const openApiJson = JSON.stringify(openApi, null, 2);
	fs$1.writeFileSync(path$1.join(projectDir, outputFile), openApiJson);
	console.log(`✅ OpenAPI specification generated successfully at ${outputFile}`);
}

//#endregion
export { generateOpenApi };
