// @motiadev/core/dist/src/logger.mjs
import { prettyPrint } from "./pretty-print.mjs";

//#region src/logger.ts
const LEVELS = {
	NOTSET: 0,
	DEBUG: 10,
	INFO: 20,
	WARNING: 30,
	ERROR: 40,
	CRITICAL: 50
};
const levelMap = {
	debug: LEVELS.DEBUG,
	info: LEVELS.INFO,
	warn: LEVELS.WARNING,
	warning: LEVELS.WARNING,
	error: LEVELS.ERROR,
	critical: LEVELS.CRITICAL
};
const getLogLevel = () => {
	return levelMap[process.env.LOG_LEVEL ?? "info"] ?? LEVELS.INFO;
};
const shouldLog = (messageLevel) => {
	return messageLevel >= getLogLevel();
};
var Logger = class Logger {
	constructor(isVerbose = false, meta = {}, coreListeners = []) {
		this.isVerbose = isVerbose;
		this.meta = meta;
		this.coreListeners = coreListeners;
		this.listeners = [];
	}
	child(meta) {
		return new Logger(this.isVerbose, {
			...this.meta,
			...meta
		}, this.coreListeners);
	}
	_log(level, msg, args) {
		const time = Date.now();
		const meta = {
			...this.meta,
			...args ?? {}
		};
		prettyPrint({
			level,
			time,
			msg,
			...meta
		}, !this.isVerbose);
		this.coreListeners.forEach((listener) => listener(level, msg, meta));
		this.listeners.forEach((listener) => listener(level, msg, meta));
	}
	info(message, args) {
		if (shouldLog(LEVELS.INFO)) this._log("info", message, args);
	}
	error(message, args) {
		if (shouldLog(LEVELS.ERROR)) this._log("error", message, args);
	}
	debug(message, args) {
		if (shouldLog(LEVELS.DEBUG)) this._log("debug", message, args);
	}
	warn(message, args) {
		if (shouldLog(LEVELS.WARNING)) this._log("warn", message, args);
	}
	log(args) {
		const level = args.level ?? "info";
		if (!shouldLog(levelMap[level] ?? LEVELS.INFO)) return;
		this._log(level, args.msg, args);
	}
	addListener(listener) {
		this.listeners.push(listener);
	}
	removeListener(listener) {
		const index = this.listeners.indexOf(listener);
		if (index !== -1) {
			this.listeners.splice(index, 1);
			return true;
		}
		return false;
	}
};
const globalLogger = new Logger();

//#endregion
export { Logger, globalLogger };
