// @motiadev/core/dist/src/pretty-print.mjs
import pc from "picocolors";

//#region src/pretty-print.ts
const stepTag = (step) => pc.bold(pc.cyan(step));
const timestampTag = (timestamp) => pc.gray(timestamp);
const traceIdTag = (traceId) => pc.gray(traceId);
const levelTags = {
	error: pc.red("[ERROR]"),
	info: pc.blue("[INFO]"),
	warn: pc.yellow("[WARN]"),
	debug: pc.gray("[DEBUG]"),
	trace: pc.gray("[TRACE]")
};
const numericTag = (value) => pc.green(value);
const stringTag = (value) => pc.cyan(value);
const booleanTag = (value) => pc.blue(value);
const arrayBrackets = ["[", "]"].map((s) => pc.gray(s));
const objectBrackets = ["{", "}"].map((s) => pc.gray(s));
const prettyPrintObject = (obj, depth = 0, parentIsLast = false, prefix = "") => {
	const tab = prefix + (depth === 0 ? "" : parentIsLast ? "│ " : "│ ");
	if (depth > 2) return `${tab} └ ${pc.gray("[...]")}`;
	const entries = Object.entries(obj);
	return entries.map(([key, value], index) => {
		const isLast = index === entries.length - 1;
		const isObject = typeof value === "object" && value !== null;
		const prefix$1 = isLast && !isObject ? "└" : "├";
		if (isObject) {
			const subObject = prettyPrintObject(value, depth + 1, isLast, tab);
			const [start, end] = Array.isArray(value) ? arrayBrackets : objectBrackets;
			return `${tab}${prefix$1} ${key}: ${start}\n${subObject}\n${tab}${isLast ? "└" : "│"} ${end}`;
		}
		let printedValue = value;
		if (typeof value === "number") printedValue = numericTag(String(value));
		else if (typeof value === "boolean") printedValue = booleanTag(String(value));
		else if (typeof value === "string") printedValue = stringTag(value);
		return `${tab}${prefix$1} ${key}: ${printedValue}`;
	}).join("\n");
};
const prettyPrint = (json, excludeDetails = false) => {
	const { time, traceId, msg, flows, level, step, ...details } = json;
	const levelTag = levelTags[level];
	const timestamp = timestampTag(`[${new Date(time).toLocaleTimeString()}]`);
	const objectHasKeys = Object.keys(details).length > 0;
	console.log(`${timestamp} ${traceIdTag(traceId)} ${levelTag} ${stepTag(step)} ${msg}`);
	if (objectHasKeys && !excludeDetails) console.log(prettyPrintObject(details));
};

//#endregion
export { prettyPrint };
