// @motiadev/core/dist/src/step-handlers.mjs
import { getQueueConfigWithDefaults } from "./infrastructure-validator/defaults.mjs";
import { globalLogger } from "./logger.mjs";
import { callStepFile } from "./call-step-file.mjs";
import { validateInfrastructureConfig } from "./infrastructure-validator/validations.mjs";
import { validateEventInput } from "./validate-event-input.mjs";

//#region src/step-handlers.ts
const createStepHandlers = (motia, eventAdapter) => {
	const eventSteps = motia.lockedData.eventSteps();
	const handlerMap = /* @__PURE__ */ new Map();
	globalLogger.debug(`[step handler] creating step handlers for ${eventSteps.length} steps`);
	const removeLogger = (event) => {
		const { logger, tracer, ...rest } = event;
		return rest;
	};
	const createHandler = (step) => {
		const { config, filePath } = step;
		const { subscribes, name } = config;
		globalLogger.debug("[step handler] establishing step subscriptions", {
			filePath,
			step: step.config.name
		});
		if (config.infrastructure) {
			globalLogger.debug("[step handler] validating infrastructure config", {
				step: name,
				infrastructure: config.infrastructure
			});
			const validationResult = validateInfrastructureConfig(config.infrastructure);
			if (!validationResult.success && validationResult.errors) {
				globalLogger.error("[step handler] Infrastructure configuration validation failed", {
					step: name,
					filePath,
					errors: validationResult.errors
				});
				return;
			}
			globalLogger.debug("[step handler] infrastructure config validated successfully", { step: name });
		}
		const queueConfig = getQueueConfigWithDefaults(config.infrastructure);
		const handlers = [];
		subscribes.forEach(async (subscribe) => {
			const handler = async (event) => {
				const { data, traceId, flows } = event;
				const logger = event.logger ? event.logger.child({ step: step.config.name }) : motia.loggerFactory.create({
					traceId,
					flows: flows || [],
					stepName: step.config.name
				});
				const tracer = event.tracer ? event.tracer.child(step, logger) : await motia.tracerFactory.attachToTrace(traceId, step, logger);
				globalLogger.debug("[step handler] received event", {
					event: removeLogger(event),
					step: name
				});
				validateEventInput(step, event, motia);
				await callStepFile({
					step,
					data,
					traceId,
					tracer,
					logger,
					infrastructure: config.infrastructure
				}, motia);
			};
			const subscriptionHandle = await eventAdapter.subscribe(subscribe, name, handler, queueConfig);
			handlers.push(subscriptionHandle);
		});
		handlerMap.set(filePath, handlers);
	};
	const removeHandler = (step) => {
		const { filePath } = step;
		const handlers = handlerMap.get(filePath);
		if (handlers) {
			handlers.forEach((subscriptionHandle) => {
				eventAdapter.unsubscribe(subscriptionHandle);
				globalLogger.debug("[step handler] unsubscribed handler", {
					filePath,
					...subscriptionHandle,
					step: step.config.name
				});
			});
			handlerMap.delete(filePath);
		}
	};
	eventSteps.forEach(createHandler);
	return {
		removeHandler,
		createHandler
	};
};

//#endregion
export { createStepHandlers };
