// @motiadev/plugin-states/dist/plugin.js
//#region src/api.ts
const api = (motia) => {
	motia.registerApi({
		method: "GET",
		path: "/__motia/state"
	}, async (req) => {
		try {
			const groupId = req.queryParams.groupId;
			const filter = req.queryParams.filter ? JSON.parse(req.queryParams.filter) : void 0;
			return {
				status: 200,
				body: await motia.state.items({
					groupId,
					filter
				})
			};
		} catch (error) {
			return {
				status: 500,
				body: { error: error instanceof Error ? error.message : "Unknown error" }
			};
		}
	});
	motia.registerApi({
		method: "POST",
		path: "/__motia/state"
	}, async (req, ctx) => {
		try {
			const { key, groupId, value } = req.body;
			await ctx.state.set(groupId, key, value);
			return {
				status: 200,
				body: {
					key,
					groupId,
					value
				}
			};
		} catch (error) {
			return {
				status: 500,
				body: { error: error instanceof Error ? error.message : "Unknown error" }
			};
		}
	});
	motia.registerApi({
		method: "POST",
		path: "/__motia/state/delete"
	}, async (req, ctx) => {
		try {
			for (const id of req.body.ids) {
				const [groupId, ...keyParts] = id.split(":");
				const key = keyParts.join(":");
				await ctx.state.delete(groupId, key);
			}
			return {
				status: 204,
				body: ""
			};
		} catch (error) {
			return {
				status: 500,
				body: { error: error instanceof Error ? error.message : "Unknown error" }
			};
		}
	});
};

//#endregion
//#region src/plugin.ts
function plugin(motia) {
	api(motia);
	return { workbench: [{
		packageName: "@motiadev/plugin-states",
		cssImports: ["@motiadev/plugin-states/dist/styles.css"],
		label: "States",
		position: "bottom",
		componentName: "StatesPage",
		labelIcon: "file"
	}] };
}

//#endregion
export { plugin as default };
