// @motiadev/stream-client-node/dist/index.mjs
import { Stream as Stream$1, StreamGroupSubscription, StreamItemSubscription, StreamSubscription } from "@motiadev/stream-client";
import { WebSocket } from "ws";

//#region src/stream-adapter.ts
var StreamSocketAdapter = class {
	constructor(address, protocols) {
		this.address = address;
		this.ws = new WebSocket(this.address, protocols);
	}
	connect() {}
	send(message) {
		this.ws.send(message);
	}
	onMessage(callback) {
		const listener = (message) => callback(message.data.toString());
		this.ws.addEventListener("message", listener);
	}
	onOpen(callback) {
		this.ws.addEventListener("open", callback);
	}
	onClose(callback) {
		this.ws.addEventListener("close", callback);
	}
	close() {
		this.ws.close();
		this.ws.removeAllListeners();
	}
	isOpen() {
		return this.ws.readyState === WebSocket.OPEN;
	}
};

//#endregion
//#region src/stream.ts
var Stream = class extends Stream$1 {
	constructor(address, options) {
		super(() => new StreamSocketAdapter(address, options?.protocols));
	}
};

//#endregion
export { Stream, StreamGroupSubscription, StreamItemSubscription, StreamSubscription };
