// motia/dist/cloud/new-deployment/utils/upload.mjs
import { distDir, maxUploadSize } from "../constants.mjs";
import { cloudApi } from "../cloud-api/index.mjs";
import fs from "fs";
import path from "path";
import axios from "axios";

//#region src/cloud/new-deployment/utils/upload.ts
const upload = async (deploymentToken, fileName, onProgress) => {
	const filePath = path.join(distDir, fileName);
	const fileStats = fs.statSync(filePath);
	const { fileInfo: { presignedUrl } } = await cloudApi.upload({
		deploymentToken,
		originalName: fileName,
		size: fileStats.size,
		mimetype: "application/zip"
	});
	await uploadToPresignedUrl(filePath, presignedUrl, fileStats.size, onProgress);
};
const uploadToPresignedUrl = async (filePath, presignedUrl, fileSize, onProgress) => {
	const fileName = path.basename(filePath);
	const fileStream = fs.createReadStream(filePath);
	await axios.put(presignedUrl, fileStream, {
		headers: {
			"Content-Type": "application/zip",
			"Content-Length": fileSize,
			"Content-Disposition": `attachment; filename="${fileName}"`
		},
		maxContentLength: maxUploadSize,
		maxBodyLength: maxUploadSize,
		onUploadProgress(progressEvent) {
			const progress = progressEvent.progress ?? 0;
			onProgress(Math.round(progress * 100));
		}
	});
};

//#endregion
export { upload };
