// @motiadev/plugin-endpoint/dist/index.js
import { c } from "react/compiler-runtime";
import { BackgroundEffect, Badge, Button, Checkbox, Input, Panel, Sidebar, Tabs, TabsContent, TabsList, TabsTrigger, cn, useThemeStore } from "@motiadev/ui";
import { Fragment, memo, useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState } from "react";
import { AlertCircle, Book, Check, ChevronDown, CircleX, Copy, Download, Loader2, Plus, Search, Trash2, X } from "lucide-react";
import { Fragment as Fragment$1, jsx, jsxs } from "react/jsx-runtime";
import { create } from "zustand";
import { createJSONStorage, devtools, persist } from "zustand/middleware";
import { useStreamGroup, useStreamItem } from "@motiadev/stream-client-react";
import { useShallow } from "zustand/react/shallow";
import ReactJson from "react18-json-view";
import "react18-json-view/src/dark.css";
import "react18-json-view/src/style.css";
import { Editor, useMonaco } from "@monaco-editor/react";
import { Prism } from "react-syntax-highlighter";
import { atomDark, oneLight } from "react-syntax-highlighter/dist/esm/styles/prism";

//#region src/components/endpoints-search.tsx
const EndpointsSearch = (t0) => {
	const $ = c(16);
	if ($[0] !== "b04fe9fcf2dc81964e7b2f3a7fac01e0ffd66006443485e8e32bf9d2d88776cf") {
		for (let $i = 0; $i < 16; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "b04fe9fcf2dc81964e7b2f3a7fac01e0ffd66006443485e8e32bf9d2d88776cf";
	}
	const { value, onChange, onClear } = t0;
	let t1;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = /* @__PURE__ */ jsx(Search, { className: "absolute left-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground/50" });
		$[1] = t1;
	} else t1 = $[1];
	let t2;
	if ($[2] !== onChange) {
		t2 = (e) => onChange(e.target.value);
		$[2] = onChange;
		$[3] = t2;
	} else t2 = $[3];
	let t3;
	if ($[4] !== t2 || $[5] !== value) {
		t3 = /* @__PURE__ */ jsx(Input, {
			variant: "shade",
			value,
			onChange: t2,
			className: "px-9! font-medium",
			placeholder: "Search by Method or Path"
		});
		$[4] = t2;
		$[5] = value;
		$[6] = t3;
	} else t3 = $[6];
	const t4 = value !== "";
	const t5 = value === "";
	let t6;
	if ($[7] !== t4 || $[8] !== t5) {
		t6 = cn("cursor-pointer absolute right-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground/50 hover:text-muted-foreground", {
			visible: t4,
			invisible: t5
		});
		$[7] = t4;
		$[8] = t5;
		$[9] = t6;
	} else t6 = $[9];
	let t7;
	if ($[10] !== onClear || $[11] !== t6) {
		t7 = /* @__PURE__ */ jsx(X, {
			className: t6,
			onClick: onClear
		});
		$[10] = onClear;
		$[11] = t6;
		$[12] = t7;
	} else t7 = $[12];
	let t8;
	if ($[13] !== t3 || $[14] !== t7) {
		t8 = /* @__PURE__ */ jsx("div", {
			className: "p-2 border-b gap-4",
			"data-testid": "endpoints-search-container",
			children: /* @__PURE__ */ jsxs("div", {
				className: "relative",
				children: [
					t1,
					t3,
					t7
				]
			})
		});
		$[13] = t3;
		$[14] = t7;
		$[15] = t8;
	} else t8 = $[15];
	return t8;
};

//#endregion
//#region src/components/endpoint-badge.tsx
const getMethodStyles = (method) => {
	switch (method.toUpperCase()) {
		case "POST": return "bg-[#258DC3]/15 text-[#258DC3]";
		case "GET": return "bg-[#709A2D]/15 text-[#709A2D]";
		case "DELETE": return "bg-[#DE2134]/15 text-[#DE2134]";
		case "PUT": return "bg-[#B9922D]/15 text-[#B9922D]";
		case "PATCH": return "bg-[#B9922D]/15 text-[#B9922D]";
		case "HEAD": return "bg-[#E221DF]/15 text-[#E221DF]";
		case "OPTIONS": return "bg-[#B9922D]/15 text-[#B9922D]";
		default: return "bg-[#258DC3]/15 text-[#258DC3]";
	}
};
const EndpointBadge = memo((t0) => {
	const $ = c(11);
	if ($[0] !== "1017d069da7af8d79b772819cd6ea273bc02186be5bf8c70750727fd63506daa") {
		for (let $i = 0; $i < 11; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "1017d069da7af8d79b772819cd6ea273bc02186be5bf8c70750727fd63506daa";
	}
	let className;
	let props;
	let variant;
	if ($[1] !== t0) {
		({className, variant, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
		$[4] = variant;
	} else {
		className = $[2];
		props = $[3];
		variant = $[4];
	}
	let t1;
	if ($[5] !== className || $[6] !== variant) {
		t1 = cn("rounded-lg px-2 py-0.5 text-xs font-mono font-bold transition-colors", getMethodStyles(variant), className);
		$[5] = className;
		$[6] = variant;
		$[7] = t1;
	} else t1 = $[7];
	let t2;
	if ($[8] !== props || $[9] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: t1,
			...props
		});
		$[8] = props;
		$[9] = t1;
		$[10] = t2;
	} else t2 = $[10];
	return t2;
});

//#endregion
//#region src/components/endpoint-path.tsx
const PathSeparator = () => {
	const $ = c(2);
	if ($[0] !== "603f796b38bb9b15093370b94bc9b921b7621c9f6baa94432316d5a5ca450c9c") {
		for (let $i = 0; $i < 2; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "603f796b38bb9b15093370b94bc9b921b7621c9f6baa94432316d5a5ca450c9c";
	}
	let t0;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t0 = /* @__PURE__ */ jsx("div", {
			className: "text-muted-foreground",
			children: "/"
		});
		$[1] = t0;
	} else t0 = $[1];
	return t0;
};
const EndpointPath = (t0) => {
	const $ = c(14);
	if ($[0] !== "603f796b38bb9b15093370b94bc9b921b7621c9f6baa94432316d5a5ca450c9c") {
		for (let $i = 0; $i < 14; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "603f796b38bb9b15093370b94bc9b921b7621c9f6baa94432316d5a5ca450c9c";
	}
	const { method, path } = t0;
	let t1;
	if ($[1] !== path) {
		const parts = path.split("/").filter(Boolean);
		const partsLength = parts.length - 1;
		let t2$1;
		if ($[3] !== partsLength) {
			t2$1 = (part, index) => {
				const isLast = index === partsLength;
				const key = `part-${part}-${index}`;
				const separator = isLast ? void 0 : /* @__PURE__ */ jsx(PathSeparator, {}, `separator-${key}`);
				if (part.startsWith(":")) return [/* @__PURE__ */ jsx("div", {
					className: "bg-[#2862FE]/20 text-[#2862FE] rounded-sm px-1 py-0.5 text-sm font-mono font-bold font-medium",
					children: part
				}, key), separator];
				return [/* @__PURE__ */ jsx("div", { children: part }, key), separator];
			};
			$[3] = partsLength;
			$[4] = t2$1;
		} else t2$1 = $[4];
		t1 = parts.flatMap(t2$1);
		$[1] = path;
		$[2] = t1;
	} else t1 = $[2];
	const pathView = t1;
	const t2 = method;
	let t3;
	if ($[5] !== method || $[6] !== t2) {
		t3 = /* @__PURE__ */ jsx(EndpointBadge, {
			variant: t2,
			children: method
		});
		$[5] = method;
		$[6] = t2;
		$[7] = t3;
	} else t3 = $[7];
	let t4;
	if ($[8] === Symbol.for("react.memo_cache_sentinel")) {
		t4 = /* @__PURE__ */ jsx(PathSeparator, {});
		$[8] = t4;
	} else t4 = $[8];
	let t5;
	if ($[9] !== pathView) {
		t5 = /* @__PURE__ */ jsxs("span", {
			className: "font-mono font-bold whitespace-nowrap flex flex-row gap-2 items-center truncate",
			children: [t4, pathView]
		});
		$[9] = pathView;
		$[10] = t5;
	} else t5 = $[10];
	let t6;
	if ($[11] !== t3 || $[12] !== t5) {
		t6 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-[auto_1fr] gap-2 items-center",
			children: [t3, t5]
		});
		$[11] = t3;
		$[12] = t5;
		$[13] = t6;
	} else t6 = $[13];
	return t6;
};

//#endregion
//#region src/components/endpoint-item.tsx
const EndpointItem = (t0) => {
	const $ = c(22);
	if ($[0] !== "29651f7b7dd25cd15fa149793a83da8eadabea6c1b2d872f1881f15627da1c3a") {
		for (let $i = 0; $i < 22; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "29651f7b7dd25cd15fa149793a83da8eadabea6c1b2d872f1881f15627da1c3a";
	}
	const { endpoint, isSelected, isLast, onSelect } = t0;
	const t1 = `endpoint-${endpoint.method}-${endpoint.path}`;
	let t2;
	if ($[1] !== endpoint.id || $[2] !== onSelect) {
		t2 = () => onSelect(endpoint.id);
		$[1] = endpoint.id;
		$[2] = onSelect;
		$[3] = t2;
	} else t2 = $[3];
	const t3 = isSelected && "bg-muted-foreground/10";
	let t4;
	if ($[4] !== t3) {
		t4 = cn("grid grid-cols-[auto_1fr] items-center justify-center px-[17px] select-none hover:bg-muted-foreground/10 cursor-pointer", t3);
		$[4] = t3;
		$[5] = t4;
	} else t4 = $[5];
	let t5;
	if ($[6] !== isLast) {
		t5 = isLast ? /* @__PURE__ */ jsx("svg", {
			width: "14",
			height: "34",
			viewBox: "0 0 12 34",
			fill: "none",
			xmlns: "http://www.w3.org/2000/svg",
			children: /* @__PURE__ */ jsx("path", {
				d: "M6.5 16.5C6.50006 19.2614 8.7386 21.4999 11.5 21.5C11.7759 21.5003 12 21.724 12 22C12 22.276 11.7759 22.4997 11.5 22.5C8.18632 22.4999 5.50006 19.8137 5.5 16.5V0H6.5V16.5Z",
				className: "fill-[#555]"
			})
		}) : /* @__PURE__ */ jsx("svg", {
			width: "14",
			height: "34",
			viewBox: "0 0 12 34",
			fill: "none",
			xmlns: "http://www.w3.org/2000/svg",
			children: /* @__PURE__ */ jsx("path", {
				d: "M6.5 0V11.5C6.50007 14.2614 8.73861 16.4999 11.5 16.5C11.7759 16.5003 12 16.724 12 17C12 17.276 11.7759 17.4997 11.5 17.5C9.41273 17.5 7.57486 16.4335 6.5 14.8164V34H5.5V0H6.5Z",
				className: "fill-[#555]"
			})
		});
		$[6] = isLast;
		$[7] = t5;
	} else t5 = $[7];
	let t6;
	if ($[8] !== endpoint.method || $[9] !== endpoint.path) {
		t6 = /* @__PURE__ */ jsx(EndpointPath, {
			method: endpoint.method,
			path: endpoint.path
		});
		$[8] = endpoint.method;
		$[9] = endpoint.path;
		$[10] = t6;
	} else t6 = $[10];
	let t7;
	if ($[11] !== endpoint.description) {
		t7 = /* @__PURE__ */ jsx("span", {
			className: "text-sm text-muted-foreground truncate",
			children: endpoint.description
		});
		$[11] = endpoint.description;
		$[12] = t7;
	} else t7 = $[12];
	let t8;
	if ($[13] !== t6 || $[14] !== t7) {
		t8 = /* @__PURE__ */ jsx("div", { children: /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-[auto_1fr] items-center gap-3 px-2",
			children: [t6, t7]
		}) });
		$[13] = t6;
		$[14] = t7;
		$[15] = t8;
	} else t8 = $[15];
	let t9;
	if ($[16] !== t1 || $[17] !== t2 || $[18] !== t4 || $[19] !== t5 || $[20] !== t8) {
		t9 = /* @__PURE__ */ jsxs("div", {
			"data-testid": t1,
			onClick: t2,
			className: t4,
			children: [t5, t8]
		});
		$[16] = t1;
		$[17] = t2;
		$[18] = t4;
		$[19] = t5;
		$[20] = t8;
		$[21] = t9;
	} else t9 = $[21];
	return t9;
};

//#endregion
//#region src/components/flow-group.tsx
const FlowGroup = (t0) => {
	const $ = c(41);
	if ($[0] !== "a80b3cb43bb62d012f095b7847fb01d92f2a4fcd29fc05244f6ae839f1f0b9de") {
		for (let $i = 0; $i < 41; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "a80b3cb43bb62d012f095b7847fb01d92f2a4fcd29fc05244f6ae839f1f0b9de";
	}
	const { flow, endpoints, isOpen, isSelected, onToggle, onClearSelection, selectedEndpointId, onSelectEndpoint } = t0;
	const t1 = `flow-group-${flow}`;
	let t2;
	if ($[1] !== flow || $[2] !== isSelected || $[3] !== onClearSelection || $[4] !== onToggle) {
		t2 = () => {
			if (isSelected) onClearSelection();
			onToggle(flow);
		};
		$[1] = flow;
		$[2] = isSelected;
		$[3] = onClearSelection;
		$[4] = onToggle;
		$[5] = t2;
	} else t2 = $[5];
	let t3;
	if ($[6] !== isOpen) {
		t3 = cn("w-4 h-4 transition-transform duration-300 text-[#555]", { "rotate-180 mt-2.5": isOpen });
		$[6] = isOpen;
		$[7] = t3;
	} else t3 = $[7];
	let t4;
	if ($[8] !== t3) {
		t4 = /* @__PURE__ */ jsx(ChevronDown, {
			className: t3,
			strokeWidth: 1.5
		});
		$[8] = t3;
		$[9] = t4;
	} else t4 = $[9];
	const t5 = !isOpen;
	let t6;
	if ($[10] !== t5) {
		t6 = cn("ml-[7px]", { hidden: t5 });
		$[10] = t5;
		$[11] = t6;
	} else t6 = $[11];
	let t7;
	if ($[12] === Symbol.for("react.memo_cache_sentinel")) {
		t7 = /* @__PURE__ */ jsx("path", {
			d: "M1.5 1C1.5 0.723858 1.27614 0.5 1 0.5C0.723858 0.5 0.5 0.723858 0.5 1H1H1.5ZM1 1H0.5V10H1H1.5V1H1Z",
			className: "fill-[#555]"
		});
		$[12] = t7;
	} else t7 = $[12];
	let t8;
	if ($[13] !== t6) {
		t8 = /* @__PURE__ */ jsx("svg", {
			xmlns: "http://www.w3.org/2000/svg",
			width: "2",
			height: "10",
			viewBox: "0 0 2 10",
			fill: "none",
			className: t6,
			children: t7
		});
		$[13] = t6;
		$[14] = t8;
	} else t8 = $[14];
	let t9;
	if ($[15] !== t4 || $[16] !== t8) {
		t9 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-1",
			children: [t4, t8]
		});
		$[15] = t4;
		$[16] = t8;
		$[17] = t9;
	} else t9 = $[17];
	let t10;
	if ($[18] !== flow) {
		t10 = /* @__PURE__ */ jsx("h3", {
			className: "text-sm font-medium text-left",
			children: flow
		});
		$[18] = flow;
		$[19] = t10;
	} else t10 = $[19];
	let t11;
	if ($[20] !== t1 || $[21] !== t10 || $[22] !== t2 || $[23] !== t9) {
		t11 = /* @__PURE__ */ jsxs("button", {
			"data-testid": t1,
			className: "w-full grid grid-cols-[auto_1fr] items-center gap-3 hover:bg-muted/40 cursor-pointer min-h-8.5 select-none hover:bg-muted-foreground/10 px-4",
			onClick: t2,
			children: [t9, t10]
		});
		$[20] = t1;
		$[21] = t10;
		$[22] = t2;
		$[23] = t9;
		$[24] = t11;
	} else t11 = $[24];
	const t12 = !isOpen;
	let t13;
	if ($[25] !== t12) {
		t13 = cn("grid grid-cols-1 items-center justify-center", { hidden: t12 });
		$[25] = t12;
		$[26] = t13;
	} else t13 = $[26];
	let t14;
	if ($[27] !== endpoints || $[28] !== onSelectEndpoint || $[29] !== selectedEndpointId) {
		let t15$1;
		if ($[31] !== endpoints.length || $[32] !== onSelectEndpoint || $[33] !== selectedEndpointId) {
			t15$1 = (endpoint, index) => /* @__PURE__ */ jsx(EndpointItem, {
				endpoint,
				isSelected: selectedEndpointId === endpoint.id,
				isLast: index === endpoints.length - 1,
				onSelect: onSelectEndpoint
			}, endpoint.id);
			$[31] = endpoints.length;
			$[32] = onSelectEndpoint;
			$[33] = selectedEndpointId;
			$[34] = t15$1;
		} else t15$1 = $[34];
		t14 = endpoints.map(t15$1);
		$[27] = endpoints;
		$[28] = onSelectEndpoint;
		$[29] = selectedEndpointId;
		$[30] = t14;
	} else t14 = $[30];
	let t15;
	if ($[35] !== t13 || $[36] !== t14) {
		t15 = /* @__PURE__ */ jsx("div", {
			className: t13,
			children: t14
		});
		$[35] = t13;
		$[36] = t14;
		$[37] = t15;
	} else t15 = $[37];
	let t16;
	if ($[38] !== t11 || $[39] !== t15) {
		t16 = /* @__PURE__ */ jsxs("div", {
			className: "pt-2",
			children: [t11, t15]
		});
		$[38] = t11;
		$[39] = t15;
		$[40] = t16;
	} else t16 = $[40];
	return t16;
};

//#endregion
//#region src/hooks/use-endpoint-configuration.ts
const defaultHeaders = {
	CONTENT_TYPE: {
		name: "Content-Type",
		value: "application/json",
		active: true
	},
	USER_AGENT: {
		name: "User-Agent",
		value: "Motia/1.0",
		active: true
	},
	ACCEPT: {
		name: "Accept",
		value: "application/json",
		active: true
	}
};
const getHeadersSelector = (state) => {
	const selectedEndpointId = state.selectedEndpointId;
	if (selectedEndpointId) return state.headers[selectedEndpointId] || defaultHeaders;
	return defaultHeaders;
};
const getBodyIsValidSelector = (state) => {
	const selectedEndpointId = state.selectedEndpointId;
	if (selectedEndpointId) return state.bodyIsValid[selectedEndpointId];
	return true;
};
const getBodySelector = (state) => {
	const selectedEndpointId = state.selectedEndpointId;
	if (selectedEndpointId) return state.body[selectedEndpointId] || "";
	return "";
};
const getResponseSelector = (state) => {
	const selectedEndpointId = state.selectedEndpointId;
	if (selectedEndpointId) return state.response[selectedEndpointId] || void 0;
};
const getQueryParamsSelector = (state) => {
	const selectedEndpointId = state.selectedEndpointId;
	if (selectedEndpointId) return state.queryParams[selectedEndpointId] || {};
	return {};
};
const getPathParamsSelector = (state) => {
	const selectedEndpointId = state.selectedEndpointId;
	if (selectedEndpointId) return state.pathParams[selectedEndpointId] || {};
	return {};
};
const useEndpointConfiguration = create()(persist(devtools((set) => ({
	selectedEndpointId: "",
	headers: {},
	body: {},
	bodyIsValid: {},
	response: {},
	queryParams: {},
	pathParams: {},
	flowGroupStatus: {},
	toggleFlowGroupStatus: (flow) => set((state) => ({ flowGroupStatus: {
		...state.flowGroupStatus,
		[flow]: !state.flowGroupStatus[flow]
	} })),
	setSelectedEndpointId: (selectedEndpointId) => set({ selectedEndpointId }),
	setQueryParams: (queryParams) => set((state) => ({ queryParams: {
		...state.queryParams,
		[state.selectedEndpointId]: queryParams
	} })),
	removeQueryParams: (key) => set((state) => {
		const newQueryParams = { ...state.queryParams[state.selectedEndpointId] };
		delete newQueryParams[key];
		return { queryParams: {
			...state.queryParams,
			[state.selectedEndpointId]: newQueryParams
		} };
	}),
	setPathParams: (pathParams) => set((state) => ({ pathParams: {
		...state.pathParams,
		[state.selectedEndpointId]: pathParams
	} })),
	removePathParams: (key) => set((state) => {
		const newPathParams = { ...state.pathParams[state.selectedEndpointId] };
		delete newPathParams[key];
		return { pathParams: {
			...state.pathParams,
			[state.selectedEndpointId]: newPathParams
		} };
	}),
	setHeaders: (headers) => set((state) => ({ headers: {
		...state.headers,
		[state.selectedEndpointId]: headers
	} })),
	setResponse: async (response, startTime) => {
		if (!response) {
			set((state) => ({ response: {
				...state.response,
				[state.selectedEndpointId]: void 0
			} }));
			return;
		}
		let body;
		try {
			const contentType = response?.headers.get("content-type") ?? "";
			if (contentType.includes("text/")) body = response ? await response.text() : void 0;
			else if (contentType.includes("application/json")) body = response ? await response.json() : void 0;
			else {
				body = response ? await response.blob() : void 0;
				body.toJSON = () => "Preview not available";
			}
		} catch (error) {
			console.error("Error setting response:", error);
		}
		set((state) => ({ response: {
			...state.response,
			[state.selectedEndpointId]: {
				executionTime: Date.now() - startTime,
				statusCode: response?.status,
				headers: response?.headers ? Object.fromEntries(response.headers.entries()) : {},
				body
			}
		} }));
	},
	setBody: (body) => set((state) => ({ body: {
		...state.body,
		[state.selectedEndpointId]: body
	} })),
	removeHeaders: (key) => set((state) => {
		const newHeaders = { ...state.headers[state.selectedEndpointId] || defaultHeaders };
		delete newHeaders[key];
		return { headers: {
			...state.headers,
			[state.selectedEndpointId]: newHeaders
		} };
	}),
	setBodyIsValid: (bodyIsValid) => set((state) => ({ bodyIsValid: {
		...state.bodyIsValid,
		[state.selectedEndpointId]: bodyIsValid
	} }))
})), {
	name: "motia-endpoint-configuration",
	storage: createJSONStorage(() => localStorage)
}));

//#endregion
//#region src/hooks/use-get-endpoints.ts
const useGetEndpoints = () => {
	const $ = c(7);
	if ($[0] !== "cadd8d3432dfc6d2432c646eaf1f90c044b70a708974ecf5f816ef36004b4da0") {
		for (let $i = 0; $i < 7; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cadd8d3432dfc6d2432c646eaf1f90c044b70a708974ecf5f816ef36004b4da0";
	}
	let t0;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t0 = {
			streamName: "__motia.api-endpoints",
			groupId: "default"
		};
		$[1] = t0;
	} else t0 = $[1];
	const { data: endpoints } = useStreamGroup(t0);
	let t1;
	if ($[2] !== endpoints) {
		t1 = endpoints.reduce(_temp$8, {});
		$[2] = endpoints;
		$[3] = t1;
	} else t1 = $[3];
	const groupedEndpoints = t1;
	let t2;
	if ($[4] !== endpoints || $[5] !== groupedEndpoints) {
		t2 = {
			endpoints,
			groupedEndpoints
		};
		$[4] = endpoints;
		$[5] = groupedEndpoints;
		$[6] = t2;
	} else t2 = $[6];
	return t2;
};
function _temp$8(acc, endpoint) {
	if (endpoint.flows && endpoint.flows.length > 0) endpoint.flows.forEach((flow) => {
		acc[flow] = acc[flow] || [];
		acc[flow].push(endpoint);
	});
	else {
		acc["no-flow"] = acc["no-flow"] || [];
		acc["no-flow"].push(endpoint);
	}
	return acc;
}

//#endregion
//#region src/hooks/use-path-params.ts
const usePathParams = (path) => {
	return path.match(/:(\w+)/g)?.map((param) => param.slice(1)) ?? [];
};

//#endregion
//#region src/hooks/utils.ts
const resolveRef = (ref, rootSchema) => {
	if (!rootSchema || !ref.startsWith("#/$defs/")) return;
	const defName = ref.replace("#/$defs/", "");
	return rootSchema.$defs?.[defName];
};
/**
* Converts a schema to a JSON object with default values
* Supports both TypeScript step schemas (inline definitions) and Python step schemas ($defs/$ref)
* @param schema - The schema to convert
* @param rootSchema - Optional root schema containing $defs for $ref resolution. Defaults to schema if not provided.
*/
const convertSchemaToJson = (schema, rootSchema) => {
	if (!schema) return {};
	const effectiveRootSchema = rootSchema ?? schema;
	if (schema.$ref) {
		const resolvedSchema = resolveRef(schema.$ref, effectiveRootSchema);
		if (resolvedSchema) return convertSchemaToJson(resolvedSchema, effectiveRootSchema);
		return {};
	}
	if (schema.anyOf && Array.isArray(schema.anyOf)) {
		const nonNullSchema = schema.anyOf.find((item) => item && item.type !== "null");
		if (nonNullSchema) return convertSchemaToJson(nonNullSchema, effectiveRootSchema);
		return null;
	}
	if (schema.type === "object") {
		const result = {};
		if (schema.properties) Object.entries(schema.properties).forEach(([key, value]) => {
			result[key] = convertSchemaToJson(value, effectiveRootSchema);
		});
		return result;
	}
	switch (schema.type) {
		case "array": return schema.items ? [convertSchemaToJson(schema.items, effectiveRootSchema)] : [];
		case "string": return schema.enum?.[0] ?? schema.description ?? "string";
		case "number": return schema.description ?? 0;
		case "integer": return 0;
		case "boolean": return schema.description ?? false;
		case "null": return null;
		default: return;
	}
};

//#endregion
//#region src/hooks/use-json-schema-to-json.tsx
const useJsonSchemaToJson = (schema) => {
	const $ = c(6);
	if ($[0] !== "ba29f15b5046bda288d763530dc1c4ad1bc3136b6950394508cecd61132b94a3") {
		for (let $i = 0; $i < 6; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "ba29f15b5046bda288d763530dc1c4ad1bc3136b6950394508cecd61132b94a3";
	}
	const [body, setBody] = useState("");
	let t0;
	let t1;
	if ($[1] !== schema) {
		t0 = () => {
			if (schema) setBody(JSON.stringify(convertSchemaToJson(schema), null, 2));
		};
		t1 = [schema];
		$[1] = schema;
		$[2] = t0;
		$[3] = t1;
	} else {
		t0 = $[2];
		t1 = $[3];
	}
	useEffect(t0, t1);
	let t2;
	if ($[4] !== body) {
		t2 = {
			body,
			setBody
		};
		$[4] = body;
		$[5] = t2;
	} else t2 = $[5];
	return t2;
};

//#endregion
//#region src/spec/spec-body-panel.tsx
const EndpointBodyPanel = (t0) => {
	const $ = c(9);
	if ($[0] !== "b8eff7f7ce8e9189baa5c639bfd152e5d0c4e2073eede72b1020744bec7ba947") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "b8eff7f7ce8e9189baa5c639bfd152e5d0c4e2073eede72b1020744bec7ba947";
	}
	const { endpoint, panelName, value } = t0;
	if (![
		"post",
		"put",
		"patch"
	].includes(endpoint.method.toLowerCase())) return null;
	const t1 = `endpoint-body-panel__${panelName}`;
	let t2;
	if ($[1] !== value) {
		t2 = value ? JSON.parse(value) : {};
		$[1] = value;
		$[2] = t2;
	} else t2 = $[2];
	let t3;
	if ($[3] === Symbol.for("react.memo_cache_sentinel")) {
		t3 = { backgroundColor: "transparent" };
		$[3] = t3;
	} else t3 = $[3];
	let t4;
	if ($[4] !== t2) {
		t4 = /* @__PURE__ */ jsx(ReactJson, {
			src: t2,
			theme: "default",
			enableClipboard: false,
			style: t3
		});
		$[4] = t2;
		$[5] = t4;
	} else t4 = $[5];
	let t5;
	if ($[6] !== t1 || $[7] !== t4) {
		t5 = /* @__PURE__ */ jsx(Panel, {
			title: "Body",
			size: "sm",
			contentClassName: "p-0",
			"data-testid": t1,
			children: t4
		});
		$[6] = t1;
		$[7] = t4;
		$[8] = t5;
	} else t5 = $[8];
	return t5;
};

//#endregion
//#region src/spec/spec-path-params-panel.tsx
const EndpointPathParamsPanel = (t0) => {
	const $ = c(6);
	if ($[0] !== "4b99c635b467b8c34956e7f3473f52521462aab54a3d1e16ce1af5ee24ede587") {
		for (let $i = 0; $i < 6; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "4b99c635b467b8c34956e7f3473f52521462aab54a3d1e16ce1af5ee24ede587";
	}
	const { endpoint } = t0;
	const pathParams = usePathParams(endpoint.path);
	if (!pathParams.length) return null;
	let t1;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = { gridTemplateColumns: "1fr 2fr" };
		$[1] = t1;
	} else t1 = $[1];
	let t2;
	if ($[2] !== pathParams) {
		t2 = pathParams.map(_temp$7);
		$[2] = pathParams;
		$[3] = t2;
	} else t2 = $[3];
	let t3;
	if ($[4] !== t2) {
		t3 = /* @__PURE__ */ jsx(Panel, {
			title: "Path params",
			size: "sm",
			variant: "default",
			children: /* @__PURE__ */ jsx("div", {
				className: "grid gap-3",
				style: t1,
				children: t2
			})
		});
		$[4] = t2;
		$[5] = t3;
	} else t3 = $[5];
	return t3;
};
function _temp$7(param) {
	return /* @__PURE__ */ jsx("div", {
		className: "font-bold leading-[36px] flex text-xs",
		children: param
	}, param);
}

//#endregion
//#region src/spec/spec-query-params-panel.tsx
const EndpointQueryParamsPanel = (t0) => {
	const $ = c(6);
	if ($[0] !== "4e1364928f69469968044d4855a06c351178a75725f12806c407b9bdbee699b0") {
		for (let $i = 0; $i < 6; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "4e1364928f69469968044d4855a06c351178a75725f12806c407b9bdbee699b0";
	}
	const { endpoint } = t0;
	if (!endpoint.queryParams?.length) return null;
	let t1;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = {
			gridTemplateColumns: "1fr 2fr",
			gridTemplateRows: "1fr 1fr"
		};
		$[1] = t1;
	} else t1 = $[1];
	let t2;
	if ($[2] !== endpoint.queryParams) {
		t2 = endpoint.queryParams.map(_temp$6);
		$[2] = endpoint.queryParams;
		$[3] = t2;
	} else t2 = $[3];
	let t3;
	if ($[4] !== t2) {
		t3 = /* @__PURE__ */ jsx(Panel, {
			title: "Query params",
			size: "sm",
			variant: "outlined",
			children: /* @__PURE__ */ jsx("div", {
				className: "grid gap-3",
				style: t1,
				children: t2
			})
		});
		$[4] = t2;
		$[5] = t3;
	} else t3 = $[5];
	return t3;
};
function _temp$6(param) {
	return /* @__PURE__ */ jsxs(Fragment, { children: [/* @__PURE__ */ jsx("div", {
		className: "font-bold leading-[36px] flex text-xs",
		children: param.name
	}), /* @__PURE__ */ jsx("div", {
		className: "flex items-center text-xs ",
		children: /* @__PURE__ */ jsx("span", { children: param.description })
	})] }, param.name);
}

//#endregion
//#region src/spec/spec-response-schema.tsx
const EndpointResponseSchemaItem = (t0) => {
	const $ = c(10);
	if ($[0] !== "e4de7b36c48b33b1d8a3aa52a415bcc5b35556fbd21c0f6bc47983adc48a77ce") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "e4de7b36c48b33b1d8a3aa52a415bcc5b35556fbd21c0f6bc47983adc48a77ce";
	}
	const { responseCode, bodySchema } = t0;
	const theme = useThemeStore(_temp$5);
	let t1;
	if ($[1] !== bodySchema) {
		t1 = convertSchemaToJson(bodySchema);
		$[1] = bodySchema;
		$[2] = t1;
	} else t1 = $[2];
	const schema = t1;
	const description = typeof bodySchema.description === "string" ? bodySchema.description : "";
	let t2;
	if ($[3] !== description || $[4] !== schema || $[5] !== theme) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: "text-xs font-mono rounded-lg whitespace-pre-wrap",
			children: schema ? /* @__PURE__ */ jsx(ReactJson, {
				src: schema,
				dark: theme === "dark",
				enableClipboard: false,
				style: { backgroundColor: "transparent" }
			}) : /* @__PURE__ */ jsx("div", {
				className: "text-xs font-mono rounded-lg whitespace-pre-wrap p-4",
				children: description
			})
		});
		$[3] = description;
		$[4] = schema;
		$[5] = theme;
		$[6] = t2;
	} else t2 = $[6];
	let t3;
	if ($[7] !== responseCode || $[8] !== t2) {
		t3 = /* @__PURE__ */ jsx(TabsContent, {
			value: responseCode,
			className: "border-t",
			children: t2
		}, responseCode);
		$[7] = responseCode;
		$[8] = t2;
		$[9] = t3;
	} else t3 = $[9];
	return t3;
};
const EndpointResponseSchema = (t0) => {
	const $ = c(11);
	if ($[0] !== "e4de7b36c48b33b1d8a3aa52a415bcc5b35556fbd21c0f6bc47983adc48a77ce") {
		for (let $i = 0; $i < 11; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "e4de7b36c48b33b1d8a3aa52a415bcc5b35556fbd21c0f6bc47983adc48a77ce";
	}
	const { items } = t0;
	if (items.length === 0) return null;
	const t1 = items[0].responseCode;
	let t2;
	if ($[1] !== items) {
		t2 = items.map(_temp2);
		$[1] = items;
		$[2] = t2;
	} else t2 = $[2];
	let t3;
	if ($[3] !== t2) {
		t3 = /* @__PURE__ */ jsx("div", {
			className: "flex items-center justify-between bg-card",
			children: /* @__PURE__ */ jsx(TabsList, {
				className: "bg-transparent p-0",
				children: t2
			})
		});
		$[3] = t2;
		$[4] = t3;
	} else t3 = $[4];
	let t4;
	if ($[5] !== items) {
		t4 = items.map(_temp3);
		$[5] = items;
		$[6] = t4;
	} else t4 = $[6];
	let t5;
	if ($[7] !== items[0].responseCode || $[8] !== t3 || $[9] !== t4) {
		t5 = /* @__PURE__ */ jsx("div", {
			className: "flex flex-col rounded-lg border",
			children: /* @__PURE__ */ jsxs(Tabs, {
				defaultValue: t1,
				children: [t3, t4]
			})
		});
		$[7] = items[0].responseCode;
		$[8] = t3;
		$[9] = t4;
		$[10] = t5;
	} else t5 = $[10];
	return t5;
};
function _temp$5(store) {
	return store.theme;
}
function _temp2(item) {
	return /* @__PURE__ */ jsx(TabsTrigger, {
		value: item.responseCode,
		className: "text-xs font-bold cursor-pointer",
		children: item.responseCode
	}, item.responseCode);
}
function _temp3(props) {
	return /* @__PURE__ */ jsx(EndpointResponseSchemaItem, { ...props }, props.responseCode);
}

//#endregion
//#region src/spec/spec-side-panel.tsx
const SpecSidePanel = (t0) => {
	const $ = c(26);
	if ($[0] !== "1460e2628753bf80967063f35b14988afbd9dd8b96c8778ac3b5ebba7a833fc1") {
		for (let $i = 0; $i < 26; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "1460e2628753bf80967063f35b14988afbd9dd8b96c8778ac3b5ebba7a833fc1";
	}
	const { endpoint, onClose } = t0;
	const { body } = useJsonSchemaToJson(endpoint.bodySchema);
	let t1;
	if ($[1] !== endpoint.method || $[2] !== endpoint.path) {
		t1 = /* @__PURE__ */ jsx(EndpointPath, {
			method: endpoint.method,
			path: endpoint.path
		});
		$[1] = endpoint.method;
		$[2] = endpoint.path;
		$[3] = t1;
	} else t1 = $[3];
	let t2;
	if ($[4] !== onClose) {
		t2 = /* @__PURE__ */ jsx(X, {
			className: "cursor-pointer w-4 h-4",
			onClick: onClose
		});
		$[4] = onClose;
		$[5] = t2;
	} else t2 = $[5];
	let t3;
	if ($[6] !== onClose || $[7] !== t2) {
		t3 = [{
			icon: t2,
			onClick: onClose
		}];
		$[6] = onClose;
		$[7] = t2;
		$[8] = t3;
	} else t3 = $[8];
	let t4;
	let t5;
	if ($[9] !== endpoint) {
		t4 = /* @__PURE__ */ jsx(EndpointPathParamsPanel, { endpoint });
		t5 = /* @__PURE__ */ jsx(EndpointQueryParamsPanel, { endpoint });
		$[9] = endpoint;
		$[10] = t4;
		$[11] = t5;
	} else {
		t4 = $[10];
		t5 = $[11];
	}
	let t6;
	if ($[12] !== body || $[13] !== endpoint) {
		t6 = /* @__PURE__ */ jsx(EndpointBodyPanel, {
			endpoint,
			panelName: "details",
			value: body
		});
		$[12] = body;
		$[13] = endpoint;
		$[14] = t6;
	} else t6 = $[14];
	let t7;
	if ($[15] !== endpoint.responseSchema) {
		t7 = /* @__PURE__ */ jsx(EndpointResponseSchema, { items: Object.entries(endpoint?.responseSchema ?? {}).map(_temp$4) });
		$[15] = endpoint.responseSchema;
		$[16] = t7;
	} else t7 = $[16];
	let t8;
	if ($[17] !== endpoint.description || $[18] !== onClose || $[19] !== t1 || $[20] !== t3 || $[21] !== t4 || $[22] !== t5 || $[23] !== t6 || $[24] !== t7) {
		t8 = /* @__PURE__ */ jsxs(Sidebar, {
			initialWidth: 600,
			subtitle: endpoint.description,
			title: t1,
			onClose,
			actions: t3,
			children: [
				t4,
				t5,
				t6,
				t7
			]
		});
		$[17] = endpoint.description;
		$[18] = onClose;
		$[19] = t1;
		$[20] = t3;
		$[21] = t4;
		$[22] = t5;
		$[23] = t6;
		$[24] = t7;
		$[25] = t8;
	} else t8 = $[25];
	return t8;
};
function _temp$4(t0) {
	const [status, schema] = t0;
	return {
		responseCode: status,
		bodySchema: schema
	};
}

//#endregion
//#region src/components/json-editor.tsx
const JsonEditor = (t0) => {
	const $ = c(41);
	if ($[0] !== "0b3355d4554a672310b1abdad893ca8974afca90f2334dd0d80cf3d8e960c55e") {
		for (let $i = 0; $i < 41; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "0b3355d4554a672310b1abdad893ca8974afca90f2334dd0d80cf3d8e960c55e";
	}
	const { value, schema, onChange, onValidate, language: t1, readOnly: t2 } = t0;
	const language = t1 === void 0 ? "json" : t1;
	const readOnly = t2 === void 0 ? false : t2;
	const monaco = useMonaco();
	const editorTheme = useThemeStore(_temp$3) === "dark" ? "transparent-dark" : "transparent-light";
	const [editor, setEditor] = useState(null);
	const resizeAnimationFrameRef = useRef(null);
	const isValidatingRef = useRef(false);
	let t3;
	let t4;
	if ($[1] !== monaco) {
		t3 = () => {
			if (!monaco) return;
			monaco.editor.defineTheme("transparent-light", {
				base: "vs",
				inherit: true,
				rules: [],
				colors: {
					"editor.background": "#00000000",
					"editor.lineHighlightBackground": "#00000000",
					"editorLineNumber.foreground": "#999999",
					"editorLineNumber.activeForeground": "#000000",
					focusBorder: "#00000000",
					"widget.border": "#00000000",
					"editor.border": "#00000000",
					"editorWidget.border": "#00000000"
				}
			});
			monaco.editor.defineTheme("transparent-dark", {
				base: "vs-dark",
				inherit: true,
				rules: [],
				colors: {
					"editor.background": "#00000000",
					"editor.lineHighlightBackground": "#00000000",
					"editorLineNumber.foreground": "#666666",
					"editorLineNumber.activeForeground": "#ffffff",
					focusBorder: "#00000000",
					"widget.border": "#00000000",
					"editor.border": "#00000000",
					"editorWidget.border": "#00000000"
				}
			});
		};
		t4 = [monaco];
		$[1] = monaco;
		$[2] = t3;
		$[3] = t4;
	} else {
		t3 = $[2];
		t4 = $[3];
	}
	useLayoutEffect(t3, t4);
	let t5;
	let t6;
	if ($[4] !== monaco || $[5] !== schema) {
		t5 = () => {
			if (!monaco) return;
			monaco.languages.typescript.javascriptDefaults.setCompilerOptions({ isolatedModules: true });
			monaco.languages.json.jsonDefaults.setDiagnosticsOptions({ schemas: schema ? [{
				uri: window.location.href,
				fileMatch: ["*"],
				schema
			}] : [] });
		};
		t6 = [monaco, schema];
		$[4] = monaco;
		$[5] = schema;
		$[6] = t5;
		$[7] = t6;
	} else {
		t5 = $[6];
		t6 = $[7];
	}
	useEffect(t5, t6);
	let t7;
	let t8;
	if ($[8] !== editorTheme || $[9] !== monaco) {
		t7 = () => {
			if (!monaco) return;
			monaco.editor.setTheme(editorTheme);
		};
		t8 = [monaco, editorTheme];
		$[8] = editorTheme;
		$[9] = monaco;
		$[10] = t7;
		$[11] = t8;
	} else {
		t7 = $[10];
		t8 = $[11];
	}
	useLayoutEffect(t7, t8);
	let t10;
	let t9;
	if ($[12] !== editor) {
		t9 = () => {
			if (!editor) return;
			const container = editor.getContainerDomNode().parentElement?.parentElement;
			if (!container) return;
			const handleResize = () => {
				if (resizeAnimationFrameRef.current !== null) cancelAnimationFrame(resizeAnimationFrameRef.current);
				resizeAnimationFrameRef.current = requestAnimationFrame(() => {
					const { width, height } = container.getBoundingClientRect();
					editor.layout({
						width,
						height
					});
					resizeAnimationFrameRef.current = null;
				});
			};
			handleResize();
			const resizeObserver = new ResizeObserver(handleResize);
			resizeObserver.observe(container);
			return () => {
				resizeObserver.disconnect();
				if (resizeAnimationFrameRef.current !== null) cancelAnimationFrame(resizeAnimationFrameRef.current);
			};
		};
		t10 = [editor];
		$[12] = editor;
		$[13] = t10;
		$[14] = t9;
	} else {
		t10 = $[13];
		t9 = $[14];
	}
	useEffect(t9, t10);
	let t11;
	let t12;
	if ($[15] !== editor || $[16] !== monaco || $[17] !== onValidate || $[18] !== schema || $[19] !== value) {
		t11 = () => {
			if (!editor || !monaco || !onValidate || isValidatingRef.current) return;
			const model = editor.getModel();
			if (!model) return;
			if (schema && !value) {
				isValidatingRef.current = true;
				onValidate(false);
				isValidatingRef.current = false;
				return;
			}
			isValidatingRef.current = true;
			const timeoutId = setTimeout(() => {
				onValidate(monaco.editor.getModelMarkers({ resource: model.uri }).length === 0);
				isValidatingRef.current = false;
			}, 100);
			return () => {
				clearTimeout(timeoutId);
				isValidatingRef.current = false;
			};
		};
		t12 = [
			editor,
			monaco,
			onValidate,
			value,
			schema
		];
		$[15] = editor;
		$[16] = monaco;
		$[17] = onValidate;
		$[18] = schema;
		$[19] = value;
		$[20] = t11;
		$[21] = t12;
	} else {
		t11 = $[20];
		t12 = $[21];
	}
	useEffect(t11, t12);
	let t13;
	if ($[22] !== schema) {
		t13 = schema ? JSON.stringify(schema) : "no-schema";
		$[22] = schema;
		$[23] = t13;
	} else t13 = $[23];
	const editorKey = t13;
	let t14;
	if ($[24] !== onChange) {
		t14 = (value_0) => {
			onChange?.(value_0 ?? "");
		};
		$[24] = onChange;
		$[25] = t14;
	} else t14 = $[25];
	let t15;
	if ($[26] !== onValidate || $[27] !== schema || $[28] !== value) {
		t15 = (markers_0) => {
			if (!onValidate || isValidatingRef.current) return;
			isValidatingRef.current = true;
			if (schema && !value) onValidate(false);
			else onValidate(markers_0.length === 0);
			isValidatingRef.current = false;
		};
		$[26] = onValidate;
		$[27] = schema;
		$[28] = value;
		$[29] = t15;
	} else t15 = $[29];
	let t16;
	if ($[30] === Symbol.for("react.memo_cache_sentinel")) {
		t16 = { enabled: false };
		$[30] = t16;
	} else t16 = $[30];
	let t17;
	if ($[31] !== readOnly) {
		t17 = {
			automaticLayout: false,
			readOnly,
			scrollBeyondLastLine: false,
			minimap: t16,
			overviewRulerLanes: 0
		};
		$[31] = readOnly;
		$[32] = t17;
	} else t17 = $[32];
	let t18;
	if ($[33] !== editorKey || $[34] !== editorTheme || $[35] !== language || $[36] !== t14 || $[37] !== t15 || $[38] !== t17 || $[39] !== value) {
		t18 = /* @__PURE__ */ jsx(Editor, {
			"data-testid": "json-editor",
			language,
			value,
			loading: "",
			theme: editorTheme,
			onMount: setEditor,
			onChange: t14,
			onValidate: t15,
			options: t17
		}, editorKey);
		$[33] = editorKey;
		$[34] = editorTheme;
		$[35] = language;
		$[36] = t14;
		$[37] = t15;
		$[38] = t17;
		$[39] = value;
		$[40] = t18;
	} else t18 = $[40];
	return t18;
};
function _temp$3(state) {
	return state.theme;
}

//#endregion
//#region src/play/side-panel-body-tab.tsx
const SidePanelBodyTab = memo((t0) => {
	const $ = c(21);
	if ($[0] !== "15cdaf91612bb3c1ef34320b0f01a80a135e66606fb87beb7a4a2f7f95e53a69") {
		for (let $i = 0; $i < 21; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "15cdaf91612bb3c1ef34320b0f01a80a135e66606fb87beb7a4a2f7f95e53a69";
	}
	const { schema } = t0;
	const { setBody, setBodyIsValid } = useEndpointConfiguration();
	const bodyIsValid = useEndpointConfiguration(useShallow(getBodyIsValidSelector));
	const body = useEndpointConfiguration(getBodySelector);
	const previousSchemaRef = useRef(schema);
	const userClearedRef = useRef(false);
	let t1;
	let t2;
	if ($[1] !== body || $[2] !== schema || $[3] !== setBody || $[4] !== setBodyIsValid) {
		t1 = () => {
			const schemaChanged = previousSchemaRef.current !== schema;
			previousSchemaRef.current = schema;
			if (schemaChanged) userClearedRef.current = false;
			if (schema && !body && !userClearedRef.current) {
				setBody(JSON.stringify(convertSchemaToJson(schema), null, 2));
				return;
			}
			if (!schema) {
				setBody(body || "");
				setBodyIsValid(true);
			}
		};
		t2 = [
			schema,
			body,
			setBody,
			setBodyIsValid
		];
		$[1] = body;
		$[2] = schema;
		$[3] = setBody;
		$[4] = setBodyIsValid;
		$[5] = t1;
		$[6] = t2;
	} else {
		t1 = $[5];
		t2 = $[6];
	}
	useEffect(t1, t2);
	let t3;
	if ($[7] !== schema || $[8] !== setBody || $[9] !== setBodyIsValid) {
		t3 = (value) => {
			setBody(value);
			if (schema && !value) {
				setBodyIsValid(false);
				userClearedRef.current = true;
			} else userClearedRef.current = false;
		};
		$[7] = schema;
		$[8] = setBody;
		$[9] = setBodyIsValid;
		$[10] = t3;
	} else t3 = $[10];
	const handleBodyChange = t3;
	let t4;
	if ($[11] !== body || $[12] !== handleBodyChange || $[13] !== schema || $[14] !== setBodyIsValid) {
		t4 = /* @__PURE__ */ jsx(JsonEditor, {
			value: body,
			schema,
			onChange: handleBodyChange,
			onValidate: setBodyIsValid
		});
		$[11] = body;
		$[12] = handleBodyChange;
		$[13] = schema;
		$[14] = setBodyIsValid;
		$[15] = t4;
	} else t4 = $[15];
	let t5;
	if ($[16] !== bodyIsValid) {
		t5 = bodyIsValid === false && /* @__PURE__ */ jsxs("div", {
			className: "absolute bottom-0 left-0 right-0 border-t border-border p-3 text-sm dark:text-yellow-500 text-yellow-700 flex items-center gap-1 font-medium",
			"data-testid": "endpoint-body-tab-invalid",
			children: [/* @__PURE__ */ jsx(CircleX, { className: "w-4 h-4" }), "The body payload is invalid"]
		});
		$[16] = bodyIsValid;
		$[17] = t5;
	} else t5 = $[17];
	let t6;
	if ($[18] !== t4 || $[19] !== t5) {
		t6 = /* @__PURE__ */ jsxs("div", {
			className: "max-h-full h-full relative",
			children: [t4, t5]
		});
		$[18] = t4;
		$[19] = t5;
		$[20] = t6;
	} else t6 = $[20];
	return t6;
});

//#endregion
//#region src/components/configuration-list-item.tsx
const ConfigurationListItem = memo((t0) => {
	const $ = c(59);
	if ($[0] !== "49b457e4243ed5368ffc15725d84d290ce5fe7173cb1a11ee0a1be10ae181259") {
		for (let $i = 0; $i < 59; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "49b457e4243ed5368ffc15725d84d290ce5fe7173cb1a11ee0a1be10ae181259";
	}
	const { value, id, onUpdate, onRemove, required: t1 } = t0;
	const required = t1 === void 0 ? false : t1;
	const [activeConfiguration, setActiveConfiguration] = useState(null);
	const [valueToDelete, setValueToDelete] = useState(null);
	const deleteTimeout = useRef(void 0);
	const isActive = activeConfiguration === id;
	let t2;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = (key) => {
			clearTimeout(deleteTimeout.current);
			setValueToDelete(key);
			deleteTimeout.current = setTimeout(() => {
				setValueToDelete(null);
			}, 5e3);
		};
		$[1] = t2;
	} else t2 = $[1];
	const onDelete = t2;
	let t3;
	if ($[2] !== id || $[3] !== required) {
		t3 = () => {
			if (required) return;
			setActiveConfiguration(id);
		};
		$[2] = id;
		$[3] = required;
		$[4] = t3;
	} else t3 = $[4];
	const onMouseOver = t3;
	let t4;
	if ($[5] !== required) {
		t4 = () => {
			if (required) return;
			setActiveConfiguration(null);
		};
		$[5] = required;
		$[6] = t4;
	} else t4 = $[6];
	const onMouseLeave = t4;
	const t5 = isActive && "bg-card";
	let t6;
	if ($[7] !== t5) {
		t6 = cn("grid grid-cols-[auto_1fr_auto] items-center py-2 px-4 bg-muted/30 rounded-lg gap-2", t5);
		$[7] = t5;
		$[8] = t6;
	} else t6 = $[8];
	let t7;
	if ($[9] !== id || $[10] !== onUpdate) {
		t7 = (checked) => onUpdate(id, "active", checked);
		$[9] = id;
		$[10] = onUpdate;
		$[11] = t7;
	} else t7 = $[11];
	let t8;
	if ($[12] === Symbol.for("react.memo_cache_sentinel")) {
		t8 = /* @__PURE__ */ jsx(Check, { className: "h-3 w-3" });
		$[12] = t8;
	} else t8 = $[12];
	let t9;
	if ($[13] !== required || $[14] !== t7 || $[15] !== value.active) {
		t9 = /* @__PURE__ */ jsx(Checkbox, {
			checked: value.active,
			onCheckedChange: t7,
			disabled: required,
			children: t8
		});
		$[13] = required;
		$[14] = t7;
		$[15] = value.active;
		$[16] = t9;
	} else t9 = $[16];
	const t10 = !value.active;
	const t11 = required ? "outlineReadonly" : "outline";
	const t12 = `name-${id}`;
	let t13;
	if ($[17] !== id || $[18] !== onUpdate) {
		t13 = (e) => onUpdate(id, "name", e.target.value);
		$[17] = id;
		$[18] = onUpdate;
		$[19] = t13;
	} else t13 = $[19];
	let t14;
	if ($[20] !== required || $[21] !== t10 || $[22] !== t11 || $[23] !== t12 || $[24] !== t13 || $[25] !== value.name) {
		t14 = /* @__PURE__ */ jsx(Input, {
			readOnly: required,
			disabled: t10,
			variant: t11,
			id: t12,
			defaultValue: value.name,
			onBlur: t13,
			placeholder: "Name",
			className: "h-8"
		});
		$[20] = required;
		$[21] = t10;
		$[22] = t11;
		$[23] = t12;
		$[24] = t13;
		$[25] = value.name;
		$[26] = t14;
	} else t14 = $[26];
	const t15 = !value.active;
	const t16 = `value-${id}`;
	let t17;
	if ($[27] !== id || $[28] !== onUpdate) {
		t17 = (e_0) => onUpdate(id, "value", e_0.target.value);
		$[27] = id;
		$[28] = onUpdate;
		$[29] = t17;
	} else t17 = $[29];
	let t18;
	if ($[30] !== t15 || $[31] !== t16 || $[32] !== t17 || $[33] !== value.value) {
		t18 = /* @__PURE__ */ jsx(Input, {
			disabled: t15,
			variant: "outline",
			id: t16,
			defaultValue: value.value,
			onBlur: t17,
			placeholder: "Value",
			className: "h-8"
		});
		$[30] = t15;
		$[31] = t16;
		$[32] = t17;
		$[33] = value.value;
		$[34] = t18;
	} else t18 = $[34];
	let t19;
	if ($[35] !== t14 || $[36] !== t18) {
		t19 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-2 gap-2",
			children: [t14, t18]
		});
		$[35] = t14;
		$[36] = t18;
		$[37] = t19;
	} else t19 = $[37];
	let t20;
	if ($[38] !== id || $[39] !== onRemove || $[40] !== valueToDelete) {
		t20 = () => {
			if (valueToDelete === id) onRemove?.(id);
			else onDelete(id);
		};
		$[38] = id;
		$[39] = onRemove;
		$[40] = valueToDelete;
		$[41] = t20;
	} else t20 = $[41];
	const t21 = isActive && "opacity-100";
	let t22;
	if ($[42] !== t21) {
		t22 = cn("h-6 w-6 opacity-0 transition-opacity duration-200", t21);
		$[42] = t21;
		$[43] = t22;
	} else t22 = $[43];
	const t23 = valueToDelete === id && isActive ? "Click again to remove" : "Remove";
	let t24;
	if ($[44] !== id || $[45] !== valueToDelete) {
		t24 = valueToDelete === id ? /* @__PURE__ */ jsx(AlertCircle, { className: "h-3 w-3 text-destructive" }) : /* @__PURE__ */ jsx(Trash2, { className: "h-3 w-3" });
		$[44] = id;
		$[45] = valueToDelete;
		$[46] = t24;
	} else t24 = $[46];
	let t25;
	if ($[47] !== t20 || $[48] !== t22 || $[49] !== t23 || $[50] !== t24) {
		t25 = /* @__PURE__ */ jsx(Button, {
			variant: "ghost",
			size: "icon",
			onClick: t20,
			className: t22,
			title: t23,
			children: t24
		});
		$[47] = t20;
		$[48] = t22;
		$[49] = t23;
		$[50] = t24;
		$[51] = t25;
	} else t25 = $[51];
	let t26;
	if ($[52] !== onMouseLeave || $[53] !== onMouseOver || $[54] !== t19 || $[55] !== t25 || $[56] !== t6 || $[57] !== t9) {
		t26 = /* @__PURE__ */ jsxs("div", {
			className: t6,
			onMouseOver,
			onMouseLeave,
			children: [
				t9,
				t19,
				t25
			]
		});
		$[52] = onMouseLeave;
		$[53] = onMouseOver;
		$[54] = t19;
		$[55] = t25;
		$[56] = t6;
		$[57] = t9;
		$[58] = t26;
	} else t26 = $[58];
	return t26;
});

//#endregion
//#region src/play/side-panel-headers-tab.tsx
const SidePanelHeadersTab = () => {
	const { setHeaders, removeHeaders } = useEndpointConfiguration();
	const headers = useEndpointConfiguration(useShallow(getHeadersSelector));
	const addHeader = useCallback(() => {
		const newHeader = {
			name: "",
			value: "",
			active: true
		};
		setHeaders({
			...headers,
			[(/* @__PURE__ */ new Date()).getTime().toString()]: newHeader
		});
	}, [headers, setHeaders]);
	const updateHeader = useCallback((key, field, value) => {
		if (!key) return;
		setHeaders({
			...headers,
			[key]: {
				...headers[key],
				[field]: value
			}
		});
	}, [headers, setHeaders]);
	return /* @__PURE__ */ jsxs("div", {
		className: "h-full max-h-full grid grid-rows-[auto_1fr]",
		children: [/* @__PURE__ */ jsx("div", {
			className: "grid px-4 border-b h-10 items-center grid-cols-[auto_1fr]",
			children: /* @__PURE__ */ jsxs(Button, {
				size: "sm",
				onClick: addHeader,
				children: [/* @__PURE__ */ jsx(Plus, { className: "h-3 w-3" }), "Add"]
			})
		}), /* @__PURE__ */ jsxs("div", {
			className: "p-2",
			children: [Object.entries(headers).map(([key_0, header]) => /* @__PURE__ */ jsx(ConfigurationListItem, {
				value: header,
				id: key_0,
				onUpdate: updateHeader,
				onRemove: removeHeaders
			}, key_0)), Object.entries(headers).length === 0 && /* @__PURE__ */ jsx("div", {
				className: "grid grid-cols-1 items-center h-full",
				children: /* @__PURE__ */ jsx("div", {
					className: "text-sm text-muted-foreground text-center",
					children: "There are no headers in this endpoint"
				})
			})]
		})]
	});
};

//#endregion
//#region src/hooks/use-path-url.ts
const queryParamsSelector = (state) => Object.values(getQueryParamsSelector(state)).filter((param) => param.active && param.value !== "" && param.name !== "");
const usePathUrl = (path) => {
	const $ = c(6);
	if ($[0] !== "7e3e8f17b13eb81e00060ed9f39b6042cd1d077a2d43f0c27437d6309314f3aa") {
		for (let $i = 0; $i < 6; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "7e3e8f17b13eb81e00060ed9f39b6042cd1d077a2d43f0c27437d6309314f3aa";
	}
	const pathParams = useEndpointConfiguration(useShallow(getPathParamsSelector));
	const queryParams = useEndpointConfiguration(useShallow(queryParamsSelector));
	let t0;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t0 = /:(\w+)/g;
		$[1] = t0;
	} else t0 = $[1];
	let t1;
	if ($[2] !== pathParams) {
		t1 = (match, p1) => pathParams[p1]?.value || match;
		$[2] = pathParams;
		$[3] = t1;
	} else t1 = $[3];
	const url = path.replace(t0, t1);
	let t2;
	if ($[4] !== queryParams) {
		t2 = queryParams.length > 0 ? `?${queryParams.map(_temp$2).join("&")}` : "";
		$[4] = queryParams;
		$[5] = t2;
	} else t2 = $[5];
	return url + t2;
};
function _temp$2(param) {
	return `${param.name}=${param.value}`;
}

//#endregion
//#region src/components/endpoint-path-preview.tsx
const EndpointPathPreview = (t0) => {
	const $ = c(17);
	if ($[0] !== "d28e3e9329232a2b4ad7f30814822a8f9a40f9e19a3e5c4e80f13d7dbd0fed07") {
		for (let $i = 0; $i < 17; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "d28e3e9329232a2b4ad7f30814822a8f9a40f9e19a3e5c4e80f13d7dbd0fed07";
	}
	const { path, baseUrl: t1 } = t0;
	const fullUrl = `${t1 === void 0 ? window.location.origin : t1}${usePathUrl(path)}`;
	const [copied, setCopied] = useState(false);
	let t2;
	if ($[1] !== fullUrl) {
		t2 = async () => {
			try {
				await navigator.clipboard.writeText(fullUrl);
				setCopied(true);
				setTimeout(() => setCopied(false), 2e3);
			} catch (t3$1) {
				const err = t3$1;
				console.error("Failed to copy URL:", err);
			}
		};
		$[1] = fullUrl;
		$[2] = t2;
	} else t2 = $[2];
	const handleCopy = t2;
	let t3;
	if ($[3] === Symbol.for("react.memo_cache_sentinel")) {
		t3 = /* @__PURE__ */ jsx("div", {
			className: "text-sm font-medium py-2",
			children: "URL Preview"
		});
		$[3] = t3;
	} else t3 = $[3];
	let t4;
	if ($[4] !== fullUrl) {
		t4 = /* @__PURE__ */ jsx("div", {
			className: "min-w-0",
			children: /* @__PURE__ */ jsx("div", {
				className: "font-medium text-xs text-muted-foreground truncate",
				title: fullUrl,
				children: fullUrl
			})
		});
		$[4] = fullUrl;
		$[5] = t4;
	} else t4 = $[5];
	const t5 = copied ? "text-green-400" : "text-muted-foreground";
	let t6;
	if ($[6] !== t5) {
		t6 = cn("w-3 h-3 grid place-items-center transition-colors cursor-pointer", t5);
		$[6] = t5;
		$[7] = t6;
	} else t6 = $[7];
	let t7;
	if ($[8] !== copied) {
		t7 = copied ? /* @__PURE__ */ jsx(Check, { className: "w-2.5 h-2.5" }) : /* @__PURE__ */ jsx(Copy, { className: "w-2.5 h-2.5" });
		$[8] = copied;
		$[9] = t7;
	} else t7 = $[9];
	let t8;
	if ($[10] !== handleCopy || $[11] !== t6 || $[12] !== t7) {
		t8 = /* @__PURE__ */ jsx(Button, {
			onClick: handleCopy,
			className: t6,
			variant: "icon",
			size: "sm",
			"aria-label": "Copy URL",
			children: t7
		});
		$[10] = handleCopy;
		$[11] = t6;
		$[12] = t7;
		$[13] = t8;
	} else t8 = $[13];
	let t9;
	if ($[14] !== t4 || $[15] !== t8) {
		t9 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-[auto_1fr] items-center gap-5 px-5 relative border-b border-border",
			children: [t3, /* @__PURE__ */ jsxs("div", {
				className: "bg-muted-foreground/10 box-border grid grid-cols-[1fr_auto] gap-1 h-6 items-center px-2 py-1 rounded border border-border",
				children: [t4, t8]
			})]
		});
		$[14] = t4;
		$[15] = t8;
		$[16] = t9;
	} else t9 = $[16];
	return t9;
};

//#endregion
//#region src/play/side-panel-params-tab.tsx
const SidePanelParamsTab = ({ path }) => {
	const { setQueryParams, removeQueryParams, setPathParams } = useEndpointConfiguration();
	const queryParams = useEndpointConfiguration(useShallow(getQueryParamsSelector));
	const pathParams = useEndpointConfiguration(useShallow(getPathParamsSelector));
	const pathParamsConfig = useMemo(() => {
		return path.match(/:(\w+)/g)?.map((param) => {
			return {
				name: param.slice(1),
				value: pathParams[param.slice(1)]?.value ?? "",
				active: true
			};
		}) ?? [];
	}, [path]);
	const addParam = useCallback(() => {
		const newParam = {
			name: "",
			value: "",
			active: true
		};
		setQueryParams({
			...queryParams,
			[(/* @__PURE__ */ new Date()).getTime().toString()]: newParam
		});
	}, [queryParams, setQueryParams]);
	const updateParam = useCallback((key, field, value) => {
		if (!key) return;
		setQueryParams({
			...queryParams,
			[key]: {
				...queryParams[key],
				[field]: value
			}
		});
	}, [queryParams, setQueryParams]);
	const updatePathParam = useCallback((key_0, field_0, value_0) => {
		if (!key_0) return;
		setPathParams({
			...pathParams,
			[key_0]: {
				...pathParams[key_0],
				[field_0]: value_0
			}
		});
	}, [pathParams, setPathParams]);
	return /* @__PURE__ */ jsxs("div", {
		className: "h-full grid grid-rows-[auto_auto_1fr]",
		children: [
			/* @__PURE__ */ jsx("div", {
				className: "grid px-4 border-b h-10 items-center grid-cols-[auto_1fr]",
				children: /* @__PURE__ */ jsxs(Button, {
					size: "sm",
					onClick: addParam,
					children: [/* @__PURE__ */ jsx(Plus, { className: "h-3 w-3" }), "Add"]
				})
			}),
			/* @__PURE__ */ jsx(EndpointPathPreview, { path }),
			/* @__PURE__ */ jsxs("div", {
				className: "grid grid-rows-[1fr_1fr]",
				children: [pathParamsConfig.length > 0 && /* @__PURE__ */ jsxs("div", {
					className: "p-2",
					children: [/* @__PURE__ */ jsx("div", {
						className: "text-sm font-medium pl-3",
						children: "Path variables"
					}), pathParamsConfig.map((pathName) => /* @__PURE__ */ jsx(ConfigurationListItem, {
						value: {
							name: pathName.name,
							value: pathName.value,
							active: pathName.active
						},
						id: pathName.name,
						required: true,
						onUpdate: updatePathParam
					}, pathName.name))]
				}), /* @__PURE__ */ jsxs("div", {
					className: "p-2 border-b border-border",
					children: [
						/* @__PURE__ */ jsx("div", {
							className: "text-sm font-medium pl-3",
							children: "Query parameters"
						}),
						Object.entries(queryParams).map(([key_1, param_0]) => /* @__PURE__ */ jsx(ConfigurationListItem, {
							value: param_0,
							id: key_1,
							onUpdate: updateParam,
							onRemove: removeQueryParams
						}, key_1)),
						Object.entries(queryParams).length === 0 && /* @__PURE__ */ jsx("div", {
							className: "grid grid-cols-1 items-center h-full",
							children: /* @__PURE__ */ jsx("div", {
								className: "text-sm text-muted-foreground text-center",
								children: "There are no query params in this endpoint"
							})
						})
					]
				})]
			})
		]
	});
};

//#endregion
//#region src/hooks/use-state-stream.ts
const useStateStream = (object) => {
	const $ = c(15);
	if ($[0] !== "285f0ed50e2dd4fc3015728ea5aa13d2c7f564b6cdec2adc1e1f9b84c6cc298a") {
		for (let $i = 0; $i < 15; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "285f0ed50e2dd4fc3015728ea5aa13d2c7f564b6cdec2adc1e1f9b84c6cc298a";
	}
	let t0;
	if ($[1] !== object) {
		t0 = typeof object === "string" ? {} : object || {};
		$[1] = object;
		$[2] = t0;
	} else t0 = $[2];
	let __motia;
	let rest;
	if ($[3] !== t0) {
		({__motia, ...rest} = t0);
		$[3] = t0;
		$[4] = __motia;
		$[5] = rest;
	} else {
		__motia = $[4];
		rest = $[5];
	}
	const { data } = useStreamItem(__motia);
	const originalData = Array.isArray(object) ? object : rest || object;
	if (typeof object === "string") {
		let t1$1;
		if ($[6] !== object) {
			t1$1 = {
				data: object,
				originalData: object,
				isStreamed: false
			};
			$[6] = object;
			$[7] = t1$1;
		} else t1$1 = $[7];
		return t1$1;
	}
	let t1;
	if ($[8] !== data || $[9] !== originalData) {
		t1 = JSON.stringify(data || originalData, null, 2);
		$[8] = data;
		$[9] = originalData;
		$[10] = t1;
	} else t1 = $[10];
	const t2 = !!__motia;
	let t3;
	if ($[11] !== originalData || $[12] !== t1 || $[13] !== t2) {
		t3 = {
			data: t1,
			originalData,
			isStreamed: t2
		};
		$[11] = originalData;
		$[12] = t1;
		$[13] = t2;
		$[14] = t3;
	} else t3 = $[14];
	return t3;
};

//#endregion
//#region src/play/reponse-code/utils.ts
const statusMessages = {
	100: "Continue",
	101: "Switching Protocols",
	102: "Processing",
	103: "Early Hints",
	200: "OK",
	201: "Created",
	202: "Accepted",
	203: "Non-Authoritative Information",
	204: "No Content",
	205: "Reset Content",
	206: "Partial Content",
	207: "Multi-Status",
	208: "Already Reported",
	226: "IM Used",
	300: "Multiple Choices",
	301: "Moved Permanently",
	302: "Found",
	303: "See Other",
	304: "Not Modified",
	305: "Use Proxy",
	307: "Temporary Redirect",
	308: "Permanent Redirect",
	400: "Bad Request",
	401: "Unauthorized",
	402: "Payment Required",
	403: "Forbidden",
	404: "Not Found",
	405: "Method Not Allowed",
	406: "Not Acceptable",
	407: "Proxy Authentication Required",
	408: "Request Timeout",
	409: "Conflict",
	410: "Gone",
	411: "Length Required",
	412: "Precondition Failed",
	413: "Payload Too Large",
	414: "URI Too Long",
	415: "Unsupported Media Type",
	416: "Range Not Satisfiable",
	417: "Expectation Failed",
	418: "I'm a teapot",
	421: "Misdirected Request",
	422: "Unprocessable Entity",
	423: "Locked",
	424: "Failed Dependency",
	425: "Too Early",
	426: "Upgrade Required",
	428: "Precondition Required",
	429: "Too Many Requests",
	431: "Request Header Fields Too Large",
	451: "Unavailable For Legal Reasons",
	500: "Internal Server Error",
	501: "Not Implemented",
	502: "Bad Gateway",
	503: "Service Unavailable",
	504: "Gateway Timeout",
	505: "HTTP Version Not Supported",
	506: "Variant Also Negotiates",
	507: "Insufficient Storage",
	508: "Loop Detected",
	510: "Not Extended",
	511: "Network Authentication Required"
};
const getStatusMessage = (statusCode) => {
	return statusMessages[statusCode] || "Unknown";
};

//#endregion
//#region src/play/response-code.tsx
const ResponseCode = (t0) => {
	const $ = c(13);
	if ($[0] !== "0a8e4c92b858ecb81d25d971cc42d057e6dc5a30e6757fd87f3755a59a3871b2") {
		for (let $i = 0; $i < 13; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "0a8e4c92b858ecb81d25d971cc42d057e6dc5a30e6757fd87f3755a59a3871b2";
	}
	const { statusCode } = t0;
	let t1;
	if ($[1] !== statusCode) {
		t1 = getStatusMessage(statusCode);
		$[1] = statusCode;
		$[2] = t1;
	} else t1 = $[2];
	const statusMessage = t1;
	const isSuccess = statusCode > 0 && statusCode < 400;
	const isWarning = statusCode >= 400 && statusCode < 500;
	const isError = statusCode >= 500;
	const t2 = isWarning && "dark:bg-[#EAB71F]/20 dark:text-[#EAB71F] bg-[#EAB71F] text-white";
	const t3 = isError && "dark:bg-[#F8367D]/20 dark:text-[#F8367D] bg-[#F8367D] text-white";
	const t4 = isSuccess && "dark:bg-accent-200 dark:text-primary bg-accent text-white";
	let t5;
	if ($[3] !== t2 || $[4] !== t3 || $[5] !== t4) {
		t5 = cn("px-2 py-1 rounded-sm flex items-center gap-1", t2, t3, t4);
		$[3] = t2;
		$[4] = t3;
		$[5] = t4;
		$[6] = t5;
	} else t5 = $[6];
	let t6;
	if ($[7] !== statusCode) {
		t6 = /* @__PURE__ */ jsx("span", {
			className: "font-bold font-mono",
			children: statusCode
		});
		$[7] = statusCode;
		$[8] = t6;
	} else t6 = $[8];
	let t7;
	if ($[9] !== statusMessage || $[10] !== t5 || $[11] !== t6) {
		t7 = /* @__PURE__ */ jsxs("div", {
			className: t5,
			children: [
				t6,
				" ",
				statusMessage
			]
		});
		$[9] = statusMessage;
		$[10] = t5;
		$[11] = t6;
		$[12] = t7;
	} else t7 = $[12];
	return t7;
};

//#endregion
//#region src/play/response-content.tsx
const codeTagProps = { style: {
	fontFamily: "DM Mono, monospace",
	fontSize: "16px"
} };
const customStyle = {
	margin: 0,
	borderRadius: 0,
	padding: 0
};
const useLanguage = (contentType) => {
	if (contentType?.includes("json")) return "json";
	if (contentType?.includes("html")) return "html";
	return "text";
};
const ResponseContent = (t0) => {
	const $ = c(14);
	if ($[0] !== "51b73d23c1291874f81ba55f0a6e81eecd3ad0cede326e91ed482cd0ab995a8f") {
		for (let $i = 0; $i < 14; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "51b73d23c1291874f81ba55f0a6e81eecd3ad0cede326e91ed482cd0ab995a8f";
	}
	const { code, blob, contentType } = t0;
	const themeStyle = useThemeStore(_temp$1) === "dark" ? atomDark : oneLight;
	const language = useLanguage(contentType);
	let t1;
	if ($[1] !== blob) {
		t1 = () => {
			if (blob) {
				const url = URL.createObjectURL(blob);
				window.open(url, "_blank");
			}
		};
		$[1] = blob;
		$[2] = t1;
	} else t1 = $[2];
	const onDownload = t1;
	let t2;
	if ($[3] !== blob || $[4] !== onDownload) {
		t2 = blob && /* @__PURE__ */ jsx("div", {
			className: "flex p-8 justify-center",
			children: /* @__PURE__ */ jsxs(Button, {
				variant: "default",
				onClick: onDownload,
				children: [/* @__PURE__ */ jsx(Download, { className: "h-4 w-4" }), " Download"]
			})
		});
		$[3] = blob;
		$[4] = onDownload;
		$[5] = t2;
	} else t2 = $[5];
	let t3;
	if ($[6] !== blob || $[7] !== code || $[8] !== language || $[9] !== themeStyle) {
		t3 = !blob && code && /* @__PURE__ */ jsx(Prism, {
			showLineNumbers: true,
			language,
			style: themeStyle,
			codeTagProps,
			customStyle,
			wrapLines: true,
			children: code
		});
		$[6] = blob;
		$[7] = code;
		$[8] = language;
		$[9] = themeStyle;
		$[10] = t3;
	} else t3 = $[10];
	let t4;
	if ($[11] !== t2 || $[12] !== t3) {
		t4 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col h-full overflow-y-auto",
			children: [t2, t3]
		});
		$[11] = t2;
		$[12] = t3;
		$[13] = t4;
	} else t4 = $[13];
	return t4;
};
function _temp$1(state) {
	return state.theme;
}

//#endregion
//#region src/play/side-panel-response.tsx
const SidePanelResponse = memo(() => {
	const $ = c(29);
	if ($[0] !== "c41d9d0c255bfb25883f458c9673498cbc698b0e44623ab8537e78707a43852c") {
		for (let $i = 0; $i < 29; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "c41d9d0c255bfb25883f458c9673498cbc698b0e44623ab8537e78707a43852c";
	}
	const { setResponse } = useEndpointConfiguration();
	const response = useEndpointConfiguration(getResponseSelector);
	const { data } = useStateStream(response?.body instanceof Blob ? void 0 : response?.body);
	const [activeTab, setActiveTab] = useState("preview");
	let t0;
	if ($[1] !== setResponse) {
		t0 = () => setResponse(void 0);
		$[1] = setResponse;
		$[2] = t0;
	} else t0 = $[2];
	const onClose = t0;
	if (!response) return null;
	let t1;
	if ($[3] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = (value) => setActiveTab(value);
		$[3] = t1;
	} else t1 = $[3];
	let t2;
	if ($[4] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsxs(TabsList, { children: [/* @__PURE__ */ jsx(TabsTrigger, {
			value: "preview",
			className: "cursor-pointer",
			children: "Preview"
		}), /* @__PURE__ */ jsx(TabsTrigger, {
			value: "headers",
			className: "grid grid-cols-[auto_auto] gap-2 items-center cursor-pointer",
			children: "Headers"
		})] });
		$[4] = t2;
	} else t2 = $[4];
	let t3;
	if ($[5] === Symbol.for("react.memo_cache_sentinel")) {
		t3 = /* @__PURE__ */ jsx(X, { className: "h-4 w-4" });
		$[5] = t3;
	} else t3 = $[5];
	let t4;
	if ($[6] !== onClose) {
		t4 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-[1fr_auto] items-center h-10 border-b px-5 bg-card",
			children: [t2, /* @__PURE__ */ jsx(Button, {
				variant: "ghost",
				size: "icon",
				onClick: onClose,
				children: t3
			})]
		});
		$[6] = onClose;
		$[7] = t4;
	} else t4 = $[7];
	let t5;
	if ($[8] !== response.statusCode) {
		t5 = /* @__PURE__ */ jsx(ResponseCode, { statusCode: response.statusCode });
		$[8] = response.statusCode;
		$[9] = t5;
	} else t5 = $[9];
	let t6;
	if ($[10] !== response.executionTime) {
		t6 = !!response.executionTime && /* @__PURE__ */ jsxs("div", {
			className: "text-muted-foreground bg-muted-foreground/10 px-2 py-1 rounded-sm",
			children: [response.executionTime, "ms"]
		});
		$[10] = response.executionTime;
		$[11] = t6;
	} else t6 = $[11];
	let t7;
	if ($[12] !== t5 || $[13] !== t6) {
		t7 = /* @__PURE__ */ jsx("div", {
			className: "sticky bottom-0 border-b border-border p-3 text-sm flex items-center gap-1 font-medium",
			children: /* @__PURE__ */ jsxs("div", {
				className: "flex flex-row items-center flex-1 gap-3",
				children: [t5, t6]
			})
		});
		$[12] = t5;
		$[13] = t6;
		$[14] = t7;
	} else t7 = $[14];
	const t8 = response?.body instanceof Blob ? response.body : void 0;
	const t9 = response.headers["content-type"];
	let t10;
	if ($[15] !== data || $[16] !== t8 || $[17] !== t9) {
		t10 = /* @__PURE__ */ jsx(TabsContent, {
			value: "preview",
			children: /* @__PURE__ */ jsx(ResponseContent, {
				code: data,
				blob: t8,
				contentType: t9
			})
		});
		$[15] = data;
		$[16] = t8;
		$[17] = t9;
		$[18] = t10;
	} else t10 = $[18];
	let t11;
	if ($[19] !== response.headers) {
		t11 = response.headers && Object.entries(response.headers).map(_temp);
		$[19] = response.headers;
		$[20] = t11;
	} else t11 = $[20];
	let t12;
	if ($[21] !== t11) {
		t12 = /* @__PURE__ */ jsx(TabsContent, {
			value: "headers",
			children: /* @__PURE__ */ jsx("div", {
				className: "grid grid-cols-[auto_minmax(0,1fr)] gap-4 p-4 overflow-auto auto-rows-max h-full",
				children: t11
			})
		});
		$[21] = t11;
		$[22] = t12;
	} else t12 = $[22];
	let t13;
	if ($[23] !== activeTab || $[24] !== t10 || $[25] !== t12 || $[26] !== t4 || $[27] !== t7) {
		t13 = /* @__PURE__ */ jsxs(Tabs, {
			value: activeTab,
			onValueChange: t1,
			className: "border-l border-border",
			"data-testid": "endpoint-response-container",
			children: [
				t4,
				t7,
				t10,
				t12
			]
		});
		$[23] = activeTab;
		$[24] = t10;
		$[25] = t12;
		$[26] = t4;
		$[27] = t7;
		$[28] = t13;
	} else t13 = $[28];
	return t13;
});
function _temp(t0) {
	const [key, value_0] = t0;
	return /* @__PURE__ */ jsxs(Fragment$1, { children: [/* @__PURE__ */ jsx("span", {
		className: "font-bold text-sm h-8 items-center grid whitespace-nowrap",
		children: key
	}), /* @__PURE__ */ jsx("span", {
		className: "text-sm text-muted-foreground h-8 items-center grid whitespace-nowrap",
		children: value_0
	})] });
}

//#endregion
//#region src/play/trigger-button.tsx
const TriggerButton = memo(({ method, path }) => {
	const { setResponse } = useEndpointConfiguration();
	const headers = useEndpointConfiguration(useShallow(getHeadersSelector));
	const body = useEndpointConfiguration(useShallow(getBodySelector));
	const pathUrl = usePathUrl(path);
	const [isLoading, setIsLoading] = useState(false);
	const onClick = async () => {
		try {
			setIsLoading(true);
			const _headers = Object.values(headers).filter((header) => header.active && header.name !== "" && header.value !== "").reduce((acc, header_0) => {
				acc[header_0.name.toLowerCase()] = header_0.value;
				return acc;
			}, {});
			const startTime = Date.now();
			setResponse(await fetch(pathUrl, {
				method,
				headers: _headers,
				body: [
					"GET",
					"DELETE",
					"HEAD",
					"OPTIONS"
				].includes(method) ? null : body
			}), startTime);
		} catch (error) {
			console.error("Error triggering endpoint:", error);
		} finally {
			setIsLoading(false);
		}
	};
	return /* @__PURE__ */ jsx(Button, {
		variant: "accent",
		size: "sm",
		onClick,
		disabled: isLoading,
		"data-testid": "endpoint-play-button",
		children: isLoading ? /* @__PURE__ */ jsx(Loader2, { className: "animate-spin" }) : "Send"
	});
});

//#endregion
//#region src/play/side-panel.tsx
const headersCountSelector = (state) => Object.keys(getHeadersSelector(state)).length;
const hasResponseSelector = (state) => getResponseSelector(state) !== void 0;
const paramsCountSelector = (state) => Object.keys(getQueryParamsSelector(state)).length;
const SidePanel = memo((t0) => {
	const $ = c(62);
	if ($[0] !== "0020adea1162c54cf5aae85d8c5ed34a7262d6cfdf90ef6b8d89094328918944") {
		for (let $i = 0; $i < 62; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "0020adea1162c54cf5aae85d8c5ed34a7262d6cfdf90ef6b8d89094328918944";
	}
	const { endpoint, onClose } = t0;
	const [activeTab, setActiveTab] = useState(endpoint.method === "GET" || endpoint.method === "DELETE" ? "params" : "body");
	const [isSpecOpen, setIsSpecOpen] = useState(false);
	const headersCount = useEndpointConfiguration(useShallow(headersCountSelector));
	const hasResponse = useEndpointConfiguration(useShallow(hasResponseSelector));
	const pathParamsCount = usePathParams(endpoint.path).length;
	const paramsCount = useEndpointConfiguration(useShallow(paramsCountSelector)) + pathParamsCount;
	let t1;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = /* @__PURE__ */ jsx(BackgroundEffect, {});
		$[1] = t1;
	} else t1 = $[1];
	let t2;
	if ($[2] !== endpoint.method || $[3] !== endpoint.path) {
		t2 = /* @__PURE__ */ jsx(EndpointPath, {
			method: endpoint.method,
			path: endpoint.path
		});
		$[2] = endpoint.method;
		$[3] = endpoint.path;
		$[4] = t2;
	} else t2 = $[4];
	let t3;
	if ($[5] !== endpoint.description) {
		t3 = endpoint.description && /* @__PURE__ */ jsx("p", {
			className: "text-sm text-muted-foreground",
			children: endpoint.description
		});
		$[5] = endpoint.description;
		$[6] = t3;
	} else t3 = $[6];
	let t4;
	if ($[7] !== t2 || $[8] !== t3) {
		t4 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-rows-2 gap-2",
			children: [t2, t3]
		});
		$[7] = t2;
		$[8] = t3;
		$[9] = t4;
	} else t4 = $[9];
	let t5;
	if ($[10] !== isSpecOpen) {
		t5 = () => setIsSpecOpen(!isSpecOpen);
		$[10] = isSpecOpen;
		$[11] = t5;
	} else t5 = $[11];
	let t6;
	if ($[12] === Symbol.for("react.memo_cache_sentinel")) {
		t6 = /* @__PURE__ */ jsx(Book, {});
		$[12] = t6;
	} else t6 = $[12];
	let t7;
	if ($[13] !== t5) {
		t7 = /* @__PURE__ */ jsx("div", {
			className: "flex items-end justify-end",
			children: /* @__PURE__ */ jsx(Button, {
				variant: "icon",
				size: "icon",
				onClick: t5,
				"data-testid": "endpoint-spec-button",
				children: t6
			})
		});
		$[13] = t5;
		$[14] = t7;
	} else t7 = $[14];
	let t8;
	if ($[15] === Symbol.for("react.memo_cache_sentinel")) {
		t8 = /* @__PURE__ */ jsx(X, {});
		$[15] = t8;
	} else t8 = $[15];
	let t9;
	if ($[16] !== onClose) {
		t9 = /* @__PURE__ */ jsx(Button, {
			variant: "icon",
			size: "icon",
			onClick: onClose,
			children: t8
		});
		$[16] = onClose;
		$[17] = t9;
	} else t9 = $[17];
	let t10;
	if ($[18] !== t4 || $[19] !== t7 || $[20] !== t9) {
		t10 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-[1fr_1fr_auto] items-start gap-4 px-5 py-4 border-b bg-card w-full",
			children: [
				t4,
				t7,
				t9
			]
		});
		$[18] = t4;
		$[19] = t7;
		$[20] = t9;
		$[21] = t10;
	} else t10 = $[21];
	const t11 = !hasResponse && "grid-cols-1";
	let t12;
	if ($[22] !== t11) {
		t12 = cn("grid grid-cols-[minmax(380px,1fr)_minmax(auto,1fr)]", t11);
		$[22] = t11;
		$[23] = t12;
	} else t12 = $[23];
	let t13;
	if ($[24] === Symbol.for("react.memo_cache_sentinel")) {
		t13 = (value) => setActiveTab(value);
		$[24] = t13;
	} else t13 = $[24];
	let t14;
	if ($[25] !== paramsCount) {
		t14 = /* @__PURE__ */ jsxs(TabsTrigger, {
			value: "params",
			className: "grid grid-cols-[auto_auto] gap-2 items-center cursor-pointer",
			"data-testid": "endpoint-params-tab",
			children: ["Params", /* @__PURE__ */ jsx(Badge, {
				variant: "outline",
				className: "h-4 px-1.5 text-xs",
				children: paramsCount
			})]
		});
		$[25] = paramsCount;
		$[26] = t14;
	} else t14 = $[26];
	let t15;
	if ($[27] === Symbol.for("react.memo_cache_sentinel")) {
		t15 = /* @__PURE__ */ jsx(TabsTrigger, {
			value: "body",
			className: "cursor-pointer",
			"data-testid": "endpoint-body-tab",
			children: "Body"
		});
		$[27] = t15;
	} else t15 = $[27];
	let t16;
	if ($[28] !== headersCount) {
		t16 = /* @__PURE__ */ jsxs(TabsTrigger, {
			value: "headers",
			className: "grid grid-cols-[auto_auto] gap-2 items-center cursor-pointer",
			"data-testid": "endpoint-headers-tab",
			children: ["Headers", /* @__PURE__ */ jsx(Badge, {
				variant: "outline",
				className: "h-4 px-1.5 text-xs",
				children: headersCount
			})]
		});
		$[28] = headersCount;
		$[29] = t16;
	} else t16 = $[29];
	let t17;
	if ($[30] !== t14 || $[31] !== t16) {
		t17 = /* @__PURE__ */ jsxs(TabsList, { children: [
			t14,
			t15,
			t16
		] });
		$[30] = t14;
		$[31] = t16;
		$[32] = t17;
	} else t17 = $[32];
	const t18 = endpoint.method;
	let t19;
	if ($[33] !== endpoint.path) {
		t19 = endpoint.path.toString();
		$[33] = endpoint.path;
		$[34] = t19;
	} else t19 = $[34];
	let t20;
	if ($[35] !== endpoint.method || $[36] !== t19) {
		t20 = /* @__PURE__ */ jsx(TriggerButton, {
			method: t18,
			path: t19
		});
		$[35] = endpoint.method;
		$[36] = t19;
		$[37] = t20;
	} else t20 = $[37];
	let t21;
	if ($[38] !== t17 || $[39] !== t20) {
		t21 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-[1fr_auto] items-center h-10 border-b px-5 bg-card",
			children: [t17, t20]
		});
		$[38] = t17;
		$[39] = t20;
		$[40] = t21;
	} else t21 = $[40];
	let t22;
	if ($[41] !== endpoint.path) {
		t22 = /* @__PURE__ */ jsx(TabsContent, {
			value: "params",
			children: /* @__PURE__ */ jsx(SidePanelParamsTab, { path: endpoint.path })
		});
		$[41] = endpoint.path;
		$[42] = t22;
	} else t22 = $[42];
	let t23;
	if ($[43] !== endpoint.bodySchema) {
		t23 = /* @__PURE__ */ jsx(TabsContent, {
			value: "body",
			children: /* @__PURE__ */ jsx(SidePanelBodyTab, { schema: endpoint.bodySchema })
		});
		$[43] = endpoint.bodySchema;
		$[44] = t23;
	} else t23 = $[44];
	let t24;
	if ($[45] === Symbol.for("react.memo_cache_sentinel")) {
		t24 = /* @__PURE__ */ jsx(TabsContent, {
			value: "headers",
			children: /* @__PURE__ */ jsx(SidePanelHeadersTab, {})
		});
		$[45] = t24;
	} else t24 = $[45];
	let t25;
	if ($[46] !== activeTab || $[47] !== t21 || $[48] !== t22 || $[49] !== t23) {
		t25 = /* @__PURE__ */ jsxs(Tabs, {
			value: activeTab,
			onValueChange: t13,
			children: [
				t21,
				t22,
				t23,
				t24
			]
		});
		$[46] = activeTab;
		$[47] = t21;
		$[48] = t22;
		$[49] = t23;
		$[50] = t25;
	} else t25 = $[50];
	let t26;
	if ($[51] === Symbol.for("react.memo_cache_sentinel")) {
		t26 = /* @__PURE__ */ jsx(SidePanelResponse, {});
		$[51] = t26;
	} else t26 = $[51];
	let t27;
	if ($[52] !== t12 || $[53] !== t25) {
		t27 = /* @__PURE__ */ jsxs("div", {
			className: t12,
			children: [t25, t26]
		});
		$[52] = t12;
		$[53] = t25;
		$[54] = t27;
	} else t27 = $[54];
	let t28;
	if ($[55] !== endpoint || $[56] !== isSpecOpen) {
		t28 = isSpecOpen && /* @__PURE__ */ jsx(SpecSidePanel, {
			endpoint,
			onClose: () => setIsSpecOpen(false)
		});
		$[55] = endpoint;
		$[56] = isSpecOpen;
		$[57] = t28;
	} else t28 = $[57];
	let t29;
	if ($[58] !== t10 || $[59] !== t27 || $[60] !== t28) {
		t29 = /* @__PURE__ */ jsxs("div", {
			className: "isolate grid grid-cols-1 overflow-y-auto min-w-0 grid-rows-[auto_1fr] border-l border-border",
			"data-testid": "endpoint-details-panel",
			children: [
				t1,
				t10,
				t27,
				t28
			]
		});
		$[58] = t10;
		$[59] = t27;
		$[60] = t28;
		$[61] = t29;
	} else t29 = $[61];
	return t29;
});

//#endregion
//#region src/endpoints-page.tsx
const EndpointsPage = () => {
	const $ = c(33);
	if ($[0] !== "198b8272767e037dafaba995069a2aa9f7c2ab3fe909e2acb55498bc6d00f981") {
		for (let $i = 0; $i < 33; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "198b8272767e037dafaba995069a2aa9f7c2ab3fe909e2acb55498bc6d00f981";
	}
	const { endpoints, groupedEndpoints } = useGetEndpoints();
	const { selectedEndpointId, setSelectedEndpointId, toggleFlowGroupStatus, flowGroupStatus } = useEndpointConfiguration();
	let t0;
	if ($[1] !== endpoints || $[2] !== selectedEndpointId) {
		let t1$1;
		if ($[4] !== selectedEndpointId) {
			t1$1 = (endpoint) => endpoint.id === selectedEndpointId;
			$[4] = selectedEndpointId;
			$[5] = t1$1;
		} else t1$1 = $[5];
		t0 = endpoints.find(t1$1);
		$[1] = endpoints;
		$[2] = selectedEndpointId;
		$[3] = t0;
	} else t0 = $[3];
	const selectedEndpoint = t0;
	let t1;
	if ($[6] !== setSelectedEndpointId) {
		t1 = () => {
			setSelectedEndpointId("");
		};
		$[6] = setSelectedEndpointId;
		$[7] = t1;
	} else t1 = $[7];
	const onClose = t1;
	const [search, setSearch] = useState("");
	let t2;
	if ($[8] !== groupedEndpoints || $[9] !== search) {
		t2 = Object.entries(groupedEndpoints).filter((t3$1) => {
			const [, endpoints_0] = t3$1;
			return endpoints_0.some((endpoint_0) => endpoint_0.method?.toLowerCase().includes(search.toLowerCase()) || endpoint_0.path?.toLowerCase().includes(search.toLowerCase()));
		});
		$[8] = groupedEndpoints;
		$[9] = search;
		$[10] = t2;
	} else t2 = $[10];
	const filteredEndpoints = t2;
	const t3 = selectedEndpoint ? "grid-cols-[300px_1fr] " : "grid-cols-1";
	let t4;
	if ($[11] !== t3) {
		t4 = cn("grid h-full max-h-full", t3);
		$[11] = t3;
		$[12] = t4;
	} else t4 = $[12];
	let t5;
	if ($[13] === Symbol.for("react.memo_cache_sentinel")) {
		t5 = () => setSearch("");
		$[13] = t5;
	} else t5 = $[13];
	let t6;
	if ($[14] !== search) {
		t6 = /* @__PURE__ */ jsx(EndpointsSearch, {
			value: search,
			onChange: setSearch,
			onClear: t5
		});
		$[14] = search;
		$[15] = t6;
	} else t6 = $[15];
	let t7;
	if ($[16] !== filteredEndpoints || $[17] !== flowGroupStatus || $[18] !== search || $[19] !== selectedEndpointId || $[20] !== setSelectedEndpointId || $[21] !== toggleFlowGroupStatus) {
		t7 = /* @__PURE__ */ jsx("div", {
			className: "grid grid-cols-1 auto-rows-max overflow-auto min-w-0",
			children: filteredEndpoints.map((t8$1) => {
				const [flow, endpoints_1] = t8$1;
				const isSelected = endpoints_1.some((endpoint_1) => endpoint_1.id === selectedEndpointId);
				return /* @__PURE__ */ jsx(FlowGroup, {
					flow,
					endpoints: endpoints_1,
					isOpen: !flowGroupStatus[flow] || isSelected || search !== "",
					isSelected,
					onToggle: toggleFlowGroupStatus,
					onClearSelection: () => setSelectedEndpointId(""),
					selectedEndpointId,
					onSelectEndpoint: setSelectedEndpointId
				}, flow);
			})
		});
		$[16] = filteredEndpoints;
		$[17] = flowGroupStatus;
		$[18] = search;
		$[19] = selectedEndpointId;
		$[20] = setSelectedEndpointId;
		$[21] = toggleFlowGroupStatus;
		$[22] = t7;
	} else t7 = $[22];
	let t8;
	if ($[23] !== t6 || $[24] !== t7) {
		t8 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-rows-[auto_1fr] h-full overflow-auto min-w-0",
			children: [t6, t7]
		});
		$[23] = t6;
		$[24] = t7;
		$[25] = t8;
	} else t8 = $[25];
	let t9;
	if ($[26] !== onClose || $[27] !== selectedEndpoint) {
		t9 = selectedEndpoint && /* @__PURE__ */ jsx(SidePanel, {
			endpoint: selectedEndpoint,
			onClose
		});
		$[26] = onClose;
		$[27] = selectedEndpoint;
		$[28] = t9;
	} else t9 = $[28];
	let t10;
	if ($[29] !== t4 || $[30] !== t8 || $[31] !== t9) {
		t10 = /* @__PURE__ */ jsxs("div", {
			className: t4,
			"data-testid": "endpoints-list",
			children: [t8, t9]
		});
		$[29] = t4;
		$[30] = t8;
		$[31] = t9;
		$[32] = t10;
	} else t10 = $[32];
	return t10;
};

//#endregion
export { EndpointsPage };
