// @motiadev/plugin-endpoint/dist/plugin.js
import { StreamAdapter, isApiStep } from "@motiadev/core";

//#region src/api/endpoints-stream.ts
const mapEndpoint = (step) => {
	return {
		id: step.filePath,
		method: step.config.method,
		path: step.config.path,
		description: step.config.description,
		queryParams: step.config.queryParams,
		responseSchema: step.config.responseSchema,
		bodySchema: step.config.bodySchema,
		flows: step.config.flows
	};
};
var EndpointsStream = class extends StreamAdapter {
	constructor(lockedData) {
		super("__motia.api-endpoints");
		this.lockedData = lockedData;
	}
	async get(_groupId, id) {
		const endpoint = this.lockedData.apiSteps().find((step) => step.filePath === id);
		return endpoint ? mapEndpoint(endpoint) : null;
	}
	async delete(id) {
		return { id };
	}
	async set(_, __, data) {
		return data;
	}
	async getGroup() {
		return this.lockedData.apiSteps().map(mapEndpoint);
	}
};

//#endregion
//#region src/api/index.ts
function plugin({ lockedData }) {
	const stream = lockedData.createStream({
		filePath: "__motia.api-endpoints.ts",
		hidden: true,
		config: {
			name: "__motia.api-endpoints",
			baseConfig: {
				storageType: "custom",
				factory: () => new EndpointsStream(lockedData)
			},
			schema: null
		}
	})();
	const apiStepCreated = (step) => {
		if (isApiStep(step)) stream.set("default", step.filePath, {
			id: step.filePath,
			method: step.config.method,
			path: step.config.path,
			description: step.config.description,
			queryParams: step.config.queryParams
		});
	};
	const apiStepUpdated = (step) => {
		if (isApiStep(step)) stream.set("default", step.filePath, mapEndpoint(step));
	};
	const apiStepRemoved = (step) => {
		if (isApiStep(step)) stream.delete("default", step.filePath);
	};
	lockedData.onStep("step-created", apiStepCreated);
	lockedData.onStep("step-updated", apiStepUpdated);
	lockedData.onStep("step-removed", apiStepRemoved);
	return { workbench: [{
		packageName: "@motiadev/plugin-endpoint",
		cssImports: ["@motiadev/plugin-endpoint/dist/styles.css"],
		label: "Endpoints",
		position: "top",
		componentName: "EndpointsPage",
		labelIcon: "link-2"
	}] };
}

//#endregion
//#region src/plugin.ts
var plugin_default = plugin;

//#endregion
export { plugin_default as default };
