// @motiadev/adapter-redis-streams/dist/redis-stream-adapter-manager.mjs
import { RedisStreamAdapter } from "./redis-stream-adapter.mjs";
import { createClient } from "redis";

//#region src/redis-stream-adapter-manager.ts
function isRedisClient(input) {
	return typeof input === "object" && "isOpen" in input && "connect" in input;
}
var RedisStreamAdapterManager = class {
	constructor(redisConnection, options) {
		this.connected = false;
		this.config = {
			keyPrefix: options?.keyPrefix || "motia:stream:",
			socketKeepAlive: options?.socketKeepAlive ?? true
		};
		if (isRedisClient(redisConnection)) {
			this.client = redisConnection;
			this.isExternalClient = true;
			this.connected = this.client.isOpen;
		} else {
			const config = {
				...redisConnection,
				socket: {
					...redisConnection.socket || {},
					keepAlive: this.config.socketKeepAlive,
					noDelay: true
				}
			};
			this.isExternalClient = false;
			this.client = createClient(config);
			this.client.on("error", (err) => {
				if (this.connected) console.error("[Redis Stream Manager] Client error:", err?.message);
			});
			this.client.on("connect", () => {
				this.connected = true;
			});
			this.client.on("disconnect", () => {
				console.warn("[Redis Stream Manager] Disconnected");
				this.connected = false;
			});
			this.connect();
		}
	}
	async connect() {
		if (!this.connected && !this.isExternalClient) try {
			await this.client.connect();
		} catch (error) {
			console.error("[Redis Stream Manager] Failed to connect:", error);
			throw error;
		}
	}
	createStream(streamName) {
		return new RedisStreamAdapter(streamName, this.config, this.client);
	}
	getClient() {
		return this.client;
	}
	async shutdown() {
		if (!this.isExternalClient && this.client.isOpen) await this.client.quit();
	}
};

//#endregion
export { RedisStreamAdapterManager };
