// @motiadev/core/dist/src/adapters/defaults/state/file-state-adapter.mjs
import { filterItem, inferType } from "./utils.mjs";
import fs from "fs";
import * as path$1 from "path";

//#region src/adapters/defaults/state/file-state-adapter.ts
var FileStateAdapter = class {
	constructor(config) {
		this.filePath = path$1.join(config.filePath, "motia.state.json");
		this.init();
	}
	init() {
		const dir = this.filePath.replace("motia.state.json", "");
		try {
			fs.realpathSync(dir);
		} catch {
			fs.mkdirSync(dir, { recursive: true });
		}
		try {
			fs.readFileSync(this.filePath, "utf-8");
		} catch {
			fs.writeFileSync(this.filePath, JSON.stringify({}), "utf-8");
		}
	}
	async getGroup(groupId) {
		const data = this._readFile();
		return Object.entries(data).filter(([key]) => key.startsWith(groupId)).map(([, value]) => JSON.parse(value));
	}
	async get(traceId, key) {
		const data = this._readFile();
		const fullKey = this._makeKey(traceId, key);
		return data[fullKey] ? JSON.parse(data[fullKey]) : null;
	}
	async set(traceId, key, value) {
		const data = this._readFile();
		const fullKey = this._makeKey(traceId, key);
		data[fullKey] = JSON.stringify(value);
		this._writeFile(data);
		return value;
	}
	async delete(traceId, key) {
		const data = this._readFile();
		const fullKey = this._makeKey(traceId, key);
		const value = await this.get(traceId, key);
		if (value) {
			delete data[fullKey];
			this._writeFile(data);
		}
		return value;
	}
	async clear(traceId) {
		const data = this._readFile();
		const pattern = this._makeKey(traceId, "");
		for (const key in data) if (key.startsWith(pattern)) delete data[key];
		this._writeFile(data);
	}
	async keys(traceId) {
		const data = this._readFile();
		return Object.keys(data).filter((key) => key.startsWith(this._makeKey(traceId, ""))).map((key) => key.replace(this._makeKey(traceId, ""), ""));
	}
	async traceIds() {
		const data = this._readFile();
		const traceIds = /* @__PURE__ */ new Set();
		Object.keys(data).forEach((key) => traceIds.add(key.split(":")[0]));
		return Array.from(traceIds);
	}
	async cleanup() {}
	async items(input) {
		const data = this._readFile();
		return Object.entries(data).map(([key, value]) => {
			const [groupId, ...keyParts] = key.split(":");
			const itemKey = keyParts.join(":");
			const itemValue = JSON.parse(value);
			return {
				groupId,
				key: itemKey,
				value: itemValue,
				type: inferType(itemValue)
			};
		}).filter((item) => input.groupId ? item.groupId === input.groupId : true).filter((item) => input.filter ? filterItem(item, input.filter) : true);
	}
	_makeKey(traceId, key) {
		return `${traceId}:${key}`;
	}
	_readFile() {
		try {
			const content = fs.readFileSync(this.filePath, "utf-8");
			return JSON.parse(content);
		} catch (error) {
			this.init();
			return {};
		}
	}
	_writeFile(data) {
		try {
			fs.writeFileSync(this.filePath, JSON.stringify(data, null, 2), "utf-8");
		} catch (error) {
			this.init();
			fs.writeFileSync(this.filePath, JSON.stringify(data, null, 2), "utf-8");
		}
	}
};

//#endregion
export { FileStateAdapter };
