// motia/dist/dev.mjs
import { generateLockedData, getStepFiles, getStreamFiles } from "./generate-locked-data.mjs";
import { version } from "./version.mjs";
import { identifyUser } from "./utils/analytics.mjs";
import { activatePythonVenv } from "./utils/activate-python-env.mjs";
import { validatePythonEnvironment } from "./utils/validate-python-environment.mjs";
import { loadMotiaConfig } from "./load-motia-config.mjs";
import { deployEndpoints } from "./cloud/endpoints.mjs";
import { isTutorialDisabled, workbenchBase } from "./constants.mjs";
import { createDevWatchers } from "./dev-watchers.mjs";
import { processPlugins } from "./plugins/process-plugins.mjs";
import "./plugins/index.mjs";
import { getRedisClient, getRedisConnectionInfo, stopRedisConnection } from "./redis/connection.mjs";
import { listenWithFallback } from "./utils/listen-with-fallback.mjs";
import { createMermaidGenerator, createServer, getProjectIdentifier, trackEvent } from "@motiadev/core";
import { flush } from "@amplitude/analytics-node";
import { BullMQEventAdapter } from "@motiadev/adapter-bullmq-events";
import { RedisCronAdapter } from "@motiadev/adapter-redis-cron";
import { RedisStateAdapter } from "@motiadev/adapter-redis-state";
import { RedisStreamAdapterManager } from "@motiadev/adapter-redis-streams";

//#region src/dev.ts
const dev = async (port, hostname, disableVerbose, enableMermaid, motiaFileStorageDir) => {
	const baseDir = process.cwd();
	const isVerbose = !disableVerbose;
	identifyUser();
	const stepFiles = [...getStepFiles(baseDir), ...getStreamFiles(baseDir)];
	const hasPythonFiles = stepFiles.some((file) => file.endsWith(".py"));
	trackEvent("dev_server_started", {
		port,
		verbose_mode: isVerbose,
		mermaid_enabled: enableMermaid,
		has_python_files: hasPythonFiles,
		total_step_files: stepFiles.length,
		project_name: getProjectIdentifier(baseDir)
	});
	const pythonValidation = await validatePythonEnvironment({
		baseDir,
		hasPythonFiles
	});
	if (!pythonValidation.success) process.exit(1);
	if (pythonValidation.hasPythonFiles) {
		activatePythonVenv({
			baseDir,
			isVerbose
		});
		trackEvent("python_environment_activated");
	}
	const motiaFileStoragePath = motiaFileStorageDir || ".motia";
	const appConfig = await loadMotiaConfig(baseDir);
	const redisClient = await getRedisClient(motiaFileStoragePath, appConfig);
	const adapters = {
		eventAdapter: appConfig.adapters?.events || new BullMQEventAdapter({
			connection: getRedisConnectionInfo(),
			prefix: "motia:events"
		}),
		cronAdapter: appConfig.adapters?.cron || new RedisCronAdapter(redisClient),
		streamAdapter: appConfig.adapters?.streams || new RedisStreamAdapterManager(redisClient)
	};
	const lockedData = await generateLockedData({
		projectDir: baseDir,
		streamAdapter: adapters.streamAdapter,
		redisClient,
		streamAuth: appConfig.streamAuth
	});
	const motiaServer = createServer(lockedData, appConfig.adapters?.state || new RedisStateAdapter(redisClient), { isVerbose }, adapters, appConfig.app);
	const watcher = createDevWatchers(lockedData, motiaServer, motiaServer.motiaEventManager, motiaServer.cronManager);
	const plugins = await processPlugins(motiaServer);
	if (enableMermaid) {
		createMermaidGenerator(baseDir).initialize(lockedData);
		trackEvent("mermaid_generator_initialized");
	}
	deployEndpoints(motiaServer, lockedData);
	motiaServer.app.get("/__motia", (_, res) => {
		const meta = {
			version,
			isDev: true,
			isTutorialDisabled,
			workbenchBase
		};
		res.header("Access-Control-Allow-Origin", "*").header("Access-Control-Allow-Private-Network", "true").status(200).json(meta);
	});
	trackEvent("dev_server_ready", {
		port,
		flows_count: lockedData.flows?.length || 0,
		steps_count: lockedData.activeSteps?.length || 0,
		flows: Object.keys(lockedData.flows || {}),
		steps: lockedData.activeSteps.map((step) => step.config.name),
		streams: Object.keys(lockedData.getStreams() || {}),
		runtime_version: version,
		environment: process.env.NODE_ENV || "development"
	});
	const { applyMiddleware } = await import("@motiadev/workbench/middleware");
	await applyMiddleware({
		app: motiaServer.app,
		port,
		workbenchBase,
		plugins: plugins.flatMap((item) => item.workbench)
	});
	const actualPort = await listenWithFallback(motiaServer.server, port, hostname);
	if (actualPort !== port) console.log(`âš ï¸  Port ${port} was in use, using port ${actualPort} instead`);
	console.log("ðŸš€ Server ready and listening on port", actualPort);
	console.log(`ðŸ”— Open http://localhost:${actualPort}${workbenchBase} to open workbench ðŸ› ï¸`);
	process.on("SIGTERM", async () => {
		trackEvent("dev_server_shutdown", { reason: "SIGTERM" });
		motiaServer.server.close();
		await watcher.stop();
		await stopRedisConnection();
		await flush().promise;
		process.exit(0);
	});
	process.on("SIGINT", async () => {
		trackEvent("dev_server_shutdown", { reason: "SIGINT" });
		motiaServer.server.close();
		await watcher.stop();
		await stopRedisConnection();
		await flush().promise;
		process.exit(0);
	});
};

//#endregion
export { dev };
