// motia/dist/start.mjs
import { generateLockedData, getStepFiles, getStreamFiles } from "./generate-locked-data.mjs";
import { version } from "./version.mjs";
import { activatePythonVenv } from "./utils/activate-python-env.mjs";
import { validatePythonEnvironment } from "./utils/validate-python-environment.mjs";
import { loadMotiaConfig } from "./load-motia-config.mjs";
import { workbenchBase } from "./constants.mjs";
import { processPlugins } from "./plugins/process-plugins.mjs";
import "./plugins/index.mjs";
import { getRedisClient, getRedisConnectionInfo, stopRedisConnection } from "./redis/connection.mjs";
import { listenWithFallback } from "./utils/listen-with-fallback.mjs";
import { createServer } from "@motiadev/core";
import path from "path";
import { BullMQEventAdapter } from "@motiadev/adapter-bullmq-events";
import { RedisCronAdapter } from "@motiadev/adapter-redis-cron";
import { RedisStateAdapter } from "@motiadev/adapter-redis-state";
import { RedisStreamAdapterManager } from "@motiadev/adapter-redis-streams";

//#region src/start.ts
const start = async (port, hostname, disableVerbose, motiaFileStorageDir) => {
	const baseDir = process.cwd();
	const isVerbose = !disableVerbose;
	const pythonValidation = await validatePythonEnvironment({
		baseDir,
		hasPythonFiles: [...getStepFiles(baseDir), ...getStreamFiles(baseDir)].some((file) => file.endsWith(".py"))
	});
	if (!pythonValidation.success) process.exit(1);
	if (pythonValidation.hasPythonFiles) {
		console.log("âš™ï¸ Activating Python environment...");
		activatePythonVenv({
			baseDir,
			isVerbose
		});
	}
	const motiaFileStoragePath = motiaFileStorageDir || ".motia";
	const dotMotia = path.join(baseDir, motiaFileStoragePath);
	const appConfig = await loadMotiaConfig(baseDir);
	const redisClient = await getRedisClient(dotMotia, appConfig);
	const adapters = {
		eventAdapter: appConfig.adapters?.events || new BullMQEventAdapter({ connection: getRedisConnectionInfo() }),
		cronAdapter: appConfig.adapters?.cron || new RedisCronAdapter(redisClient),
		streamAdapter: appConfig.adapters?.streams || new RedisStreamAdapterManager(redisClient)
	};
	const motiaServer = createServer(await generateLockedData({
		projectDir: baseDir,
		streamAdapter: adapters.streamAdapter,
		redisClient,
		streamAuth: appConfig.streamAuth
	}), appConfig.adapters?.state || new RedisStateAdapter(redisClient), {
		isVerbose,
		isDev: false,
		version
	}, adapters, appConfig.app);
	const plugins = await processPlugins(motiaServer);
	if (!process.env.MOTIA_DOCKER_DISABLE_WORKBENCH) {
		const { applyMiddleware } = await import("@motiadev/workbench/middleware");
		await applyMiddleware({
			app: motiaServer.app,
			port,
			workbenchBase,
			plugins: plugins.flatMap((item) => item.workbench)
		});
	}
	const actualPort = await listenWithFallback(motiaServer.server, port, hostname);
	if (actualPort !== port) console.log(`âš ï¸  Port ${port} was in use, using port ${actualPort} instead`);
	console.log("ðŸš€ Server ready and listening on port", actualPort);
	console.log(`ðŸ”— Open http://${hostname}:${actualPort}${workbenchBase} to open workbench ðŸ› ï¸`);
	process.on("SIGTERM", async () => {
		motiaServer.server.close();
		await stopRedisConnection();
		process.exit(0);
	});
	process.on("SIGINT", async () => {
		motiaServer.server.close();
		await stopRedisConnection();
		process.exit(0);
	});
};

//#endregion
export { start };
