// @motiadev/plugin-logs/dist/index.js
import { Button, Input, LevelDot, Sidebar, cn } from "@motiadev/ui";
import { useVirtualizer } from "@tanstack/react-virtual";
import { Filter, Search, Trash, X } from "lucide-react";
import { useMemo, useRef, useState } from "react";
import { c } from "react/compiler-runtime";
import { useStreamGroup } from "@motiadev/stream-client-react";
import { create } from "zustand";
import ReactJson from "react18-json-view";
import "react18-json-view/src/dark.css";
import "react18-json-view/src/style.css";
import { Fragment, jsx, jsxs } from "react/jsx-runtime";

//#region src/stores/use-logs-store.ts
const useLogsStore = create()((set) => ({
	logs: [],
	selectedLogId: void 0,
	setLogs: (logs) => set(() => {
		return { logs: [...Array.isArray(logs) ? logs : []].reverse() };
	}),
	resetLogs: () => {
		set({ logs: [] });
	},
	selectLogId: (logId) => set({ selectedLogId: logId })
}));

//#endregion
//#region src/hooks/use-logs-stream.ts
const streamName = "__motia.logs";
const groupId = "default";
const useLogsStream = () => {
	const $ = c(3);
	if ($[0] !== "07a885904ac486305cf3393f9436986fc94bf9cf48b0fc1eba70eadcdf49ee10") {
		for (let $i = 0; $i < 3; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "07a885904ac486305cf3393f9436986fc94bf9cf48b0fc1eba70eadcdf49ee10";
	}
	const setData = useLogsStore(_temp);
	let t0;
	if ($[1] !== setData) {
		t0 = {
			streamName,
			groupId,
			setData
		};
		$[1] = setData;
		$[2] = t0;
	} else t0 = $[2];
	useStreamGroup(t0);
};
function _temp(state) {
	return state.setLogs;
}

//#endregion
//#region src/utils/format-timestamp.ts
const formatTimestamp = (time) => {
	const date = new Date(Number(time));
	return `${date.toLocaleDateString("en-US", {
		year: void 0,
		month: "short",
		day: "2-digit"
	})}, ${date.toLocaleTimeString("en-US", {
		hour: "2-digit",
		minute: "2-digit",
		second: "2-digit",
		hourCycle: "h24"
	})}.${date.getMilliseconds().toString().padStart(3, "0")}`;
};

//#endregion
//#region src/components/log-detail.tsx
const defaultProps = [
	"id",
	"msg",
	"time",
	"level",
	"step",
	"flows",
	"traceId"
];
const LogDetail = ({ log, onClose }) => {
	const [hasOtherProps, setHasOtherProps] = useState(false);
	const otherPropsObject = useMemo(() => {
		if (!log) return null;
		const otherProps = Object.keys(log ?? {}).filter((key) => !defaultProps.includes(key));
		setHasOtherProps(otherProps.length > 0);
		return otherProps.reduce((acc, key_0) => {
			acc[key_0] = log[key_0];
			return acc;
		}, {});
	}, [log]);
	if (!log) return null;
	return /* @__PURE__ */ jsx(Sidebar, {
		onClose,
		title: "Logs Details",
		subtitle: "Details including custom properties",
		actions: [{
			icon: /* @__PURE__ */ jsx(X, {}),
			onClick: onClose,
			label: "Close"
		}],
		details: [
			{
				label: "Level",
				value: /* @__PURE__ */ jsxs("div", {
					className: "flex items-center gap-2",
					children: [/* @__PURE__ */ jsx(LevelDot, { level: log.level }), /* @__PURE__ */ jsx("div", {
						className: "capitalize",
						children: log.level
					})]
				})
			},
			{
				label: "Time",
				value: formatTimestamp(log.time)
			},
			{
				label: "Step",
				value: log.step
			},
			{
				label: "Flows",
				value: log.flows.join(", ")
			},
			{
				label: "Trace ID",
				value: log.traceId
			}
		],
		children: hasOtherProps && /* @__PURE__ */ jsx(ReactJson, {
			src: otherPropsObject,
			theme: "default",
			enableClipboard: true
		})
	});
};

//#endregion
//#region src/components/logs-page.tsx
const ROW_HEIGHT = 40;
const LogsPage = () => {
	useLogsStream();
	const logs = useLogsStore((state) => state.logs);
	const resetLogs = useLogsStore((state_0) => state_0.resetLogs);
	const selectedLogId = useLogsStore((state_1) => state_1.selectedLogId);
	const selectLogId = useLogsStore((state_2) => state_2.selectLogId);
	const selectedLog = useMemo(() => selectedLogId ? logs.find((log) => log.id === selectedLogId) : void 0, [logs, selectedLogId]);
	const [search, setSearch] = useState("");
	const filteredLogs = useMemo(() => {
		return logs.filter((log_0) => {
			return String(log_0.msg || "").toLowerCase().includes(search.toLowerCase()) || String(log_0.traceId || "").toLowerCase().includes(search.toLowerCase()) || String(log_0.step || "").toLowerCase().includes(search.toLowerCase());
		});
	}, [logs, search]);
	const scrollContainerRef = useRef(null);
	const virtualizer = useVirtualizer({
		count: filteredLogs.length,
		getScrollElement: () => scrollContainerRef.current,
		estimateSize: () => ROW_HEIGHT,
		overscan: 5
	});
	const virtualItems = virtualizer.getVirtualItems();
	return /* @__PURE__ */ jsxs(Fragment, { children: [/* @__PURE__ */ jsxs("div", {
		className: "grid grid-rows-[auto_1fr] h-full",
		"data-testid": "logs-container",
		children: [/* @__PURE__ */ jsxs("div", {
			className: "flex p-2 border-b gap-2",
			"data-testid": "logs-search-container",
			children: [/* @__PURE__ */ jsxs("div", {
				className: "flex-1 relative",
				children: [
					/* @__PURE__ */ jsx(Input, {
						variant: "shade",
						value: search,
						onChange: (e) => setSearch(e.target.value),
						className: "px-9! font-medium",
						placeholder: "Search by Trace ID or Message"
					}),
					/* @__PURE__ */ jsx(Search, { className: "absolute left-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground/50" }),
					/* @__PURE__ */ jsx(X, {
						className: "cursor-pointer absolute right-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground/50 hover:text-muted-foreground",
						onClick: () => setSearch("")
					})
				]
			}), /* @__PURE__ */ jsxs(Button, {
				variant: "default",
				onClick: resetLogs,
				className: "h-[34px]",
				children: [/* @__PURE__ */ jsx(Trash, {}), " Clear"]
			})]
		}), /* @__PURE__ */ jsx("div", {
			ref: scrollContainerRef,
			className: "overflow-auto h-full",
			children: /* @__PURE__ */ jsx("div", {
				className: "relative w-full",
				style: { height: virtualizer.getTotalSize() },
				children: virtualItems.map((virtualRow) => {
					const log_1 = filteredLogs[virtualRow.index];
					if (!log_1) return null;
					const index = virtualRow.index;
					return /* @__PURE__ */ jsxs("div", {
						"data-testid": "log-row",
						"data-index": virtualRow.index,
						ref: virtualizer.measureElement,
						role: "row",
						tabIndex: 0,
						className: cn("absolute left-0 w-full flex items-center font-mono font-semibold cursor-pointer text-sm", {
							"bg-muted-foreground/10 hover:bg-muted-foreground/20": selectedLogId === log_1.id,
							"hover:bg-muted-foreground/10": selectedLogId !== log_1.id
						}),
						style: {
							height: ROW_HEIGHT,
							transform: `translateY(${virtualRow.start}px)`
						},
						onClick: () => selectLogId(log_1.id),
						onKeyDown: (e_0) => {
							if (e_0.key === "Enter" || e_0.key === " ") selectLogId(log_1.id);
						},
						children: [
							/* @__PURE__ */ jsxs("div", {
								"data-testid": `time-${index}`,
								role: "cell",
								className: "whitespace-nowrap flex items-center gap-2 text-muted-foreground p-2 shrink-0",
								children: [/* @__PURE__ */ jsx(LevelDot, { level: log_1.level }), formatTimestamp(log_1.time)]
							}),
							/* @__PURE__ */ jsxs("div", {
								className: "flex items-center shrink-0",
								children: [/* @__PURE__ */ jsx("span", {
									"data-testid": `trace-${log_1.traceId}`,
									className: "whitespace-nowrap text-muted-foreground p-2 font-mono font-semibold text-sm",
									children: log_1.traceId
								}), /* @__PURE__ */ jsx("button", {
									type: "button",
									"data-testid": `trace-filter-${log_1.traceId}`,
									"aria-label": `Filter by trace ${log_1.traceId}`,
									title: `Filter by trace ${log_1.traceId}`,
									className: "p-1 rounded hover:bg-muted-foreground/20 text-muted-foreground hover:text-primary transition-colors cursor-pointer",
									onClick: (e_1) => {
										e_1.stopPropagation();
										setSearch(log_1.traceId);
									},
									children: /* @__PURE__ */ jsx(Filter, { className: "w-3 h-3" })
								})]
							}),
							/* @__PURE__ */ jsx("div", {
								"data-testid": `step-${index}`,
								role: "cell",
								title: log_1.step,
								className: "whitespace-nowrap p-2 shrink-0",
								children: log_1.step
							}),
							/* @__PURE__ */ jsx("div", {
								"data-testid": `msg-${index}`,
								role: "cell",
								title: log_1.msg,
								className: "whitespace-nowrap max-w-[500px] truncate p-2 flex-1",
								children: log_1.msg
							})
						]
					}, log_1.id);
				})
			})
		})]
	}), /* @__PURE__ */ jsx(LogDetail, {
		log: selectedLog,
		onClose: () => selectLogId(void 0)
	})] });
};

//#endregion
export { LogsPage };
