// motia/dist/cloud/new-deployment/listeners/build-printer.mjs
import { CLIOutputManager } from "../../cli-output-manager.mjs";
import { prettyBytes } from "../utils/pretty-bytes.mjs";
import { Printer } from "@motiadev/core";
import pc from "picocolors";

//#region src/cloud/new-deployment/listeners/build-printer.ts
const building = pc.yellow("➜ [BUILDING]");
const built = pc.green("✓ [BUILT]");
const failed = pc.red("✘ [FAILED]");
const skipped = pc.gray("- [SKIPPED]");
const baseTag = (tag) => pc.bold(pc.magenta(tag));
const streamTag = baseTag("Stream");
const stepTag = baseTag("Step");
const routerTag = baseTag("Router");
var BuildPrinter = class {
	constructor() {
		this.printer = new Printer(process.cwd());
		this.output = new CLIOutputManager();
	}
	getLanguage(language) {
		if (language === "python") return pc.bold(pc.blue("Python"));
		else if (language === "node") return pc.bold(pc.green("Node"));
		else if (language === "ruby") return pc.bold(pc.red("Ruby"));
		return pc.bold(pc.gray("Unknown"));
	}
	getStepLanguage(step) {
		if (step.filePath.endsWith(".py")) return this.getLanguage("python");
		else if (step.filePath.endsWith(".js") || step.filePath.endsWith(".ts")) return this.getLanguage("node");
		else if (step.filePath.endsWith(".rb")) return this.getLanguage("ruby");
		return this.getLanguage("unknown");
	}
	printStepBuilding(step, progressMessage) {
		const stepLanguage = this.getStepLanguage(step);
		const stepType = this.printer.getStepType(step);
		const stepPath = this.printer.getStepPath(step);
		this.output.log(step.filePath, (message) => {
			message.append(`${building} ${stepTag} ${stepLanguage} ${stepType} ${stepPath}`);
			if (progressMessage) message.append(pc.yellow(progressMessage));
		});
	}
	printStepBuilt(step, size) {
		const stepLanguage = this.getStepLanguage(step);
		const stepType = this.printer.getStepType(step);
		const stepPath = this.printer.getStepPath(step);
		this.output.log(step.filePath, (message) => message.append(`${built} ${stepTag} ${stepLanguage} ${stepType} ${stepPath}`).append(`${prettyBytes(size)}`, "gray"));
	}
	printApiRouterBuilding(language) {
		const fileName = `router-${language}.zip`;
		const coloredFileName = pc.bold(pc.cyan(fileName));
		this.output.log(fileName, (message) => message.append(routerTag).append(building).append(this.getLanguage(language)).append(coloredFileName));
	}
	printApiRouterBuilt(language, size) {
		const fileName = `router-${language}.zip`;
		const coloredFileName = pc.bold(pc.cyan(fileName));
		this.output.log(fileName, (message) => message.append(built).append(routerTag).append(this.getLanguage(language)).append(coloredFileName).append(prettyBytes(size), "gray"));
	}
	printStepFailed(step, error) {
		const stepLanguage = this.getStepLanguage(step);
		const stepType = this.printer.getStepType(step);
		const stepPath = this.printer.getStepPath(step);
		this.output.log(step.filePath, (message) => message.append(`${failed} ${stepTag} ${stepLanguage} ${stepType} ${stepPath}`).append(error.message, "red"));
	}
	printStepSkipped(step, reason) {
		const stepLanguage = this.getStepLanguage(step);
		const stepType = this.printer.getStepType(step);
		const stepPath = this.printer.getStepPath(step);
		this.output.log(step.filePath, (message) => message.append(`${skipped} ${stepTag} ${stepLanguage} ${stepType} ${stepPath}`).append(reason, "yellow"));
	}
	printStreamCreated(stream) {
		const streamPath = this.getStreamPath(stream);
		const streamName = pc.gray(`[${stream.config.name}]`);
		this.output.log(stream.filePath, (message) => message.append(`${built} ${streamTag} ${streamName} ${streamPath}`));
	}
	getStreamPath(stream) {
		return pc.bold(pc.cyan(this.printer.getRelativePath(stream.filePath)));
	}
};

//#endregion
export { BuildPrinter };
