// motia/dist/cloud/new-deployment/listeners/printer.mjs
import { CLIOutputManager } from "../../cli-output-manager.mjs";
import { projectDir } from "../constants.mjs";
import { Printer } from "@motiadev/core";
import pc from "picocolors";

//#region src/cloud/new-deployment/listeners/printer.ts
const uploading = pc.yellow("➜ [UPLOADING]");
const uploaded = pc.green("✓ [UPLOADED]");
var DeployPrinter = class {
	constructor() {
		this.printer = new Printer(projectDir);
		this.output = new CLIOutputManager();
	}
	getLanguage(language) {
		if (language === "node") return pc.bold(pc.green("Node"));
		else if (language === "python") return pc.bold(pc.blue("Python"));
		else if (language === "ruby") return pc.bold(pc.red("Ruby"));
		return pc.bold(pc.gray("Unknown"));
	}
	getStepLanguage(stepConfig) {
		if (stepConfig.filePath.endsWith(".py")) return this.getLanguage("python");
		else if (stepConfig.filePath.endsWith(".js") || stepConfig.filePath.endsWith(".ts")) return this.getLanguage("node");
		else if (stepConfig.filePath.endsWith(".rb")) return this.getLanguage("ruby");
		return this.getLanguage("unknown");
	}
	getStepType(stepConfig) {
		if (stepConfig.config.type === "api") return pc.gray("(API)");
		else if (stepConfig.config.type === "event") return pc.gray("(Event)");
		else if (stepConfig.config.type === "cron") return pc.gray("(Cron)");
		else if (stepConfig.config.type === "noop") return pc.gray("(Noop)");
		return pc.gray("(Unknown)");
	}
	getRelativePath(path) {
		return pc.bold(pc.cyan(this.printer.getRelativePath(path)));
	}
	getStepPath(stepConfig) {
		return this.getRelativePath(stepConfig.filePath);
	}
	printStepUploading(stepConfig) {
		const stepLanguage = this.getStepLanguage(stepConfig);
		const stepType = this.getStepType(stepConfig);
		const stepPath = this.getStepPath(stepConfig);
		const stepTag = this.printer.stepTag;
		this.output.log(stepConfig.filePath, (message) => {
			message.append(`${uploading} ${stepLanguage} ${stepTag} ${stepType} ${stepPath}`);
		});
	}
	printStepUploaded(stepConfig) {
		const stepLanguage = this.getStepLanguage(stepConfig);
		const stepType = this.getStepType(stepConfig);
		const stepPath = this.getStepPath(stepConfig);
		const stepTag = this.printer.stepTag;
		this.output.log(stepConfig.filePath, (message) => {
			message.append(`${uploaded} ${stepLanguage} ${stepTag} ${stepType} ${stepPath}`);
		});
	}
	printRouterUploading(language, routerPath) {
		const stepLanguage = this.getLanguage(language);
		const stepPath = this.getRelativePath(routerPath);
		this.output.log(routerPath, (message) => message.append(`${uploading} ${stepLanguage} ${stepPath}`));
	}
	printRouterUploaded(language, routerPath) {
		const stepLanguage = this.getLanguage(language);
		const stepPath = this.getRelativePath(routerPath);
		this.output.log(routerPath, (message) => message.append(`${uploaded} ${stepLanguage} ${stepPath}`));
	}
	printConfigurationUploading() {
		this.output.log("upload-config", (message) => message.tag("progress").append(`${uploading} Configuration`));
	}
	printConfigurationUploaded() {
		this.output.log("upload-config", (message) => message.tag("success").append(`${uploaded} Configuration`));
	}
};

//#endregion
export { DeployPrinter };
