// @motiadev/plugin-bullmq/dist/index.js
import { c } from "react/compiler-runtime";
import { memo, useCallback, useEffect, useState } from "react";
import { useStreamGroup } from "@motiadev/stream-client-react";
import { create } from "zustand";
import { QueryClient, QueryClientProvider, useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
import { jsx, jsxs } from "react/jsx-runtime";
import { Button, DropdownMenu, DropdownMenuContent, DropdownMenuItem, DropdownMenuSeparator, DropdownMenuTrigger, Input, Sidebar, Table, TableBody, TableCell, TableHead, TableHeader, TableRow, Tabs, TabsList, TabsTrigger, Tooltip, TooltipContent, TooltipProvider, TooltipTrigger, cn } from "@motiadev/ui";
import { formatDistanceToNow } from "date-fns";
import { AlertTriangle, ArrowUpRight, Layers, MoreVertical, Pause, Play, RefreshCw, Search, Skull, Trash, X } from "lucide-react";
import JsonView from "react18-json-view";
import "react18-json-view/src/style.css";

//#region src/stores/use-bullmq-store.ts
const initialState = {
	queues: [],
	selectedQueue: null,
	selectedJob: null,
	selectedStatus: "waiting",
	error: null,
	searchQuery: "",
	jobDetailOpen: false
};
const useBullMQStore = create((set) => ({
	...initialState,
	setQueues: (queues) => set({ queues }),
	setSelectedQueue: (queue) => set({
		selectedQueue: queue,
		selectedJob: null
	}),
	updateSelectedQueueStats: (queue) => set({ selectedQueue: queue }),
	setSelectedJob: (job) => set({ selectedJob: job }),
	setSelectedStatus: (status) => set({ selectedStatus: status }),
	setError: (error) => set({ error }),
	setSearchQuery: (query) => set({ searchQuery: query }),
	setJobDetailOpen: (open) => set({ jobDetailOpen: open }),
	reset: () => set(initialState)
}));

//#endregion
//#region src/hooks/use-queues.ts
const STREAM_NAME = "__motia.bullmq-queues";
const useQueues = () => {
	const { queues, setQueues, setError, error, selectedQueue, setSelectedQueue } = useBullMQStore();
	const { data: streamQueues } = useStreamGroup({
		streamName: STREAM_NAME,
		groupId: "default"
	});
	useEffect(() => {
		if (streamQueues.length > 0) {
			setQueues(streamQueues);
			if (selectedQueue) {
				const updatedQueue = streamQueues.find((q) => q.name === selectedQueue.name);
				if (updatedQueue) {
					const currentStats = JSON.stringify(selectedQueue.stats);
					const newStats = JSON.stringify(updatedQueue.stats);
					const pausedChanged = selectedQueue.isPaused !== updatedQueue.isPaused;
					if (currentStats !== newStats || pausedChanged) setSelectedQueue(updatedQueue);
				}
			}
		}
	}, [
		streamQueues,
		setQueues,
		selectedQueue,
		setSelectedQueue
	]);
	return {
		queues,
		error,
		refreshQueue: useCallback(async (name) => {
			try {
				const response = await fetch(`/__motia/bullmq/queues/${encodeURIComponent(name)}`);
				if (!response.ok) throw new Error("Failed to fetch queue");
				return await response.json();
			} catch {
				return null;
			}
		}, []),
		pauseQueue: useCallback(async (name_0) => {
			try {
				await fetch(`/__motia/bullmq/queues/${encodeURIComponent(name_0)}/pause`, { method: "POST" });
			} catch (err) {
				setError(err instanceof Error ? err.message : "Failed to pause queue");
			}
		}, [setError]),
		resumeQueue: useCallback(async (name_1) => {
			try {
				await fetch(`/__motia/bullmq/queues/${encodeURIComponent(name_1)}/resume`, { method: "POST" });
			} catch (err_0) {
				setError(err_0 instanceof Error ? err_0.message : "Failed to resume queue");
			}
		}, [setError]),
		cleanQueue: useCallback(async (name_2, status, grace = 0, limit = 1e3) => {
			try {
				await fetch(`/__motia/bullmq/queues/${encodeURIComponent(name_2)}/clean`, {
					method: "POST",
					headers: { "Content-Type": "application/json" },
					body: JSON.stringify({
						status,
						grace,
						limit
					})
				});
			} catch (err_1) {
				setError(err_1 instanceof Error ? err_1.message : "Failed to clean queue");
			}
		}, [setError]),
		drainQueue: useCallback(async (name_3) => {
			try {
				await fetch(`/__motia/bullmq/queues/${encodeURIComponent(name_3)}/drain`, { method: "POST" });
			} catch (err_2) {
				setError(err_2 instanceof Error ? err_2.message : "Failed to drain queue");
			}
		}, [setError])
	};
};

//#endregion
//#region src/providers/query-provider.tsx
const queryClient = new QueryClient({ defaultOptions: { queries: {
	staleTime: 5e3,
	refetchOnWindowFocus: false,
	retry: 1
} } });
const QueryProvider = (t0) => {
	const $ = c(3);
	if ($[0] !== "72268343e8809a517b650d94aea39cf9f299784a2a89cdc3b8e9a0d84a7244b8") {
		for (let $i = 0; $i < 3; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "72268343e8809a517b650d94aea39cf9f299784a2a89cdc3b8e9a0d84a7244b8";
	}
	const { children } = t0;
	let t1;
	if ($[1] !== children) {
		t1 = /* @__PURE__ */ jsx(QueryClientProvider, {
			client: queryClient,
			children
		});
		$[1] = children;
		$[2] = t1;
	} else t1 = $[2];
	return t1;
};

//#endregion
//#region src/hooks/use-jobs-mutations.ts
const retryJobFn = async ({ queueName, jobId }) => {
	if (!(await fetch(`/__motia/bullmq/queues/${encodeURIComponent(queueName)}/jobs/${encodeURIComponent(jobId)}/retry`, { method: "POST" })).ok) throw new Error("Failed to retry job");
};
const removeJobFn = async ({ queueName, jobId }) => {
	if (!(await fetch(`/__motia/bullmq/queues/${encodeURIComponent(queueName)}/jobs/${encodeURIComponent(jobId)}/remove`, { method: "POST" })).ok) throw new Error("Failed to remove job");
};
const promoteJobFn = async ({ queueName, jobId }) => {
	if (!(await fetch(`/__motia/bullmq/queues/${encodeURIComponent(queueName)}/jobs/${encodeURIComponent(jobId)}/promote`, { method: "POST" })).ok) throw new Error("Failed to promote job");
};
const retryFromDLQFn = async ({ queueName, jobId }) => {
	if (!(await fetch(`/__motia/bullmq/dlq/${encodeURIComponent(queueName)}/retry/${encodeURIComponent(jobId)}`, { method: "POST" })).ok) throw new Error("Failed to retry from DLQ");
};
const retryAllFromDLQFn = async ({ queueName }) => {
	if (!(await fetch(`/__motia/bullmq/dlq/${encodeURIComponent(queueName)}/retry-all`, { method: "POST" })).ok) throw new Error("Failed to retry all from DLQ");
};
const clearDLQFn = async ({ queueName }) => {
	if (!(await fetch(`/__motia/bullmq/dlq/${encodeURIComponent(queueName)}/clear`, { method: "POST" })).ok) throw new Error("Failed to clear DLQ");
};
const useRetryJob = () => {
	const $ = c(8);
	if ($[0] !== "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177") {
		for (let $i = 0; $i < 8; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177";
	}
	const queryClient$1 = useQueryClient();
	const setError = useBullMQStore(_temp$7);
	let t0;
	if ($[1] !== queryClient$1) {
		t0 = (_, t1$1) => {
			const { queueName } = t1$1;
			queryClient$1.invalidateQueries({ queryKey: ["jobs", queueName] });
		};
		$[1] = queryClient$1;
		$[2] = t0;
	} else t0 = $[2];
	let t1;
	if ($[3] !== setError) {
		t1 = (error) => {
			setError(error instanceof Error ? error.message : "Failed to retry job");
		};
		$[3] = setError;
		$[4] = t1;
	} else t1 = $[4];
	let t2;
	if ($[5] !== t0 || $[6] !== t1) {
		t2 = {
			mutationFn: retryJobFn,
			onSuccess: t0,
			onError: t1
		};
		$[5] = t0;
		$[6] = t1;
		$[7] = t2;
	} else t2 = $[7];
	return useMutation(t2);
};
const useRemoveJob = () => {
	const $ = c(8);
	if ($[0] !== "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177") {
		for (let $i = 0; $i < 8; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177";
	}
	const queryClient$1 = useQueryClient();
	const setError = useBullMQStore(_temp2$7);
	let t0;
	if ($[1] !== queryClient$1) {
		t0 = (_, t1$1) => {
			const { queueName } = t1$1;
			queryClient$1.invalidateQueries({ queryKey: ["jobs", queueName] });
		};
		$[1] = queryClient$1;
		$[2] = t0;
	} else t0 = $[2];
	let t1;
	if ($[3] !== setError) {
		t1 = (error) => {
			setError(error instanceof Error ? error.message : "Failed to remove job");
		};
		$[3] = setError;
		$[4] = t1;
	} else t1 = $[4];
	let t2;
	if ($[5] !== t0 || $[6] !== t1) {
		t2 = {
			mutationFn: removeJobFn,
			onSuccess: t0,
			onError: t1
		};
		$[5] = t0;
		$[6] = t1;
		$[7] = t2;
	} else t2 = $[7];
	return useMutation(t2);
};
const usePromoteJob = () => {
	const $ = c(8);
	if ($[0] !== "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177") {
		for (let $i = 0; $i < 8; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177";
	}
	const queryClient$1 = useQueryClient();
	const setError = useBullMQStore(_temp3$4);
	let t0;
	if ($[1] !== queryClient$1) {
		t0 = (_, t1$1) => {
			const { queueName } = t1$1;
			queryClient$1.invalidateQueries({ queryKey: ["jobs", queueName] });
		};
		$[1] = queryClient$1;
		$[2] = t0;
	} else t0 = $[2];
	let t1;
	if ($[3] !== setError) {
		t1 = (error) => {
			setError(error instanceof Error ? error.message : "Failed to promote job");
		};
		$[3] = setError;
		$[4] = t1;
	} else t1 = $[4];
	let t2;
	if ($[5] !== t0 || $[6] !== t1) {
		t2 = {
			mutationFn: promoteJobFn,
			onSuccess: t0,
			onError: t1
		};
		$[5] = t0;
		$[6] = t1;
		$[7] = t2;
	} else t2 = $[7];
	return useMutation(t2);
};
const useRetryFromDLQ = () => {
	const $ = c(8);
	if ($[0] !== "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177") {
		for (let $i = 0; $i < 8; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177";
	}
	const queryClient$1 = useQueryClient();
	const setError = useBullMQStore(_temp4$3);
	let t0;
	if ($[1] !== queryClient$1) {
		t0 = (_, t1$1) => {
			const { queueName } = t1$1;
			queryClient$1.invalidateQueries({ queryKey: ["dlq-jobs", queueName] });
		};
		$[1] = queryClient$1;
		$[2] = t0;
	} else t0 = $[2];
	let t1;
	if ($[3] !== setError) {
		t1 = (error) => {
			setError(error instanceof Error ? error.message : "Failed to retry from DLQ");
		};
		$[3] = setError;
		$[4] = t1;
	} else t1 = $[4];
	let t2;
	if ($[5] !== t0 || $[6] !== t1) {
		t2 = {
			mutationFn: retryFromDLQFn,
			onSuccess: t0,
			onError: t1
		};
		$[5] = t0;
		$[6] = t1;
		$[7] = t2;
	} else t2 = $[7];
	return useMutation(t2);
};
const useRetryAllFromDLQ = () => {
	const $ = c(8);
	if ($[0] !== "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177") {
		for (let $i = 0; $i < 8; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177";
	}
	const queryClient$1 = useQueryClient();
	const setError = useBullMQStore(_temp5$3);
	let t0;
	if ($[1] !== queryClient$1) {
		t0 = (_, t1$1) => {
			const { queueName } = t1$1;
			queryClient$1.invalidateQueries({ queryKey: ["dlq-jobs", queueName] });
		};
		$[1] = queryClient$1;
		$[2] = t0;
	} else t0 = $[2];
	let t1;
	if ($[3] !== setError) {
		t1 = (error) => {
			setError(error instanceof Error ? error.message : "Failed to retry all from DLQ");
		};
		$[3] = setError;
		$[4] = t1;
	} else t1 = $[4];
	let t2;
	if ($[5] !== t0 || $[6] !== t1) {
		t2 = {
			mutationFn: retryAllFromDLQFn,
			onSuccess: t0,
			onError: t1
		};
		$[5] = t0;
		$[6] = t1;
		$[7] = t2;
	} else t2 = $[7];
	return useMutation(t2);
};
const useClearDLQ = () => {
	const $ = c(8);
	if ($[0] !== "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177") {
		for (let $i = 0; $i < 8; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dda4e584b854b7387aa6e222ff1915ce152f6bfdf2fc4ff1beda4b29692e9177";
	}
	const queryClient$1 = useQueryClient();
	const setError = useBullMQStore(_temp6$1);
	let t0;
	if ($[1] !== queryClient$1) {
		t0 = (_, t1$1) => {
			const { queueName } = t1$1;
			queryClient$1.invalidateQueries({ queryKey: ["dlq-jobs", queueName] });
		};
		$[1] = queryClient$1;
		$[2] = t0;
	} else t0 = $[2];
	let t1;
	if ($[3] !== setError) {
		t1 = (error) => {
			setError(error instanceof Error ? error.message : "Failed to clear DLQ");
		};
		$[3] = setError;
		$[4] = t1;
	} else t1 = $[4];
	let t2;
	if ($[5] !== t0 || $[6] !== t1) {
		t2 = {
			mutationFn: clearDLQFn,
			onSuccess: t0,
			onError: t1
		};
		$[5] = t0;
		$[6] = t1;
		$[7] = t2;
	} else t2 = $[7];
	return useMutation(t2);
};
function _temp$7(state) {
	return state.setError;
}
function _temp2$7(state) {
	return state.setError;
}
function _temp3$4(state) {
	return state.setError;
}
function _temp4$3(state) {
	return state.setError;
}
function _temp5$3(state) {
	return state.setError;
}
function _temp6$1(state) {
	return state.setError;
}

//#endregion
//#region src/hooks/use-jobs-query.ts
const fetchJobs = async (queueName, status, start = 0, end = 100) => {
	const params = new URLSearchParams({
		status,
		start: String(start),
		end: String(end)
	});
	const response = await fetch(`/__motia/bullmq/queues/${encodeURIComponent(queueName)}/jobs?${params}`);
	if (!response.ok) throw new Error("Failed to fetch jobs");
	return (await response.json()).jobs;
};
const fetchDLQJobs = async (queueName, start = 0, end = 100) => {
	const params = new URLSearchParams({
		start: String(start),
		end: String(end)
	});
	const response = await fetch(`/__motia/bullmq/dlq/${encodeURIComponent(queueName)}/jobs?${params}`);
	if (!response.ok) return [];
	return (await response.json()).jobs;
};
const useJobsQuery = () => {
	const $ = c(14);
	if ($[0] !== "62345f058f3013b9a25310d4c1435bcda4709648cddbee0fee4472c79e98da42") {
		for (let $i = 0; $i < 14; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "62345f058f3013b9a25310d4c1435bcda4709648cddbee0fee4472c79e98da42";
	}
	const selectedQueue = useBullMQStore(_temp$6);
	const selectedStatus = useBullMQStore(_temp2$6);
	const queueName = selectedQueue?.name;
	let t0;
	if ($[1] !== selectedQueue) {
		t0 = selectedQueue ? JSON.stringify(selectedQueue.stats) : null;
		$[1] = selectedQueue;
		$[2] = t0;
	} else t0 = $[2];
	const statsKey = t0;
	let t1;
	if ($[3] !== queueName || $[4] !== selectedStatus || $[5] !== statsKey) {
		t1 = [
			"jobs",
			queueName,
			selectedStatus,
			statsKey
		];
		$[3] = queueName;
		$[4] = selectedStatus;
		$[5] = statsKey;
		$[6] = t1;
	} else t1 = $[6];
	let t2;
	if ($[7] !== queueName || $[8] !== selectedStatus) {
		t2 = () => fetchJobs(queueName, selectedStatus);
		$[7] = queueName;
		$[8] = selectedStatus;
		$[9] = t2;
	} else t2 = $[9];
	const t3 = !!queueName;
	let t4;
	if ($[10] !== t1 || $[11] !== t2 || $[12] !== t3) {
		t4 = {
			queryKey: t1,
			queryFn: t2,
			enabled: t3,
			staleTime: 5e3
		};
		$[10] = t1;
		$[11] = t2;
		$[12] = t3;
		$[13] = t4;
	} else t4 = $[13];
	return useQuery(t4);
};
const useDLQJobsQuery = (queueName) => {
	const $ = c(8);
	if ($[0] !== "62345f058f3013b9a25310d4c1435bcda4709648cddbee0fee4472c79e98da42") {
		for (let $i = 0; $i < 8; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "62345f058f3013b9a25310d4c1435bcda4709648cddbee0fee4472c79e98da42";
	}
	let t0;
	let t1;
	if ($[1] !== queueName) {
		t0 = ["dlq-jobs", queueName];
		t1 = () => fetchDLQJobs(queueName);
		$[1] = queueName;
		$[2] = t0;
		$[3] = t1;
	} else {
		t0 = $[2];
		t1 = $[3];
	}
	const t2 = !!queueName;
	let t3;
	if ($[4] !== t0 || $[5] !== t1 || $[6] !== t2) {
		t3 = {
			queryKey: t0,
			queryFn: t1,
			enabled: t2
		};
		$[4] = t0;
		$[5] = t1;
		$[6] = t2;
		$[7] = t3;
	} else t3 = $[7];
	return useQuery(t3);
};
function _temp$6(state) {
	return state.selectedQueue;
}
function _temp2$6(state_0) {
	return state_0.selectedStatus;
}

//#endregion
//#region src/components/dlq-panel.tsx
const DLQJobDetailContent = memo((t0) => {
	const $ = c(30);
	if ($[0] !== "9c642c700580e607dc6114475cae1bdbb21b1057857ec3c5d1e46ac6c107027a") {
		for (let $i = 0; $i < 30; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "9c642c700580e607dc6114475cae1bdbb21b1057857ec3c5d1e46ac6c107027a";
	}
	const { job, onRetry } = t0;
	let t1;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = /* @__PURE__ */ jsx(RefreshCw, { className: "mr-2 h-4 w-4" });
		$[1] = t1;
	} else t1 = $[1];
	let t2;
	if ($[2] !== onRetry) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: "flex gap-2",
			children: /* @__PURE__ */ jsxs(Button, {
				variant: "outline",
				size: "sm",
				onClick: onRetry,
				children: [t1, "Retry"]
			})
		});
		$[2] = onRetry;
		$[3] = t2;
	} else t2 = $[3];
	let t3;
	if ($[4] === Symbol.for("react.memo_cache_sentinel")) {
		t3 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Job ID"
		});
		$[4] = t3;
	} else t3 = $[4];
	let t4;
	if ($[5] !== job.id) {
		t4 = /* @__PURE__ */ jsxs("div", { children: [t3, /* @__PURE__ */ jsx("div", {
			className: "font-mono text-xs",
			children: job.id
		})] });
		$[5] = job.id;
		$[6] = t4;
	} else t4 = $[6];
	let t5;
	if ($[7] === Symbol.for("react.memo_cache_sentinel")) {
		t5 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Original Job ID"
		});
		$[7] = t5;
	} else t5 = $[7];
	const t6 = job.originalJobId || "-";
	let t7;
	if ($[8] !== t6) {
		t7 = /* @__PURE__ */ jsxs("div", { children: [t5, /* @__PURE__ */ jsx("div", {
			className: "font-mono text-xs",
			children: t6
		})] });
		$[8] = t6;
		$[9] = t7;
	} else t7 = $[9];
	let t8;
	if ($[10] === Symbol.for("react.memo_cache_sentinel")) {
		t8 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Attempts Made"
		});
		$[10] = t8;
	} else t8 = $[10];
	let t9;
	if ($[11] !== job.attemptsMade) {
		t9 = /* @__PURE__ */ jsxs("div", { children: [t8, /* @__PURE__ */ jsx("div", {
			className: "font-semibold",
			children: job.attemptsMade
		})] });
		$[11] = job.attemptsMade;
		$[12] = t9;
	} else t9 = $[12];
	let t10;
	if ($[13] === Symbol.for("react.memo_cache_sentinel")) {
		t10 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Failed At"
		});
		$[13] = t10;
	} else t10 = $[13];
	let t11;
	if ($[14] !== job.failureTimestamp) {
		t11 = formatDistanceToNow(job.failureTimestamp, { addSuffix: true });
		$[14] = job.failureTimestamp;
		$[15] = t11;
	} else t11 = $[15];
	let t12;
	if ($[16] !== t11) {
		t12 = /* @__PURE__ */ jsxs("div", { children: [t10, /* @__PURE__ */ jsx("div", { children: t11 })] });
		$[16] = t11;
		$[17] = t12;
	} else t12 = $[17];
	let t13;
	if ($[18] !== t12 || $[19] !== t4 || $[20] !== t7 || $[21] !== t9) {
		t13 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-2 gap-4 text-sm",
			children: [
				t4,
				t7,
				t9,
				t12
			]
		});
		$[18] = t12;
		$[19] = t4;
		$[20] = t7;
		$[21] = t9;
		$[22] = t13;
	} else t13 = $[22];
	let t14;
	if ($[23] === Symbol.for("react.memo_cache_sentinel")) {
		t14 = /* @__PURE__ */ jsx("div", {
			className: "text-sm font-semibold text-destructive mb-2",
			children: "Failure Reason"
		});
		$[23] = t14;
	} else t14 = $[23];
	let t15;
	if ($[24] !== job.failureReason) {
		t15 = /* @__PURE__ */ jsxs("div", { children: [t14, /* @__PURE__ */ jsx("div", {
			className: "font-mono text-sm bg-destructive/10 p-3 rounded text-destructive",
			children: job.failureReason
		})] });
		$[24] = job.failureReason;
		$[25] = t15;
	} else t15 = $[25];
	let t16;
	if ($[26] !== t13 || $[27] !== t15 || $[28] !== t2) {
		t16 = /* @__PURE__ */ jsxs("div", {
			className: "space-y-4",
			children: [
				t2,
				t13,
				t15
			]
		});
		$[26] = t13;
		$[27] = t15;
		$[28] = t2;
		$[29] = t16;
	} else t16 = $[29];
	return t16;
});
DLQJobDetailContent.displayName = "DLQJobDetailContent";
const DLQJobDataTab = memo((t0) => {
	const $ = c(3);
	if ($[0] !== "9c642c700580e607dc6114475cae1bdbb21b1057857ec3c5d1e46ac6c107027a") {
		for (let $i = 0; $i < 3; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "9c642c700580e607dc6114475cae1bdbb21b1057857ec3c5d1e46ac6c107027a";
	}
	const { data } = t0;
	const t1 = data;
	let t2;
	if ($[1] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: "bg-muted/30 p-4 rounded overflow-auto max-h-[400px]",
			children: /* @__PURE__ */ jsx(JsonView, {
				src: t1,
				theme: "atom",
				collapsed: 2
			})
		});
		$[1] = t1;
		$[2] = t2;
	} else t2 = $[2];
	return t2;
});
DLQJobDataTab.displayName = "DLQJobDataTab";
const DLQPanel = memo(() => {
	const $ = c(66);
	if ($[0] !== "9c642c700580e607dc6114475cae1bdbb21b1057857ec3c5d1e46ac6c107027a") {
		for (let $i = 0; $i < 66; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "9c642c700580e607dc6114475cae1bdbb21b1057857ec3c5d1e46ac6c107027a";
	}
	const selectedQueue = useBullMQStore(_temp$5);
	const { data: t0, isLoading, refetch } = useDLQJobsQuery(selectedQueue?.isDLQ ? selectedQueue.name : void 0);
	let t1;
	if ($[1] !== t0) {
		t1 = t0 === void 0 ? [] : t0;
		$[1] = t0;
		$[2] = t1;
	} else t1 = $[2];
	const dlqJobs = t1;
	const retryFromDLQMutation = useRetryFromDLQ();
	const retryAllFromDLQMutation = useRetryAllFromDLQ();
	const clearDLQMutation = useClearDLQ();
	const [selectedDlqJob, setSelectedDlqJob] = useState(null);
	let t2;
	if ($[3] !== retryFromDLQMutation || $[4] !== selectedQueue) {
		t2 = (jobId) => {
			if (!selectedQueue) return;
			retryFromDLQMutation.mutate({
				queueName: selectedQueue.name,
				jobId
			});
		};
		$[3] = retryFromDLQMutation;
		$[4] = selectedQueue;
		$[5] = t2;
	} else t2 = $[5];
	const handleRetry = t2;
	let t3;
	if ($[6] !== retryAllFromDLQMutation || $[7] !== selectedQueue) {
		t3 = () => {
			if (!selectedQueue) return;
			retryAllFromDLQMutation.mutate({ queueName: selectedQueue.name });
		};
		$[6] = retryAllFromDLQMutation;
		$[7] = selectedQueue;
		$[8] = t3;
	} else t3 = $[8];
	const handleRetryAll = t3;
	let t4;
	if ($[9] !== clearDLQMutation || $[10] !== selectedQueue) {
		t4 = () => {
			if (!selectedQueue) return;
			clearDLQMutation.mutate({ queueName: selectedQueue.name });
		};
		$[9] = clearDLQMutation;
		$[10] = selectedQueue;
		$[11] = t4;
	} else t4 = $[11];
	const handleClear = t4;
	let t5;
	if ($[12] === Symbol.for("react.memo_cache_sentinel")) {
		t5 = () => {
			setSelectedDlqJob(null);
		};
		$[12] = t5;
	} else t5 = $[12];
	const handleCloseSidePanel = t5;
	let t6;
	if ($[13] !== handleRetry || $[14] !== selectedDlqJob) {
		t6 = () => {
			if (selectedDlqJob) {
				handleRetry(selectedDlqJob.id);
				setSelectedDlqJob(null);
			}
		};
		$[13] = handleRetry;
		$[14] = selectedDlqJob;
		$[15] = t6;
	} else t6 = $[15];
	const handleRetryAndClose = t6;
	if (!selectedQueue?.isDLQ) return null;
	let t7;
	if ($[16] !== selectedQueue.displayName) {
		t7 = /* @__PURE__ */ jsx("h2", {
			className: "font-semibold text-lg",
			children: selectedQueue.displayName
		});
		$[16] = selectedQueue.displayName;
		$[17] = t7;
	} else t7 = $[17];
	const t8 = dlqJobs.length !== 1 ? "s" : "";
	let t9;
	if ($[18] !== dlqJobs.length || $[19] !== t8) {
		t9 = /* @__PURE__ */ jsxs("p", {
			className: "text-sm text-muted-foreground",
			children: [
				dlqJobs.length,
				" failed job",
				t8,
				" in dead letter queue"
			]
		});
		$[18] = dlqJobs.length;
		$[19] = t8;
		$[20] = t9;
	} else t9 = $[20];
	let t10;
	if ($[21] !== t7 || $[22] !== t9) {
		t10 = /* @__PURE__ */ jsxs("div", { children: [t7, t9] });
		$[21] = t7;
		$[22] = t9;
		$[23] = t10;
	} else t10 = $[23];
	let t11;
	if ($[24] !== refetch) {
		t11 = () => refetch();
		$[24] = refetch;
		$[25] = t11;
	} else t11 = $[25];
	const t12 = `h-4 w-4 ${isLoading ? "animate-spin" : ""}`;
	let t13;
	if ($[26] !== t12) {
		t13 = /* @__PURE__ */ jsx(RefreshCw, { className: t12 });
		$[26] = t12;
		$[27] = t13;
	} else t13 = $[27];
	let t14;
	if ($[28] !== isLoading || $[29] !== t11 || $[30] !== t13) {
		t14 = /* @__PURE__ */ jsx(TooltipTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsx(Button, {
				variant: "ghost",
				size: "icon",
				onClick: t11,
				disabled: isLoading,
				children: t13
			})
		});
		$[28] = isLoading;
		$[29] = t11;
		$[30] = t13;
		$[31] = t14;
	} else t14 = $[31];
	let t15;
	if ($[32] === Symbol.for("react.memo_cache_sentinel")) {
		t15 = /* @__PURE__ */ jsx(TooltipContent, { children: /* @__PURE__ */ jsx("p", { children: "Refresh DLQ jobs list" }) });
		$[32] = t15;
	} else t15 = $[32];
	let t16;
	if ($[33] !== t14) {
		t16 = /* @__PURE__ */ jsxs(Tooltip, { children: [t14, t15] });
		$[33] = t14;
		$[34] = t16;
	} else t16 = $[34];
	const t17 = dlqJobs.length === 0;
	let t18;
	if ($[35] === Symbol.for("react.memo_cache_sentinel")) {
		t18 = /* @__PURE__ */ jsx(RefreshCw, { className: "mr-2 h-4 w-4" });
		$[35] = t18;
	} else t18 = $[35];
	let t19;
	if ($[36] !== handleRetryAll || $[37] !== t17) {
		t19 = /* @__PURE__ */ jsx(TooltipTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsxs(Button, {
				variant: "outline",
				size: "sm",
				onClick: handleRetryAll,
				disabled: t17,
				children: [t18, "Retry All"]
			})
		});
		$[36] = handleRetryAll;
		$[37] = t17;
		$[38] = t19;
	} else t19 = $[38];
	let t20;
	if ($[39] === Symbol.for("react.memo_cache_sentinel")) {
		t20 = /* @__PURE__ */ jsx(TooltipContent, { children: /* @__PURE__ */ jsx("p", { children: "Re-queue all failed jobs to the original queue" }) });
		$[39] = t20;
	} else t20 = $[39];
	let t21;
	if ($[40] !== t19) {
		t21 = /* @__PURE__ */ jsxs(Tooltip, { children: [t19, t20] });
		$[40] = t19;
		$[41] = t21;
	} else t21 = $[41];
	const t22 = dlqJobs.length === 0;
	let t23;
	if ($[42] === Symbol.for("react.memo_cache_sentinel")) {
		t23 = /* @__PURE__ */ jsx(Trash, { className: "mr-2 h-4 w-4" });
		$[42] = t23;
	} else t23 = $[42];
	let t24;
	if ($[43] !== handleClear || $[44] !== t22) {
		t24 = /* @__PURE__ */ jsx(TooltipTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsxs(Button, {
				variant: "outline",
				size: "sm",
				onClick: handleClear,
				disabled: t22,
				className: "text-destructive",
				children: [t23, "Clear All"]
			})
		});
		$[43] = handleClear;
		$[44] = t22;
		$[45] = t24;
	} else t24 = $[45];
	let t25;
	if ($[46] === Symbol.for("react.memo_cache_sentinel")) {
		t25 = /* @__PURE__ */ jsx(TooltipContent, { children: /* @__PURE__ */ jsx("p", { children: "Permanently delete all jobs from DLQ" }) });
		$[46] = t25;
	} else t25 = $[46];
	let t26;
	if ($[47] !== t24) {
		t26 = /* @__PURE__ */ jsxs(Tooltip, { children: [t24, t25] });
		$[47] = t24;
		$[48] = t26;
	} else t26 = $[48];
	let t27;
	if ($[49] !== t16 || $[50] !== t21 || $[51] !== t26) {
		t27 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center gap-2",
			children: [
				t16,
				t21,
				t26
			]
		});
		$[49] = t16;
		$[50] = t21;
		$[51] = t26;
		$[52] = t27;
	} else t27 = $[52];
	let t28;
	if ($[53] !== t10 || $[54] !== t27) {
		t28 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center justify-between p-3 border-b border-border",
			children: [t10, t27]
		});
		$[53] = t10;
		$[54] = t27;
		$[55] = t28;
	} else t28 = $[55];
	let t29;
	if ($[56] !== dlqJobs || $[57] !== handleRetry) {
		t29 = dlqJobs.length === 0 ? /* @__PURE__ */ jsx("div", {
			className: "flex items-center justify-center flex-1 text-muted-foreground",
			children: "No jobs in dead letter queue"
		}) : /* @__PURE__ */ jsx("div", {
			className: "flex-1 overflow-auto",
			children: /* @__PURE__ */ jsxs(Table, { children: [/* @__PURE__ */ jsx(TableHeader, {
				className: "sticky top-0 bg-background/95 backdrop-blur-sm",
				children: /* @__PURE__ */ jsxs(TableRow, { children: [
					/* @__PURE__ */ jsx(TableHead, {
						className: "w-[180px]",
						children: "Job ID"
					}),
					/* @__PURE__ */ jsx(TableHead, {
						className: "w-[180px]",
						children: "Original Job"
					}),
					/* @__PURE__ */ jsx(TableHead, { children: "Failure Reason" }),
					/* @__PURE__ */ jsx(TableHead, {
						className: "w-[100px]",
						children: "Attempts"
					}),
					/* @__PURE__ */ jsx(TableHead, {
						className: "w-[150px]",
						children: "Failed At"
					}),
					/* @__PURE__ */ jsx(TableHead, {
						className: "w-[100px]",
						children: "Actions"
					})
				] })
			}), /* @__PURE__ */ jsx(TableBody, { children: dlqJobs.map((job) => /* @__PURE__ */ jsxs(TableRow, {
				className: "cursor-pointer hover:bg-muted-foreground/10",
				onClick: () => setSelectedDlqJob(job),
				children: [
					/* @__PURE__ */ jsx(TableCell, {
						className: "font-mono text-xs",
						children: job.id
					}),
					/* @__PURE__ */ jsx(TableCell, {
						className: "font-mono text-xs text-muted-foreground",
						children: job.originalJobId || "-"
					}),
					/* @__PURE__ */ jsx(TableCell, {
						className: "text-destructive text-sm truncate max-w-[300px]",
						children: job.failureReason
					}),
					/* @__PURE__ */ jsx(TableCell, { children: job.attemptsMade }),
					/* @__PURE__ */ jsx(TableCell, {
						className: "text-xs text-muted-foreground",
						children: formatDistanceToNow(job.failureTimestamp, { addSuffix: true })
					}),
					/* @__PURE__ */ jsx(TableCell, {
						onClick: _temp2$5,
						children: /* @__PURE__ */ jsxs(Tooltip, { children: [/* @__PURE__ */ jsx(TooltipTrigger, {
							asChild: true,
							children: /* @__PURE__ */ jsx(Button, {
								variant: "ghost",
								size: "sm",
								onClick: () => handleRetry(job.id),
								children: /* @__PURE__ */ jsx(RefreshCw, { className: "h-4 w-4" })
							})
						}), /* @__PURE__ */ jsx(TooltipContent, { children: /* @__PURE__ */ jsx("p", { children: "Retry this job" }) })] })
					})
				]
			}, job.id)) })] })
		});
		$[56] = dlqJobs;
		$[57] = handleRetry;
		$[58] = t29;
	} else t29 = $[58];
	let t30;
	if ($[59] !== handleRetryAndClose || $[60] !== selectedDlqJob) {
		t30 = selectedDlqJob && /* @__PURE__ */ jsx(Sidebar, {
			onClose: handleCloseSidePanel,
			title: "Dead Letter Job",
			initialWidth: 600,
			tabs: [{
				label: "Details",
				content: /* @__PURE__ */ jsx(DLQJobDetailContent, {
					job: selectedDlqJob,
					onRetry: handleRetryAndClose
				})
			}, {
				label: "Event Data",
				content: /* @__PURE__ */ jsx(DLQJobDataTab, { data: selectedDlqJob.originalEvent })
			}],
			actions: [{
				icon: /* @__PURE__ */ jsx(X, {}),
				onClick: handleCloseSidePanel,
				label: "Close"
			}]
		});
		$[59] = handleRetryAndClose;
		$[60] = selectedDlqJob;
		$[61] = t30;
	} else t30 = $[61];
	let t31;
	if ($[62] !== t28 || $[63] !== t29 || $[64] !== t30) {
		t31 = /* @__PURE__ */ jsx(TooltipProvider, { children: /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col h-full",
			children: [
				t28,
				t29,
				t30
			]
		}) });
		$[62] = t28;
		$[63] = t29;
		$[64] = t30;
		$[65] = t31;
	} else t31 = $[65];
	return t31;
});
DLQPanel.displayName = "DLQPanel";
function _temp$5(state) {
	return state.selectedQueue;
}
function _temp2$5(e) {
	return e.stopPropagation();
}

//#endregion
//#region src/components/job-detail.tsx
const JobDataTab = memo((t0) => {
	const $ = c(3);
	if ($[0] !== "3b06a00e91af27c8dc4dbe23232bee1ad67432d0e4c5ed3ab9b136e6d2a7a3cd") {
		for (let $i = 0; $i < 3; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "3b06a00e91af27c8dc4dbe23232bee1ad67432d0e4c5ed3ab9b136e6d2a7a3cd";
	}
	const { data } = t0;
	const t1 = data;
	let t2;
	if ($[1] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: "bg-muted/30 p-4 rounded",
			children: /* @__PURE__ */ jsx(JsonView, {
				src: t1,
				theme: "atom",
				collapsed: 2
			})
		});
		$[1] = t1;
		$[2] = t2;
	} else t2 = $[2];
	return t2;
});
JobDataTab.displayName = "JobDataTab";
const JobOptionsTab = memo((t0) => {
	const $ = c(3);
	if ($[0] !== "3b06a00e91af27c8dc4dbe23232bee1ad67432d0e4c5ed3ab9b136e6d2a7a3cd") {
		for (let $i = 0; $i < 3; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "3b06a00e91af27c8dc4dbe23232bee1ad67432d0e4c5ed3ab9b136e6d2a7a3cd";
	}
	const { opts } = t0;
	let t1;
	if ($[1] !== opts) {
		t1 = /* @__PURE__ */ jsx("div", {
			className: "bg-muted/30 p-4 rounded",
			children: /* @__PURE__ */ jsx(JsonView, {
				src: opts,
				theme: "atom",
				collapsed: 2
			})
		});
		$[1] = opts;
		$[2] = t1;
	} else t1 = $[2];
	return t1;
});
JobOptionsTab.displayName = "JobOptionsTab";
const JobResultTab = memo((t0) => {
	const $ = c(3);
	if ($[0] !== "3b06a00e91af27c8dc4dbe23232bee1ad67432d0e4c5ed3ab9b136e6d2a7a3cd") {
		for (let $i = 0; $i < 3; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "3b06a00e91af27c8dc4dbe23232bee1ad67432d0e4c5ed3ab9b136e6d2a7a3cd";
	}
	const { returnvalue } = t0;
	const t1 = returnvalue;
	let t2;
	if ($[1] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: "bg-muted/30 p-4 rounded",
			children: /* @__PURE__ */ jsx(JsonView, {
				src: t1,
				theme: "atom",
				collapsed: 2
			})
		});
		$[1] = t1;
		$[2] = t2;
	} else t2 = $[2];
	return t2;
});
JobResultTab.displayName = "JobResultTab";
const JobErrorTab = memo((t0) => {
	const $ = c(9);
	if ($[0] !== "3b06a00e91af27c8dc4dbe23232bee1ad67432d0e4c5ed3ab9b136e6d2a7a3cd") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "3b06a00e91af27c8dc4dbe23232bee1ad67432d0e4c5ed3ab9b136e6d2a7a3cd";
	}
	const { failedReason, stacktrace } = t0;
	let t1;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = /* @__PURE__ */ jsx("div", {
			className: "text-sm font-semibold text-destructive mb-1",
			children: "Error Message"
		});
		$[1] = t1;
	} else t1 = $[1];
	let t2;
	if ($[2] !== failedReason) {
		t2 = /* @__PURE__ */ jsxs("div", { children: [t1, /* @__PURE__ */ jsx("div", {
			className: "font-mono text-sm bg-destructive/10 p-3 rounded text-destructive",
			children: failedReason
		})] });
		$[2] = failedReason;
		$[3] = t2;
	} else t2 = $[3];
	let t3;
	if ($[4] !== stacktrace) {
		t3 = stacktrace && stacktrace.length > 0 && /* @__PURE__ */ jsxs("div", { children: [/* @__PURE__ */ jsx("div", {
			className: "text-sm font-semibold text-muted-foreground mb-1",
			children: "Stack Trace"
		}), /* @__PURE__ */ jsx("pre", {
			className: "font-mono text-xs bg-muted p-3 rounded overflow-auto max-h-[300px]",
			children: stacktrace.join("\n")
		})] });
		$[4] = stacktrace;
		$[5] = t3;
	} else t3 = $[5];
	let t4;
	if ($[6] !== t2 || $[7] !== t3) {
		t4 = /* @__PURE__ */ jsxs("div", {
			className: "space-y-4",
			children: [t2, t3]
		});
		$[6] = t2;
		$[7] = t3;
		$[8] = t4;
	} else t4 = $[8];
	return t4;
});
JobErrorTab.displayName = "JobErrorTab";
const JobDetail = memo(() => {
	const $ = c(92);
	if ($[0] !== "3b06a00e91af27c8dc4dbe23232bee1ad67432d0e4c5ed3ab9b136e6d2a7a3cd") {
		for (let $i = 0; $i < 92; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "3b06a00e91af27c8dc4dbe23232bee1ad67432d0e4c5ed3ab9b136e6d2a7a3cd";
	}
	const selectedJob = useBullMQStore(_temp$4);
	const selectedQueue = useBullMQStore(_temp2$4);
	const jobDetailOpen = useBullMQStore(_temp3$3);
	const setJobDetailOpen = useBullMQStore(_temp4$2);
	const setSelectedJob = useBullMQStore(_temp5$2);
	const retryJobMutation = useRetryJob();
	const removeJobMutation = useRemoveJob();
	const promoteJobMutation = usePromoteJob();
	let t0;
	if ($[1] !== setJobDetailOpen || $[2] !== setSelectedJob) {
		t0 = () => {
			setJobDetailOpen(false);
			setSelectedJob(null);
		};
		$[1] = setJobDetailOpen;
		$[2] = setSelectedJob;
		$[3] = t0;
	} else t0 = $[3];
	const handleClose = t0;
	let t1;
	if ($[4] !== handleClose || $[5] !== retryJobMutation || $[6] !== selectedJob || $[7] !== selectedQueue) {
		t1 = () => {
			if (!selectedQueue || !selectedJob) return;
			retryJobMutation.mutate({
				queueName: selectedQueue.name,
				jobId: selectedJob.id
			});
			handleClose();
		};
		$[4] = handleClose;
		$[5] = retryJobMutation;
		$[6] = selectedJob;
		$[7] = selectedQueue;
		$[8] = t1;
	} else t1 = $[8];
	const handleRetry = t1;
	let t2;
	if ($[9] !== handleClose || $[10] !== removeJobMutation || $[11] !== selectedJob || $[12] !== selectedQueue) {
		t2 = () => {
			if (!selectedQueue || !selectedJob) return;
			removeJobMutation.mutate({
				queueName: selectedQueue.name,
				jobId: selectedJob.id
			});
			handleClose();
		};
		$[9] = handleClose;
		$[10] = removeJobMutation;
		$[11] = selectedJob;
		$[12] = selectedQueue;
		$[13] = t2;
	} else t2 = $[13];
	const handleRemove = t2;
	let t3;
	if ($[14] !== handleClose || $[15] !== promoteJobMutation || $[16] !== selectedJob || $[17] !== selectedQueue) {
		t3 = () => {
			if (!selectedQueue || !selectedJob) return;
			promoteJobMutation.mutate({
				queueName: selectedQueue.name,
				jobId: selectedJob.id
			});
			handleClose();
		};
		$[14] = handleClose;
		$[15] = promoteJobMutation;
		$[16] = selectedJob;
		$[17] = selectedQueue;
		$[18] = t3;
	} else t3 = $[18];
	const handlePromote = t3;
	if (!jobDetailOpen || !selectedJob || !selectedQueue) return null;
	const hasError = !!selectedJob.failedReason;
	const isDelayed = selectedJob.delay && selectedJob.delay > 0;
	const hasReturnValue = selectedJob.returnvalue !== void 0 && selectedJob.returnvalue !== null;
	let t4;
	if ($[19] !== selectedJob.data) {
		t4 = {
			label: "Data",
			content: /* @__PURE__ */ jsx(JobDataTab, { data: selectedJob.data })
		};
		$[19] = selectedJob.data;
		$[20] = t4;
	} else t4 = $[20];
	let t5;
	if ($[21] !== selectedJob.opts) {
		t5 = {
			label: "Options",
			content: /* @__PURE__ */ jsx(JobOptionsTab, { opts: selectedJob.opts })
		};
		$[21] = selectedJob.opts;
		$[22] = t5;
	} else t5 = $[22];
	let t6;
	if ($[23] !== hasReturnValue || $[24] !== selectedJob.returnvalue) {
		t6 = hasReturnValue ? [{
			label: "Result",
			content: /* @__PURE__ */ jsx(JobResultTab, { returnvalue: selectedJob.returnvalue })
		}] : [];
		$[23] = hasReturnValue;
		$[24] = selectedJob.returnvalue;
		$[25] = t6;
	} else t6 = $[25];
	let t7;
	if ($[26] !== hasError || $[27] !== selectedJob.failedReason || $[28] !== selectedJob.stacktrace) {
		t7 = hasError ? [{
			label: "Error",
			content: /* @__PURE__ */ jsx(JobErrorTab, {
				failedReason: selectedJob.failedReason,
				stacktrace: selectedJob.stacktrace
			})
		}] : [];
		$[26] = hasError;
		$[27] = selectedJob.failedReason;
		$[28] = selectedJob.stacktrace;
		$[29] = t7;
	} else t7 = $[29];
	let t8;
	if ($[30] !== t4 || $[31] !== t5 || $[32] !== t6 || $[33] !== t7) {
		t8 = [
			t4,
			t5,
			...t6,
			...t7
		];
		$[30] = t4;
		$[31] = t5;
		$[32] = t6;
		$[33] = t7;
		$[34] = t8;
	} else t8 = $[34];
	const tabs = t8;
	let t9;
	if ($[35] !== hasError || $[36] !== isDelayed || $[37] !== selectedJob.finishedOn || $[38] !== selectedJob.processedOn) {
		t9 = hasError ? /* @__PURE__ */ jsx("span", {
			className: "text-destructive",
			children: "Failed"
		}) : selectedJob.finishedOn ? /* @__PURE__ */ jsx("span", {
			className: "text-green-500",
			children: "Completed"
		}) : selectedJob.processedOn ? /* @__PURE__ */ jsx("span", {
			className: "text-yellow-500",
			children: "Processing"
		}) : isDelayed ? /* @__PURE__ */ jsx("span", {
			className: "text-purple-500",
			children: "Delayed"
		}) : /* @__PURE__ */ jsx("span", {
			className: "text-blue-500",
			children: "Waiting"
		});
		$[35] = hasError;
		$[36] = isDelayed;
		$[37] = selectedJob.finishedOn;
		$[38] = selectedJob.processedOn;
		$[39] = t9;
	} else t9 = $[39];
	const statusElement = t9;
	const t10 = selectedJob.name;
	let t11;
	if ($[40] === Symbol.for("react.memo_cache_sentinel")) {
		t11 = /* @__PURE__ */ jsx(X, {});
		$[40] = t11;
	} else t11 = $[40];
	let t12;
	if ($[41] !== handleClose) {
		t12 = [{
			icon: t11,
			onClick: handleClose,
			label: "Close"
		}];
		$[41] = handleClose;
		$[42] = t12;
	} else t12 = $[42];
	let t13;
	if ($[43] !== handleRetry || $[44] !== hasError) {
		t13 = hasError && /* @__PURE__ */ jsxs(Button, {
			variant: "outline",
			size: "sm",
			onClick: handleRetry,
			children: [/* @__PURE__ */ jsx(RefreshCw, { className: "mr-2 h-4 w-4" }), "Retry"]
		});
		$[43] = handleRetry;
		$[44] = hasError;
		$[45] = t13;
	} else t13 = $[45];
	let t14;
	if ($[46] !== handlePromote || $[47] !== isDelayed) {
		t14 = isDelayed && /* @__PURE__ */ jsxs(Button, {
			variant: "outline",
			size: "sm",
			onClick: handlePromote,
			children: [/* @__PURE__ */ jsx(ArrowUpRight, { className: "mr-2 h-4 w-4" }), "Promote"]
		});
		$[46] = handlePromote;
		$[47] = isDelayed;
		$[48] = t14;
	} else t14 = $[48];
	let t15;
	if ($[49] === Symbol.for("react.memo_cache_sentinel")) {
		t15 = /* @__PURE__ */ jsx(Trash, { className: "mr-2 h-4 w-4" });
		$[49] = t15;
	} else t15 = $[49];
	let t16;
	if ($[50] !== handleRemove) {
		t16 = /* @__PURE__ */ jsxs(Button, {
			variant: "outline",
			size: "sm",
			onClick: handleRemove,
			className: "text-destructive",
			children: [t15, "Remove"]
		});
		$[50] = handleRemove;
		$[51] = t16;
	} else t16 = $[51];
	let t17;
	if ($[52] !== t13 || $[53] !== t14 || $[54] !== t16) {
		t17 = /* @__PURE__ */ jsxs("div", {
			className: "flex gap-2",
			children: [
				t13,
				t14,
				t16
			]
		});
		$[52] = t13;
		$[53] = t14;
		$[54] = t16;
		$[55] = t17;
	} else t17 = $[55];
	let t18;
	if ($[56] === Symbol.for("react.memo_cache_sentinel")) {
		t18 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Job ID"
		});
		$[56] = t18;
	} else t18 = $[56];
	let t19;
	if ($[57] !== selectedJob.id) {
		t19 = /* @__PURE__ */ jsxs("div", { children: [t18, /* @__PURE__ */ jsx("div", {
			className: "font-mono text-xs",
			children: selectedJob.id
		})] });
		$[57] = selectedJob.id;
		$[58] = t19;
	} else t19 = $[58];
	let t20;
	if ($[59] === Symbol.for("react.memo_cache_sentinel")) {
		t20 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Status"
		});
		$[59] = t20;
	} else t20 = $[59];
	let t21;
	if ($[60] !== statusElement) {
		t21 = /* @__PURE__ */ jsxs("div", { children: [t20, /* @__PURE__ */ jsx("div", {
			className: "font-semibold",
			children: statusElement
		})] });
		$[60] = statusElement;
		$[61] = t21;
	} else t21 = $[61];
	let t22;
	if ($[62] === Symbol.for("react.memo_cache_sentinel")) {
		t22 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Created"
		});
		$[62] = t22;
	} else t22 = $[62];
	let t23;
	if ($[63] !== selectedJob.timestamp) {
		t23 = formatDistanceToNow(selectedJob.timestamp, { addSuffix: true });
		$[63] = selectedJob.timestamp;
		$[64] = t23;
	} else t23 = $[64];
	let t24;
	if ($[65] !== t23) {
		t24 = /* @__PURE__ */ jsxs("div", { children: [t22, /* @__PURE__ */ jsx("div", { children: t23 })] });
		$[65] = t23;
		$[66] = t24;
	} else t24 = $[66];
	let t25;
	if ($[67] === Symbol.for("react.memo_cache_sentinel")) {
		t25 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Attempts"
		});
		$[67] = t25;
	} else t25 = $[67];
	let t26;
	if ($[68] !== selectedJob.attemptsMade) {
		t26 = /* @__PURE__ */ jsxs("div", { children: [t25, /* @__PURE__ */ jsx("div", {
			className: "font-semibold",
			children: selectedJob.attemptsMade
		})] });
		$[68] = selectedJob.attemptsMade;
		$[69] = t26;
	} else t26 = $[69];
	let t27;
	if ($[70] === Symbol.for("react.memo_cache_sentinel")) {
		t27 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Progress"
		});
		$[70] = t27;
	} else t27 = $[70];
	const t28 = typeof selectedJob.progress === "number" ? `${selectedJob.progress}%` : "-";
	let t29;
	if ($[71] !== t28) {
		t29 = /* @__PURE__ */ jsxs("div", { children: [t27, /* @__PURE__ */ jsx("div", {
			className: "font-semibold",
			children: t28
		})] });
		$[71] = t28;
		$[72] = t29;
	} else t29 = $[72];
	let t30;
	if ($[73] === Symbol.for("react.memo_cache_sentinel")) {
		t30 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Delay"
		});
		$[73] = t30;
	} else t30 = $[73];
	const t31 = selectedJob.delay ? `${selectedJob.delay}ms` : "-";
	let t32;
	if ($[74] !== t31) {
		t32 = /* @__PURE__ */ jsxs("div", { children: [t30, /* @__PURE__ */ jsx("div", {
			className: "font-semibold",
			children: t31
		})] });
		$[74] = t31;
		$[75] = t32;
	} else t32 = $[75];
	let t33;
	if ($[76] !== t19 || $[77] !== t21 || $[78] !== t24 || $[79] !== t26 || $[80] !== t29 || $[81] !== t32) {
		t33 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-2 gap-4 text-sm",
			children: [
				t19,
				t21,
				t24,
				t26,
				t29,
				t32
			]
		});
		$[76] = t19;
		$[77] = t21;
		$[78] = t24;
		$[79] = t26;
		$[80] = t29;
		$[81] = t32;
		$[82] = t33;
	} else t33 = $[82];
	let t34;
	if ($[83] !== t17 || $[84] !== t33) {
		t34 = /* @__PURE__ */ jsxs("div", {
			className: "space-y-4",
			children: [t17, t33]
		});
		$[83] = t17;
		$[84] = t33;
		$[85] = t34;
	} else t34 = $[85];
	let t35;
	if ($[86] !== handleClose || $[87] !== selectedJob.name || $[88] !== t12 || $[89] !== t34 || $[90] !== tabs) {
		t35 = /* @__PURE__ */ jsx(Sidebar, {
			onClose: handleClose,
			title: t10,
			initialWidth: 600,
			tabs,
			actions: t12,
			children: t34
		});
		$[86] = handleClose;
		$[87] = selectedJob.name;
		$[88] = t12;
		$[89] = t34;
		$[90] = tabs;
		$[91] = t35;
	} else t35 = $[91];
	return t35;
});
JobDetail.displayName = "JobDetail";
function _temp$4(state) {
	return state.selectedJob;
}
function _temp2$4(state_0) {
	return state_0.selectedQueue;
}
function _temp3$3(state_1) {
	return state_1.jobDetailOpen;
}
function _temp4$2(state_2) {
	return state_2.setJobDetailOpen;
}
function _temp5$2(state_3) {
	return state_3.setSelectedJob;
}

//#endregion
//#region src/components/jobs-table.tsx
const JobRow = memo((t0) => {
	const $ = c(50);
	if ($[0] !== "35e57f18a759ee630724f8001ffcc43b791357fe1eb5c9a6692b76ef0e6628e1") {
		for (let $i = 0; $i < 50; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "35e57f18a759ee630724f8001ffcc43b791357fe1eb5c9a6692b76ef0e6628e1";
	}
	const { job, queueName, onSelect, isSelected } = t0;
	const retryJobMutation = useRetryJob();
	const removeJobMutation = useRemoveJob();
	const promoteJobMutation = usePromoteJob();
	let t1;
	if ($[1] !== job.id || $[2] !== queueName || $[3] !== retryJobMutation) {
		t1 = (e) => {
			e.stopPropagation();
			retryJobMutation.mutate({
				queueName,
				jobId: job.id
			});
		};
		$[1] = job.id;
		$[2] = queueName;
		$[3] = retryJobMutation;
		$[4] = t1;
	} else t1 = $[4];
	const handleRetry = t1;
	let t2;
	if ($[5] !== job.id || $[6] !== queueName || $[7] !== removeJobMutation) {
		t2 = (e_0) => {
			e_0.stopPropagation();
			removeJobMutation.mutate({
				queueName,
				jobId: job.id
			});
		};
		$[5] = job.id;
		$[6] = queueName;
		$[7] = removeJobMutation;
		$[8] = t2;
	} else t2 = $[8];
	const handleRemove = t2;
	let t3;
	if ($[9] !== job.id || $[10] !== promoteJobMutation || $[11] !== queueName) {
		t3 = (e_1) => {
			e_1.stopPropagation();
			promoteJobMutation.mutate({
				queueName,
				jobId: job.id
			});
		};
		$[9] = job.id;
		$[10] = promoteJobMutation;
		$[11] = queueName;
		$[12] = t3;
	} else t3 = $[12];
	const handlePromote = t3;
	const t4 = isSelected ? "bg-muted-foreground/10 hover:bg-muted-foreground/20" : "hover:bg-muted-foreground/10";
	let t5;
	if ($[13] !== t4) {
		t5 = cn("cursor-pointer border-0", t4);
		$[13] = t4;
		$[14] = t5;
	} else t5 = $[14];
	let t6;
	if ($[15] !== job.id) {
		t6 = /* @__PURE__ */ jsx(TableCell, {
			className: "font-mono text-xs",
			children: job.id
		});
		$[15] = job.id;
		$[16] = t6;
	} else t6 = $[16];
	let t7;
	if ($[17] !== job.name) {
		t7 = /* @__PURE__ */ jsx(TableCell, {
			className: "font-medium",
			children: job.name
		});
		$[17] = job.name;
		$[18] = t7;
	} else t7 = $[18];
	let t8;
	if ($[19] !== job.timestamp) {
		t8 = formatDistanceToNow(job.timestamp, { addSuffix: true });
		$[19] = job.timestamp;
		$[20] = t8;
	} else t8 = $[20];
	let t9;
	if ($[21] !== t8) {
		t9 = /* @__PURE__ */ jsx(TableCell, {
			className: "text-xs text-muted-foreground",
			children: t8
		});
		$[21] = t8;
		$[22] = t9;
	} else t9 = $[22];
	let t10;
	if ($[23] !== job.attemptsMade) {
		t10 = /* @__PURE__ */ jsx(TableCell, { children: /* @__PURE__ */ jsx("span", {
			className: "text-xs",
			children: job.attemptsMade
		}) });
		$[23] = job.attemptsMade;
		$[24] = t10;
	} else t10 = $[24];
	let t11;
	if ($[25] !== job.progress) {
		t11 = /* @__PURE__ */ jsx(TableCell, { children: typeof job.progress === "number" ? /* @__PURE__ */ jsxs("div", {
			className: "flex items-center gap-2",
			children: [/* @__PURE__ */ jsx("div", {
				className: "w-16 h-1.5 bg-muted rounded-full overflow-hidden",
				children: /* @__PURE__ */ jsx("div", {
					className: "h-full bg-primary transition-all",
					style: { width: `${Math.min(100, job.progress)}%` }
				})
			}), /* @__PURE__ */ jsxs("span", {
				className: "text-xs text-muted-foreground",
				children: [job.progress, "%"]
			})]
		}) : /* @__PURE__ */ jsx("span", {
			className: "text-xs text-muted-foreground",
			children: "-"
		}) });
		$[25] = job.progress;
		$[26] = t11;
	} else t11 = $[26];
	let t12;
	if ($[27] === Symbol.for("react.memo_cache_sentinel")) {
		t12 = /* @__PURE__ */ jsx(DropdownMenuTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsx(Button, {
				variant: "ghost",
				size: "icon",
				className: "h-8 w-8",
				children: /* @__PURE__ */ jsx(MoreVertical, { className: "h-4 w-4" })
			})
		});
		$[27] = t12;
	} else t12 = $[27];
	let t13;
	if ($[28] !== handleRetry || $[29] !== job.failedReason) {
		t13 = job.failedReason && /* @__PURE__ */ jsxs(DropdownMenuItem, {
			onClick: handleRetry,
			children: [/* @__PURE__ */ jsx(RefreshCw, { className: "mr-2 h-4 w-4" }), "Retry"]
		});
		$[28] = handleRetry;
		$[29] = job.failedReason;
		$[30] = t13;
	} else t13 = $[30];
	let t14;
	if ($[31] !== handlePromote || $[32] !== job.delay) {
		t14 = job.delay && job.delay > 0 && /* @__PURE__ */ jsxs(DropdownMenuItem, {
			onClick: handlePromote,
			children: [/* @__PURE__ */ jsx(ArrowUpRight, { className: "mr-2 h-4 w-4" }), "Promote"]
		});
		$[31] = handlePromote;
		$[32] = job.delay;
		$[33] = t14;
	} else t14 = $[33];
	let t15;
	if ($[34] === Symbol.for("react.memo_cache_sentinel")) {
		t15 = /* @__PURE__ */ jsx(Trash, { className: "mr-2 h-4 w-4" });
		$[34] = t15;
	} else t15 = $[34];
	let t16;
	if ($[35] !== handleRemove) {
		t16 = /* @__PURE__ */ jsxs(DropdownMenuItem, {
			onClick: handleRemove,
			className: "text-destructive",
			children: [t15, "Remove"]
		});
		$[35] = handleRemove;
		$[36] = t16;
	} else t16 = $[36];
	let t17;
	if ($[37] !== t13 || $[38] !== t14 || $[39] !== t16) {
		t17 = /* @__PURE__ */ jsx(TableCell, {
			onClick: _temp$3,
			children: /* @__PURE__ */ jsxs(DropdownMenu, { children: [t12, /* @__PURE__ */ jsxs(DropdownMenuContent, {
				align: "end",
				className: "bg-background text-foreground",
				children: [
					t13,
					t14,
					t16
				]
			})] })
		});
		$[37] = t13;
		$[38] = t14;
		$[39] = t16;
		$[40] = t17;
	} else t17 = $[40];
	let t18;
	if ($[41] !== onSelect || $[42] !== t10 || $[43] !== t11 || $[44] !== t17 || $[45] !== t5 || $[46] !== t6 || $[47] !== t7 || $[48] !== t9) {
		t18 = /* @__PURE__ */ jsxs(TableRow, {
			onClick: onSelect,
			className: t5,
			children: [
				t6,
				t7,
				t9,
				t10,
				t11,
				t17
			]
		});
		$[41] = onSelect;
		$[42] = t10;
		$[43] = t11;
		$[44] = t17;
		$[45] = t5;
		$[46] = t6;
		$[47] = t7;
		$[48] = t9;
		$[49] = t18;
	} else t18 = $[49];
	return t18;
});
JobRow.displayName = "JobRow";
const JobsTable = memo(() => {
	const $ = c(21);
	if ($[0] !== "35e57f18a759ee630724f8001ffcc43b791357fe1eb5c9a6692b76ef0e6628e1") {
		for (let $i = 0; $i < 21; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "35e57f18a759ee630724f8001ffcc43b791357fe1eb5c9a6692b76ef0e6628e1";
	}
	const { data: t0, isLoading } = useJobsQuery();
	let t1;
	if ($[1] !== t0) {
		t1 = t0 === void 0 ? [] : t0;
		$[1] = t0;
		$[2] = t1;
	} else t1 = $[2];
	const jobs = t1;
	const selectedQueue = useBullMQStore(_temp2$3);
	const selectedJob = useBullMQStore(_temp3$2);
	const setSelectedJob = useBullMQStore(_temp4$1);
	const setJobDetailOpen = useBullMQStore(_temp5$1);
	let t2;
	if ($[3] !== setJobDetailOpen || $[4] !== setSelectedJob) {
		t2 = (job) => {
			setSelectedJob(job);
			setJobDetailOpen(true);
		};
		$[3] = setJobDetailOpen;
		$[4] = setSelectedJob;
		$[5] = t2;
	} else t2 = $[5];
	const handleSelectJob = t2;
	if (!selectedQueue) {
		let t3$1;
		if ($[6] === Symbol.for("react.memo_cache_sentinel")) {
			t3$1 = /* @__PURE__ */ jsx("div", {
				className: "flex items-center justify-center h-full text-muted-foreground",
				children: "Select a queue to view jobs"
			});
			$[6] = t3$1;
		} else t3$1 = $[6];
		return t3$1;
	}
	if (isLoading) {
		let t3$1;
		if ($[7] === Symbol.for("react.memo_cache_sentinel")) {
			t3$1 = /* @__PURE__ */ jsx("div", {
				className: "flex items-center justify-center h-full text-muted-foreground",
				children: "Loading jobs..."
			});
			$[7] = t3$1;
		} else t3$1 = $[7];
		return t3$1;
	}
	if (jobs.length === 0) {
		let t3$1;
		if ($[8] === Symbol.for("react.memo_cache_sentinel")) {
			t3$1 = /* @__PURE__ */ jsx("div", {
				className: "flex items-center justify-center h-full text-muted-foreground",
				children: "No jobs in this status"
			});
			$[8] = t3$1;
		} else t3$1 = $[8];
		return t3$1;
	}
	let t3;
	if ($[9] === Symbol.for("react.memo_cache_sentinel")) {
		t3 = /* @__PURE__ */ jsx(TableHeader, {
			className: "sticky top-0 bg-background/95 backdrop-blur-sm",
			children: /* @__PURE__ */ jsxs(TableRow, { children: [
				/* @__PURE__ */ jsx(TableHead, {
					className: "w-[200px]",
					children: "Job ID"
				}),
				/* @__PURE__ */ jsx(TableHead, { children: "Name" }),
				/* @__PURE__ */ jsx(TableHead, {
					className: "w-[150px]",
					children: "Created"
				}),
				/* @__PURE__ */ jsx(TableHead, {
					className: "w-[80px]",
					children: "Attempts"
				}),
				/* @__PURE__ */ jsx(TableHead, {
					className: "w-[140px]",
					children: "Progress"
				}),
				/* @__PURE__ */ jsx(TableHead, {
					className: "w-[60px]",
					children: "Actions"
				})
			] })
		});
		$[9] = t3;
	} else t3 = $[9];
	let t4;
	if ($[10] !== handleSelectJob || $[11] !== jobs || $[12] !== selectedJob?.id || $[13] !== selectedQueue) {
		let t5$1;
		if ($[15] !== handleSelectJob || $[16] !== selectedJob?.id || $[17] !== selectedQueue) {
			t5$1 = (job_0) => /* @__PURE__ */ jsx(JobRow, {
				job: job_0,
				queueName: selectedQueue.name,
				onSelect: () => handleSelectJob(job_0),
				isSelected: selectedJob?.id === job_0.id
			}, job_0.id);
			$[15] = handleSelectJob;
			$[16] = selectedJob?.id;
			$[17] = selectedQueue;
			$[18] = t5$1;
		} else t5$1 = $[18];
		t4 = jobs.map(t5$1);
		$[10] = handleSelectJob;
		$[11] = jobs;
		$[12] = selectedJob?.id;
		$[13] = selectedQueue;
		$[14] = t4;
	} else t4 = $[14];
	let t5;
	if ($[19] !== t4) {
		t5 = /* @__PURE__ */ jsxs(Table, { children: [t3, /* @__PURE__ */ jsx(TableBody, { children: t4 })] });
		$[19] = t4;
		$[20] = t5;
	} else t5 = $[20];
	return t5;
});
JobsTable.displayName = "JobsTable";
function _temp$3(e_2) {
	return e_2.stopPropagation();
}
function _temp2$3(state) {
	return state.selectedQueue;
}
function _temp3$2(state_0) {
	return state_0.selectedJob;
}
function _temp4$1(state_1) {
	return state_1.setSelectedJob;
}
function _temp5$1(state_2) {
	return state_2.setJobDetailOpen;
}

//#endregion
//#region src/components/queue-detail.tsx
const STATUS_TABS = [
	{
		value: "waiting",
		label: "Waiting"
	},
	{
		value: "active",
		label: "Active"
	},
	{
		value: "completed",
		label: "Completed"
	},
	{
		value: "failed",
		label: "Failed"
	},
	{
		value: "delayed",
		label: "Delayed"
	}
];
const QueueDetail = memo(() => {
	const $ = c(115);
	if ($[0] !== "ec763ba5d94c93fbec6c5d401f02a104160374351538eec66a12a09486a09276") {
		for (let $i = 0; $i < 115; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "ec763ba5d94c93fbec6c5d401f02a104160374351538eec66a12a09486a09276";
	}
	const queryClient$1 = useQueryClient();
	const selectedQueue = useBullMQStore(_temp$2);
	const selectedStatus = useBullMQStore(_temp2$2);
	const setSelectedStatus = useBullMQStore(_temp3$1);
	const { pauseQueue, resumeQueue, cleanQueue, drainQueue } = useQueues();
	let t0;
	if ($[1] !== pauseQueue || $[2] !== selectedQueue) {
		t0 = async () => {
			if (!selectedQueue) return;
			await pauseQueue(selectedQueue.name);
		};
		$[1] = pauseQueue;
		$[2] = selectedQueue;
		$[3] = t0;
	} else t0 = $[3];
	const handlePause = t0;
	let t1;
	if ($[4] !== resumeQueue || $[5] !== selectedQueue) {
		t1 = async () => {
			if (!selectedQueue) return;
			await resumeQueue(selectedQueue.name);
		};
		$[4] = resumeQueue;
		$[5] = selectedQueue;
		$[6] = t1;
	} else t1 = $[6];
	const handleResume = t1;
	let t2;
	if ($[7] !== cleanQueue || $[8] !== selectedQueue) {
		t2 = async () => {
			if (!selectedQueue) return;
			await cleanQueue(selectedQueue.name, "completed", 0, 1e3);
		};
		$[7] = cleanQueue;
		$[8] = selectedQueue;
		$[9] = t2;
	} else t2 = $[9];
	const handleCleanCompleted = t2;
	let t3;
	if ($[10] !== cleanQueue || $[11] !== selectedQueue) {
		t3 = async () => {
			if (!selectedQueue) return;
			await cleanQueue(selectedQueue.name, "failed", 0, 1e3);
		};
		$[10] = cleanQueue;
		$[11] = selectedQueue;
		$[12] = t3;
	} else t3 = $[12];
	const handleCleanFailed = t3;
	let t4;
	if ($[13] !== drainQueue || $[14] !== selectedQueue) {
		t4 = async () => {
			if (!selectedQueue) return;
			await drainQueue(selectedQueue.name);
		};
		$[13] = drainQueue;
		$[14] = selectedQueue;
		$[15] = t4;
	} else t4 = $[15];
	const handleDrain = t4;
	let t5;
	if ($[16] !== queryClient$1 || $[17] !== selectedQueue) {
		t5 = () => {
			if (!selectedQueue) return;
			queryClient$1.invalidateQueries({ queryKey: ["jobs", selectedQueue.name] });
		};
		$[16] = queryClient$1;
		$[17] = selectedQueue;
		$[18] = t5;
	} else t5 = $[18];
	const handleRefresh = t5;
	if (!selectedQueue) {
		let t6$1;
		if ($[19] === Symbol.for("react.memo_cache_sentinel")) {
			t6$1 = /* @__PURE__ */ jsx("div", {
				className: "flex items-center justify-center h-full text-muted-foreground",
				children: "Select a queue from the list to view details"
			});
			$[19] = t6$1;
		} else t6$1 = $[19];
		return t6$1;
	}
	let t6;
	if ($[20] !== selectedQueue.displayName) {
		t6 = /* @__PURE__ */ jsx("h2", {
			className: "font-semibold text-lg",
			children: selectedQueue.displayName
		});
		$[20] = selectedQueue.displayName;
		$[21] = t6;
	} else t6 = $[21];
	let t7;
	if ($[22] !== selectedQueue.isPaused) {
		t7 = selectedQueue.isPaused && /* @__PURE__ */ jsx("span", {
			className: "px-2 py-0.5 text-xs rounded bg-yellow-500/20 text-yellow-600",
			children: "Paused"
		});
		$[22] = selectedQueue.isPaused;
		$[23] = t7;
	} else t7 = $[23];
	let t8;
	if ($[24] !== t6 || $[25] !== t7) {
		t8 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center gap-3",
			children: [t6, t7]
		});
		$[24] = t6;
		$[25] = t7;
		$[26] = t8;
	} else t8 = $[26];
	let t9;
	if ($[27] === Symbol.for("react.memo_cache_sentinel")) {
		t9 = /* @__PURE__ */ jsx(RefreshCw, { className: "h-4 w-4" });
		$[27] = t9;
	} else t9 = $[27];
	let t10;
	if ($[28] !== handleRefresh) {
		t10 = /* @__PURE__ */ jsx(TooltipTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsx(Button, {
				variant: "ghost",
				size: "icon",
				onClick: handleRefresh,
				children: t9
			})
		});
		$[28] = handleRefresh;
		$[29] = t10;
	} else t10 = $[29];
	let t11;
	if ($[30] === Symbol.for("react.memo_cache_sentinel")) {
		t11 = /* @__PURE__ */ jsx(TooltipContent, { children: /* @__PURE__ */ jsx("p", { children: "Refresh queue stats and jobs list" }) });
		$[30] = t11;
	} else t11 = $[30];
	let t12;
	if ($[31] !== t10) {
		t12 = /* @__PURE__ */ jsxs(Tooltip, { children: [t10, t11] });
		$[31] = t10;
		$[32] = t12;
	} else t12 = $[32];
	let t13;
	if ($[33] !== handlePause || $[34] !== handleResume || $[35] !== selectedQueue.isPaused) {
		t13 = selectedQueue.isPaused ? /* @__PURE__ */ jsxs(Tooltip, { children: [/* @__PURE__ */ jsx(TooltipTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsxs(Button, {
				variant: "ghost",
				size: "sm",
				onClick: handleResume,
				children: [/* @__PURE__ */ jsx(Play, { className: "mr-2 h-4 w-4" }), "Resume"]
			})
		}), /* @__PURE__ */ jsx(TooltipContent, { children: /* @__PURE__ */ jsx("p", { children: "Resume processing jobs in this queue" }) })] }) : /* @__PURE__ */ jsxs(Tooltip, { children: [/* @__PURE__ */ jsx(TooltipTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsxs(Button, {
				variant: "ghost",
				size: "sm",
				onClick: handlePause,
				children: [/* @__PURE__ */ jsx(Pause, { className: "mr-2 h-4 w-4" }), "Pause"]
			})
		}), /* @__PURE__ */ jsx(TooltipContent, { children: /* @__PURE__ */ jsx("p", { children: "Stop workers from picking up new jobs" }) })] });
		$[33] = handlePause;
		$[34] = handleResume;
		$[35] = selectedQueue.isPaused;
		$[36] = t13;
	} else t13 = $[36];
	let t14;
	if ($[37] === Symbol.for("react.memo_cache_sentinel")) {
		t14 = /* @__PURE__ */ jsx(DropdownMenuTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsx(Button, {
				variant: "ghost",
				size: "icon",
				children: /* @__PURE__ */ jsx(MoreVertical, { className: "h-4 w-4" })
			})
		});
		$[37] = t14;
	} else t14 = $[37];
	let t15;
	if ($[38] === Symbol.for("react.memo_cache_sentinel")) {
		t15 = /* @__PURE__ */ jsx(Trash, { className: "mr-2 h-4 w-4" });
		$[38] = t15;
	} else t15 = $[38];
	let t16;
	if ($[39] !== handleCleanCompleted) {
		t16 = /* @__PURE__ */ jsxs(DropdownMenuItem, {
			onClick: handleCleanCompleted,
			children: [t15, "Clean Completed"]
		});
		$[39] = handleCleanCompleted;
		$[40] = t16;
	} else t16 = $[40];
	let t17;
	if ($[41] === Symbol.for("react.memo_cache_sentinel")) {
		t17 = /* @__PURE__ */ jsx(Trash, { className: "mr-2 h-4 w-4" });
		$[41] = t17;
	} else t17 = $[41];
	let t18;
	if ($[42] !== handleCleanFailed) {
		t18 = /* @__PURE__ */ jsxs(DropdownMenuItem, {
			onClick: handleCleanFailed,
			children: [t17, "Clean Failed"]
		});
		$[42] = handleCleanFailed;
		$[43] = t18;
	} else t18 = $[43];
	let t19;
	if ($[44] === Symbol.for("react.memo_cache_sentinel")) {
		t19 = /* @__PURE__ */ jsx(DropdownMenuSeparator, {});
		$[44] = t19;
	} else t19 = $[44];
	let t20;
	if ($[45] === Symbol.for("react.memo_cache_sentinel")) {
		t20 = /* @__PURE__ */ jsx(Trash, { className: "mr-2 h-4 w-4" });
		$[45] = t20;
	} else t20 = $[45];
	let t21;
	if ($[46] !== handleDrain) {
		t21 = /* @__PURE__ */ jsxs(DropdownMenuItem, {
			onClick: handleDrain,
			className: "text-destructive",
			children: [t20, "Drain Queue"]
		});
		$[46] = handleDrain;
		$[47] = t21;
	} else t21 = $[47];
	let t22;
	if ($[48] !== t16 || $[49] !== t18 || $[50] !== t21) {
		t22 = /* @__PURE__ */ jsx(TooltipTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsxs(DropdownMenu, { children: [t14, /* @__PURE__ */ jsxs(DropdownMenuContent, {
				align: "end",
				className: "bg-background text-foreground",
				children: [
					t16,
					t18,
					t19,
					t21
				]
			})] })
		});
		$[48] = t16;
		$[49] = t18;
		$[50] = t21;
		$[51] = t22;
	} else t22 = $[51];
	let t23;
	if ($[52] === Symbol.for("react.memo_cache_sentinel")) {
		t23 = /* @__PURE__ */ jsx(TooltipContent, { children: /* @__PURE__ */ jsx("p", { children: "More actions" }) });
		$[52] = t23;
	} else t23 = $[52];
	let t24;
	if ($[53] !== t22) {
		t24 = /* @__PURE__ */ jsxs(Tooltip, { children: [t22, t23] });
		$[53] = t22;
		$[54] = t24;
	} else t24 = $[54];
	let t25;
	if ($[55] !== t12 || $[56] !== t13 || $[57] !== t24) {
		t25 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center gap-2",
			children: [
				t12,
				t13,
				t24
			]
		});
		$[55] = t12;
		$[56] = t13;
		$[57] = t24;
		$[58] = t25;
	} else t25 = $[58];
	let t26;
	if ($[59] !== t25 || $[60] !== t8) {
		t26 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center justify-between p-3 border-b border-border",
			children: [t8, t25]
		});
		$[59] = t25;
		$[60] = t8;
		$[61] = t26;
	} else t26 = $[61];
	let t27;
	if ($[62] !== selectedQueue.stats.waiting) {
		t27 = /* @__PURE__ */ jsx("div", {
			className: "text-2xl font-bold text-blue-500",
			children: selectedQueue.stats.waiting
		});
		$[62] = selectedQueue.stats.waiting;
		$[63] = t27;
	} else t27 = $[63];
	let t28;
	if ($[64] === Symbol.for("react.memo_cache_sentinel")) {
		t28 = /* @__PURE__ */ jsx("div", {
			className: "text-xs text-muted-foreground",
			children: "Waiting"
		});
		$[64] = t28;
	} else t28 = $[64];
	let t29;
	if ($[65] !== t27) {
		t29 = /* @__PURE__ */ jsxs("div", { children: [t27, t28] });
		$[65] = t27;
		$[66] = t29;
	} else t29 = $[66];
	let t30;
	if ($[67] !== selectedQueue.stats.active) {
		t30 = /* @__PURE__ */ jsx("div", {
			className: "text-2xl font-bold text-yellow-500",
			children: selectedQueue.stats.active
		});
		$[67] = selectedQueue.stats.active;
		$[68] = t30;
	} else t30 = $[68];
	let t31;
	if ($[69] === Symbol.for("react.memo_cache_sentinel")) {
		t31 = /* @__PURE__ */ jsx("div", {
			className: "text-xs text-muted-foreground",
			children: "Active"
		});
		$[69] = t31;
	} else t31 = $[69];
	let t32;
	if ($[70] !== t30) {
		t32 = /* @__PURE__ */ jsxs("div", { children: [t30, t31] });
		$[70] = t30;
		$[71] = t32;
	} else t32 = $[71];
	let t33;
	if ($[72] !== selectedQueue.stats.completed) {
		t33 = /* @__PURE__ */ jsx("div", {
			className: "text-2xl font-bold text-green-500",
			children: selectedQueue.stats.completed
		});
		$[72] = selectedQueue.stats.completed;
		$[73] = t33;
	} else t33 = $[73];
	let t34;
	if ($[74] === Symbol.for("react.memo_cache_sentinel")) {
		t34 = /* @__PURE__ */ jsx("div", {
			className: "text-xs text-muted-foreground",
			children: "Completed"
		});
		$[74] = t34;
	} else t34 = $[74];
	let t35;
	if ($[75] !== t33) {
		t35 = /* @__PURE__ */ jsxs("div", { children: [t33, t34] });
		$[75] = t33;
		$[76] = t35;
	} else t35 = $[76];
	let t36;
	if ($[77] !== selectedQueue.stats.failed) {
		t36 = /* @__PURE__ */ jsx("div", {
			className: "text-2xl font-bold text-destructive",
			children: selectedQueue.stats.failed
		});
		$[77] = selectedQueue.stats.failed;
		$[78] = t36;
	} else t36 = $[78];
	let t37;
	if ($[79] === Symbol.for("react.memo_cache_sentinel")) {
		t37 = /* @__PURE__ */ jsx("div", {
			className: "text-xs text-muted-foreground",
			children: "Failed"
		});
		$[79] = t37;
	} else t37 = $[79];
	let t38;
	if ($[80] !== t36) {
		t38 = /* @__PURE__ */ jsxs("div", { children: [t36, t37] });
		$[80] = t36;
		$[81] = t38;
	} else t38 = $[81];
	let t39;
	if ($[82] !== selectedQueue.stats.delayed) {
		t39 = /* @__PURE__ */ jsx("div", {
			className: "text-2xl font-bold text-purple-500",
			children: selectedQueue.stats.delayed
		});
		$[82] = selectedQueue.stats.delayed;
		$[83] = t39;
	} else t39 = $[83];
	let t40;
	if ($[84] === Symbol.for("react.memo_cache_sentinel")) {
		t40 = /* @__PURE__ */ jsx("div", {
			className: "text-xs text-muted-foreground",
			children: "Delayed"
		});
		$[84] = t40;
	} else t40 = $[84];
	let t41;
	if ($[85] !== t39) {
		t41 = /* @__PURE__ */ jsxs("div", { children: [t39, t40] });
		$[85] = t39;
		$[86] = t41;
	} else t41 = $[86];
	let t42;
	if ($[87] !== selectedQueue.stats.paused) {
		t42 = /* @__PURE__ */ jsx("div", {
			className: "text-2xl font-bold text-muted-foreground",
			children: selectedQueue.stats.paused
		});
		$[87] = selectedQueue.stats.paused;
		$[88] = t42;
	} else t42 = $[88];
	let t43;
	if ($[89] === Symbol.for("react.memo_cache_sentinel")) {
		t43 = /* @__PURE__ */ jsx("div", {
			className: "text-xs text-muted-foreground",
			children: "Paused"
		});
		$[89] = t43;
	} else t43 = $[89];
	let t44;
	if ($[90] !== t42) {
		t44 = /* @__PURE__ */ jsxs("div", { children: [t42, t43] });
		$[90] = t42;
		$[91] = t44;
	} else t44 = $[91];
	let t45;
	if ($[92] !== t29 || $[93] !== t32 || $[94] !== t35 || $[95] !== t38 || $[96] !== t41 || $[97] !== t44) {
		t45 = /* @__PURE__ */ jsx("div", {
			className: "p-3 border-b border-border",
			children: /* @__PURE__ */ jsxs("div", {
				className: "grid grid-cols-6 gap-4 text-center",
				children: [
					t29,
					t32,
					t35,
					t38,
					t41,
					t44
				]
			})
		});
		$[92] = t29;
		$[93] = t32;
		$[94] = t35;
		$[95] = t38;
		$[96] = t41;
		$[97] = t44;
		$[98] = t45;
	} else t45 = $[98];
	let t46;
	if ($[99] !== setSelectedStatus) {
		t46 = (v) => setSelectedStatus(v);
		$[99] = setSelectedStatus;
		$[100] = t46;
	} else t46 = $[100];
	let t47;
	if ($[101] !== selectedQueue.stats || $[102] !== selectedStatus) {
		t47 = STATUS_TABS.map((tab) => /* @__PURE__ */ jsxs(TabsTrigger, {
			value: tab.value,
			className: cn("relative", selectedStatus === tab.value && "after:absolute after:bottom-0 after:left-0 after:right-0 after:h-0.5 after:bg-primary"),
			children: [tab.label, selectedQueue.stats[tab.value] > 0 && /* @__PURE__ */ jsx("span", {
				className: "ml-1.5 px-1.5 py-0.5 text-[10px] rounded-full bg-muted",
				children: selectedQueue.stats[tab.value]
			})]
		}, tab.value));
		$[101] = selectedQueue.stats;
		$[102] = selectedStatus;
		$[103] = t47;
	} else t47 = $[103];
	let t48;
	if ($[104] !== t47) {
		t48 = /* @__PURE__ */ jsx("div", {
			className: "px-3 pt-2 border-b border-border",
			children: /* @__PURE__ */ jsx(TabsList, { children: t47 })
		});
		$[104] = t47;
		$[105] = t48;
	} else t48 = $[105];
	let t49;
	if ($[106] === Symbol.for("react.memo_cache_sentinel")) {
		t49 = /* @__PURE__ */ jsx("div", {
			className: "flex-1 overflow-auto",
			children: /* @__PURE__ */ jsx(JobsTable, {})
		});
		$[106] = t49;
	} else t49 = $[106];
	let t50;
	if ($[107] !== selectedStatus || $[108] !== t46 || $[109] !== t48) {
		t50 = /* @__PURE__ */ jsxs(Tabs, {
			value: selectedStatus,
			onValueChange: t46,
			className: "flex-1 flex flex-col",
			children: [t48, t49]
		});
		$[107] = selectedStatus;
		$[108] = t46;
		$[109] = t48;
		$[110] = t50;
	} else t50 = $[110];
	let t51;
	if ($[111] !== t26 || $[112] !== t45 || $[113] !== t50) {
		t51 = /* @__PURE__ */ jsx(TooltipProvider, { children: /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col h-full",
			children: [
				t26,
				t45,
				t50
			]
		}) });
		$[111] = t26;
		$[112] = t45;
		$[113] = t50;
		$[114] = t51;
	} else t51 = $[114];
	return t51;
});
QueueDetail.displayName = "QueueDetail";
function _temp$2(state) {
	return state.selectedQueue;
}
function _temp2$2(state_0) {
	return state_0.selectedStatus;
}
function _temp3$1(state_1) {
	return state_1.setSelectedStatus;
}

//#endregion
//#region src/components/queue-list.tsx
const QueueItem = memo((t0) => {
	const $ = c(54);
	if ($[0] !== "44e6229799ef8371f1e191c6f3e3ef70ea83c9489840337cf1d87e846b1bcc7c") {
		for (let $i = 0; $i < 54; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "44e6229799ef8371f1e191c6f3e3ef70ea83c9489840337cf1d87e846b1bcc7c";
	}
	const { queue, isSelected, onClick } = t0;
	const totalJobs = queue.stats.waiting + queue.stats.active + queue.stats.delayed + queue.stats.prioritized;
	const hasFailedJobs = queue.stats.failed > 0;
	const t1 = isSelected ? "bg-muted-foreground/10" : "hover:bg-muted/70";
	let t2;
	if ($[1] !== t1) {
		t2 = cn("w-full text-left p-3 transition-colors border-b border-border", t1);
		$[1] = t1;
		$[2] = t2;
	} else t2 = $[2];
	let t3;
	if ($[3] !== queue.isDLQ) {
		t3 = queue.isDLQ ? /* @__PURE__ */ jsx(Skull, { className: "w-4 h-4 text-destructive" }) : /* @__PURE__ */ jsx(Layers, { className: "w-4 h-4 text-muted-foreground" });
		$[3] = queue.isDLQ;
		$[4] = t3;
	} else t3 = $[4];
	let t4;
	if ($[5] !== queue.displayName) {
		t4 = /* @__PURE__ */ jsx("span", {
			className: "font-semibold text-sm truncate flex-1",
			children: queue.displayName
		});
		$[5] = queue.displayName;
		$[6] = t4;
	} else t4 = $[6];
	let t5;
	if ($[7] !== queue.isPaused) {
		t5 = queue.isPaused && /* @__PURE__ */ jsx(Pause, { className: "w-3 h-3 text-yellow-500" });
		$[7] = queue.isPaused;
		$[8] = t5;
	} else t5 = $[8];
	let t6;
	if ($[9] !== hasFailedJobs) {
		t6 = hasFailedJobs && /* @__PURE__ */ jsx(AlertTriangle, { className: "w-3 h-3 text-destructive" });
		$[9] = hasFailedJobs;
		$[10] = t6;
	} else t6 = $[10];
	let t7;
	if ($[11] !== t3 || $[12] !== t4 || $[13] !== t5 || $[14] !== t6) {
		t7 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center gap-2 mb-1",
			children: [
				t3,
				t4,
				t5,
				t6
			]
		});
		$[11] = t3;
		$[12] = t4;
		$[13] = t5;
		$[14] = t6;
		$[15] = t7;
	} else t7 = $[15];
	let t8;
	if ($[16] === Symbol.for("react.memo_cache_sentinel")) {
		t8 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Wait"
		});
		$[16] = t8;
	} else t8 = $[16];
	const t9 = queue.stats.waiting > 0 && "text-blue-500";
	let t10;
	if ($[17] !== t9) {
		t10 = cn("font-mono", t9);
		$[17] = t9;
		$[18] = t10;
	} else t10 = $[18];
	let t11;
	if ($[19] !== queue.stats.waiting || $[20] !== t10) {
		t11 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col",
			children: [t8, /* @__PURE__ */ jsx("span", {
				className: t10,
				children: queue.stats.waiting
			})]
		});
		$[19] = queue.stats.waiting;
		$[20] = t10;
		$[21] = t11;
	} else t11 = $[21];
	let t12;
	if ($[22] === Symbol.for("react.memo_cache_sentinel")) {
		t12 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Active"
		});
		$[22] = t12;
	} else t12 = $[22];
	const t13 = queue.stats.active > 0 && "text-yellow-500";
	let t14;
	if ($[23] !== t13) {
		t14 = cn("font-mono", t13);
		$[23] = t13;
		$[24] = t14;
	} else t14 = $[24];
	let t15;
	if ($[25] !== queue.stats.active || $[26] !== t14) {
		t15 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col",
			children: [t12, /* @__PURE__ */ jsx("span", {
				className: t14,
				children: queue.stats.active
			})]
		});
		$[25] = queue.stats.active;
		$[26] = t14;
		$[27] = t15;
	} else t15 = $[27];
	let t16;
	if ($[28] === Symbol.for("react.memo_cache_sentinel")) {
		t16 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Done"
		});
		$[28] = t16;
	} else t16 = $[28];
	const t17 = queue.stats.completed > 0 && "text-green-500";
	let t18;
	if ($[29] !== t17) {
		t18 = cn("font-mono", t17);
		$[29] = t17;
		$[30] = t18;
	} else t18 = $[30];
	let t19;
	if ($[31] !== queue.stats.completed || $[32] !== t18) {
		t19 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col",
			children: [t16, /* @__PURE__ */ jsx("span", {
				className: t18,
				children: queue.stats.completed
			})]
		});
		$[31] = queue.stats.completed;
		$[32] = t18;
		$[33] = t19;
	} else t19 = $[33];
	let t20;
	if ($[34] === Symbol.for("react.memo_cache_sentinel")) {
		t20 = /* @__PURE__ */ jsx("span", {
			className: "text-muted-foreground",
			children: "Failed"
		});
		$[34] = t20;
	} else t20 = $[34];
	const t21 = queue.stats.failed > 0 && "text-destructive";
	let t22;
	if ($[35] !== t21) {
		t22 = cn("font-mono", t21);
		$[35] = t21;
		$[36] = t22;
	} else t22 = $[36];
	let t23;
	if ($[37] !== queue.stats.failed || $[38] !== t22) {
		t23 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col",
			children: [t20, /* @__PURE__ */ jsx("span", {
				className: t22,
				children: queue.stats.failed
			})]
		});
		$[37] = queue.stats.failed;
		$[38] = t22;
		$[39] = t23;
	} else t23 = $[39];
	let t24;
	if ($[40] !== t11 || $[41] !== t15 || $[42] !== t19 || $[43] !== t23) {
		t24 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-4 gap-1 text-xs",
			children: [
				t11,
				t15,
				t19,
				t23
			]
		});
		$[40] = t11;
		$[41] = t15;
		$[42] = t19;
		$[43] = t23;
		$[44] = t24;
	} else t24 = $[44];
	let t25;
	if ($[45] !== queue.stats.delayed || $[46] !== totalJobs) {
		t25 = (queue.stats.delayed > 0 || totalJobs > 0) && /* @__PURE__ */ jsxs("div", {
			className: "flex gap-2 mt-1 text-xs text-muted-foreground",
			children: [queue.stats.delayed > 0 && /* @__PURE__ */ jsxs("span", { children: [queue.stats.delayed, " delayed"] }), totalJobs > 0 && /* @__PURE__ */ jsxs("span", {
				className: "ml-auto",
				children: [totalJobs, " pending"]
			})]
		});
		$[45] = queue.stats.delayed;
		$[46] = totalJobs;
		$[47] = t25;
	} else t25 = $[47];
	let t26;
	if ($[48] !== onClick || $[49] !== t2 || $[50] !== t24 || $[51] !== t25 || $[52] !== t7) {
		t26 = /* @__PURE__ */ jsxs("button", {
			type: "button",
			onClick,
			className: t2,
			children: [
				t7,
				t24,
				t25
			]
		});
		$[48] = onClick;
		$[49] = t2;
		$[50] = t24;
		$[51] = t25;
		$[52] = t7;
		$[53] = t26;
	} else t26 = $[53];
	return t26;
});
QueueItem.displayName = "QueueItem";
const QueueList = memo(() => {
	const $ = c(38);
	if ($[0] !== "44e6229799ef8371f1e191c6f3e3ef70ea83c9489840337cf1d87e846b1bcc7c") {
		for (let $i = 0; $i < 38; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "44e6229799ef8371f1e191c6f3e3ef70ea83c9489840337cf1d87e846b1bcc7c";
	}
	const queues = useBullMQStore(_temp$1);
	const selectedQueue = useBullMQStore(_temp2$1);
	const setSelectedQueue = useBullMQStore(_temp3);
	const searchQuery = useBullMQStore(_temp4);
	const setSearchQuery = useBullMQStore(_temp5);
	let t0;
	bb0: {
		if (!searchQuery) {
			t0 = queues;
			break bb0;
		}
		let t1$1;
		if ($[1] !== queues || $[2] !== searchQuery) {
			const query = searchQuery.toLowerCase();
			t1$1 = queues.filter((q) => q.name.toLowerCase().includes(query) || q.displayName.toLowerCase().includes(query));
			$[1] = queues;
			$[2] = searchQuery;
			$[3] = t1$1;
		} else t1$1 = $[3];
		t0 = t1$1;
	}
	const filteredQueues = t0;
	let t1;
	if ($[4] !== filteredQueues) {
		t1 = filteredQueues.filter(_temp6);
		$[4] = filteredQueues;
		$[5] = t1;
	} else t1 = $[5];
	const regularQueues = t1;
	let t2;
	if ($[6] !== filteredQueues) {
		t2 = filteredQueues.filter(_temp7);
		$[6] = filteredQueues;
		$[7] = t2;
	} else t2 = $[7];
	const dlqQueues = t2;
	let t3;
	if ($[8] !== setSearchQuery) {
		t3 = (e) => setSearchQuery(e.target.value);
		$[8] = setSearchQuery;
		$[9] = t3;
	} else t3 = $[9];
	let t4;
	if ($[10] !== searchQuery || $[11] !== t3) {
		t4 = /* @__PURE__ */ jsx(Input, {
			variant: "shade",
			value: searchQuery,
			onChange: t3,
			className: "px-9! font-medium text-sm",
			placeholder: "Search queues..."
		});
		$[10] = searchQuery;
		$[11] = t3;
		$[12] = t4;
	} else t4 = $[12];
	let t5;
	if ($[13] === Symbol.for("react.memo_cache_sentinel")) {
		t5 = /* @__PURE__ */ jsx(Search, { className: "absolute left-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground/50" });
		$[13] = t5;
	} else t5 = $[13];
	let t6;
	if ($[14] !== searchQuery || $[15] !== setSearchQuery) {
		t6 = searchQuery && /* @__PURE__ */ jsx(X, {
			className: "cursor-pointer absolute right-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground/50 hover:text-muted-foreground",
			onClick: () => setSearchQuery("")
		});
		$[14] = searchQuery;
		$[15] = setSearchQuery;
		$[16] = t6;
	} else t6 = $[16];
	let t7;
	if ($[17] !== t4 || $[18] !== t6) {
		t7 = /* @__PURE__ */ jsx("div", {
			className: "p-2 border-b border-border",
			children: /* @__PURE__ */ jsxs("div", {
				className: "relative",
				children: [
					t4,
					t5,
					t6
				]
			})
		});
		$[17] = t4;
		$[18] = t6;
		$[19] = t7;
	} else t7 = $[19];
	let t8;
	if ($[20] !== regularQueues || $[21] !== selectedQueue?.name || $[22] !== setSelectedQueue) {
		t8 = regularQueues.length > 0 && /* @__PURE__ */ jsxs("div", { children: [/* @__PURE__ */ jsxs("div", {
			className: "px-3 py-2 text-xs font-semibold text-muted-foreground uppercase tracking-wider bg-muted/30",
			children: [
				"Queues (",
				regularQueues.length,
				")"
			]
		}), regularQueues.map((queue) => /* @__PURE__ */ jsx(QueueItem, {
			queue,
			isSelected: selectedQueue?.name === queue.name,
			onClick: () => setSelectedQueue(queue)
		}, queue.name))] });
		$[20] = regularQueues;
		$[21] = selectedQueue?.name;
		$[22] = setSelectedQueue;
		$[23] = t8;
	} else t8 = $[23];
	let t9;
	if ($[24] !== dlqQueues || $[25] !== selectedQueue?.name || $[26] !== setSelectedQueue) {
		t9 = dlqQueues.length > 0 && /* @__PURE__ */ jsxs("div", { children: [/* @__PURE__ */ jsxs("div", {
			className: "px-3 py-2 text-xs font-semibold text-muted-foreground uppercase tracking-wider bg-destructive/10",
			children: [
				"Dead Letter Queues (",
				dlqQueues.length,
				")"
			]
		}), dlqQueues.map((queue_0) => /* @__PURE__ */ jsx(QueueItem, {
			queue: queue_0,
			isSelected: selectedQueue?.name === queue_0.name,
			onClick: () => setSelectedQueue(queue_0)
		}, queue_0.name))] });
		$[24] = dlqQueues;
		$[25] = selectedQueue?.name;
		$[26] = setSelectedQueue;
		$[27] = t9;
	} else t9 = $[27];
	let t10;
	if ($[28] !== filteredQueues.length || $[29] !== searchQuery) {
		t10 = filteredQueues.length === 0 && /* @__PURE__ */ jsx("div", {
			className: "p-4 text-center text-muted-foreground text-sm",
			children: searchQuery ? "No queues match your search" : "No queues found"
		});
		$[28] = filteredQueues.length;
		$[29] = searchQuery;
		$[30] = t10;
	} else t10 = $[30];
	let t11;
	if ($[31] !== t10 || $[32] !== t8 || $[33] !== t9) {
		t11 = /* @__PURE__ */ jsxs("div", {
			className: "flex-1 overflow-auto",
			children: [
				t8,
				t9,
				t10
			]
		});
		$[31] = t10;
		$[32] = t8;
		$[33] = t9;
		$[34] = t11;
	} else t11 = $[34];
	let t12;
	if ($[35] !== t11 || $[36] !== t7) {
		t12 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col h-full",
			children: [t7, t11]
		});
		$[35] = t11;
		$[36] = t7;
		$[37] = t12;
	} else t12 = $[37];
	return t12;
});
QueueList.displayName = "QueueList";
function _temp$1(state) {
	return state.queues;
}
function _temp2$1(state_0) {
	return state_0.selectedQueue;
}
function _temp3(state_1) {
	return state_1.setSelectedQueue;
}
function _temp4(state_2) {
	return state_2.searchQuery;
}
function _temp5(state_3) {
	return state_3.setSearchQuery;
}
function _temp6(q_0) {
	return !q_0.isDLQ;
}
function _temp7(q_1) {
	return q_1.isDLQ;
}

//#endregion
//#region src/components/queues-page.tsx
const QueuesPageContent = memo(() => {
	const $ = c(12);
	if ($[0] !== "d90f8688a2f661493c4a0312f1c954fe541dc3821309a6ba650a56e59fcac492") {
		for (let $i = 0; $i < 12; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "d90f8688a2f661493c4a0312f1c954fe541dc3821309a6ba650a56e59fcac492";
	}
	const selectedQueue = useBullMQStore(_temp);
	const updateSelectedQueueStats = useBullMQStore(_temp2);
	const { queues } = useQueues();
	let t0;
	let t1;
	if ($[1] !== queues || $[2] !== selectedQueue || $[3] !== updateSelectedQueueStats) {
		t0 = () => {
			if (selectedQueue) {
				const updatedQueue = queues.find((q) => q.name === selectedQueue.name);
				if (updatedQueue) updateSelectedQueueStats(updatedQueue);
			}
		};
		t1 = [
			queues,
			selectedQueue,
			updateSelectedQueueStats
		];
		$[1] = queues;
		$[2] = selectedQueue;
		$[3] = updateSelectedQueueStats;
		$[4] = t0;
		$[5] = t1;
	} else {
		t0 = $[4];
		t1 = $[5];
	}
	useEffect(t0, t1);
	let t2;
	if ($[6] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: "border-r border-border overflow-hidden",
			children: /* @__PURE__ */ jsx(QueueList, {})
		});
		$[6] = t2;
	} else t2 = $[6];
	let t3;
	if ($[7] !== selectedQueue?.isDLQ) {
		t3 = /* @__PURE__ */ jsx("div", {
			className: "overflow-hidden",
			children: selectedQueue?.isDLQ ? /* @__PURE__ */ jsx(DLQPanel, {}) : /* @__PURE__ */ jsx(QueueDetail, {})
		});
		$[7] = selectedQueue?.isDLQ;
		$[8] = t3;
	} else t3 = $[8];
	let t4;
	if ($[9] === Symbol.for("react.memo_cache_sentinel")) {
		t4 = /* @__PURE__ */ jsx(JobDetail, {});
		$[9] = t4;
	} else t4 = $[9];
	let t5;
	if ($[10] !== t3) {
		t5 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-cols-[300px_1fr] h-full overflow-hidden",
			children: [
				t2,
				t3,
				t4
			]
		});
		$[10] = t3;
		$[11] = t5;
	} else t5 = $[11];
	return t5;
});
QueuesPageContent.displayName = "QueuesPageContent";
const QueuesPage = memo(() => {
	const $ = c(2);
	if ($[0] !== "d90f8688a2f661493c4a0312f1c954fe541dc3821309a6ba650a56e59fcac492") {
		for (let $i = 0; $i < 2; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "d90f8688a2f661493c4a0312f1c954fe541dc3821309a6ba650a56e59fcac492";
	}
	let t0;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t0 = /* @__PURE__ */ jsx(QueryProvider, { children: /* @__PURE__ */ jsx(QueuesPageContent, {}) });
		$[1] = t0;
	} else t0 = $[1];
	return t0;
});
QueuesPage.displayName = "QueuesPage";
function _temp(state) {
	return state.selectedQueue;
}
function _temp2(state_0) {
	return state_0.updateSelectedQueueStats;
}

//#endregion
export { QueuesPage };
