// @motiadev/core/dist/src/streams/flows-config-stream.mjs
import { StreamAdapter } from "../adapters/interfaces/stream-adapter.interface.mjs";
import fs from "fs";

//#region src/streams/flows-config-stream.ts
var FlowsConfigStream = class extends StreamAdapter {
	constructor(configPath) {
		super("__motia.flows-config");
		this.configPath = configPath;
		this.config = [];
	}
	getConfig() {
		if (this.config.length === 0) {
			if (fs.existsSync(this.configPath)) {
				this.config = JSON.parse(fs.readFileSync(this.configPath, "utf8") || "[]");
				if (!Array.isArray(this.config)) {
					this.config = [];
					this.setConfig(this.config);
				}
			}
		}
		return this.config;
	}
	setConfig(config) {
		this.config = config;
		fs.writeFileSync(this.configPath, JSON.stringify(config, null, 2), "utf8");
	}
	async get(_, id) {
		return this.getConfig().find((flow) => flow.id === id) || null;
	}
	async delete(_) {
		return null;
	}
	async set(_, id, data) {
		const updatedConfig = this.getConfig().filter((item) => item.id !== id);
		updatedConfig.push(data);
		this.setConfig(updatedConfig);
		return data;
	}
	async getGroup() {
		return this.getConfig();
	}
};

//#endregion
export { FlowsConfigStream };
