// @motiadev/core/dist/src/endpoints/flows-endpoint.mjs
import { generateFlow } from "../helper/flows-helper.mjs";
import { FlowsStream } from "../streams/flows-stream.mjs";
import { z } from "zod";

//#region src/endpoints/flows-endpoint.ts
const flowsEndpoint = (lockedData) => {
	const flowsStream = lockedData.createStream({
		filePath: "__motia.flows",
		hidden: true,
		config: {
			name: "__motia.flows",
			schema: z.object({
				id: z.string(),
				name: z.string(),
				steps: z.any(),
				edges: z.any()
			}),
			baseConfig: {
				storageType: "custom",
				factory: () => new FlowsStream(lockedData)
			}
		}
	}, { disableTypeCreation: true })();
	lockedData.on("flow-created", (flowId) => {
		const flow = lockedData.flows[flowId];
		const response = generateFlow(flowId, flow.steps);
		flowsStream.set("default", flowId, response);
	});
	lockedData.on("flow-updated", (flowId) => {
		const flow = lockedData.flows[flowId];
		const response = generateFlow(flowId, flow.steps);
		flowsStream.set("default", flowId, response);
	});
	lockedData.on("flow-removed", (flowId) => flowsStream.delete("default", flowId));
};

//#endregion
export { flowsEndpoint };
