// @motiadev/core/dist/src/endpoints/flows-config-endpoint.mjs
import { FlowsConfigStream } from "../streams/flows-config-stream.mjs";
import path from "path";
import { z } from "zod";

//#region src/endpoints/flows-config-endpoint.ts
const flowsConfigEndpoint = (app, baseDir, lockedData) => {
	const stream = new FlowsConfigStream(path.join(baseDir, "motia-workbench.json"));
	lockedData.createStream({
		filePath: "__motia.flowsConfig",
		hidden: true,
		config: {
			name: "__motia.flowsConfig",
			schema: z.object({
				name: z.string(),
				steps: z.any(),
				edges: z.any()
			}),
			baseConfig: {
				storageType: "custom",
				factory: () => stream
			}
		}
	}, { disableTypeCreation: true })();
	app.post("/__motia/flows/:id/config", async (req, res) => {
		const newFlowConfig = req.body;
		try {
			await stream.set("default", newFlowConfig.id, newFlowConfig);
			res.status(200).send({ message: "Flow config saved successfully" });
		} catch (error) {
			console.error("Error saving flow config:", error.message);
			res.status(500).json({ error: "Failed to save flow config" });
		}
	});
};

//#endregion
export { flowsConfigEndpoint };
