// motia/dist/cloud/build/builders/python/python-data/get-dependencies-from-file.mjs
import { STANDARD_LIB_MODULES } from "./constants.mjs";
import { PythonCompilationError } from "./python-errors.mjs";
import { CharStreams, CommonTokenStream } from "antlr4ts";
import { createVisitor } from "python-ast";
import { Python3Lexer } from "python-ast/dist/parser/Python3Lexer.js";
import { Python3Parser } from "python-ast/dist/parser/Python3Parser.js";

//#region src/cloud/build/builders/python/python-data/get-dependencies-from-file.ts
function parse(source, sourceName) {
	const parser = new Python3Parser(new CommonTokenStream(new Python3Lexer(CharStreams.fromString(source, sourceName))));
	parser.removeErrorListeners();
	parser.addErrorListener({ syntaxError: (_recognizer, _offendingSymbol, line, charPositionInLine, msg) => {
		throw new PythonCompilationError(sourceName, `${msg} at line ${line}:${charPositionInLine}`);
	} });
	return parser.file_input();
}
const getDependenciesFromFile = (content, path, externalDependenciesMap) => {
	const result = parse(content + "\n", path);
	const modulesSet = /* @__PURE__ */ new Set();
	createVisitor({
		visitImport_from: (ctx) => {
			const name = ctx.dotted_name();
			if (name) {
				let dots = ctx.DOT().map((dot) => dot.text).join("");
				const dotsText = ctx.getChild(1)?.text ?? "";
				if (!dots && dotsText[0] === ".") dots = dotsText;
				modulesSet.add(dots.concat(name.text));
			}
		},
		visitImport_name: (ctx) => {
			const names = ctx.dotted_as_names();
			modulesSet.add(names.dotted_as_name(0)?.getChild(0)?.text ?? names.text);
		}
	}).visit(result);
	const dependencies = {
		standardLibDependencies: /* @__PURE__ */ new Set(),
		externalDependencies: /* @__PURE__ */ new Set(),
		projectDependencies: /* @__PURE__ */ new Set()
	};
	for (const module of modulesSet) {
		const [moduleName] = module.split(".");
		if (module[0] === ".") dependencies.projectDependencies.add(module);
		else if (STANDARD_LIB_MODULES.has(module)) dependencies.standardLibDependencies.add(module);
		else if (externalDependenciesMap[module] || externalDependenciesMap[moduleName]) dependencies.externalDependencies.add(module);
		else dependencies.projectDependencies.add(module);
	}
	return dependencies;
};

//#endregion
export { getDependenciesFromFile };
