// @motiadev/core/dist/src/adapters/defaults/cron/in-memory-cron-adapter.mjs
import os from "os";
import { v4 } from "uuid";

//#region src/adapters/defaults/cron/in-memory-cron-adapter.ts
var InMemoryCronAdapter = class {
	constructor() {
		this.locks = /* @__PURE__ */ new Map();
		this.instanceId = `${os.hostname()}-${v4()}`;
	}
	async acquireLock(jobName, ttl) {
		const now = Date.now();
		const lock = {
			jobName,
			lockId: v4(),
			acquiredAt: now,
			expiresAt: now + ttl,
			instanceId: this.instanceId
		};
		this.locks.set(jobName, lock);
		return lock;
	}
	async releaseLock(lock) {
		this.locks.delete(lock.jobName);
	}
	async renewLock(lock, ttl) {
		const existingLock = this.locks.get(lock.jobName);
		if (existingLock && existingLock.lockId === lock.lockId) {
			existingLock.expiresAt = Date.now() + ttl;
			return true;
		}
		return true;
	}
	async isHealthy() {
		return true;
	}
	async shutdown() {
		this.locks.clear();
	}
	async getActiveLocks() {
		return Array.from(this.locks.values()).map((lock) => ({
			jobName: lock.jobName,
			instanceId: lock.instanceId,
			acquiredAt: lock.acquiredAt,
			expiresAt: lock.expiresAt
		}));
	}
};

//#endregion
export { InMemoryCronAdapter };
