// @motiadev/ui/dist/components/ui/badge.js
import { cn } from "../../lib/utils.js";
import { c } from "react/compiler-runtime";
import { jsx } from "react/jsx-runtime";
import { cva } from "class-variance-authority";

//#region src/components/ui/badge.tsx
const badgeVariants = cva("inline-flex items-center rounded-full px-4 py-1 text-xs font-bold transition-colors", {
	variants: { variant: {
		info: "dark:bg-accent-100 dark:text-accent-1000 bg-accent-200 text-accent-900 capitalize",
		success: "bg-accent-1000 text-white",
		error: "bg-destructive/10 text-destructive capitalize",
		default: "dark:bg-gray-800/30 dark:text-gray-500 bg-gray-100 text-gray-800",
		outline: "border border-gray-200 dark:border-gray-700",
		warning: "border-transparent bg-amber-300 text-amber-950 hover:bg-amber/80"
	} },
	defaultVariants: { variant: "default" }
});
const Badge = (t0) => {
	const $ = c(11);
	if ($[0] !== "16ccda96cbfb6921771b65e8b9c3472148a756b8545b2ac6fa3ecbfb332c1bd6") {
		for (let $i = 0; $i < 11; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "16ccda96cbfb6921771b65e8b9c3472148a756b8545b2ac6fa3ecbfb332c1bd6";
	}
	let className;
	let props;
	let variant;
	if ($[1] !== t0) {
		({className, variant, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
		$[4] = variant;
	} else {
		className = $[2];
		props = $[3];
		variant = $[4];
	}
	let t1;
	if ($[5] !== className || $[6] !== variant) {
		t1 = cn(badgeVariants({ variant }), className);
		$[5] = className;
		$[6] = variant;
		$[7] = t1;
	} else t1 = $[7];
	let t2;
	if ($[8] !== props || $[9] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: t1,
			...props
		});
		$[8] = props;
		$[9] = t1;
		$[10] = t2;
	} else t2 = $[10];
	return t2;
};
Badge.displayName = "Badge";

//#endregion
export { Badge };
