// @motiadev/ui/dist/components/ui/breadcrumb.js
import { cn } from "../../lib/utils.js";
import { Button } from "./button.js";
import { DropdownMenu, DropdownMenuContent, DropdownMenuItem, DropdownMenuTrigger } from "./dropdown-menu.js";
import { c } from "react/compiler-runtime";
import { jsx, jsxs } from "react/jsx-runtime";
import { ChevronsUpDown } from "lucide-react";
import * as React$1 from "react";

//#region src/components/ui/breadcrumb.tsx
const BreadcrumbItem = (t0) => {
	const $ = c(19);
	if ($[0] !== "ce99450bed3ba349f13f3a196f9e2966c468ab2f6e5f14a8811670951dfe38e3") {
		for (let $i = 0; $i < 19; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "ce99450bed3ba349f13f3a196f9e2966c468ab2f6e5f14a8811670951dfe38e3";
	}
	const { label, onClick, isLast, dropdownItems, icon } = t0;
	if (dropdownItems?.length) {
		let t1$1;
		if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
			t1$1 = /* @__PURE__ */ jsx(ChevronsUpDown, { className: "size-4" });
			$[1] = t1$1;
		} else t1$1 = $[1];
		let t2;
		if ($[2] !== icon || $[3] !== label) {
			t2 = /* @__PURE__ */ jsx(DropdownMenuTrigger, {
				asChild: true,
				children: /* @__PURE__ */ jsxs(Button, {
					variant: "ghost",
					size: "sm",
					className: "p-1 gap-2",
					children: [
						icon,
						label,
						" ",
						t1$1
					]
				})
			});
			$[2] = icon;
			$[3] = label;
			$[4] = t2;
		} else t2 = $[4];
		let t3;
		if ($[5] !== dropdownItems) {
			t3 = dropdownItems.map(_temp);
			$[5] = dropdownItems;
			$[6] = t3;
		} else t3 = $[6];
		let t4;
		if ($[7] !== t3) {
			t4 = /* @__PURE__ */ jsx(DropdownMenuContent, {
				className: "bg-background text-foreground",
				children: t3
			});
			$[7] = t3;
			$[8] = t4;
		} else t4 = $[8];
		let t5;
		if ($[9] !== t2 || $[10] !== t4) {
			t5 = /* @__PURE__ */ jsxs(DropdownMenu, { children: [t2, t4] });
			$[9] = t2;
			$[10] = t4;
			$[11] = t5;
		} else t5 = $[11];
		return t5;
	}
	if (isLast) {
		let t1$1;
		if ($[12] !== icon || $[13] !== label) {
			t1$1 = /* @__PURE__ */ jsx(Button, {
				variant: "ghost",
				size: "sm",
				className: "cursor-default hover:bg-transparent hover:text-muted-transparent p-1",
				asChild: true,
				children: /* @__PURE__ */ jsxs("div", {
					className: "flex items-center gap-2",
					children: [icon, label]
				})
			});
			$[12] = icon;
			$[13] = label;
			$[14] = t1$1;
		} else t1$1 = $[14];
		return t1$1;
	}
	let t1;
	if ($[15] !== icon || $[16] !== label || $[17] !== onClick) {
		t1 = /* @__PURE__ */ jsxs(Button, {
			onClick,
			variant: "ghost",
			size: "sm",
			className: "p-1 gap-2",
			children: [icon, label]
		});
		$[15] = icon;
		$[16] = label;
		$[17] = onClick;
		$[18] = t1;
	} else t1 = $[18];
	return t1;
};
BreadcrumbItem.displayName = "BreadcrumbItem";
const Breadcrumb = (t0) => {
	const $ = c(16);
	if ($[0] !== "ce99450bed3ba349f13f3a196f9e2966c468ab2f6e5f14a8811670951dfe38e3") {
		for (let $i = 0; $i < 16; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "ce99450bed3ba349f13f3a196f9e2966c468ab2f6e5f14a8811670951dfe38e3";
	}
	let className;
	let items;
	let props;
	if ($[1] !== t0) {
		({className, items, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = items;
		$[4] = props;
	} else {
		className = $[2];
		items = $[3];
		props = $[4];
	}
	const lastIndex = items.length - 1;
	let t1;
	if ($[5] !== className) {
		t1 = cn("flex items-center", className);
		$[5] = className;
		$[6] = t1;
	} else t1 = $[6];
	let t2;
	if ($[7] !== items || $[8] !== lastIndex) {
		let t3$1;
		if ($[10] !== lastIndex) {
			t3$1 = (item, index) => /* @__PURE__ */ jsxs(React$1.Fragment, { children: [index !== 0 && /* @__PURE__ */ jsx("span", {
				className: "mx-1.5 text-muted-foreground",
				children: "/"
			}, `${item.label}-separator`), /* @__PURE__ */ jsx(BreadcrumbItem, {
				label: item.label,
				icon: item.icon,
				isLast: index === lastIndex,
				dropdownItems: item.dropdownItems
			}, `${item.label}-item`)] }, `${item.label}-fragment`);
			$[10] = lastIndex;
			$[11] = t3$1;
		} else t3$1 = $[11];
		t2 = items.map(t3$1);
		$[7] = items;
		$[8] = lastIndex;
		$[9] = t2;
	} else t2 = $[9];
	let t3;
	if ($[12] !== props || $[13] !== t1 || $[14] !== t2) {
		t3 = /* @__PURE__ */ jsx("div", {
			className: t1,
			...props,
			children: t2
		});
		$[12] = props;
		$[13] = t1;
		$[14] = t2;
		$[15] = t3;
	} else t3 = $[15];
	return t3;
};
Breadcrumb.displayName = "Breadcrumb";
function _temp(item) {
	return /* @__PURE__ */ jsxs(DropdownMenuItem, {
		className: "cursor-pointer gap-2",
		onClick: item.onClick,
		children: [item.icon, item.label]
	}, `dropdown-${item.label}`);
}

//#endregion
export { Breadcrumb, BreadcrumbItem };
