// @motiadev/ui/dist/components/ui/button.js
import { cn } from "../../lib/utils.js";
import { c } from "react/compiler-runtime";
import { jsx } from "react/jsx-runtime";
import { cva } from "class-variance-authority";
import { forwardRef } from "react";
import { Slot } from "@radix-ui/react-slot";

//#region src/components/ui/button.tsx
const buttonVariants = cva("inline-flex items-center cursor-pointer justify-center gap-2 whitespace-nowrap rounded-md font-semibold transition-all duration-200 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0", {
	variants: {
		variant: {
			default: "bg-input text-foreground hover:bg-border active:bg-card",
			accent: "bg-accent text-accent-foreground hover:bg-accent/90 active:bg-accent/80",
			light: "bg-background text-foreground border border-border hover:bg-muted active:bg-muted/80",
			secondary: "bg-secondary text-foreground shadow-sm hover:opacity-90 active:opacity-80",
			destructive: "bg-destructive text-destructive-foreground shadow-sm hover:opacity-90 active:opacity-80",
			outline: "border border-border bg-transparent hover:bg-muted hover:text-muted-foreground text-foreground",
			ghost: "hover:bg-muted-foreground/20 text-foreground",
			link: "text-primary underline-offset-4 hover:underline text-primary-foreground",
			icon: "hover:bg-muted-foreground/20 text-foreground rounded-full bg-muted-foreground/10"
		},
		size: {
			sm: "h-6 px-3 text-xs [&_svg]:size-3",
			default: "h-9 px-4 text-sm [&_svg]:size-4",
			md: "h-9 px-4 text-sm [&_svg]:size-4",
			lg: "h-11 px-6 text-base [&_svg]:size-5",
			icon: "h-4 w-4 [&_svg]:size-4"
		}
	},
	defaultVariants: {
		variant: "default",
		size: "default"
	}
});
const Button = forwardRef((t0, ref) => {
	const $ = c(16);
	if ($[0] !== "dc9d813c9a7cd4549d9765c8ac1c4cb94d9637ac4c1593debc6787ad7268f2a6") {
		for (let $i = 0; $i < 16; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dc9d813c9a7cd4549d9765c8ac1c4cb94d9637ac4c1593debc6787ad7268f2a6";
	}
	let className;
	let props;
	let size;
	let t1;
	let variant;
	if ($[1] !== t0) {
		({className, variant, size, asChild: t1, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
		$[4] = size;
		$[5] = t1;
		$[6] = variant;
	} else {
		className = $[2];
		props = $[3];
		size = $[4];
		t1 = $[5];
		variant = $[6];
	}
	const Comp = (t1 === void 0 ? false : t1) ? Slot : "button";
	let t2;
	if ($[7] !== className || $[8] !== size || $[9] !== variant) {
		t2 = cn(buttonVariants({
			variant,
			size
		}), className);
		$[7] = className;
		$[8] = size;
		$[9] = variant;
		$[10] = t2;
	} else t2 = $[10];
	let t3;
	if ($[11] !== Comp || $[12] !== props || $[13] !== ref || $[14] !== t2) {
		t3 = /* @__PURE__ */ jsx(Comp, {
			className: t2,
			ref,
			...props
		});
		$[11] = Comp;
		$[12] = props;
		$[13] = ref;
		$[14] = t2;
		$[15] = t3;
	} else t3 = $[15];
	return t3;
});
Button.displayName = "Button";

//#endregion
export { Button };
