// @motiadev/ui/dist/components/ui/collapsible-panel.js
import { cn } from "../../lib/utils.js";
import { Button } from "./button.js";
import { Container, ContainerContent, ContainerHeader } from "./container.js";
import { Tabs } from "./tabs.js";
import { c } from "react/compiler-runtime";
import { Fragment, jsx, jsxs } from "react/jsx-runtime";
import { ChevronDown, Equal } from "lucide-react";
import React, { useRef, useState } from "react";
import { Panel, PanelGroup, PanelResizeHandle } from "react-resizable-panels";

//#region src/components/ui/collapsible-panel.tsx
const PanelControls = (t0) => {
	const $ = c(10);
	if ($[0] !== "dcfed46e4c606b67cf4b4e5ebb8482758fc50e61c44c1de1b82c7a068d5c0653") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dcfed46e4c606b67cf4b4e5ebb8482758fc50e61c44c1de1b82c7a068d5c0653";
	}
	const { header, isCollapsed, onToggle, variant: t1 } = t0;
	const variant = t1 === void 0 ? "default" : t1;
	let t2;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx("div", { className: "flex-1" });
		$[1] = t2;
	} else t2 = $[1];
	let t3;
	if ($[2] !== isCollapsed || $[3] !== onToggle || $[4] !== variant) {
		t3 = onToggle && /* @__PURE__ */ jsx(Button, {
			variant: "default",
			onClick: onToggle,
			"aria-label": isCollapsed ? "Expand panel" : "Collapse panel",
			className: cn("h-8 px-3 [&_svg]:size-5", { "mr-5": variant === "tabs" }),
			children: /* @__PURE__ */ jsx(ChevronDown, { className: cn("transition-transform", { "-rotate-180": !isCollapsed }) })
		});
		$[2] = isCollapsed;
		$[3] = onToggle;
		$[4] = variant;
		$[5] = t3;
	} else t3 = $[5];
	let t4;
	if ($[6] !== header || $[7] !== t3 || $[8] !== variant) {
		t4 = /* @__PURE__ */ jsxs(ContainerHeader, {
			variant,
			children: [
				header,
				t2,
				t3
			]
		});
		$[6] = header;
		$[7] = t3;
		$[8] = variant;
		$[9] = t4;
	} else t4 = $[9];
	return t4;
};
const CustomResizeHandle = () => {
	const $ = c(2);
	if ($[0] !== "dcfed46e4c606b67cf4b4e5ebb8482758fc50e61c44c1de1b82c7a068d5c0653") {
		for (let $i = 0; $i < 2; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dcfed46e4c606b67cf4b4e5ebb8482758fc50e61c44c1de1b82c7a068d5c0653";
	}
	let t0;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t0 = /* @__PURE__ */ jsx("div", {
			className: "relative",
			children: /* @__PURE__ */ jsx("div", {
				className: "absolute inset-0 flex items-center justify-center",
				children: /* @__PURE__ */ jsx(PanelResizeHandle, {
					className: "group relative",
					children: /* @__PURE__ */ jsx("div", {
						className: "flex h-6 w-6 items-center justify-center rounded-full bg-background border border-border",
						children: /* @__PURE__ */ jsx(Equal, { className: "h-4 w-4 text-muted-foreground" })
					})
				})
			})
		});
		$[1] = t0;
	} else t0 = $[1];
	return t0;
};
const CollapsiblePanel = (t0) => {
	const $ = c(28);
	if ($[0] !== "dcfed46e4c606b67cf4b4e5ebb8482758fc50e61c44c1de1b82c7a068d5c0653") {
		for (let $i = 0; $i < 28; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dcfed46e4c606b67cf4b4e5ebb8482758fc50e61c44c1de1b82c7a068d5c0653";
	}
	const { id, header, children, withResizeHandle, className, variant, defaultTab, onTabChange } = t0;
	const [isCollapsed, setIsCollapsed] = useState(false);
	const panelRef = useRef(null);
	let t1;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = () => {
			const panel = panelRef.current;
			if (!panel) return;
			try {
				if (panel.isCollapsed()) panel.expand();
				else panel.collapse();
			} catch (t2$1) {
				const error = t2$1;
				console.warn("Failed to toggle panel:", error);
			}
		};
		$[1] = t1;
	} else t1 = $[1];
	const onToggle = t1;
	let t2;
	if ($[2] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = () => setIsCollapsed(true);
		$[2] = t2;
	} else t2 = $[2];
	const onCollapse = t2;
	let t3;
	if ($[3] === Symbol.for("react.memo_cache_sentinel")) {
		t3 = () => setIsCollapsed(false);
		$[3] = t3;
	} else t3 = $[3];
	const onExpand = t3;
	let t4;
	bb0: {
		let t5$1;
		if ($[4] !== header || $[5] !== isCollapsed || $[6] !== variant) {
			t5$1 = /* @__PURE__ */ jsx(PanelControls, {
				header,
				isCollapsed,
				variant,
				onToggle
			});
			$[4] = header;
			$[5] = isCollapsed;
			$[6] = variant;
			$[7] = t5$1;
		} else t5$1 = $[7];
		let t6$1;
		if ($[8] !== children) {
			t6$1 = /* @__PURE__ */ jsx(ContainerContent, { children });
			$[8] = children;
			$[9] = t6$1;
		} else t6$1 = $[9];
		let t7$1;
		if ($[10] !== t5$1 || $[11] !== t6$1) {
			t7$1 = /* @__PURE__ */ jsxs(Container, {
				className: "h-full",
				children: [t5$1, t6$1]
			});
			$[10] = t5$1;
			$[11] = t6$1;
			$[12] = t7$1;
		} else t7$1 = $[12];
		const container = t7$1;
		if (variant == "tabs") {
			let t8$1;
			if ($[13] !== container || $[14] !== defaultTab || $[15] !== onTabChange) {
				t8$1 = /* @__PURE__ */ jsx(Tabs, {
					className: "h-full",
					defaultValue: defaultTab,
					onValueChange: onTabChange,
					children: container
				});
				$[13] = container;
				$[14] = defaultTab;
				$[15] = onTabChange;
				$[16] = t8$1;
			} else t8$1 = $[16];
			t4 = t8$1;
			break bb0;
		}
		t4 = container;
	}
	const view = t4;
	let t5;
	if ($[17] !== className) {
		t5 = cn("min-h-[42px]", className);
		$[17] = className;
		$[18] = t5;
	} else t5 = $[18];
	let t6;
	if ($[19] !== id || $[20] !== t5 || $[21] !== view) {
		t6 = /* @__PURE__ */ jsx(Panel, {
			id,
			collapsible: true,
			ref: panelRef,
			className: t5,
			onCollapse,
			onExpand,
			children: view
		});
		$[19] = id;
		$[20] = t5;
		$[21] = view;
		$[22] = t6;
	} else t6 = $[22];
	let t7;
	if ($[23] !== withResizeHandle) {
		t7 = withResizeHandle && /* @__PURE__ */ jsx(CustomResizeHandle, {});
		$[23] = withResizeHandle;
		$[24] = t7;
	} else t7 = $[24];
	let t8;
	if ($[25] !== t6 || $[26] !== t7) {
		t8 = /* @__PURE__ */ jsxs(Fragment, { children: [t6, t7] });
		$[25] = t6;
		$[26] = t7;
		$[27] = t8;
	} else t8 = $[27];
	return t8;
};
const CollapsiblePanelGroup = (t0) => {
	const $ = c(13);
	if ($[0] !== "dcfed46e4c606b67cf4b4e5ebb8482758fc50e61c44c1de1b82c7a068d5c0653") {
		for (let $i = 0; $i < 13; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "dcfed46e4c606b67cf4b4e5ebb8482758fc50e61c44c1de1b82c7a068d5c0653";
	}
	let children;
	let props;
	if ($[1] !== t0) {
		({children, ...props} = t0);
		$[1] = t0;
		$[2] = children;
		$[3] = props;
	} else {
		children = $[2];
		props = $[3];
	}
	const [resizeHandleCount, setResizeHandleCount] = useState(0);
	let t1;
	if ($[4] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = (sizes) => {
			setResizeHandleCount(sizes.length);
		};
		$[4] = t1;
	} else t1 = $[4];
	let t2;
	if ($[5] !== children || $[6] !== resizeHandleCount) {
		let t3$1;
		if ($[8] !== resizeHandleCount) {
			t3$1 = (child, index) => {
				if (React.isValidElement(child)) {
					const last = resizeHandleCount - 1 === index;
					const hasResizeHandle = resizeHandleCount > 1;
					return React.cloneElement(child, { withResizeHandle: hasResizeHandle && !last });
				}
				return child;
			};
			$[8] = resizeHandleCount;
			$[9] = t3$1;
		} else t3$1 = $[9];
		t2 = React.Children.map(children, t3$1);
		$[5] = children;
		$[6] = resizeHandleCount;
		$[7] = t2;
	} else t2 = $[7];
	let t3;
	if ($[10] !== props || $[11] !== t2) {
		t3 = /* @__PURE__ */ jsx(PanelGroup, {
			onLayout: t1,
			...props,
			children: t2
		});
		$[10] = props;
		$[11] = t2;
		$[12] = t3;
	} else t3 = $[12];
	return t3;
};

//#endregion
export { CollapsiblePanel, CollapsiblePanelGroup };
