// @motiadev/ui/dist/components/ui/container.js
import { BackgroundEffect } from "./background-effect.js";
import { cn } from "../../lib/utils.js";
import { c } from "react/compiler-runtime";
import { jsx, jsxs } from "react/jsx-runtime";
import { cva } from "class-variance-authority";
import { forwardRef } from "react";

//#region src/components/ui/container.tsx
const Container = forwardRef((t0, ref) => {
	const $ = c(12);
	if ($[0] !== "8800013f57da6767f5e0125610a795ba251523cf41afeeea0834c2ee7f15f882") {
		for (let $i = 0; $i < 12; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "8800013f57da6767f5e0125610a795ba251523cf41afeeea0834c2ee7f15f882";
	}
	let children;
	let className;
	let props;
	if ($[1] !== t0) {
		({className, children, ...props} = t0);
		$[1] = t0;
		$[2] = children;
		$[3] = className;
		$[4] = props;
	} else {
		children = $[2];
		className = $[3];
		props = $[4];
	}
	let t1;
	if ($[5] !== className) {
		t1 = cn("relative border border-border text-foreground rounded-lg overflow-hidden flex flex-col h-full", className);
		$[5] = className;
		$[6] = t1;
	} else t1 = $[6];
	let t2;
	if ($[7] !== children || $[8] !== props || $[9] !== ref || $[10] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			ref,
			className: t1,
			...props,
			children
		});
		$[7] = children;
		$[8] = props;
		$[9] = ref;
		$[10] = t1;
		$[11] = t2;
	} else t2 = $[11];
	return t2;
});
Container.displayName = "Container";
const containerHeaderVariants = cva("isolate w-full bg-card flex min-h-10 items-center", {
	variants: { variant: {
		default: "px-5 py-2 items-center",
		tabs: "p-0"
	} },
	defaultVariants: { variant: "default" }
});
const ContainerHeader = forwardRef((t0, ref) => {
	const $ = c(15);
	if ($[0] !== "8800013f57da6767f5e0125610a795ba251523cf41afeeea0834c2ee7f15f882") {
		for (let $i = 0; $i < 15; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "8800013f57da6767f5e0125610a795ba251523cf41afeeea0834c2ee7f15f882";
	}
	let children;
	let className;
	let props;
	let variant;
	if ($[1] !== t0) {
		({className, variant, children, ...props} = t0);
		$[1] = t0;
		$[2] = children;
		$[3] = className;
		$[4] = props;
		$[5] = variant;
	} else {
		children = $[2];
		className = $[3];
		props = $[4];
		variant = $[5];
	}
	let t1;
	if ($[6] !== className || $[7] !== variant) {
		t1 = cn(containerHeaderVariants({
			variant,
			className
		}));
		$[6] = className;
		$[7] = variant;
		$[8] = t1;
	} else t1 = $[8];
	let t2;
	if ($[9] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx(BackgroundEffect, {});
		$[9] = t2;
	} else t2 = $[9];
	let t3;
	if ($[10] !== children || $[11] !== props || $[12] !== ref || $[13] !== t1) {
		t3 = /* @__PURE__ */ jsxs("div", {
			ref,
			className: t1,
			...props,
			children: [t2, children]
		});
		$[10] = children;
		$[11] = props;
		$[12] = ref;
		$[13] = t1;
		$[14] = t3;
	} else t3 = $[14];
	return t3;
});
ContainerHeader.displayName = "ContainerHeader";
const ContainerContent = forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "8800013f57da6767f5e0125610a795ba251523cf41afeeea0834c2ee7f15f882") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "8800013f57da6767f5e0125610a795ba251523cf41afeeea0834c2ee7f15f882";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("flex-1 overflow-auto bg-default border-border border-t", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			ref,
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
ContainerContent.displayName = "ContainerContent";

//#endregion
export { Container, ContainerContent, ContainerHeader };
