// @motiadev/ui/dist/components/ui/dropdown-menu.js
import { cn } from "../../lib/utils.js";
import { c } from "react/compiler-runtime";
import { jsx, jsxs } from "react/jsx-runtime";
import { CheckIcon, ChevronRightIcon, CircleIcon } from "lucide-react";
import * as DropdownMenuPrimitive from "@radix-ui/react-dropdown-menu";

//#region src/components/ui/dropdown-menu.tsx
const DropdownMenu = (t0) => {
	const $ = c(5);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 5; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let props;
	if ($[1] !== t0) {
		({...props} = t0);
		$[1] = t0;
		$[2] = props;
	} else props = $[2];
	let t1;
	if ($[3] !== props) {
		t1 = /* @__PURE__ */ jsx(DropdownMenuPrimitive.Root, {
			"data-slot": "dropdown-menu",
			...props
		});
		$[3] = props;
		$[4] = t1;
	} else t1 = $[4];
	return t1;
};
const DropdownMenuTrigger = (t0) => {
	const $ = c(5);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 5; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let props;
	if ($[1] !== t0) {
		({...props} = t0);
		$[1] = t0;
		$[2] = props;
	} else props = $[2];
	let t1;
	if ($[3] !== props) {
		t1 = /* @__PURE__ */ jsx(DropdownMenuPrimitive.Trigger, {
			"data-slot": "dropdown-menu-trigger",
			...props
		});
		$[3] = props;
		$[4] = t1;
	} else t1 = $[4];
	return t1;
};
const DropdownMenuContent = (t0) => {
	const $ = c(11);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 11; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let className;
	let props;
	let t1;
	if ($[1] !== t0) {
		({className, sideOffset: t1, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
		$[4] = t1;
	} else {
		className = $[2];
		props = $[3];
		t1 = $[4];
	}
	const sideOffset = t1 === void 0 ? 4 : t1;
	let t2;
	if ($[5] !== className) {
		t2 = cn("bg-popover text-popover-foreground data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 max-h-(--radix-dropdown-menu-content-available-height) min-w-[8rem] origin-(--radix-dropdown-menu-content-transform-origin) overflow-x-hidden overflow-y-auto rounded-md border p-1 shadow-md", className);
		$[5] = className;
		$[6] = t2;
	} else t2 = $[6];
	let t3;
	if ($[7] !== props || $[8] !== sideOffset || $[9] !== t2) {
		t3 = /* @__PURE__ */ jsx(DropdownMenuPrimitive.Portal, { children: /* @__PURE__ */ jsx(DropdownMenuPrimitive.Content, {
			"data-slot": "dropdown-menu-content",
			sideOffset,
			className: t2,
			...props
		}) });
		$[7] = props;
		$[8] = sideOffset;
		$[9] = t2;
		$[10] = t3;
	} else t3 = $[10];
	return t3;
};
const DropdownMenuGroup = (t0) => {
	const $ = c(5);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 5; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let props;
	if ($[1] !== t0) {
		({...props} = t0);
		$[1] = t0;
		$[2] = props;
	} else props = $[2];
	let t1;
	if ($[3] !== props) {
		t1 = /* @__PURE__ */ jsx(DropdownMenuPrimitive.Group, {
			"data-slot": "dropdown-menu-group",
			...props
		});
		$[3] = props;
		$[4] = t1;
	} else t1 = $[4];
	return t1;
};
const DropdownMenuItem = (t0) => {
	const $ = c(13);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 13; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let className;
	let inset;
	let props;
	let t1;
	if ($[1] !== t0) {
		({className, inset, variant: t1, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = inset;
		$[4] = props;
		$[5] = t1;
	} else {
		className = $[2];
		inset = $[3];
		props = $[4];
		t1 = $[5];
	}
	const variant = t1 === void 0 ? "default" : t1;
	let t2;
	if ($[6] !== className) {
		t2 = cn("focus:bg-accent focus:text-accent-foreground data-[variant=destructive]:text-destructive data-[variant=destructive]:focus:bg-destructive/10 dark:data-[variant=destructive]:focus:bg-destructive/20 data-[variant=destructive]:focus:text-destructive data-[variant=destructive]:*:[svg]:!text-destructive [&_svg:not([class*='text-'])]:text-muted-foreground relative flex cursor-default items-center gap-2 rounded-sm px-2 py-1.5 text-sm outline-hidden select-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 data-[inset]:pl-8 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4", className);
		$[6] = className;
		$[7] = t2;
	} else t2 = $[7];
	let t3;
	if ($[8] !== inset || $[9] !== props || $[10] !== t2 || $[11] !== variant) {
		t3 = /* @__PURE__ */ jsx(DropdownMenuPrimitive.Item, {
			"data-slot": "dropdown-menu-item",
			"data-inset": inset,
			"data-variant": variant,
			className: t2,
			...props
		});
		$[8] = inset;
		$[9] = props;
		$[10] = t2;
		$[11] = variant;
		$[12] = t3;
	} else t3 = $[12];
	return t3;
};
const DropdownMenuCheckboxItem = (t0) => {
	const $ = c(14);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 14; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let checked;
	let children;
	let className;
	let props;
	if ($[1] !== t0) {
		({className, children, checked, ...props} = t0);
		$[1] = t0;
		$[2] = checked;
		$[3] = children;
		$[4] = className;
		$[5] = props;
	} else {
		checked = $[2];
		children = $[3];
		className = $[4];
		props = $[5];
	}
	let t1;
	if ($[6] !== className) {
		t1 = cn("focus:bg-accent focus:text-accent-foreground relative flex cursor-default items-center gap-2 rounded-sm py-1.5 pr-2 pl-8 text-sm outline-hidden select-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4", className);
		$[6] = className;
		$[7] = t1;
	} else t1 = $[7];
	let t2;
	if ($[8] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx("span", {
			className: "pointer-events-none absolute left-2 flex size-3.5 items-center justify-center",
			children: /* @__PURE__ */ jsx(DropdownMenuPrimitive.ItemIndicator, { children: /* @__PURE__ */ jsx(CheckIcon, { className: "size-4" }) })
		});
		$[8] = t2;
	} else t2 = $[8];
	let t3;
	if ($[9] !== checked || $[10] !== children || $[11] !== props || $[12] !== t1) {
		t3 = /* @__PURE__ */ jsxs(DropdownMenuPrimitive.CheckboxItem, {
			"data-slot": "dropdown-menu-checkbox-item",
			className: t1,
			checked,
			...props,
			children: [t2, children]
		});
		$[9] = checked;
		$[10] = children;
		$[11] = props;
		$[12] = t1;
		$[13] = t3;
	} else t3 = $[13];
	return t3;
};
const DropdownMenuRadioGroup = (t0) => {
	const $ = c(5);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 5; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let props;
	if ($[1] !== t0) {
		({...props} = t0);
		$[1] = t0;
		$[2] = props;
	} else props = $[2];
	let t1;
	if ($[3] !== props) {
		t1 = /* @__PURE__ */ jsx(DropdownMenuPrimitive.RadioGroup, {
			"data-slot": "dropdown-menu-radio-group",
			...props
		});
		$[3] = props;
		$[4] = t1;
	} else t1 = $[4];
	return t1;
};
const DropdownMenuRadioItem = (t0) => {
	const $ = c(12);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 12; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let children;
	let className;
	let props;
	if ($[1] !== t0) {
		({className, children, ...props} = t0);
		$[1] = t0;
		$[2] = children;
		$[3] = className;
		$[4] = props;
	} else {
		children = $[2];
		className = $[3];
		props = $[4];
	}
	let t1;
	if ($[5] !== className) {
		t1 = cn("focus:bg-accent focus:text-accent-foreground relative flex cursor-default items-center gap-2 rounded-sm py-1.5 pr-2 pl-8 text-sm outline-hidden select-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4", className);
		$[5] = className;
		$[6] = t1;
	} else t1 = $[6];
	let t2;
	if ($[7] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx("span", {
			className: "pointer-events-none absolute left-2 flex size-3.5 items-center justify-center",
			children: /* @__PURE__ */ jsx(DropdownMenuPrimitive.ItemIndicator, { children: /* @__PURE__ */ jsx(CircleIcon, { className: "size-2 fill-current" }) })
		});
		$[7] = t2;
	} else t2 = $[7];
	let t3;
	if ($[8] !== children || $[9] !== props || $[10] !== t1) {
		t3 = /* @__PURE__ */ jsxs(DropdownMenuPrimitive.RadioItem, {
			"data-slot": "dropdown-menu-radio-item",
			className: t1,
			...props,
			children: [t2, children]
		});
		$[8] = children;
		$[9] = props;
		$[10] = t1;
		$[11] = t3;
	} else t3 = $[11];
	return t3;
};
const DropdownMenuLabel = (t0) => {
	const $ = c(11);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 11; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let className;
	let inset;
	let props;
	if ($[1] !== t0) {
		({className, inset, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = inset;
		$[4] = props;
	} else {
		className = $[2];
		inset = $[3];
		props = $[4];
	}
	let t1;
	if ($[5] !== className) {
		t1 = cn("px-2 py-1.5 text-sm font-medium data-[inset]:pl-8", className);
		$[5] = className;
		$[6] = t1;
	} else t1 = $[6];
	let t2;
	if ($[7] !== inset || $[8] !== props || $[9] !== t1) {
		t2 = /* @__PURE__ */ jsx(DropdownMenuPrimitive.Label, {
			"data-slot": "dropdown-menu-label",
			"data-inset": inset,
			className: t1,
			...props
		});
		$[7] = inset;
		$[8] = props;
		$[9] = t1;
		$[10] = t2;
	} else t2 = $[10];
	return t2;
};
const DropdownMenuSeparator = (t0) => {
	const $ = c(9);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("bg-border -mx-1 my-1 h-px", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== t1) {
		t2 = /* @__PURE__ */ jsx(DropdownMenuPrimitive.Separator, {
			"data-slot": "dropdown-menu-separator",
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = t1;
		$[8] = t2;
	} else t2 = $[8];
	return t2;
};
const DropdownMenuSub = (t0) => {
	const $ = c(5);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 5; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let props;
	if ($[1] !== t0) {
		({...props} = t0);
		$[1] = t0;
		$[2] = props;
	} else props = $[2];
	let t1;
	if ($[3] !== props) {
		t1 = /* @__PURE__ */ jsx(DropdownMenuPrimitive.Sub, {
			"data-slot": "dropdown-menu-sub",
			...props
		});
		$[3] = props;
		$[4] = t1;
	} else t1 = $[4];
	return t1;
};
const DropdownMenuSubTrigger = (t0) => {
	const $ = c(14);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 14; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let children;
	let className;
	let inset;
	let props;
	if ($[1] !== t0) {
		({className, inset, children, ...props} = t0);
		$[1] = t0;
		$[2] = children;
		$[3] = className;
		$[4] = inset;
		$[5] = props;
	} else {
		children = $[2];
		className = $[3];
		inset = $[4];
		props = $[5];
	}
	let t1;
	if ($[6] !== className) {
		t1 = cn("focus:bg-accent focus:text-accent-foreground data-[state=open]:bg-accent data-[state=open]:text-accent-foreground flex cursor-default items-center rounded-sm px-2 py-1.5 text-sm outline-hidden select-none data-[inset]:pl-8", className);
		$[6] = className;
		$[7] = t1;
	} else t1 = $[7];
	let t2;
	if ($[8] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx(ChevronRightIcon, { className: "ml-auto size-4" });
		$[8] = t2;
	} else t2 = $[8];
	let t3;
	if ($[9] !== children || $[10] !== inset || $[11] !== props || $[12] !== t1) {
		t3 = /* @__PURE__ */ jsxs(DropdownMenuPrimitive.SubTrigger, {
			"data-slot": "dropdown-menu-sub-trigger",
			"data-inset": inset,
			className: t1,
			...props,
			children: [children, t2]
		});
		$[9] = children;
		$[10] = inset;
		$[11] = props;
		$[12] = t1;
		$[13] = t3;
	} else t3 = $[13];
	return t3;
};
const DropdownMenuSubContent = (t0) => {
	const $ = c(9);
	if ($[0] !== "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "cc5176081ffdf21e1c4aaae4b991a5a2bb0db137f11d07543b3f43f2bf26fe40";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("bg-popover text-popover-foreground data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 min-w-[8rem] origin-(--radix-dropdown-menu-content-transform-origin) overflow-hidden rounded-md border p-1 shadow-lg", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== t1) {
		t2 = /* @__PURE__ */ jsx(DropdownMenuPrimitive.SubContent, {
			"data-slot": "dropdown-menu-sub-content",
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = t1;
		$[8] = t2;
	} else t2 = $[8];
	return t2;
};

//#endregion
export { DropdownMenu, DropdownMenuCheckboxItem, DropdownMenuContent, DropdownMenuGroup, DropdownMenuItem, DropdownMenuLabel, DropdownMenuRadioGroup, DropdownMenuRadioItem, DropdownMenuSeparator, DropdownMenuSub, DropdownMenuSubContent, DropdownMenuSubTrigger, DropdownMenuTrigger };
