// @motiadev/ui/dist/components/ui/empty.js
import { cn } from "../../lib/utils.js";
import { c } from "react/compiler-runtime";
import { jsx } from "react/jsx-runtime";
import { cva } from "class-variance-authority";

//#region src/components/ui/empty.tsx
function Empty(t0) {
	const $ = c(9);
	if ($[0] !== "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("flex min-w-0 flex-1 flex-col items-center justify-center gap-6 text-balance rounded-lg border-dashed p-6 text-center md:p-12", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			"data-slot": "empty",
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = t1;
		$[8] = t2;
	} else t2 = $[8];
	return t2;
}
function EmptyHeader(t0) {
	const $ = c(9);
	if ($[0] !== "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("flex max-w-sm flex-col items-center gap-2 text-center", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			"data-slot": "empty-header",
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = t1;
		$[8] = t2;
	} else t2 = $[8];
	return t2;
}
const emptyMediaVariants = cva("mb-2 flex shrink-0 items-center justify-center [&_svg]:pointer-events-none [&_svg]:shrink-0", {
	variants: { variant: {
		default: "bg-transparent",
		icon: "bg-dark-50 dark:bg-light-50 text-foreground flex size-10 shrink-0 items-center justify-center rounded-lg [&_svg:not([class*='size-'])]:size-6"
	} },
	defaultVariants: { variant: "default" }
});
function EmptyMedia(t0) {
	const $ = c(12);
	if ($[0] !== "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539") {
		for (let $i = 0; $i < 12; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539";
	}
	let className;
	let props;
	let t1;
	if ($[1] !== t0) {
		({className, variant: t1, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
		$[4] = t1;
	} else {
		className = $[2];
		props = $[3];
		t1 = $[4];
	}
	const variant = t1 === void 0 ? "default" : t1;
	let t2;
	if ($[5] !== className || $[6] !== variant) {
		t2 = cn(emptyMediaVariants({
			variant,
			className
		}));
		$[5] = className;
		$[6] = variant;
		$[7] = t2;
	} else t2 = $[7];
	let t3;
	if ($[8] !== props || $[9] !== t2 || $[10] !== variant) {
		t3 = /* @__PURE__ */ jsx("div", {
			"data-slot": "empty-icon",
			"data-variant": variant,
			className: t2,
			...props
		});
		$[8] = props;
		$[9] = t2;
		$[10] = variant;
		$[11] = t3;
	} else t3 = $[11];
	return t3;
}
function EmptyTitle(t0) {
	const $ = c(9);
	if ($[0] !== "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("text-lg font-medium tracking-tight", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			"data-slot": "empty-title",
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = t1;
		$[8] = t2;
	} else t2 = $[8];
	return t2;
}
function EmptyDescription(t0) {
	const $ = c(9);
	if ($[0] !== "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("text-muted-foreground [&>a:hover]:text-primary text-sm/relaxed [&>a]:underline [&>a]:underline-offset-4", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			"data-slot": "empty-description",
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = t1;
		$[8] = t2;
	} else t2 = $[8];
	return t2;
}
function EmptyContent(t0) {
	const $ = c(9);
	if ($[0] !== "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f8af5c7bbd5b7b466051ad6c6ba7c28263cee5f1ff855c30b732e93e6441e539";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("flex w-full min-w-0 max-w-sm flex-col items-center gap-4 text-balance text-sm", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			"data-slot": "empty-content",
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = t1;
		$[8] = t2;
	} else t2 = $[8];
	return t2;
}

//#endregion
export { Empty, EmptyContent, EmptyDescription, EmptyHeader, EmptyMedia, EmptyTitle };
