// @motiadev/ui/dist/components/ui/input.js
import { cn } from "../../lib/utils.js";
import { c } from "react/compiler-runtime";
import { jsx } from "react/jsx-runtime";
import { cva } from "class-variance-authority";
import { forwardRef } from "react";

//#region src/components/ui/input.tsx
const inputVariants = cva(cn("flex w-full rounded-md px-3 py-2 text-base transition-colors outline-none resize-none", "field-sizing-content", "aria-invalid:border-destructive", "disabled:cursor-not-allowed disabled:opacity-50", "text-sm font-medium leading-[1.3] tracking-[-0.01em]"), {
	variants: { variant: {
		default: "bg-input border border-border hover:border-border focus-visible:border-ring text-foreground placeholder:text-muted-foreground focus-visible:ring-2 focus-visible:ring-ring/20",
		shade: "bg-muted-foreground/10 text-muted-foreground hover:bg-muted-foreground/12 focus:bg-muted-foreground/15",
		outline: "focus-visible:border-border focus-visible:bg-card hover:border-border hover:bg-card bg-transparent text-foreground border border-transparent",
		outlineReadonly: "bg-transparent text-foreground border border-transparent"
	} },
	defaultVariants: { variant: "default" }
});
const Input = forwardRef((t0, ref) => {
	const $ = c(14);
	if ($[0] !== "ed2073a4c548690e4b490ee4a53cd2431c8f22f07636cbbde7ce31719476c874") {
		for (let $i = 0; $i < 14; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "ed2073a4c548690e4b490ee4a53cd2431c8f22f07636cbbde7ce31719476c874";
	}
	let className;
	let props;
	let type;
	let variant;
	if ($[1] !== t0) {
		({className, type, variant, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
		$[4] = type;
		$[5] = variant;
	} else {
		className = $[2];
		props = $[3];
		type = $[4];
		variant = $[5];
	}
	let t1;
	if ($[6] !== className || $[7] !== variant) {
		t1 = inputVariants({
			variant,
			className
		});
		$[6] = className;
		$[7] = variant;
		$[8] = t1;
	} else t1 = $[8];
	let t2;
	if ($[9] !== props || $[10] !== ref || $[11] !== t1 || $[12] !== type) {
		t2 = /* @__PURE__ */ jsx("input", {
			type,
			className: t1,
			ref,
			...props
		});
		$[9] = props;
		$[10] = ref;
		$[11] = t1;
		$[12] = type;
		$[13] = t2;
	} else t2 = $[13];
	return t2;
});
Input.displayName = "Input";

//#endregion
export { Input };
