// @motiadev/ui/dist/components/ui/level-dot.js
import { c } from "react/compiler-runtime";
import { jsx } from "react/jsx-runtime";
import { cva } from "class-variance-authority";

//#region src/components/ui/level-dot.tsx
const badgeVariants = cva("text-xs font-medium tracking-wide rounded-full h-[6px] w-[6px] m-[4px] outline-[2px]", { variants: { variant: {
	info: "bg-[#2862FE] outline-[#2862FE]/20",
	trace: "bg-[#2862FE] outline-[#2862FE]/20",
	debug: "bg-[#2862FE] outline-[#2862FE]/20",
	error: "bg-[#E22A6D] outline-[#E22A6D]/20",
	fatal: "bg-[#E22A6D] outline-[#E22A6D]/20",
	warn: "bg-[#F59F0B] outline-[#F59F0B]/20"
} } });
const LevelDot = (t0) => {
	const $ = c(5);
	if ($[0] !== "287e910b40d2d16a0ea10c860e841aec14db3697fd5534011c819d5b26df8ac1") {
		for (let $i = 0; $i < 5; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "287e910b40d2d16a0ea10c860e841aec14db3697fd5534011c819d5b26df8ac1";
	}
	const { level } = t0;
	const t1 = level;
	let t2;
	if ($[1] !== t1) {
		t2 = badgeVariants({ variant: t1 });
		$[1] = t1;
		$[2] = t2;
	} else t2 = $[2];
	let t3;
	if ($[3] !== t2) {
		t3 = /* @__PURE__ */ jsx("div", { className: t2 });
		$[3] = t2;
		$[4] = t3;
	} else t3 = $[4];
	return t3;
};

//#endregion
export { LevelDot };
