// @motiadev/ui/dist/components/ui/panel.js
import { BackgroundEffect } from "./background-effect.js";
import { cn } from "../../lib/utils.js";
import { Button } from "./button.js";
import { Tabs, TabsContent, TabsList, TabsTrigger } from "./tabs.js";
import { c } from "react/compiler-runtime";
import { Fragment, jsx, jsxs } from "react/jsx-runtime";

//#region src/components/ui/panel.tsx
const panelVariants = {
	default: "bg-transparent border border-border",
	outlined: "bg-transparent border border-border",
	filled: "bg-muted border border-transparent",
	ghost: "bg-transparent border-transparent shadow-none"
};
const PanelDetailItem = (t0) => {
	const $ = c(15);
	if ($[0] !== "b3b7236dd5fadd4d6212298d092e19d59c580762f7498864f6e887d5ae5a53c3") {
		for (let $i = 0; $i < 15; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "b3b7236dd5fadd4d6212298d092e19d59c580762f7498864f6e887d5ae5a53c3";
	}
	const { label, value, highlighted } = t0;
	let t1;
	if ($[1] !== label) {
		t1 = /* @__PURE__ */ jsx("div", {
			className: "flex items-center h-8 shrink-0",
			children: /* @__PURE__ */ jsx("span", {
				className: "text-sm font-medium text-foreground tracking-[-0.25px] w-24 truncate",
				children: label
			})
		});
		$[1] = label;
		$[2] = t1;
	} else t1 = $[2];
	const t2 = highlighted && "bg-secondary";
	let t3;
	if ($[3] !== t2) {
		t3 = cn("flex-1 rounded-lg px-2 py-1 min-h-6", t2);
		$[3] = t2;
		$[4] = t3;
	} else t3 = $[4];
	let t4;
	if ($[5] !== value) {
		t4 = typeof value === "string" ? /* @__PURE__ */ jsx("span", {
			className: "text-sm font-medium text-muted-foreground tracking-[-0.25px] leading-tight",
			children: value
		}) : value;
		$[5] = value;
		$[6] = t4;
	} else t4 = $[6];
	let t5;
	if ($[7] !== t4) {
		t5 = /* @__PURE__ */ jsx("div", {
			className: "flex items-center min-h-6",
			children: t4
		});
		$[7] = t4;
		$[8] = t5;
	} else t5 = $[8];
	let t6;
	if ($[9] !== t3 || $[10] !== t5) {
		t6 = /* @__PURE__ */ jsx("div", {
			className: t3,
			children: t5
		});
		$[9] = t3;
		$[10] = t5;
		$[11] = t6;
	} else t6 = $[11];
	let t7;
	if ($[12] !== t1 || $[13] !== t6) {
		t7 = /* @__PURE__ */ jsxs("div", {
			className: "flex gap-4 items-start",
			children: [t1, t6]
		});
		$[12] = t1;
		$[13] = t6;
		$[14] = t7;
	} else t7 = $[14];
	return t7;
};
const Panel = (t0) => {
	const $ = c(48);
	if ($[0] !== "b3b7236dd5fadd4d6212298d092e19d59c580762f7498864f6e887d5ae5a53c3") {
		for (let $i = 0; $i < 48; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "b3b7236dd5fadd4d6212298d092e19d59c580762f7498864f6e887d5ae5a53c3";
	}
	const { "data-testid": dataTestId, title, subtitle, details, actions, className, children, size, variant: t1, contentClassName, tabs } = t0;
	const variant = t1 === void 0 ? "default" : t1;
	const hasTabs = tabs && tabs.length > 0;
	let t2;
	bb0: {
		let t3$1;
		if ($[1] !== hasTabs || $[2] !== tabs || $[3] !== variant) {
			t3$1 = hasTabs && /* @__PURE__ */ jsx(TabsList, {
				className: cn("bg-card border-b border-border px-1 pt-5", { "bg-transparent": variant === "ghost" }),
				children: tabs?.map(_temp)
			});
			$[1] = hasTabs;
			$[2] = tabs;
			$[3] = variant;
			$[4] = t3$1;
		} else t3$1 = $[4];
		let t4$1;
		if ($[5] !== contentClassName) {
			t4$1 = cn("flex flex-col gap-2 p-4 h-full", contentClassName);
			$[5] = contentClassName;
			$[6] = t4$1;
		} else t4$1 = $[6];
		let t5$1;
		if ($[7] !== details) {
			t5$1 = details?.map(_temp2);
			$[7] = details;
			$[8] = t5$1;
		} else t5$1 = $[8];
		let t6$1;
		if ($[9] !== hasTabs || $[10] !== tabs) {
			t6$1 = hasTabs && tabs.map(_temp3);
			$[9] = hasTabs;
			$[10] = tabs;
			$[11] = t6$1;
		} else t6$1 = $[11];
		let t7$1;
		if ($[12] !== children || $[13] !== t4$1 || $[14] !== t5$1 || $[15] !== t6$1) {
			t7$1 = /* @__PURE__ */ jsxs("div", {
				className: t4$1,
				children: [
					t5$1,
					t6$1,
					children
				]
			});
			$[12] = children;
			$[13] = t4$1;
			$[14] = t5$1;
			$[15] = t6$1;
			$[16] = t7$1;
		} else t7$1 = $[16];
		let t8$1;
		if ($[17] !== t3$1 || $[18] !== t7$1) {
			t8$1 = /* @__PURE__ */ jsxs(Fragment, { children: [t3$1, t7$1] });
			$[17] = t3$1;
			$[18] = t7$1;
			$[19] = t8$1;
		} else t8$1 = $[19];
		const _view = t8$1;
		if (hasTabs) {
			const t9$1 = tabs?.[0]?.label;
			let t10$1;
			if ($[20] !== _view || $[21] !== t9$1) {
				t10$1 = /* @__PURE__ */ jsx(Tabs, {
					defaultValue: t9$1,
					children: _view
				});
				$[20] = _view;
				$[21] = t9$1;
				$[22] = t10$1;
			} else t10$1 = $[22];
			t2 = t10$1;
			break bb0;
		}
		t2 = _view;
	}
	const content = t2;
	const t3 = panelVariants[variant];
	let t4;
	if ($[23] !== className || $[24] !== t3) {
		t4 = cn("relative isolate size-full backdrop-blur-[48px] backdrop-filter", "text-foreground", "rounded-lg overflow-hidden", t3, className);
		$[23] = className;
		$[24] = t3;
		$[25] = t4;
	} else t4 = $[25];
	let t5;
	if ($[26] === Symbol.for("react.memo_cache_sentinel")) {
		t5 = /* @__PURE__ */ jsx(BackgroundEffect, {});
		$[26] = t5;
	} else t5 = $[26];
	const t6 = variant === "ghost";
	let t7;
	if ($[27] !== hasTabs || $[28] !== t6) {
		t7 = cn("relative shrink-0 w-full border-b border-border bg-card", {
			"bg-transparent": t6,
			"border-b-0": hasTabs
		});
		$[27] = hasTabs;
		$[28] = t6;
		$[29] = t7;
	} else t7 = $[29];
	let t8;
	if ($[30] !== actions || $[31] !== hasTabs || $[32] !== size || $[33] !== subtitle || $[34] !== title) {
		t8 = title && /* @__PURE__ */ jsxs("div", {
			className: cn("flex flex-col gap-1 px-5 py-4", {
				"px-4 py-3": size === "sm",
				"px-5 py-4": size === "md",
				"pb-0": hasTabs
			}),
			children: [/* @__PURE__ */ jsxs("div", {
				className: "flex items-center w-full",
				children: [/* @__PURE__ */ jsx("div", {
					className: cn("font-semibold text-foreground tracking-[-0.25px] leading-tight flex-1", size === "sm" ? "text-xs" : "text-base"),
					children: title
				}), actions && actions.length > 0 && /* @__PURE__ */ jsx("div", {
					className: "flex items-center gap-1",
					children: actions.map(_temp4)
				})]
			}), subtitle && /* @__PURE__ */ jsx("p", {
				className: "text-sm font-medium text-muted-foreground tracking-[-0.25px] leading-tight",
				children: subtitle
			})]
		});
		$[30] = actions;
		$[31] = hasTabs;
		$[32] = size;
		$[33] = subtitle;
		$[34] = title;
		$[35] = t8;
	} else t8 = $[35];
	let t9;
	if ($[36] !== t7 || $[37] !== t8) {
		t9 = /* @__PURE__ */ jsx("div", {
			className: t7,
			children: t8
		});
		$[36] = t7;
		$[37] = t8;
		$[38] = t9;
	} else t9 = $[38];
	let t10;
	if ($[39] !== content) {
		t10 = /* @__PURE__ */ jsx("div", {
			className: "flex-1 overflow-auto",
			children: content
		});
		$[39] = content;
		$[40] = t10;
	} else t10 = $[40];
	let t11;
	if ($[41] !== t10 || $[42] !== t9) {
		t11 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col size-full",
			children: [t9, t10]
		});
		$[41] = t10;
		$[42] = t9;
		$[43] = t11;
	} else t11 = $[43];
	let t12;
	if ($[44] !== dataTestId || $[45] !== t11 || $[46] !== t4) {
		t12 = /* @__PURE__ */ jsxs("div", {
			className: t4,
			"data-testid": dataTestId,
			children: [t5, t11]
		});
		$[44] = dataTestId;
		$[45] = t11;
		$[46] = t4;
		$[47] = t12;
	} else t12 = $[47];
	return t12;
};
Panel.displayName = "Panel";
function _temp(tab) {
	return /* @__PURE__ */ jsx(TabsTrigger, {
		value: tab.label,
		"data-testid": tab["data-testid"],
		className: "cursor-pointer",
		children: tab.labelComponent || tab.label
	}, tab.label);
}
function _temp2(detail, index) {
	return /* @__PURE__ */ jsx(PanelDetailItem, { ...detail }, index);
}
function _temp3(tab_0) {
	return /* @__PURE__ */ jsx(TabsContent, {
		value: tab_0.label,
		children: tab_0.content
	}, tab_0.label);
}
function _temp4(action, index_0) {
	return /* @__PURE__ */ jsx(Button, {
		onClick: action.onClick,
		variant: "ghost",
		className: cn(action.active && "bg-muted-foreground/20 hover:bg-muted-foreground/30"),
		size: "icon",
		"aria-label": action.label,
		"data-testid": "close-panel",
		children: action.icon
	}, index_0);
}

//#endregion
export { Panel, PanelDetailItem };
