// @motiadev/ui/dist/components/ui/select.js
import { cn } from "../../lib/utils.js";
import { c } from "react/compiler-runtime";
import { jsx, jsxs } from "react/jsx-runtime";
import { Check, ChevronDown, ChevronUp } from "lucide-react";
import * as React$1 from "react";
import * as SelectPrimitive from "@radix-ui/react-select";

//#region src/components/ui/select.tsx
const Select = SelectPrimitive.Root;
const SelectGroup = SelectPrimitive.Group;
const SelectValue = SelectPrimitive.Value;
const SelectTrigger = React$1.forwardRef((t0, ref) => {
	const $ = c(13);
	if ($[0] !== "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a") {
		for (let $i = 0; $i < 13; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a";
	}
	let children;
	let className;
	let props;
	if ($[1] !== t0) {
		({className, children, ...props} = t0);
		$[1] = t0;
		$[2] = children;
		$[3] = className;
		$[4] = props;
	} else {
		children = $[2];
		className = $[3];
		props = $[4];
	}
	let t1;
	if ($[5] !== className) {
		t1 = cn("flex h-9 w-full items-center justify-between whitespace-nowrap rounded-md border border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background data-[placeholder]:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1", className);
		$[5] = className;
		$[6] = t1;
	} else t1 = $[6];
	let t2;
	if ($[7] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx(SelectPrimitive.Icon, {
			asChild: true,
			children: /* @__PURE__ */ jsx(ChevronDown, { className: "h-4 w-4 opacity-50" })
		});
		$[7] = t2;
	} else t2 = $[7];
	let t3;
	if ($[8] !== children || $[9] !== props || $[10] !== ref || $[11] !== t1) {
		t3 = /* @__PURE__ */ jsxs(SelectPrimitive.Trigger, {
			ref,
			className: t1,
			...props,
			children: [children, t2]
		});
		$[8] = children;
		$[9] = props;
		$[10] = ref;
		$[11] = t1;
		$[12] = t3;
	} else t3 = $[12];
	return t3;
});
SelectTrigger.displayName = SelectPrimitive.Trigger.displayName;
const SelectScrollUpButton = React$1.forwardRef((t0, ref) => {
	const $ = c(11);
	if ($[0] !== "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a") {
		for (let $i = 0; $i < 11; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("flex cursor-default items-center justify-center py-1", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx(ChevronUp, { className: "h-4 w-4" });
		$[6] = t2;
	} else t2 = $[6];
	let t3;
	if ($[7] !== props || $[8] !== ref || $[9] !== t1) {
		t3 = /* @__PURE__ */ jsx(SelectPrimitive.ScrollUpButton, {
			ref,
			className: t1,
			...props,
			children: t2
		});
		$[7] = props;
		$[8] = ref;
		$[9] = t1;
		$[10] = t3;
	} else t3 = $[10];
	return t3;
});
SelectScrollUpButton.displayName = SelectPrimitive.ScrollUpButton.displayName;
const SelectScrollDownButton = React$1.forwardRef((t0, ref) => {
	const $ = c(11);
	if ($[0] !== "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a") {
		for (let $i = 0; $i < 11; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("flex cursor-default items-center justify-center py-1", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx(ChevronDown, { className: "h-4 w-4" });
		$[6] = t2;
	} else t2 = $[6];
	let t3;
	if ($[7] !== props || $[8] !== ref || $[9] !== t1) {
		t3 = /* @__PURE__ */ jsx(SelectPrimitive.ScrollDownButton, {
			ref,
			className: t1,
			...props,
			children: t2
		});
		$[7] = props;
		$[8] = ref;
		$[9] = t1;
		$[10] = t3;
	} else t3 = $[10];
	return t3;
});
SelectScrollDownButton.displayName = SelectPrimitive.ScrollDownButton.displayName;
const SelectContent = React$1.forwardRef((t0, ref) => {
	const $ = c(22);
	if ($[0] !== "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a") {
		for (let $i = 0; $i < 22; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a";
	}
	let children;
	let className;
	let props;
	let t1;
	if ($[1] !== t0) {
		({className, children, position: t1, ...props} = t0);
		$[1] = t0;
		$[2] = children;
		$[3] = className;
		$[4] = props;
		$[5] = t1;
	} else {
		children = $[2];
		className = $[3];
		props = $[4];
		t1 = $[5];
	}
	const position = t1 === void 0 ? "popper" : t1;
	const t2 = position === "popper" && "data-[side=bottom]:translate-y-1 data-[side=left]:-translate-x-1 data-[side=right]:translate-x-1 data-[side=top]:-translate-y-1";
	let t3;
	if ($[6] !== className || $[7] !== t2) {
		t3 = cn("relative z-50 max-h-[--radix-select-content-available-height] min-w-[8rem] backdrop-blur-2xl overflow-y-auto overflow-x-hidden rounded-md border bg-popover text-popover-foreground shadow-md data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 origin-[--radix-select-content-transform-origin]", t2, className);
		$[6] = className;
		$[7] = t2;
		$[8] = t3;
	} else t3 = $[8];
	let t4;
	if ($[9] === Symbol.for("react.memo_cache_sentinel")) {
		t4 = /* @__PURE__ */ jsx(SelectScrollUpButton, {});
		$[9] = t4;
	} else t4 = $[9];
	const t5 = position === "popper" && "h-[var(--radix-select-trigger-height)] w-full min-w-[var(--radix-select-trigger-width)]";
	let t6;
	if ($[10] !== t5) {
		t6 = cn("p-1", t5);
		$[10] = t5;
		$[11] = t6;
	} else t6 = $[11];
	let t7;
	if ($[12] !== children || $[13] !== t6) {
		t7 = /* @__PURE__ */ jsx(SelectPrimitive.Viewport, {
			className: t6,
			children
		});
		$[12] = children;
		$[13] = t6;
		$[14] = t7;
	} else t7 = $[14];
	let t8;
	if ($[15] === Symbol.for("react.memo_cache_sentinel")) {
		t8 = /* @__PURE__ */ jsx(SelectScrollDownButton, {});
		$[15] = t8;
	} else t8 = $[15];
	let t9;
	if ($[16] !== position || $[17] !== props || $[18] !== ref || $[19] !== t3 || $[20] !== t7) {
		t9 = /* @__PURE__ */ jsx(SelectPrimitive.Portal, { children: /* @__PURE__ */ jsxs(SelectPrimitive.Content, {
			ref,
			className: t3,
			position,
			...props,
			children: [
				t4,
				t7,
				t8
			]
		}) });
		$[16] = position;
		$[17] = props;
		$[18] = ref;
		$[19] = t3;
		$[20] = t7;
		$[21] = t9;
	} else t9 = $[21];
	return t9;
});
SelectContent.displayName = SelectPrimitive.Content.displayName;
const SelectLabel = React$1.forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("px-2 py-1.5 text-sm font-semibold", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx(SelectPrimitive.Label, {
			ref,
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
SelectLabel.displayName = SelectPrimitive.Label.displayName;
const SelectItem = React$1.forwardRef((t0, ref) => {
	const $ = c(15);
	if ($[0] !== "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a") {
		for (let $i = 0; $i < 15; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a";
	}
	let children;
	let className;
	let props;
	if ($[1] !== t0) {
		({className, children, ...props} = t0);
		$[1] = t0;
		$[2] = children;
		$[3] = className;
		$[4] = props;
	} else {
		children = $[2];
		className = $[3];
		props = $[4];
	}
	let t1;
	if ($[5] !== className) {
		t1 = cn("relative flex w-full cursor-default select-none items-center rounded-sm py-1.5 pl-2 pr-8 text-sm outline-none focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50", className);
		$[5] = className;
		$[6] = t1;
	} else t1 = $[6];
	let t2;
	if ($[7] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx("span", {
			className: "absolute right-2 flex h-3.5 w-3.5 items-center justify-center",
			children: /* @__PURE__ */ jsx(SelectPrimitive.ItemIndicator, { children: /* @__PURE__ */ jsx(Check, { className: "h-4 w-4" }) })
		});
		$[7] = t2;
	} else t2 = $[7];
	let t3;
	if ($[8] !== children) {
		t3 = /* @__PURE__ */ jsx(SelectPrimitive.ItemText, { children });
		$[8] = children;
		$[9] = t3;
	} else t3 = $[9];
	let t4;
	if ($[10] !== props || $[11] !== ref || $[12] !== t1 || $[13] !== t3) {
		t4 = /* @__PURE__ */ jsxs(SelectPrimitive.Item, {
			ref,
			className: t1,
			...props,
			children: [t2, t3]
		});
		$[10] = props;
		$[11] = ref;
		$[12] = t1;
		$[13] = t3;
		$[14] = t4;
	} else t4 = $[14];
	return t4;
});
SelectItem.displayName = SelectPrimitive.Item.displayName;
const SelectSeparator = React$1.forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "7558ea93396792fd0b2c0cde73c662fdbfcdbc0d323b3b63f0daa9d9121f355a";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("-mx-1 my-1 h-px bg-muted", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx(SelectPrimitive.Separator, {
			ref,
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
SelectSeparator.displayName = SelectPrimitive.Separator.displayName;

//#endregion
export { Select, SelectContent, SelectItem, SelectTrigger, SelectValue };
