// @motiadev/ui/dist/components/ui/side-panel.js
import { cn } from "../../lib/utils.js";
import { Button } from "./button.js";
import { c } from "react/compiler-runtime";
import { jsx, jsxs } from "react/jsx-runtime";
import { X } from "lucide-react";

//#region src/components/ui/side-panel.tsx
const SidePanel = (t0) => {
	const $ = c(29);
	if ($[0] !== "39a5c1c37b367fbe5681e107f52c735a559918c882d3fc91fa9f52f7cdd414a8") {
		for (let $i = 0; $i < 29; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "39a5c1c37b367fbe5681e107f52c735a559918c882d3fc91fa9f52f7cdd414a8";
	}
	const { title, subtitle, actions, className, children, onClose } = t0;
	let t1;
	if ($[1] !== className) {
		t1 = cn("bg-muted-foreground/5 text-foreground border-l border-border overflow-hidden w-1/3", className);
		$[1] = className;
		$[2] = t1;
	} else t1 = $[2];
	let t2;
	if ($[3] !== title) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: "text-lg font-semibold text-foreground tracking-[-0.25px] leading-tight",
			children: title
		});
		$[3] = title;
		$[4] = t2;
	} else t2 = $[4];
	let t3;
	if ($[5] !== actions) {
		t3 = actions?.map(_temp);
		$[5] = actions;
		$[6] = t3;
	} else t3 = $[6];
	let t4;
	if ($[7] === Symbol.for("react.memo_cache_sentinel")) {
		t4 = /* @__PURE__ */ jsx(X, { className: "w-4 h-4 text-muted-foreground" });
		$[7] = t4;
	} else t4 = $[7];
	let t5;
	if ($[8] !== onClose) {
		t5 = /* @__PURE__ */ jsx(Button, {
			onClick: onClose,
			variant: "ghost",
			size: "icon",
			"aria-label": "Close",
			children: t4
		});
		$[8] = onClose;
		$[9] = t5;
	} else t5 = $[9];
	let t6;
	if ($[10] !== t3 || $[11] !== t5) {
		t6 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center gap-1",
			children: [t3, t5]
		});
		$[10] = t3;
		$[11] = t5;
		$[12] = t6;
	} else t6 = $[12];
	let t7;
	if ($[13] !== t2 || $[14] !== t6) {
		t7 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center justify-between w-full",
			children: [t2, t6]
		});
		$[13] = t2;
		$[14] = t6;
		$[15] = t7;
	} else t7 = $[15];
	let t8;
	if ($[16] !== subtitle) {
		t8 = subtitle && /* @__PURE__ */ jsx("div", {
			className: "text-md font-medium text-muted-foreground tracking-[-0.25px] leading-tight",
			children: subtitle
		});
		$[16] = subtitle;
		$[17] = t8;
	} else t8 = $[17];
	let t9;
	if ($[18] !== t7 || $[19] !== t8) {
		t9 = /* @__PURE__ */ jsx("div", {
			className: "relative shrink-0 w-full border-b border-border",
			children: /* @__PURE__ */ jsxs("div", {
				className: "flex flex-col gap-0 p-4",
				children: [t7, t8]
			})
		});
		$[18] = t7;
		$[19] = t8;
		$[20] = t9;
	} else t9 = $[20];
	let t10;
	if ($[21] !== children) {
		t10 = /* @__PURE__ */ jsx("div", {
			className: "flex-1 overflow-auto",
			children: /* @__PURE__ */ jsx("div", {
				className: "flex flex-col gap-2 p-4",
				children
			})
		});
		$[21] = children;
		$[22] = t10;
	} else t10 = $[22];
	let t11;
	if ($[23] !== t10 || $[24] !== t9) {
		t11 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col size-full",
			children: [t9, t10]
		});
		$[23] = t10;
		$[24] = t9;
		$[25] = t11;
	} else t11 = $[25];
	let t12;
	if ($[26] !== t1 || $[27] !== t11) {
		t12 = /* @__PURE__ */ jsx("div", {
			className: t1,
			children: t11
		});
		$[26] = t1;
		$[27] = t11;
		$[28] = t12;
	} else t12 = $[28];
	return t12;
};
SidePanel.displayName = "SidePanel";
function _temp(action, index) {
	return /* @__PURE__ */ jsx(Button, {
		onClick: action.onClick,
		variant: "ghost",
		size: "icon",
		"aria-label": action.label,
		children: action.icon
	}, index);
}

//#endregion
export { SidePanel };
