// @motiadev/ui/dist/components/ui/sidebar.js
import { Panel } from "./panel.js";
import { c } from "react/compiler-runtime";
import { jsx, jsxs } from "react/jsx-runtime";
import { Equal } from "lucide-react";
import { useEffect } from "react";
import { createPortal } from "react-dom";
import { useResizable } from "react-use-resizable";

//#region src/components/ui/sidebar.tsx
const APP_SIDEBAR_CONTAINER_ID = "app-sidebar-container";
const CLOSE_PREVIOUS_SIDEBAR_EVENT = "close-previous-sidebar";
const Sidebar = (t0) => {
	const $ = c(30);
	if ($[0] !== "2d038c97da230ef76b013d9bf722504b2323d8b19cafc19547e60c7328f1f3f3") {
		for (let $i = 0; $i < 30; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "2d038c97da230ef76b013d9bf722504b2323d8b19cafc19547e60c7328f1f3f3";
	}
	let containerId;
	let initialWidth;
	let onClose;
	let props;
	if ($[1] !== t0) {
		({initialWidth, onClose, containerId, ...props} = t0);
		$[1] = t0;
		$[2] = containerId;
		$[3] = initialWidth;
		$[4] = onClose;
		$[5] = props;
	} else {
		containerId = $[2];
		initialWidth = $[3];
		onClose = $[4];
		props = $[5];
	}
	let t1;
	if ($[6] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = Symbol();
		$[6] = t1;
	} else t1 = $[6];
	const sidebarId = t1;
	const t2 = initialWidth ?? 400;
	let t3;
	let t4;
	if ($[7] === Symbol.for("react.memo_cache_sentinel")) {
		t3 = _temp;
		t4 = _temp2;
		$[7] = t3;
		$[8] = t4;
	} else {
		t3 = $[7];
		t4 = $[8];
	}
	let t5;
	if ($[9] !== t2) {
		t5 = {
			lockVertical: true,
			initialWidth: t2,
			initialHeight: "100%",
			onDragStart: t3,
			onDragEnd: t4
		};
		$[9] = t2;
		$[10] = t5;
	} else t5 = $[10];
	const { getRootProps, getHandleProps } = useResizable(t5);
	let t6;
	let t7;
	if ($[11] !== onClose) {
		t6 = () => {
			const event = new CustomEvent(CLOSE_PREVIOUS_SIDEBAR_EVENT, { detail: { sidebarId } });
			window.dispatchEvent(event);
			const handleClose = (e) => {
				if (e.detail.sidebarId !== sidebarId) onClose();
			};
			window.addEventListener(CLOSE_PREVIOUS_SIDEBAR_EVENT, handleClose);
			return () => {
				window.removeEventListener(CLOSE_PREVIOUS_SIDEBAR_EVENT, handleClose);
			};
		};
		t7 = [sidebarId, onClose];
		$[11] = onClose;
		$[12] = t6;
		$[13] = t7;
	} else {
		t6 = $[12];
		t7 = $[13];
	}
	useEffect(t6, t7);
	let t8;
	if ($[14] !== getRootProps) {
		t8 = getRootProps();
		$[14] = getRootProps;
		$[15] = t8;
	} else t8 = $[15];
	let t9;
	if ($[16] !== getHandleProps) {
		t9 = getHandleProps({ reverse: true });
		$[16] = getHandleProps;
		$[17] = t9;
	} else t9 = $[17];
	let t10;
	if ($[18] === Symbol.for("react.memo_cache_sentinel")) {
		t10 = /* @__PURE__ */ jsx(Equal, { className: "rotate-90 w-4 h-4 text-muted-foreground" });
		$[18] = t10;
	} else t10 = $[18];
	let t11;
	if ($[19] !== t9) {
		t11 = /* @__PURE__ */ jsx("div", {
			...t9,
			className: "flex h-6 w-6 items-center justify-center rounded-full bg-background border border-border absolute top-1/2 -translate-y-1/2 -left-4 z-20",
			children: t10
		});
		$[19] = t9;
		$[20] = t11;
	} else t11 = $[20];
	let t12;
	if ($[21] !== props) {
		t12 = /* @__PURE__ */ jsx(Panel, {
			...props,
			variant: "outlined",
			className: "max-h-[calc(100vh-80px)] h-full",
			"data-testid": "sidebar-panel"
		});
		$[21] = props;
		$[22] = t12;
	} else t12 = $[22];
	let t13;
	if ($[23] !== t11 || $[24] !== t12 || $[25] !== t8) {
		t13 = /* @__PURE__ */ jsxs("div", {
			...t8,
			className: "pr-2 py-2 relative",
			children: [t11, t12]
		});
		$[23] = t11;
		$[24] = t12;
		$[25] = t8;
		$[26] = t13;
	} else t13 = $[26];
	let t14;
	if ($[27] !== containerId || $[28] !== t13) {
		t14 = createPortal(t13, document.querySelector(`#${containerId ?? APP_SIDEBAR_CONTAINER_ID}`));
		$[27] = containerId;
		$[28] = t13;
		$[29] = t14;
	} else t14 = $[29];
	return t14;
};
function _temp() {
	document.body.style.userSelect = "none";
}
function _temp2() {
	document.body.style.userSelect = "";
}

//#endregion
export { APP_SIDEBAR_CONTAINER_ID, Sidebar };
