// @motiadev/ui/dist/components/ui/table.js
import { cn } from "../../lib/utils.js";
import { c } from "react/compiler-runtime";
import { jsx } from "react/jsx-runtime";
import React from "react";

//#region src/components/ui/table.tsx
const Table = React.forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("w-full caption-bottom text-sm", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: "relative w-full overflow-auto",
			children: /* @__PURE__ */ jsx("table", {
				ref,
				className: t1,
				...props
			})
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
Table.displayName = "Table";
const TableHeader = React.forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("[&_tr]:border-b", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx("thead", {
			ref,
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
TableHeader.displayName = "TableHeader";
const TableBody = React.forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("[&_tr:last-child]:border-0", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx("tbody", {
			ref,
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
TableBody.displayName = "TableBody";
const TableFooter = React.forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("border-t bg-muted/50 font-medium [&>tr]:last:border-b-0", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx("tfoot", {
			ref,
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
TableFooter.displayName = "TableFooter";
const TableRow = React.forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx("tr", {
			ref,
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
TableRow.displayName = "TableRow";
const TableHead = React.forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("h-10 px-2 text-left align-middle text-md font-medium bg-muted text-muted-foreground font-bold [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx("th", {
			ref,
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
TableHead.displayName = "TableHead";
const TableCell = React.forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx("td", {
			ref,
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
TableCell.displayName = "TableCell";
const TableCaption = React.forwardRef((t0, ref) => {
	const $ = c(10);
	if ($[0] !== "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "beec69b290f0ccfec44f9ddba7d69326f6e92a8ed664f00af9e1ee8aa24ece7d";
	}
	let className;
	let props;
	if ($[1] !== t0) {
		({className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
	} else {
		className = $[2];
		props = $[3];
	}
	let t1;
	if ($[4] !== className) {
		t1 = cn("mt-4 text-sm text-muted-foreground", className);
		$[4] = className;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== props || $[7] !== ref || $[8] !== t1) {
		t2 = /* @__PURE__ */ jsx("caption", {
			ref,
			className: t1,
			...props
		});
		$[6] = props;
		$[7] = ref;
		$[8] = t1;
		$[9] = t2;
	} else t2 = $[9];
	return t2;
});
TableCaption.displayName = "TableCaption";

//#endregion
export { Table, TableBody, TableCell, TableHead, TableHeader, TableRow };
