// motia/dist/redis/connection.mjs
import { RedisMemoryManager } from "./memory-manager.mjs";
import { isExternalRedisConfig, shouldUseMemoryServer } from "./utils.mjs";
import { createClient } from "redis";

//#region src/redis/connection.ts
var RedisConnectionManager = class {
	constructor() {
		this.client = null;
		this.redisConnectionInfo = null;
	}
	async getConnectionInfo(baseDir, config) {
		if (shouldUseMemoryServer(config)) return await new RedisMemoryManager().startServer(baseDir);
		if (isExternalRedisConfig(config.redis)) return {
			host: config.redis.host,
			port: config.redis.port,
			password: config.redis.password,
			username: config.redis.username,
			db: config.redis.db
		};
		return {
			host: process.env.MOTIA_REDIS_HOST || "127.0.0.1",
			port: parseInt(process.env.MOTIA_REDIS_PORT || "6379"),
			password: process.env.MOTIA_REDIS_PASSWORD,
			username: process.env.MOTIA_REDIS_USERNAME,
			db: parseInt(process.env.MOTIA_REDIS_DB || "0")
		};
	}
	async connect(baseDir, config) {
		if (this.client) return this.client;
		this.redisConnectionInfo = await this.getConnectionInfo(baseDir, config);
		this.client = createClient({
			password: this.redisConnectionInfo.password,
			username: this.redisConnectionInfo.username,
			database: this.redisConnectionInfo.db,
			socket: {
				host: this.redisConnectionInfo.host,
				port: this.redisConnectionInfo.port,
				noDelay: true,
				keepAlive: true,
				reconnectStrategy: (retries) => {
					if (retries > 10) return /* @__PURE__ */ new Error("Redis connection retry limit exceeded");
					return Math.min(retries * 100, 3e3);
				},
				connectTimeout: 1e4
			}
		});
		await this.client.connect();
		return this.client;
	}
	async stop() {
		if (this.client?.isOpen) {
			await this.client.quit().catch(() => {
				console.error("[Redis Connection] Redis client closed");
			});
			this.client = null;
		}
	}
	isRunning() {
		return !!this.client?.isOpen;
	}
	getClient() {
		return this.client;
	}
	get connectionInfo() {
		return this.redisConnectionInfo;
	}
};
const manager = new RedisConnectionManager();
const getRedisClient = (baseDir, config) => manager.connect(baseDir, config);
const stopRedisConnection = () => manager.stop();
const getRedisConnectionInfo = () => manager.connectionInfo;

//#endregion
export { getRedisClient, getRedisConnectionInfo, stopRedisConnection };
