// motia/dist/redis/memory-manager.mjs
import { mkdirSync } from "fs";
import { RedisMemoryServer } from "redis-memory-server";

//#region src/redis/memory-manager.ts
var RedisMemoryManager = class {
	constructor() {
		this.server = null;
		this.client = null;
		this.running = false;
	}
	registerCleanupHandlers() {
		process.on("exit", async () => {
			await this.stop();
		});
		process.on("SIGTERM", async () => {
			await this.stop();
		});
		process.on("SIGINT", async () => {
			await this.stop();
		});
	}
	async startServer(baseDir) {
		if (!this.server) try {
			mkdirSync(baseDir, { recursive: true });
			const instance = { args: [
				"--appendonly",
				"yes",
				"--appendfsync",
				"everysec",
				"--save",
				"\"\"",
				"--dir",
				baseDir
			] };
			if (process.env.MOTIA_REDIS_PORT) instance.port = parseInt(process.env.MOTIA_REDIS_PORT || "6379");
			this.server = new RedisMemoryServer({ instance });
			console.log("Redis Memory Server started");
			this.running = true;
			this.registerCleanupHandlers();
		} catch (error) {
			console.error("[Redis Memory Server] Failed to start:", error);
			throw error;
		}
		return {
			host: await this.server.getHost(),
			port: await this.server.getPort()
		};
	}
	async stop() {
		if (this.server && this.running) try {
			await this.server.stop();
		} catch (error) {
			console.error("[Redis Memory Server] Error stopping:", error?.message);
		} finally {
			this.running = false;
			this.server = null;
		}
	}
	isRunning() {
		return this.running;
	}
	getClient() {
		return this.client;
	}
};

//#endregion
export { RedisMemoryManager };
