// @motiadev/plugin-observability/dist/index.js
import { c } from "react/compiler-runtime";
import { Fragment, forwardRef, memo, useEffect, useMemo, useRef, useState } from "react";
import { Badge, Button, Input, LevelDot, Sidebar, cn } from "@motiadev/ui";
import { Check, Copy, GanttChartSquare, MessageCircle, Minus, Package, Plus, Radio, ScrollText, Search, Trash, X } from "lucide-react";
import { create } from "zustand";
import { Fragment as Fragment$1, jsx, jsxs } from "react/jsx-runtime";
import { useMotiaStream, useStreamGroup } from "@motiadev/stream-client-react";
import * as PopoverPrimitive from "@radix-ui/react-popover";
import ReactJson from "react18-json-view";
import "react18-json-view/src/dark.css";
import "react18-json-view/src/style.css";
import { useVirtualizer } from "@tanstack/react-virtual";
import { formatDistanceToNow } from "date-fns";

//#region src/hooks/use-derive-trace-group.ts
const deriveTraceGroup = (meta, traces) => {
	const completedSteps = traces.filter((t) => t.status === "completed").length;
	const failedCount = traces.filter((t) => t.status === "failed").length;
	const runningCount = traces.filter((t) => t.status === "running").length;
	const status = failedCount > 0 ? "failed" : runningCount > 0 ? "running" : "completed";
	const endTimes = traces.filter((t) => t.endTime !== void 0).map((t) => t.endTime);
	const endTime = runningCount === 0 && endTimes.length > 0 ? Math.max(...endTimes) : void 0;
	const lastActivity = Math.max(meta.startTime, ...traces.map((t) => t.endTime || t.startTime), ...traces.flatMap((t) => t.events.map((e) => e.timestamp)));
	return {
		...meta,
		status,
		endTime,
		lastActivity,
		metadata: {
			completedSteps,
			activeSteps: runningCount,
			totalSteps: traces.length
		}
	};
};

//#endregion
//#region src/stores/use-observability-store.ts
const useObservabilityStore = create()((set, get) => ({
	traceGroupMetas: [],
	traces: [],
	tracesByGroupId: {},
	selectedTraceGroupId: "",
	selectedTraceId: void 0,
	search: "",
	setTraceGroupMetas: (metas) => {
		set({ traceGroupMetas: Array.isArray(metas) ? metas : [] });
	},
	setTraces: (traces) => {
		set({ traces: Array.isArray(traces) ? traces : [] });
	},
	setTracesForGroup: (groupId$1, traces) => {
		const state = get();
		const safeTraces = Array.isArray(traces) ? [...traces] : [];
		set({
			tracesByGroupId: {
				...state.tracesByGroupId,
				[groupId$1]: safeTraces
			},
			traces: state.selectedTraceGroupId === groupId$1 ? safeTraces : state.traces
		});
	},
	selectTraceGroupId: (groupId$1) => {
		const state = get();
		const traces = groupId$1 ? state.tracesByGroupId[groupId$1] || [] : [];
		set({
			selectedTraceGroupId: groupId$1 || "",
			traces
		});
	},
	selectTraceId: (traceId) => set({ selectedTraceId: traceId }),
	setSearch: (search) => set({ search }),
	clearTraces: () => {
		fetch("/__motia/trace/clear", { method: "POST" });
	},
	getTraceGroups: () => {
		const state = get();
		return state.traceGroupMetas.map((meta) => {
			const traces = state.tracesByGroupId[meta.id] || [];
			if (traces.length === 0) return {
				...meta,
				status: "running",
				lastActivity: meta.startTime,
				metadata: {
					completedSteps: 0,
					activeSteps: 0,
					totalSteps: 0
				}
			};
			return deriveTraceGroup(meta, traces);
		});
	}
}));

//#endregion
//#region src/components/search-bar.tsx
const SearchBar = memo(() => {
	const $ = c(24);
	if ($[0] !== "0d378097fa8da23c1a1ca72ef31fe4b19f0c9bbeda9092deafe087d58425242d") {
		for (let $i = 0; $i < 24; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "0d378097fa8da23c1a1ca72ef31fe4b19f0c9bbeda9092deafe087d58425242d";
	}
	const search = useObservabilityStore(_temp$8);
	const setSearch = useObservabilityStore(_temp2$5);
	const clearTraces = useObservabilityStore(_temp3$4);
	let t0;
	if ($[1] !== setSearch) {
		t0 = (e) => setSearch(e.target.value);
		$[1] = setSearch;
		$[2] = t0;
	} else t0 = $[2];
	let t1;
	if ($[3] !== search || $[4] !== t0) {
		t1 = /* @__PURE__ */ jsx(Input, {
			variant: "shade",
			value: search,
			onChange: t0,
			className: "px-9! font-medium",
			placeholder: "Search by Trace ID or Step Name"
		});
		$[3] = search;
		$[4] = t0;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx(Search, { className: "absolute left-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground/50" });
		$[6] = t2;
	} else t2 = $[6];
	const t3 = search !== "";
	const t4 = search === "";
	let t5;
	if ($[7] !== t3 || $[8] !== t4) {
		t5 = cn("cursor-pointer absolute right-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground/50 hover:text-muted-foreground", {
			visible: t3,
			invisible: t4
		});
		$[7] = t3;
		$[8] = t4;
		$[9] = t5;
	} else t5 = $[9];
	let t6;
	if ($[10] !== setSearch) {
		t6 = () => setSearch("");
		$[10] = setSearch;
		$[11] = t6;
	} else t6 = $[11];
	let t7;
	if ($[12] !== t5 || $[13] !== t6) {
		t7 = /* @__PURE__ */ jsx(X, {
			className: t5,
			onClick: t6
		});
		$[12] = t5;
		$[13] = t6;
		$[14] = t7;
	} else t7 = $[14];
	let t8;
	if ($[15] !== t1 || $[16] !== t7) {
		t8 = /* @__PURE__ */ jsxs("div", {
			className: "flex-1 relative",
			children: [
				t1,
				t2,
				t7
			]
		});
		$[15] = t1;
		$[16] = t7;
		$[17] = t8;
	} else t8 = $[17];
	let t9;
	if ($[18] === Symbol.for("react.memo_cache_sentinel")) {
		t9 = /* @__PURE__ */ jsx(Trash, {});
		$[18] = t9;
	} else t9 = $[18];
	let t10;
	if ($[19] !== clearTraces) {
		t10 = /* @__PURE__ */ jsxs(Button, {
			variant: "default",
			onClick: clearTraces,
			className: "h-[34px]",
			children: [t9, " Clear"]
		});
		$[19] = clearTraces;
		$[20] = t10;
	} else t10 = $[20];
	let t11;
	if ($[21] !== t10 || $[22] !== t8) {
		t11 = /* @__PURE__ */ jsxs("div", {
			className: "flex p-2 border-b gap-2",
			"data-testid": "logs-search-container",
			children: [t8, t10]
		});
		$[21] = t10;
		$[22] = t8;
		$[23] = t11;
	} else t11 = $[23];
	return t11;
});
SearchBar.displayName = "SearchBar";
function _temp$8(state) {
	return state.search;
}
function _temp2$5(state_0) {
	return state_0.setSearch;
}
function _temp3$4(state_1) {
	return state_1.clearTraces;
}

//#endregion
//#region src/components/trace-empty-state.tsx
const TraceEmptyState = memo(() => {
	const $ = c(2);
	if ($[0] !== "2108089e2c746ed0862d68dda608395c37fa47e1141188e212f129d7d9fe3d12") {
		for (let $i = 0; $i < 2; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "2108089e2c746ed0862d68dda608395c37fa47e1141188e212f129d7d9fe3d12";
	}
	if (useObservabilityStore(_temp$7)) return null;
	let t0;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t0 = /* @__PURE__ */ jsx("div", {
			className: "flex items-center justify-center h-full text-muted-foreground text-center",
			children: "Select a trace or trace group to view the timeline"
		});
		$[1] = t0;
	} else t0 = $[1];
	return t0;
});
TraceEmptyState.displayName = "TraceEmptyState";
function _temp$7(state) {
	return state.selectedTraceGroupId;
}

//#endregion
//#region src/hooks/use-get-endtime.ts
const useGetEndTime = (group) => {
	const $ = c(6);
	if ($[0] !== "f07b1721656d8064013e11ff58f0cb63e38b74f1034170d677e517b5cd2fc048") {
		for (let $i = 0; $i < 6; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f07b1721656d8064013e11ff58f0cb63e38b74f1034170d677e517b5cd2fc048";
	}
	const groupEndTime = group?.endTime;
	let t0;
	if ($[1] !== groupEndTime) {
		t0 = () => groupEndTime || Date.now();
		$[1] = groupEndTime;
		$[2] = t0;
	} else t0 = $[2];
	const [endTime, setEndTime] = useState(t0);
	const intervalRef = useRef(null);
	let t1;
	let t2;
	if ($[3] !== groupEndTime) {
		t1 = () => {
			if (intervalRef.current) {
				clearInterval(intervalRef.current);
				intervalRef.current = null;
			}
			if (groupEndTime) setEndTime(groupEndTime);
			else intervalRef.current = setInterval(() => setEndTime(Date.now()), 100);
			return () => {
				if (intervalRef.current) {
					clearInterval(intervalRef.current);
					intervalRef.current = null;
				}
			};
		};
		t2 = [groupEndTime];
		$[3] = groupEndTime;
		$[4] = t1;
		$[5] = t2;
	} else {
		t1 = $[4];
		t2 = $[5];
	}
	useEffect(t1, t2);
	return endTime;
};

//#endregion
//#region src/hooks/use-traces-stream.ts
const streamName$2 = "motia-trace";
const useTracesStream = () => {
	const $ = c(10);
	if ($[0] !== "812f0c24ae3cbbbcb9380336cb1996e09032dcef72ae69cf410c08f0825131a9") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "812f0c24ae3cbbbcb9380336cb1996e09032dcef72ae69cf410c08f0825131a9";
	}
	const groupId$1 = useObservabilityStore(_temp$6);
	const setData = useObservabilityStore(_temp2$4);
	const setTracesForGroup = useObservabilityStore(_temp3$3);
	const groupIdRef = useRef(groupId$1);
	let t0;
	let t1;
	if ($[1] !== groupId$1) {
		t0 = () => {
			groupIdRef.current = groupId$1;
		};
		t1 = [groupId$1];
		$[1] = groupId$1;
		$[2] = t0;
		$[3] = t1;
	} else {
		t0 = $[2];
		t1 = $[3];
	}
	useEffect(t0, t1);
	let t2;
	if ($[4] !== setData || $[5] !== setTracesForGroup) {
		t2 = (traces) => {
			const currentGroupId = groupIdRef.current;
			setData(traces);
			if (currentGroupId) setTracesForGroup(currentGroupId, traces);
		};
		$[4] = setData;
		$[5] = setTracesForGroup;
		$[6] = t2;
	} else t2 = $[6];
	let t3;
	if ($[7] !== groupId$1 || $[8] !== t2) {
		t3 = {
			streamName: streamName$2,
			groupId: groupId$1,
			setData: t2
		};
		$[7] = groupId$1;
		$[8] = t2;
		$[9] = t3;
	} else t3 = $[9];
	useStreamGroup(t3);
};
function _temp$6(state) {
	return state.selectedTraceGroupId;
}
function _temp2$4(state_0) {
	return state_0.setTraces;
}
function _temp3$3(state_1) {
	return state_1.setTracesForGroup;
}

//#endregion
//#region src/lib/utils.ts
const formatDuration = (duration) => {
	if (duration === void 0 || duration === null) return "N/A";
	if (duration < 1e3) return `${duration}ms`;
	if (duration < 6e4) return `${(duration / 1e3).toFixed(1)}s`;
	if (duration < 36e5) return `${(duration / 6e4).toFixed(1)}min`;
	return `${(duration / 36e5).toFixed(1)}h`;
};

//#endregion
//#region src/components/trace-item/trace-item.tsx
const GRADIENT_CLASSES = {
	running: "bg-[repeating-linear-gradient(140deg,#BEFE29,#BEFE29_8px,#ABE625_8px,#ABE625_16px)]",
	completed: "bg-[repeating-linear-gradient(140deg,#2862FE,#2862FE_8px,#2358E5_8px,#2358E5_16px)]",
	failed: "bg-[repeating-linear-gradient(140deg,#EA2069,#EA2069_8px,#D41E60_8px,#D41E60_16px)]"
};
const TraceItem = memo((t0) => {
	const $ = c(18);
	if ($[0] !== "3296a69d1668d9416c4211c6d1cf083f73fcfe03edbf7e74160503957cdfe795") {
		for (let $i = 0; $i < 18; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "3296a69d1668d9416c4211c6d1cf083f73fcfe03edbf7e74160503957cdfe795";
	}
	const { traceId, traceName, traceStatus, traceStartTime, traceEndTime, groupStartTime, groupEndTime, onExpand } = t0;
	let t1;
	if ($[1] !== onExpand || $[2] !== traceId) {
		t1 = () => {
			onExpand(traceId);
		};
		$[1] = onExpand;
		$[2] = traceId;
		$[3] = t1;
	} else t1 = $[3];
	const handleClick = t1;
	const t2 = GRADIENT_CLASSES[traceStatus];
	let t3;
	if ($[4] !== t2) {
		t3 = cn("h-[24px] rounded-[4px] hover:opacity-80 transition-all duration-200", t2);
		$[4] = t2;
		$[5] = t3;
	} else t3 = $[5];
	const barClassName = t3;
	const t4 = `${(traceStartTime - groupStartTime) / (groupEndTime - groupStartTime) * 100}%`;
	const t5 = traceEndTime ? `${(traceEndTime - traceStartTime) / (groupEndTime - groupStartTime) * 100}%` : `${(Date.now() - traceStartTime) / (groupEndTime - groupStartTime) * 100}%`;
	let t6;
	if ($[6] !== t4 || $[7] !== t5) {
		t6 = {
			marginLeft: t4,
			width: t5
		};
		$[6] = t4;
		$[7] = t5;
		$[8] = t6;
	} else t6 = $[8];
	const barStyle = t6;
	let t7;
	if ($[9] !== traceName) {
		t7 = /* @__PURE__ */ jsx("div", {
			className: "flex items-center min-w-[200px] max-w-[200px] h-[32px] max-h-[32px] py-4 px-2 text-sm font-semibold text-foreground sticky left-0 bg-card z-9",
			children: /* @__PURE__ */ jsx("span", {
				className: "truncate min-w-0",
				children: traceName
			})
		});
		$[9] = traceName;
		$[10] = t7;
	} else t7 = $[10];
	let t8;
	if ($[11] !== barClassName || $[12] !== barStyle) {
		t8 = /* @__PURE__ */ jsx("div", {
			className: "flex w-full flex-row items-center hover:bg-muted/50 rounded-md",
			children: /* @__PURE__ */ jsx("div", {
				className: "relative w-full h-[32px] flex items-center",
				children: /* @__PURE__ */ jsx("div", {
					className: barClassName,
					style: barStyle
				})
			})
		});
		$[11] = barClassName;
		$[12] = barStyle;
		$[13] = t8;
	} else t8 = $[13];
	let t9;
	if ($[14] !== handleClick || $[15] !== t7 || $[16] !== t8) {
		t9 = /* @__PURE__ */ jsxs("div", {
			className: "flex hover:bg-muted-foreground/10 relative cursor-pointer",
			onClick: handleClick,
			"data-testid": "trace-timeline-item",
			children: [t7, t8]
		});
		$[14] = handleClick;
		$[15] = t7;
		$[16] = t8;
		$[17] = t9;
	} else t9 = $[17];
	return t9;
});
TraceItem.displayName = "TraceItem";

//#endregion
//#region src/components/events/event-icon.tsx
const EventIcon = (t0) => {
	const $ = c(5);
	if ($[0] !== "a1e6111b39d1930b766ba2ebca3959949a34edb0ba32f492db6cf8ba92b7256a") {
		for (let $i = 0; $i < 5; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "a1e6111b39d1930b766ba2ebca3959949a34edb0ba32f492db6cf8ba92b7256a";
	}
	const { event } = t0;
	if (event.type === "log") {
		let t1;
		if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
			t1 = /* @__PURE__ */ jsx(ScrollText, { className: "w-4 h-4 text-muted-foreground" });
			$[1] = t1;
		} else t1 = $[1];
		return t1;
	} else if (event.type === "emit") {
		let t1;
		if ($[2] === Symbol.for("react.memo_cache_sentinel")) {
			t1 = /* @__PURE__ */ jsx(MessageCircle, { className: "w-4 h-4 text-muted-foreground" });
			$[2] = t1;
		} else t1 = $[2];
		return t1;
	} else if (event.type === "state") {
		let t1;
		if ($[3] === Symbol.for("react.memo_cache_sentinel")) {
			t1 = /* @__PURE__ */ jsx(Package, { className: "w-4 h-4 text-muted-foreground" });
			$[3] = t1;
		} else t1 = $[3];
		return t1;
	} else if (event.type === "stream") {
		let t1;
		if ($[4] === Symbol.for("react.memo_cache_sentinel")) {
			t1 = /* @__PURE__ */ jsx(Radio, { className: "w-4 h-4 text-muted-foreground" });
			$[4] = t1;
		} else t1 = $[4];
		return t1;
	}
};

//#endregion
//#region src/components/ui/copy-button.tsx
const CopyButton = (t0) => {
	const $ = c(12);
	if ($[0] !== "369763ad5970a7e166e1ca288cc760280bc27f103992fd9a47b59d832664d39c") {
		for (let $i = 0; $i < 12; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "369763ad5970a7e166e1ca288cc760280bc27f103992fd9a47b59d832664d39c";
	}
	const { textToCopy, className, title } = t0;
	const [copied, setCopied] = useState(false);
	let t1;
	if ($[1] !== textToCopy) {
		t1 = async () => {
			try {
				await navigator.clipboard.writeText(textToCopy);
				setCopied(true);
				setTimeout(() => setCopied(false), 2e3);
			} catch (t2$1) {
				const error = t2$1;
				console.error("Failed to copy:", error);
			}
		};
		$[1] = textToCopy;
		$[2] = t1;
	} else t1 = $[2];
	const handleCopy = t1;
	let t2;
	if ($[3] !== className) {
		t2 = cn("cursor-pointer p-3 rounded-full", className);
		$[3] = className;
		$[4] = t2;
	} else t2 = $[4];
	const t3 = title ?? (copied ? "Copied!" : "Copy");
	let t4;
	if ($[5] !== copied) {
		t4 = copied ? /* @__PURE__ */ jsx(Check, { className: "w-4 h-4 text-green-500" }) : /* @__PURE__ */ jsx(Copy, { className: "w-4 h-4 text-gray-400" });
		$[5] = copied;
		$[6] = t4;
	} else t4 = $[6];
	let t5;
	if ($[7] !== handleCopy || $[8] !== t2 || $[9] !== t3 || $[10] !== t4) {
		t5 = /* @__PURE__ */ jsx(Button, {
			variant: "icon",
			size: "icon",
			onClick: handleCopy,
			className: t2,
			title: t3,
			children: t4
		});
		$[7] = handleCopy;
		$[8] = t2;
		$[9] = t3;
		$[10] = t4;
		$[11] = t5;
	} else t5 = $[11];
	return t5;
};

//#endregion
//#region src/components/ui/popover.tsx
const Popover = (t0) => {
	const $ = c(5);
	if ($[0] !== "f0c325c78a10b590d1cfefafc6cbfda4c4948cc8b9838222f88cb7d84a6d7a8b") {
		for (let $i = 0; $i < 5; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f0c325c78a10b590d1cfefafc6cbfda4c4948cc8b9838222f88cb7d84a6d7a8b";
	}
	let props;
	if ($[1] !== t0) {
		({...props} = t0);
		$[1] = t0;
		$[2] = props;
	} else props = $[2];
	let t1;
	if ($[3] !== props) {
		t1 = /* @__PURE__ */ jsx(PopoverPrimitive.Root, {
			"data-slot": "popover",
			...props
		});
		$[3] = props;
		$[4] = t1;
	} else t1 = $[4];
	return t1;
};
const PopoverTrigger = (t0) => {
	const $ = c(5);
	if ($[0] !== "f0c325c78a10b590d1cfefafc6cbfda4c4948cc8b9838222f88cb7d84a6d7a8b") {
		for (let $i = 0; $i < 5; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f0c325c78a10b590d1cfefafc6cbfda4c4948cc8b9838222f88cb7d84a6d7a8b";
	}
	let props;
	if ($[1] !== t0) {
		({...props} = t0);
		$[1] = t0;
		$[2] = props;
	} else props = $[2];
	let t1;
	if ($[3] !== props) {
		t1 = /* @__PURE__ */ jsx(PopoverPrimitive.Trigger, {
			"data-slot": "popover-trigger",
			...props
		});
		$[3] = props;
		$[4] = t1;
	} else t1 = $[4];
	return t1;
};
const PopoverContent = (t0) => {
	const $ = c(13);
	if ($[0] !== "f0c325c78a10b590d1cfefafc6cbfda4c4948cc8b9838222f88cb7d84a6d7a8b") {
		for (let $i = 0; $i < 13; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f0c325c78a10b590d1cfefafc6cbfda4c4948cc8b9838222f88cb7d84a6d7a8b";
	}
	let className;
	let props;
	let t1;
	let t2;
	if ($[1] !== t0) {
		({className, align: t1, sideOffset: t2, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = props;
		$[4] = t1;
		$[5] = t2;
	} else {
		className = $[2];
		props = $[3];
		t1 = $[4];
		t2 = $[5];
	}
	const align = t1 === void 0 ? "center" : t1;
	const sideOffset = t2 === void 0 ? 4 : t2;
	let t3;
	if ($[6] !== className) {
		t3 = cn("bg-popover text-popover-foreground data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 w-72 origin-(--radix-popover-content-transform-origin) rounded-md border p-4 shadow-md outline-hidden", className);
		$[6] = className;
		$[7] = t3;
	} else t3 = $[7];
	let t4;
	if ($[8] !== align || $[9] !== props || $[10] !== sideOffset || $[11] !== t3) {
		t4 = /* @__PURE__ */ jsx(PopoverPrimitive.Portal, { children: /* @__PURE__ */ jsx(PopoverPrimitive.Content, {
			"data-slot": "popover-content",
			align,
			sideOffset,
			className: t3,
			...props
		}) });
		$[8] = align;
		$[9] = props;
		$[10] = sideOffset;
		$[11] = t3;
		$[12] = t4;
	} else t4 = $[12];
	return t4;
};

//#endregion
//#region src/components/events/code/utils.ts
const formatArgumentValue = (arg) => {
	if (typeof arg === "object" && arg !== null) return JSON.stringify(arg, null, 1);
	if (arg === null) return "null";
	return arg.toString();
};

//#endregion
//#region src/components/events/code/argument.tsx
const Argument = memo((t0) => {
	const $ = c(19);
	if ($[0] !== "c9966227012cab85e29278e1c8a710fd408ed34dfcad88a24d905811c73e1f19") {
		for (let $i = 0; $i < 19; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "c9966227012cab85e29278e1c8a710fd408ed34dfcad88a24d905811c73e1f19";
	}
	const { arg, popover: t1 } = t0;
	const popover = t1 === void 0 ? true : t1;
	const isObject = typeof arg === "object" && arg !== null;
	let t2;
	if ($[1] !== arg) {
		t2 = formatArgumentValue(arg);
		$[1] = arg;
		$[2] = t2;
	} else t2 = $[2];
	const value = t2;
	let t3;
	if ($[3] !== arg) {
		t3 = JSON.stringify(arg, null, 2);
		$[3] = arg;
		$[4] = t3;
	} else t3 = $[4];
	const textToCopy = t3;
	if (!popover) {
		let t4$1;
		if ($[5] !== arg || $[6] !== isObject || $[7] !== value) {
			t4$1 = /* @__PURE__ */ jsx(Fragment$1, { children: isObject ? /* @__PURE__ */ jsx(ReactJson, {
				src: arg,
				theme: "default",
				enableClipboard: false,
				style: { padding: 0 }
			}) : /* @__PURE__ */ jsx("span", {
				className: "font-mono text-blue-500",
				children: value
			}) });
			$[5] = arg;
			$[6] = isObject;
			$[7] = value;
			$[8] = t4$1;
		} else t4$1 = $[8];
		return t4$1;
	}
	let t4;
	if ($[9] !== value) {
		t4 = /* @__PURE__ */ jsx(PopoverTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsx("span", {
				className: "truncate font-mono text-blue-500 cursor-pointer inline-block",
				children: value
			})
		});
		$[9] = value;
		$[10] = t4;
	} else t4 = $[10];
	let t5;
	if ($[11] !== arg || $[12] !== isObject || $[13] !== textToCopy || $[14] !== value) {
		t5 = /* @__PURE__ */ jsx(PopoverContent, {
			className: "backdrop-blur-md w-auto max-w-2xl relative max-h-[50vh] overflow-y-auto",
			hideWhenDetached: true,
			children: isObject ? /* @__PURE__ */ jsxs("div", { children: [/* @__PURE__ */ jsx(CopyButton, {
				textToCopy,
				className: "absolute top-4 right-4",
				title: "Copy JSON"
			}), /* @__PURE__ */ jsx(ReactJson, {
				src: arg,
				theme: "default",
				enableClipboard: false,
				style: { padding: 0 }
			})] }) : /* @__PURE__ */ jsxs("div", {
				className: "flex items-center justify-between",
				children: [/* @__PURE__ */ jsx("span", { children: value }), /* @__PURE__ */ jsx(CopyButton, {
					textToCopy,
					className: "ml-2",
					title: "Copy JSON"
				})]
			})
		});
		$[11] = arg;
		$[12] = isObject;
		$[13] = textToCopy;
		$[14] = value;
		$[15] = t5;
	} else t5 = $[15];
	let t6;
	if ($[16] !== t4 || $[17] !== t5) {
		t6 = /* @__PURE__ */ jsxs(Popover, { children: [t4, t5] });
		$[16] = t4;
		$[17] = t5;
		$[18] = t6;
	} else t6 = $[18];
	return t6;
});
Argument.displayName = "Argument";

//#endregion
//#region src/components/events/code/closing-parenthesis.tsx
const ClosingParenthesis = memo((t0) => {
	const $ = c(8);
	if ($[0] !== "4ab03367b18a84c2b46e40ed2eb5f7059f595a509137274a6e998e2052c417ca") {
		for (let $i = 0; $i < 8; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "4ab03367b18a84c2b46e40ed2eb5f7059f595a509137274a6e998e2052c417ca";
	}
	const { callsQuantity, noQuantity: t1 } = t0;
	const noQuantity = t1 === void 0 ? false : t1;
	const hasCalls = callsQuantity && callsQuantity > 1;
	let t2;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsx("span", {
			className: "font-mono text-emerald-500",
			children: ")"
		});
		$[1] = t2;
	} else t2 = $[1];
	let t3;
	if ($[2] !== callsQuantity || $[3] !== hasCalls || $[4] !== noQuantity) {
		t3 = !noQuantity && hasCalls && /* @__PURE__ */ jsxs("span", {
			className: "font-mono text-muted-foreground",
			children: [" x", callsQuantity]
		});
		$[2] = callsQuantity;
		$[3] = hasCalls;
		$[4] = noQuantity;
		$[5] = t3;
	} else t3 = $[5];
	let t4;
	if ($[6] !== t3) {
		t4 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center",
			children: [t2, t3]
		});
		$[6] = t3;
		$[7] = t4;
	} else t4 = $[7];
	return t4;
});
ClosingParenthesis.displayName = "ClosingParenthesis";

//#endregion
//#region src/components/events/code/function-call-content.tsx
const FunctionCallContent = forwardRef((t0, ref) => {
	const $ = c(23);
	if ($[0] !== "f9cee2defe3a707d028f8a516d115937e9c066312490fa54845af3b1f275f18b") {
		for (let $i = 0; $i < 23; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "f9cee2defe3a707d028f8a516d115937e9c066312490fa54845af3b1f275f18b";
	}
	let className;
	let functionName;
	let objectName;
	let props;
	let topLevelClassName;
	if ($[1] !== t0) {
		({topLevelClassName, objectName, functionName, className, ...props} = t0);
		$[1] = t0;
		$[2] = className;
		$[3] = functionName;
		$[4] = objectName;
		$[5] = props;
		$[6] = topLevelClassName;
	} else {
		className = $[2];
		functionName = $[3];
		objectName = $[4];
		props = $[5];
		topLevelClassName = $[6];
	}
	let t1;
	if ($[7] !== className) {
		t1 = cn("flex items-center", className);
		$[7] = className;
		$[8] = t1;
	} else t1 = $[8];
	let t2;
	if ($[9] !== topLevelClassName) {
		t2 = topLevelClassName && /* @__PURE__ */ jsxs(Fragment$1, { children: [/* @__PURE__ */ jsx("span", {
			className: "font-mono text-pink-500",
			children: topLevelClassName
		}), /* @__PURE__ */ jsx("span", { children: "." })] });
		$[9] = topLevelClassName;
		$[10] = t2;
	} else t2 = $[10];
	let t3;
	if ($[11] !== objectName) {
		t3 = objectName && /* @__PURE__ */ jsxs(Fragment$1, { children: [/* @__PURE__ */ jsx("span", {
			className: "font-mono text-pink-500",
			children: objectName
		}), /* @__PURE__ */ jsx("span", { children: "." })] });
		$[11] = objectName;
		$[12] = t3;
	} else t3 = $[12];
	let t4;
	if ($[13] !== functionName) {
		t4 = /* @__PURE__ */ jsx("span", {
			className: "font-mono text-pink-500",
			children: functionName
		});
		$[13] = functionName;
		$[14] = t4;
	} else t4 = $[14];
	let t5;
	if ($[15] === Symbol.for("react.memo_cache_sentinel")) {
		t5 = /* @__PURE__ */ jsx("span", {
			className: "font-mono text-emerald-500",
			children: "("
		});
		$[15] = t5;
	} else t5 = $[15];
	let t6;
	if ($[16] !== props || $[17] !== ref || $[18] !== t1 || $[19] !== t2 || $[20] !== t3 || $[21] !== t4) {
		t6 = /* @__PURE__ */ jsxs("div", {
			ref,
			className: t1,
			...props,
			children: [
				t2,
				t3,
				t4,
				t5
			]
		});
		$[16] = props;
		$[17] = ref;
		$[18] = t1;
		$[19] = t2;
		$[20] = t3;
		$[21] = t4;
		$[22] = t6;
	} else t6 = $[22];
	return t6;
});
FunctionCallContent.displayName = "FunctionCallContent";

//#endregion
//#region src/components/events/code/function-call.tsx
const GRID_LAYOUT_STYLE = { gridTemplateColumns: "max-content 1fr max-content" };
const FunctionCall = memo((t0) => {
	const $ = c(49);
	if ($[0] !== "a4649595a9d35de28471f3275d991c694dbeccd05d971cc25fd0884247e778d1") {
		for (let $i = 0; $i < 49; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "a4649595a9d35de28471f3275d991c694dbeccd05d971cc25fd0884247e778d1";
	}
	const { topLevelClassName, objectName, functionName, args, callsQuantity } = t0;
	let t1;
	if ($[1] !== args) {
		t1 = args.filter(_temp$5);
		$[1] = args;
		$[2] = t1;
	} else t1 = $[2];
	const filteredArgs = t1;
	const t2 = `repeat(${filteredArgs.length * 2}, minmax(0, auto))`;
	let t3;
	if ($[3] !== t2) {
		t3 = { gridTemplateColumns: t2 };
		$[3] = t2;
		$[4] = t3;
	} else t3 = $[4];
	const argsGridStyle = t3;
	let t4;
	if ($[5] !== objectName || $[6] !== topLevelClassName) {
		t4 = [topLevelClassName, objectName].filter(Boolean);
		$[5] = objectName;
		$[6] = topLevelClassName;
		$[7] = t4;
	} else t4 = $[7];
	const prefix = t4.join(".");
	const fullPrefix = prefix ? `${prefix}.` : "";
	let t5;
	if ($[8] !== filteredArgs) {
		t5 = filteredArgs.map(_temp2$3).join(", ");
		$[8] = filteredArgs;
		$[9] = t5;
	} else t5 = $[9];
	const functionCallText = `${fullPrefix}${functionName}(${t5})`;
	let t6;
	if ($[10] !== functionName || $[11] !== objectName || $[12] !== topLevelClassName) {
		t6 = /* @__PURE__ */ jsx(PopoverTrigger, {
			asChild: true,
			children: /* @__PURE__ */ jsx(FunctionCallContent, {
				className: "cursor-pointer",
				functionName,
				topLevelClassName,
				objectName
			})
		});
		$[10] = functionName;
		$[11] = objectName;
		$[12] = topLevelClassName;
		$[13] = t6;
	} else t6 = $[13];
	let t7;
	if ($[14] !== functionName || $[15] !== objectName || $[16] !== topLevelClassName) {
		t7 = /* @__PURE__ */ jsx(FunctionCallContent, {
			functionName,
			topLevelClassName,
			objectName
		});
		$[14] = functionName;
		$[15] = objectName;
		$[16] = topLevelClassName;
		$[17] = t7;
	} else t7 = $[17];
	let t8;
	if ($[18] !== functionCallText) {
		t8 = /* @__PURE__ */ jsx(CopyButton, { textToCopy: functionCallText });
		$[18] = functionCallText;
		$[19] = t8;
	} else t8 = $[19];
	let t9;
	if ($[20] !== t7 || $[21] !== t8) {
		t9 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center justify-between gap-2",
			children: [t7, t8]
		});
		$[20] = t7;
		$[21] = t8;
		$[22] = t9;
	} else t9 = $[22];
	let t10;
	if ($[23] !== filteredArgs) {
		let t11$1;
		if ($[25] !== filteredArgs.length) {
			t11$1 = (arg_1, index) => /* @__PURE__ */ jsxs("div", {
				className: "grid grid-cols-[auto_1fr] items-center pl-4",
				children: [/* @__PURE__ */ jsx(Argument, {
					arg: arg_1,
					popover: false
				}), index < filteredArgs.length - 1 && /* @__PURE__ */ jsx("span", { children: ", " })]
			}, index);
			$[25] = filteredArgs.length;
			$[26] = t11$1;
		} else t11$1 = $[26];
		t10 = filteredArgs.map(t11$1);
		$[23] = filteredArgs;
		$[24] = t10;
	} else t10 = $[24];
	let t11;
	if ($[27] !== callsQuantity) {
		t11 = /* @__PURE__ */ jsx(ClosingParenthesis, {
			callsQuantity,
			noQuantity: true
		});
		$[27] = callsQuantity;
		$[28] = t11;
	} else t11 = $[28];
	let t12;
	if ($[29] !== t10 || $[30] !== t11 || $[31] !== t9) {
		t12 = /* @__PURE__ */ jsxs(PopoverContent, {
			className: "grid grid-rows-auto backdrop-blur-md w-auto max-w-2xl max-h-[50vh] overflow-y-auto",
			hideWhenDetached: true,
			children: [
				t9,
				t10,
				t11
			]
		});
		$[29] = t10;
		$[30] = t11;
		$[31] = t9;
		$[32] = t12;
	} else t12 = $[32];
	let t13;
	if ($[33] !== t12 || $[34] !== t6) {
		t13 = /* @__PURE__ */ jsxs(Popover, { children: [t6, t12] });
		$[33] = t12;
		$[34] = t6;
		$[35] = t13;
	} else t13 = $[35];
	let t14;
	if ($[36] !== filteredArgs) {
		let t15$1;
		if ($[38] !== filteredArgs.length) {
			t15$1 = (arg_2, index_0) => /* @__PURE__ */ jsxs(Fragment, { children: [/* @__PURE__ */ jsx(Argument, { arg: arg_2 }), index_0 < filteredArgs.length - 1 && /* @__PURE__ */ jsx("span", { children: ", " })] }, index_0);
			$[38] = filteredArgs.length;
			$[39] = t15$1;
		} else t15$1 = $[39];
		t14 = filteredArgs.map(t15$1);
		$[36] = filteredArgs;
		$[37] = t14;
	} else t14 = $[37];
	let t15;
	if ($[40] !== argsGridStyle || $[41] !== t14) {
		t15 = /* @__PURE__ */ jsx("div", {
			className: "grid items-center min-w-0",
			style: argsGridStyle,
			children: t14
		});
		$[40] = argsGridStyle;
		$[41] = t14;
		$[42] = t15;
	} else t15 = $[42];
	let t16;
	if ($[43] !== callsQuantity) {
		t16 = /* @__PURE__ */ jsx(ClosingParenthesis, { callsQuantity });
		$[43] = callsQuantity;
		$[44] = t16;
	} else t16 = $[44];
	let t17;
	if ($[45] !== t13 || $[46] !== t15 || $[47] !== t16) {
		t17 = /* @__PURE__ */ jsxs("div", {
			className: "grid overflow-hidden items-center",
			style: GRID_LAYOUT_STYLE,
			children: [
				t13,
				t15,
				t16
			]
		});
		$[45] = t13;
		$[46] = t15;
		$[47] = t16;
		$[48] = t17;
	} else t17 = $[48];
	return t17;
});
FunctionCall.displayName = "FunctionCall";
function _temp$5(arg) {
	return arg !== void 0;
}
function _temp2$3(arg_0) {
	if (typeof arg_0 === "object" && arg_0 !== null) return JSON.stringify(arg_0, null, 2);
	if (arg_0 === null) return "null";
	if (typeof arg_0 === "string") return `'${arg_0}'`;
	return String(arg_0);
}

//#endregion
//#region src/components/events/trace-emit-event.tsx
const TraceEmitEvent = memo((t0) => {
	const $ = c(6);
	if ($[0] !== "1c99155502674878739fbbcfa0a143efee75279ad69880260a5f175528b3f050") {
		for (let $i = 0; $i < 6; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "1c99155502674878739fbbcfa0a143efee75279ad69880260a5f175528b3f050";
	}
	const { event } = t0;
	let t1;
	if ($[1] !== event.data || $[2] !== event.topic) {
		t1 = [{
			topic: event.topic,
			data: event.data
		}];
		$[1] = event.data;
		$[2] = event.topic;
		$[3] = t1;
	} else t1 = $[3];
	const args = t1;
	let t2;
	if ($[4] !== args) {
		t2 = /* @__PURE__ */ jsx(FunctionCall, {
			functionName: "emit",
			args
		});
		$[4] = args;
		$[5] = t2;
	} else t2 = $[5];
	return t2;
});
TraceEmitEvent.displayName = "TraceEmitEvent";

//#endregion
//#region src/components/events/trace-log-event.tsx
const TraceLogEvent = memo((t0) => {
	const $ = c(6);
	if ($[0] !== "2ad036e2c4427811bdd655eab2630232cdaa410e5242349c3064a1c0f232d784") {
		for (let $i = 0; $i < 6; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "2ad036e2c4427811bdd655eab2630232cdaa410e5242349c3064a1c0f232d784";
	}
	const { event } = t0;
	let t1;
	if ($[1] !== event.level) {
		t1 = /* @__PURE__ */ jsx(LevelDot, { level: event.level });
		$[1] = event.level;
		$[2] = t1;
	} else t1 = $[2];
	let t2;
	if ($[3] !== event.message || $[4] !== t1) {
		t2 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center gap-2",
			children: [
				t1,
				" ",
				event.message
			]
		});
		$[3] = event.message;
		$[4] = t1;
		$[5] = t2;
	} else t2 = $[5];
	return t2;
});
TraceLogEvent.displayName = "TraceLogEvent";

//#endregion
//#region src/components/events/trace-state-event.tsx
const TraceStateEvent = memo((t0) => {
	const $ = c(8);
	if ($[0] !== "7c38f56f572803a01857cb4bed35b853c294cca3e11476480cd02e0ce7b86827") {
		for (let $i = 0; $i < 8; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "7c38f56f572803a01857cb4bed35b853c294cca3e11476480cd02e0ce7b86827";
	}
	const { event } = t0;
	let t1;
	if ($[1] !== event.data.key || $[2] !== event.data.traceId || $[3] !== event.data.value) {
		t1 = [
			event.data.traceId,
			event.data.key,
			event.data.value
		];
		$[1] = event.data.key;
		$[2] = event.data.traceId;
		$[3] = event.data.value;
		$[4] = t1;
	} else t1 = $[4];
	const args = t1;
	let t2;
	if ($[5] !== args || $[6] !== event.operation) {
		t2 = /* @__PURE__ */ jsx(FunctionCall, {
			objectName: "state",
			functionName: event.operation,
			args
		});
		$[5] = args;
		$[6] = event.operation;
		$[7] = t2;
	} else t2 = $[7];
	return t2;
});
TraceStateEvent.displayName = "TraceStateEvent";

//#endregion
//#region src/components/events/trace-stream-event.tsx
const TraceStreamEvent = memo((t0) => {
	const $ = c(10);
	if ($[0] !== "6c8ca715a79e72803fc3cde1b57bcfc9af7f426e893ce07dc6f50cf811b702b6") {
		for (let $i = 0; $i < 10; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "6c8ca715a79e72803fc3cde1b57bcfc9af7f426e893ce07dc6f50cf811b702b6";
	}
	const { event } = t0;
	const t1 = event.data.data;
	let t2;
	if ($[1] !== event.data.groupId || $[2] !== event.data.id || $[3] !== t1) {
		t2 = [
			event.data.groupId,
			event.data.id,
			t1
		];
		$[1] = event.data.groupId;
		$[2] = event.data.id;
		$[3] = t1;
		$[4] = t2;
	} else t2 = $[4];
	const args = t2;
	let t3;
	if ($[5] !== args || $[6] !== event.calls || $[7] !== event.operation || $[8] !== event.streamName) {
		t3 = /* @__PURE__ */ jsx(FunctionCall, {
			topLevelClassName: "streams",
			objectName: event.streamName,
			functionName: event.operation,
			args,
			callsQuantity: event.calls
		});
		$[5] = args;
		$[6] = event.calls;
		$[7] = event.operation;
		$[8] = event.streamName;
		$[9] = t3;
	} else t3 = $[9];
	return t3;
});
TraceStreamEvent.displayName = "TraceStreamEvent";

//#endregion
//#region src/components/events/trace-event.tsx
const TraceEvent = memo((t0) => {
	const $ = c(9);
	if ($[0] !== "bb6873064e676dc947ff1df89d486f742da27c53a44f3427557c30d4dcc50852") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "bb6873064e676dc947ff1df89d486f742da27c53a44f3427557c30d4dcc50852";
	}
	const { event } = t0;
	if (event.type === "log") {
		let t1;
		if ($[1] !== event) {
			t1 = /* @__PURE__ */ jsx(TraceLogEvent, { event });
			$[1] = event;
			$[2] = t1;
		} else t1 = $[2];
		return t1;
	} else if (event.type === "emit") {
		let t1;
		if ($[3] !== event) {
			t1 = /* @__PURE__ */ jsx(TraceEmitEvent, { event });
			$[3] = event;
			$[4] = t1;
		} else t1 = $[4];
		return t1;
	} else if (event.type === "state") {
		let t1;
		if ($[5] !== event) {
			t1 = /* @__PURE__ */ jsx(TraceStateEvent, { event });
			$[5] = event;
			$[6] = t1;
		} else t1 = $[6];
		return t1;
	} else if (event.type === "stream") {
		let t1;
		if ($[7] !== event) {
			t1 = /* @__PURE__ */ jsx(TraceStreamEvent, { event });
			$[7] = event;
			$[8] = t1;
		} else t1 = $[8];
		return t1;
	}
});
TraceEvent.displayName = "TraceEvent";

//#endregion
//#region src/components/trace-item/trace-event-item.tsx
const TraceEventItem = memo((t0) => {
	const $ = c(14);
	if ($[0] !== "6a7235dddbab5b9658f53f7dc4ccc892ec78776c6bf4ebe24fc950401299f990") {
		for (let $i = 0; $i < 14; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "6a7235dddbab5b9658f53f7dc4ccc892ec78776c6bf4ebe24fc950401299f990";
	}
	const { event, traceStartTime } = t0;
	let t1;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = /* @__PURE__ */ jsx("div", {
			className: "grid place-items-center",
			children: /* @__PURE__ */ jsx("div", { className: "w-1 h-1 rounded-full bg-emerald-500 outline outline-2 outline-emerald-500/50 -ml-[26px]" })
		});
		$[1] = t1;
	} else t1 = $[1];
	let t2;
	if ($[2] !== event) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: "grid place-items-center",
			children: /* @__PURE__ */ jsx(EventIcon, { event })
		});
		$[2] = event;
		$[3] = t2;
	} else t2 = $[3];
	const t3 = Math.floor(event.timestamp - traceStartTime);
	let t4;
	if ($[4] !== t3) {
		t4 = formatDuration(t3);
		$[4] = t3;
		$[5] = t4;
	} else t4 = $[5];
	let t5;
	if ($[6] !== t4) {
		t5 = /* @__PURE__ */ jsx("div", {
			className: "grid place-items-center",
			children: /* @__PURE__ */ jsxs("span", {
				className: "text-sm font-mono text-muted-foreground",
				children: ["+", t4]
			})
		});
		$[6] = t4;
		$[7] = t5;
	} else t5 = $[7];
	let t6;
	if ($[8] !== event) {
		t6 = /* @__PURE__ */ jsx("div", {
			className: "grid place-items-start",
			children: /* @__PURE__ */ jsx(TraceEvent, { event })
		});
		$[8] = event;
		$[9] = t6;
	} else t6 = $[9];
	let t7;
	if ($[10] !== t2 || $[11] !== t5 || $[12] !== t6) {
		t7 = /* @__PURE__ */ jsxs(Fragment, { children: [
			t1,
			t2,
			t5,
			t6
		] });
		$[10] = t2;
		$[11] = t5;
		$[12] = t6;
		$[13] = t7;
	} else t7 = $[13];
	return t7;
});
TraceEventItem.displayName = "TraceEventItem";

//#endregion
//#region src/components/trace-item/trace-item-detail.tsx
const TraceItemDetail = memo((t0) => {
	const $ = c(33);
	if ($[0] !== "1a8f3badc41dcb22bbab4d8d915d8d9b99520c2bd79570b27acfb6158532a96a") {
		for (let $i = 0; $i < 33; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "1a8f3badc41dcb22bbab4d8d915d8d9b99520c2bd79570b27acfb6158532a96a";
	}
	const { trace, onClose } = t0;
	let t1;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = /* @__PURE__ */ jsx(X, {});
		$[1] = t1;
	} else t1 = $[1];
	let t2;
	if ($[2] !== onClose) {
		t2 = [{
			icon: t1,
			onClick: onClose,
			label: "Close"
		}];
		$[2] = onClose;
		$[3] = t2;
	} else t2 = $[3];
	const actions = t2;
	const t3 = `Viewing details from step ${trace.name}`;
	let t4;
	if ($[4] !== trace.endTime || $[5] !== trace.startTime) {
		t4 = trace.endTime && /* @__PURE__ */ jsxs("span", { children: ["Duration: ", formatDuration(trace.endTime - trace.startTime)] });
		$[4] = trace.endTime;
		$[5] = trace.startTime;
		$[6] = t4;
	} else t4 = $[6];
	let t5;
	if ($[7] !== trace.entryPoint.type) {
		t5 = /* @__PURE__ */ jsx("div", {
			className: "bg-blue-500 font-bold text-xs px-[4px] py-[2px] rounded-sm text-blue-100",
			children: trace.entryPoint.type
		});
		$[7] = trace.entryPoint.type;
		$[8] = t5;
	} else t5 = $[8];
	let t6;
	if ($[9] !== trace.correlationId) {
		t6 = trace.correlationId && /* @__PURE__ */ jsxs(Badge, {
			variant: "outline",
			children: ["Correlated: ", trace.correlationId]
		});
		$[9] = trace.correlationId;
		$[10] = t6;
	} else t6 = $[10];
	let t7;
	if ($[11] !== t4 || $[12] !== t5 || $[13] !== t6) {
		t7 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center gap-4 text-sm text-muted-foreground mb-4",
			children: [
				t4,
				t5,
				t6
			]
		});
		$[11] = t4;
		$[12] = t5;
		$[13] = t6;
		$[14] = t7;
	} else t7 = $[14];
	let t8;
	if ($[15] !== trace.events || $[16] !== trace.startTime) {
		let t9$1;
		if ($[18] !== trace.startTime) {
			t9$1 = (event, index) => /* @__PURE__ */ jsx(TraceEventItem, {
				event,
				traceStartTime: trace.startTime
			}, index);
			$[18] = trace.startTime;
			$[19] = t9$1;
		} else t9$1 = $[19];
		t8 = trace.events.map(t9$1);
		$[15] = trace.events;
		$[16] = trace.startTime;
		$[17] = t8;
	} else t8 = $[17];
	let t9;
	if ($[20] !== t8) {
		t9 = /* @__PURE__ */ jsx("div", {
			className: "grid grid-cols-[auto_auto_auto_1fr] gap-x-2 gap-y-3 font-mono text-xs border-l-1 border-gray-500/40 pl-6",
			children: t8
		});
		$[20] = t8;
		$[21] = t9;
	} else t9 = $[21];
	let t10;
	if ($[22] !== t7 || $[23] !== t9) {
		t10 = /* @__PURE__ */ jsxs("div", {
			className: "px-2 overflow-auto",
			children: [t7, t9]
		});
		$[22] = t7;
		$[23] = t9;
		$[24] = t10;
	} else t10 = $[24];
	let t11;
	if ($[25] !== trace.error) {
		t11 = trace.error && /* @__PURE__ */ jsxs("div", {
			className: "p-4 bg-red-800/10",
			children: [/* @__PURE__ */ jsx("div", {
				className: "text-sm text-red-800 dark:text-red-400 font-semibold",
				children: trace.error.message
			}), /* @__PURE__ */ jsx("div", {
				className: "text-sm text-red-800 dark:text-red-400 pl-4",
				children: trace.error.stack
			})]
		});
		$[25] = trace.error;
		$[26] = t11;
	} else t11 = $[26];
	let t12;
	if ($[27] !== actions || $[28] !== onClose || $[29] !== t10 || $[30] !== t11 || $[31] !== t3) {
		t12 = /* @__PURE__ */ jsxs(Sidebar, {
			onClose,
			initialWidth: 600,
			title: "Trace Details",
			subtitle: t3,
			actions,
			children: [t10, t11]
		});
		$[27] = actions;
		$[28] = onClose;
		$[29] = t10;
		$[30] = t11;
		$[31] = t3;
		$[32] = t12;
	} else t12 = $[32];
	return t12;
});
TraceItemDetail.displayName = "TraceItemDetail";

//#endregion
//#region src/components/trace-timeline.tsx
const TraceTimeline = memo(() => {
	const $ = c(3);
	if ($[0] !== "0861cb6c9c7d40fb40c0b128d67452794264c9d58bd2fd2513753f1aaac123df") {
		for (let $i = 0; $i < 3; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "0861cb6c9c7d40fb40c0b128d67452794264c9d58bd2fd2513753f1aaac123df";
	}
	const groupId$1 = useObservabilityStore(_temp$4);
	if (!groupId$1) return null;
	let t0;
	if ($[1] !== groupId$1) {
		t0 = /* @__PURE__ */ jsx(TraceTimelineComponent, { groupId: groupId$1 });
		$[1] = groupId$1;
		$[2] = t0;
	} else t0 = $[2];
	return t0;
});
TraceTimeline.displayName = "TraceTimeline";
const TraceTimelineComponent = memo((t0) => {
	const $ = c(84);
	if ($[0] !== "0861cb6c9c7d40fb40c0b128d67452794264c9d58bd2fd2513753f1aaac123df") {
		for (let $i = 0; $i < 84; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "0861cb6c9c7d40fb40c0b128d67452794264c9d58bd2fd2513753f1aaac123df";
	}
	const { groupId: groupId$1 } = t0;
	useTracesStream();
	const traceGroupMetas = useObservabilityStore(_temp2$2);
	const traces = useObservabilityStore(_temp3$2);
	let t1;
	if ($[1] !== groupId$1 || $[2] !== traceGroupMetas || $[3] !== traces) {
		bb0: {
			let t2$1;
			if ($[5] !== groupId$1) {
				t2$1 = (m) => m.id === groupId$1;
				$[5] = groupId$1;
				$[6] = t2$1;
			} else t2$1 = $[6];
			const meta = traceGroupMetas.find(t2$1);
			if (!meta) {
				t1 = null;
				break bb0;
			}
			if (traces.length === 0) {
				t1 = {
					...meta,
					status: "running",
					lastActivity: meta.startTime,
					metadata: {
						completedSteps: 0,
						activeSteps: 0,
						totalSteps: 0
					}
				};
				break bb0;
			}
			t1 = deriveTraceGroup(meta, traces);
		}
		$[1] = groupId$1;
		$[2] = traceGroupMetas;
		$[3] = traces;
		$[4] = t1;
	} else t1 = $[4];
	const group = t1;
	const endTime = useGetEndTime(group);
	const [zoom, setZoom] = useState(100);
	const selectedTraceId = useObservabilityStore(_temp4$1);
	const selectTraceId = useObservabilityStore(_temp5);
	let t2;
	if ($[7] !== selectedTraceId || $[8] !== traces) {
		let t3$1;
		if ($[10] !== selectedTraceId) {
			t3$1 = (trace) => trace.id === selectedTraceId;
			$[10] = selectedTraceId;
			$[11] = t3$1;
		} else t3$1 = $[11];
		t2 = traces.find(t3$1);
		$[7] = selectedTraceId;
		$[8] = traces;
		$[9] = t2;
	} else t2 = $[9];
	const selectedTrace = t2;
	let t3;
	if ($[12] !== zoom) {
		t3 = () => zoom > 100 && setZoom(_temp6);
		$[12] = zoom;
		$[13] = t3;
	} else t3 = $[13];
	const zoomMinus = t3;
	let t4;
	if ($[14] !== zoom) {
		t4 = () => zoom < 200 && setZoom(_temp7);
		$[14] = zoom;
		$[15] = t4;
	} else t4 = $[15];
	const zoomPlus = t4;
	if (!group) return null;
	const t5 = `${zoom}%`;
	let t6;
	if ($[16] !== t5) {
		t6 = { width: t5 };
		$[16] = t5;
		$[17] = t6;
	} else t6 = $[17];
	const t7 = `${zoom}%`;
	let t8;
	if ($[18] !== t7) {
		t8 = { width: t7 };
		$[18] = t7;
		$[19] = t8;
	} else t8 = $[19];
	const t9 = zoom <= 100;
	let t10;
	if ($[20] === Symbol.for("react.memo_cache_sentinel")) {
		t10 = /* @__PURE__ */ jsx(Minus, { className: "w-4 h-4 cursor-pointer" });
		$[20] = t10;
	} else t10 = $[20];
	let t11;
	if ($[21] !== t9 || $[22] !== zoomMinus) {
		t11 = /* @__PURE__ */ jsx(Button, {
			variant: "icon",
			size: "sm",
			className: "px-2",
			onClick: zoomMinus,
			disabled: t9,
			children: t10
		});
		$[21] = t9;
		$[22] = zoomMinus;
		$[23] = t11;
	} else t11 = $[23];
	let t12;
	if ($[24] !== zoom) {
		t12 = /* @__PURE__ */ jsxs("span", {
			className: "min-w-12 text-center select-none text-sm font-bold text-muted-foreground",
			children: [zoom, "%"]
		});
		$[24] = zoom;
		$[25] = t12;
	} else t12 = $[25];
	const t13 = zoom >= 200;
	let t14;
	if ($[26] === Symbol.for("react.memo_cache_sentinel")) {
		t14 = /* @__PURE__ */ jsx(Plus, { className: "w-4 h-4 cursor-pointer" });
		$[26] = t14;
	} else t14 = $[26];
	let t15;
	if ($[27] !== t13 || $[28] !== zoomPlus) {
		t15 = /* @__PURE__ */ jsx(Button, {
			variant: "icon",
			size: "sm",
			className: "px-2",
			onClick: zoomPlus,
			disabled: t13,
			children: t14
		});
		$[27] = t13;
		$[28] = zoomPlus;
		$[29] = t15;
	} else t15 = $[29];
	let t16;
	if ($[30] !== t11 || $[31] !== t12 || $[32] !== t15) {
		t16 = /* @__PURE__ */ jsxs("div", {
			className: "shrink-0 w-[200px] h-[37px] flex items-center justify-center gap-2 sticky left-0 top-0 z-10 bg-card backdrop-blur-xs backdrop-filter",
			children: [
				t11,
				t12,
				t15
			]
		});
		$[30] = t11;
		$[31] = t12;
		$[32] = t15;
		$[33] = t16;
	} else t16 = $[33];
	let t17;
	if ($[34] === Symbol.for("react.memo_cache_sentinel")) {
		t17 = /* @__PURE__ */ jsx("span", { children: "0ms" });
		$[34] = t17;
	} else t17 = $[34];
	const t18 = Math.floor((endTime - group.startTime) * .25);
	let t19;
	if ($[35] !== t18) {
		t19 = formatDuration(t18);
		$[35] = t18;
		$[36] = t19;
	} else t19 = $[36];
	let t20;
	if ($[37] !== t19) {
		t20 = /* @__PURE__ */ jsx("span", { children: t19 });
		$[37] = t19;
		$[38] = t20;
	} else t20 = $[38];
	const t21 = Math.floor((endTime - group.startTime) * .5);
	let t22;
	if ($[39] !== t21) {
		t22 = formatDuration(t21);
		$[39] = t21;
		$[40] = t22;
	} else t22 = $[40];
	let t23;
	if ($[41] !== t22) {
		t23 = /* @__PURE__ */ jsx("span", { children: t22 });
		$[41] = t22;
		$[42] = t23;
	} else t23 = $[42];
	const t24 = Math.floor((endTime - group.startTime) * .75);
	let t25;
	if ($[43] !== t24) {
		t25 = formatDuration(t24);
		$[43] = t24;
		$[44] = t25;
	} else t25 = $[44];
	let t26;
	if ($[45] !== t25) {
		t26 = /* @__PURE__ */ jsx("span", { children: t25 });
		$[45] = t25;
		$[46] = t26;
	} else t26 = $[46];
	const t27 = Math.floor(endTime - group.startTime);
	let t28;
	if ($[47] !== t27) {
		t28 = formatDuration(t27);
		$[47] = t27;
		$[48] = t28;
	} else t28 = $[48];
	let t29;
	if ($[49] !== t28) {
		t29 = /* @__PURE__ */ jsx("span", { children: t28 });
		$[49] = t28;
		$[50] = t29;
	} else t29 = $[50];
	let t30;
	if ($[51] !== t20 || $[52] !== t23 || $[53] !== t26 || $[54] !== t29) {
		t30 = /* @__PURE__ */ jsxs("div", {
			className: "flex justify-between font-mono p-2 w-full text-xs text-muted-foreground bg-card",
			children: [
				t17,
				t20,
				t23,
				t26,
				t29
			]
		});
		$[51] = t20;
		$[52] = t23;
		$[53] = t26;
		$[54] = t29;
		$[55] = t30;
	} else t30 = $[55];
	let t31;
	if ($[56] !== t16 || $[57] !== t30 || $[58] !== t8) {
		t31 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-1 bg-background",
			style: t8,
			children: [t16, t30]
		});
		$[56] = t16;
		$[57] = t30;
		$[58] = t8;
		$[59] = t31;
	} else t31 = $[59];
	const t32 = `${zoom}%`;
	let t33;
	if ($[60] !== t32) {
		t33 = { width: t32 };
		$[60] = t32;
		$[61] = t33;
	} else t33 = $[61];
	let t34;
	if ($[62] !== endTime || $[63] !== group.startTime || $[64] !== selectTraceId || $[65] !== traces) {
		let t35$1;
		if ($[67] !== endTime || $[68] !== group.startTime || $[69] !== selectTraceId) {
			t35$1 = (trace_0) => /* @__PURE__ */ jsx(TraceItem, {
				traceId: trace_0.id,
				traceName: trace_0.name,
				traceStatus: trace_0.status,
				traceStartTime: trace_0.startTime,
				traceEndTime: trace_0.endTime,
				groupStartTime: group.startTime,
				groupEndTime: endTime,
				onExpand: selectTraceId
			}, trace_0.id);
			$[67] = endTime;
			$[68] = group.startTime;
			$[69] = selectTraceId;
			$[70] = t35$1;
		} else t35$1 = $[70];
		t34 = traces.map(t35$1);
		$[62] = endTime;
		$[63] = group.startTime;
		$[64] = selectTraceId;
		$[65] = traces;
		$[66] = t34;
	} else t34 = $[66];
	let t35;
	if ($[71] !== t33 || $[72] !== t34) {
		t35 = /* @__PURE__ */ jsx("div", {
			className: "flex flex-col h-full",
			style: t33,
			children: t34
		});
		$[71] = t33;
		$[72] = t34;
		$[73] = t35;
	} else t35 = $[73];
	let t36;
	if ($[74] !== t31 || $[75] !== t35 || $[76] !== t6) {
		t36 = /* @__PURE__ */ jsx("div", {
			className: "flex flex-col flex-1 relative min-h-full min-w-[1000px]",
			children: /* @__PURE__ */ jsxs("div", {
				className: "flex flex-col",
				style: t6,
				children: [t31, t35]
			})
		});
		$[74] = t31;
		$[75] = t35;
		$[76] = t6;
		$[77] = t36;
	} else t36 = $[77];
	let t37;
	if ($[78] !== selectTraceId || $[79] !== selectedTrace) {
		t37 = selectedTrace && /* @__PURE__ */ jsx(TraceItemDetail, {
			trace: selectedTrace,
			onClose: () => selectTraceId(void 0)
		});
		$[78] = selectTraceId;
		$[79] = selectedTrace;
		$[80] = t37;
	} else t37 = $[80];
	let t38;
	if ($[81] !== t36 || $[82] !== t37) {
		t38 = /* @__PURE__ */ jsxs(Fragment$1, { children: [t36, t37] });
		$[81] = t36;
		$[82] = t37;
		$[83] = t38;
	} else t38 = $[83];
	return t38;
});
TraceTimelineComponent.displayName = "TraceTimelineComponent";
function _temp$4(state) {
	return state.selectedTraceGroupId;
}
function _temp2$2(state) {
	return state.traceGroupMetas;
}
function _temp3$2(state_0) {
	return state_0.traces;
}
function _temp4$1(state_1) {
	return state_1.selectedTraceId;
}
function _temp5(state_2) {
	return state_2.selectTraceId;
}
function _temp6(prevZoom) {
	return prevZoom - 10;
}
function _temp7(prevZoom_0) {
	return prevZoom_0 + 10;
}

//#endregion
//#region src/hooks/use-all-traces-stream.ts
const streamName$1 = "motia-trace";
const useAllTracesStream = () => {
	const $ = c(7);
	if ($[0] !== "0dc3e9c8afc0b2969dcf2787b0c80362563ef31a54c1ab42e67b0ba82db2facf") {
		for (let $i = 0; $i < 7; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "0dc3e9c8afc0b2969dcf2787b0c80362563ef31a54c1ab42e67b0ba82db2facf";
	}
	const { stream } = useMotiaStream();
	const traceGroupMetas = useObservabilityStore(_temp$3);
	const setTracesForGroup = useObservabilityStore(_temp2$1);
	let t0;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t0 = /* @__PURE__ */ new Map();
		$[1] = t0;
	} else t0 = $[1];
	const subscriptionsRef = useRef(t0);
	let t1;
	let t2;
	if ($[2] !== setTracesForGroup || $[3] !== stream || $[4] !== traceGroupMetas) {
		t1 = () => {
			if (!stream) return;
			const currentGroupIds = new Set(traceGroupMetas.map(_temp3$1));
			const subscribedGroupIds = new Set(subscriptionsRef.current.keys());
			traceGroupMetas.forEach((meta_0) => {
				if (!subscriptionsRef.current.has(meta_0.id)) {
					const subscription = stream.subscribeGroup(streamName$1, meta_0.id);
					subscription.addChangeListener((data) => {
						const traces = data;
						setTracesForGroup(meta_0.id, traces);
					});
					subscriptionsRef.current.set(meta_0.id, subscription);
				}
			});
			subscribedGroupIds.forEach((groupId$1) => {
				if (!currentGroupIds.has(groupId$1)) {
					const subscription_0 = subscriptionsRef.current.get(groupId$1);
					if (subscription_0) {
						subscription_0.close();
						subscriptionsRef.current.delete(groupId$1);
					}
				}
			});
			return () => {
				subscriptionsRef.current.forEach(_temp4);
				subscriptionsRef.current.clear();
			};
		};
		t2 = [
			stream,
			traceGroupMetas,
			setTracesForGroup
		];
		$[2] = setTracesForGroup;
		$[3] = stream;
		$[4] = traceGroupMetas;
		$[5] = t1;
		$[6] = t2;
	} else {
		t1 = $[5];
		t2 = $[6];
	}
	useEffect(t1, t2);
};
function _temp$3(state) {
	return state.traceGroupMetas;
}
function _temp2$1(state_0) {
	return state_0.setTracesForGroup;
}
function _temp3$1(meta) {
	return meta.id;
}
function _temp4(subscription_1) {
	subscription_1.close();
}

//#endregion
//#region src/hooks/use-filtered-trace-groups.ts
const useFilteredTraceGroups = () => {
	const $ = c(9);
	if ($[0] !== "ec346c037792c603f78bfd5665c2f837f63d92d06f90a3f8a121adc9dc036808") {
		for (let $i = 0; $i < 9; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "ec346c037792c603f78bfd5665c2f837f63d92d06f90a3f8a121adc9dc036808";
	}
	const traceGroupMetas = useObservabilityStore(_temp$2);
	const tracesByGroupId = useObservabilityStore(_temp2);
	const search = useObservabilityStore(_temp3);
	let t0;
	if ($[1] !== search || $[2] !== traceGroupMetas || $[3] !== tracesByGroupId) {
		let t1;
		if ($[5] !== tracesByGroupId) {
			t1 = (meta) => {
				const traces = tracesByGroupId[meta.id] || [];
				if (traces.length === 0) return {
					...meta,
					status: "running",
					lastActivity: meta.startTime,
					metadata: {
						completedSteps: 0,
						activeSteps: 0,
						totalSteps: 0
					}
				};
				return deriveTraceGroup(meta, traces);
			};
			$[5] = tracesByGroupId;
			$[6] = t1;
		} else t1 = $[6];
		const traceGroups = traceGroupMetas.map(t1);
		let t2;
		if ($[7] !== search) {
			t2 = (group) => group.name.toLowerCase().includes(search.toLowerCase()) || group.id.toLowerCase().includes(search.toLowerCase());
			$[7] = search;
			$[8] = t2;
		} else t2 = $[8];
		t0 = traceGroups.filter(t2);
		$[1] = search;
		$[2] = traceGroupMetas;
		$[3] = tracesByGroupId;
		$[4] = t0;
	} else t0 = $[4];
	return t0;
};
function _temp$2(state) {
	return state.traceGroupMetas;
}
function _temp2(state_0) {
	return state_0.tracesByGroupId;
}
function _temp3(state_1) {
	return state_1.search;
}

//#endregion
//#region src/hooks/use-trace-groups-stream.ts
const streamName = "motia-trace-group";
const groupId = "default";
const useTraceGroupsStream = () => {
	const $ = c(3);
	if ($[0] !== "a55cc63ac3227553bdabe16419e817ac4d448fdd9e3663a994c2e53ebc24d222") {
		for (let $i = 0; $i < 3; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "a55cc63ac3227553bdabe16419e817ac4d448fdd9e3663a994c2e53ebc24d222";
	}
	const setData = useObservabilityStore(_temp$1);
	let t0;
	if ($[1] !== setData) {
		t0 = {
			streamName,
			groupId,
			setData
		};
		$[1] = setData;
		$[2] = t0;
	} else t0 = $[2];
	useStreamGroup(t0);
};
function _temp$1(state) {
	return state.setTraceGroupMetas;
}

//#endregion
//#region src/components/trace-status.tsx
const variantMap = {
	running: "info",
	completed: "success",
	failed: "error",
	default: "default"
};
const TraceStatusBadge = memo((t0) => {
	const $ = c(4);
	if ($[0] !== "135bb98907459b4eb8cfbaba04a1232c86a5e95742a41c0a99dcf1a140fa89df") {
		for (let $i = 0; $i < 4; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "135bb98907459b4eb8cfbaba04a1232c86a5e95742a41c0a99dcf1a140fa89df";
	}
	const { status, duration } = t0;
	const t1 = variantMap[status];
	const t2 = duration && status !== "failed" ? duration : status;
	let t3;
	if ($[1] !== t1 || $[2] !== t2) {
		t3 = /* @__PURE__ */ jsx(Badge, {
			variant: t1,
			children: t2
		});
		$[1] = t1;
		$[2] = t2;
		$[3] = t3;
	} else t3 = $[3];
	return t3;
});
TraceStatusBadge.displayName = "TraceStatusBadge";

//#endregion
//#region src/components/trace-group-item.tsx
const TraceGroupItem = memo((t0) => {
	const $ = c(44);
	if ($[0] !== "ade6a55bfd668f5f8283f490ae1f964ba2b70d3e3adecd61f24c5ddb8d2b211f") {
		for (let $i = 0; $i < 44; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "ade6a55bfd668f5f8283f490ae1f964ba2b70d3e3adecd61f24c5ddb8d2b211f";
	}
	const { groupId: groupId$1, groupName, groupStatus, groupStartTime, groupEndTime, totalSteps, activeSteps, isSelected, style } = t0;
	const selectTraceGroupId = useObservabilityStore(_temp);
	let t1;
	if ($[1] !== groupEndTime || $[2] !== groupStartTime) {
		t1 = groupEndTime ? formatDuration(groupEndTime - groupStartTime) : void 0;
		$[1] = groupEndTime;
		$[2] = groupStartTime;
		$[3] = t1;
	} else t1 = $[3];
	const duration = t1;
	let t2;
	if ($[4] !== groupId$1 || $[5] !== selectTraceGroupId) {
		t2 = () => {
			selectTraceGroupId(groupId$1);
		};
		$[4] = groupId$1;
		$[5] = selectTraceGroupId;
		$[6] = t2;
	} else t2 = $[6];
	const onSelect = t2;
	const t3 = `trace-${groupId$1}`;
	const t4 = isSelected ? "bg-muted-foreground/10" : "hover:bg-muted/70";
	let t5;
	if ($[7] !== t4) {
		t5 = cn("motia-trace-group cursor-pointer transition-colors w-full text-left", t4);
		$[7] = t4;
		$[8] = t5;
	} else t5 = $[8];
	let t6;
	if ($[9] !== groupName) {
		t6 = /* @__PURE__ */ jsx("span", {
			className: "font-semibold text-lg truncate flex-1 min-w-0",
			children: groupName
		});
		$[9] = groupName;
		$[10] = t6;
	} else t6 = $[10];
	let t7;
	if ($[11] !== duration || $[12] !== groupStatus) {
		t7 = /* @__PURE__ */ jsx(TraceStatusBadge, {
			status: groupStatus,
			duration
		});
		$[11] = duration;
		$[12] = groupStatus;
		$[13] = t7;
	} else t7 = $[13];
	let t8;
	if ($[14] !== t6 || $[15] !== t7) {
		t8 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-row justify-between items-center gap-2",
			children: [t6, t7]
		});
		$[14] = t6;
		$[15] = t7;
		$[16] = t8;
	} else t8 = $[16];
	let t9;
	if ($[17] !== groupId$1) {
		t9 = /* @__PURE__ */ jsx("div", {
			"data-testid": "trace-id",
			className: "text-xs text-muted-foreground font-mono tracking-[1px]",
			children: groupId$1
		});
		$[17] = groupId$1;
		$[18] = t9;
	} else t9 = $[18];
	let t10;
	if ($[19] !== totalSteps) {
		t10 = /* @__PURE__ */ jsxs("span", { children: [totalSteps, " steps"] });
		$[19] = totalSteps;
		$[20] = t10;
	} else t10 = $[20];
	let t11;
	if ($[21] !== t10 || $[22] !== t9) {
		t11 = /* @__PURE__ */ jsxs("div", {
			className: "flex justify-between",
			children: [t9, t10]
		});
		$[21] = t10;
		$[22] = t9;
		$[23] = t11;
	} else t11 = $[23];
	let t12;
	if ($[24] !== groupStartTime) {
		t12 = formatDistanceToNow(groupStartTime);
		$[24] = groupStartTime;
		$[25] = t12;
	} else t12 = $[25];
	let t13;
	if ($[26] !== t12) {
		t13 = /* @__PURE__ */ jsxs("div", {
			className: "flex justify-between",
			children: [t12, " ago"]
		});
		$[26] = t12;
		$[27] = t13;
	} else t13 = $[27];
	let t14;
	if ($[28] !== activeSteps) {
		t14 = activeSteps > 0 && /* @__PURE__ */ jsxs("div", {
			className: "text-blue-600",
			children: [activeSteps, " active"]
		});
		$[28] = activeSteps;
		$[29] = t14;
	} else t14 = $[29];
	let t15;
	if ($[30] !== t11 || $[31] !== t13 || $[32] !== t14) {
		t15 = /* @__PURE__ */ jsxs("div", {
			className: "text-xs text-muted-foreground space-y-1",
			children: [
				t11,
				t13,
				t14
			]
		});
		$[30] = t11;
		$[31] = t13;
		$[32] = t14;
		$[33] = t15;
	} else t15 = $[33];
	let t16;
	if ($[34] !== t15 || $[35] !== t8) {
		t16 = /* @__PURE__ */ jsxs("div", {
			className: "p-3 flex flex-col gap-1",
			children: [t8, t15]
		});
		$[34] = t15;
		$[35] = t8;
		$[36] = t16;
	} else t16 = $[36];
	let t17;
	if ($[37] !== groupId$1 || $[38] !== onSelect || $[39] !== style || $[40] !== t16 || $[41] !== t3 || $[42] !== t5) {
		t17 = /* @__PURE__ */ jsx("button", {
			type: "button",
			"data-testid": t3,
			className: t5,
			style,
			onClick: onSelect,
			children: t16
		}, groupId$1);
		$[37] = groupId$1;
		$[38] = onSelect;
		$[39] = style;
		$[40] = t16;
		$[41] = t3;
		$[42] = t5;
		$[43] = t17;
	} else t17 = $[43];
	return t17;
});
TraceGroupItem.displayName = "TraceGroupItem";
function _temp(state) {
	return state.selectTraceGroupId;
}

//#endregion
//#region src/components/traces-groups.tsx
const ROW_HEIGHT = 110;
const TracesGroups = memo(() => {
	useTraceGroupsStream();
	useAllTracesStream();
	const groups = useFilteredTraceGroups();
	const selectedGroupId = useObservabilityStore((state) => state.selectedTraceGroupId);
	const selectTraceGroupId = useObservabilityStore((state_0) => state_0.selectTraceGroupId);
	const scrollContainerRef = useRef(null);
	const groupsLength = useMemo(() => groups?.length || 0, [groups]);
	const lastRunningGroupId = useMemo(() => {
		if (!groups || groups.length === 0) return "";
		const lastGroup = groups[groups.length - 1];
		return lastGroup?.status === "running" ? lastGroup.id : "";
	}, [groups]);
	useEffect(() => {
		if (lastRunningGroupId && lastRunningGroupId !== selectedGroupId) selectTraceGroupId(lastRunningGroupId);
		else if (!lastRunningGroupId && !groupsLength && selectedGroupId) selectTraceGroupId("");
	}, [
		lastRunningGroupId,
		groupsLength,
		selectedGroupId,
		selectTraceGroupId
	]);
	const reversedGroups = useMemo(() => groups ? [...groups].reverse() : [], [groups]);
	const virtualizer = useVirtualizer({
		count: reversedGroups.length,
		getScrollElement: () => scrollContainerRef.current,
		estimateSize: () => ROW_HEIGHT,
		overscan: 5
	});
	const virtualItems = virtualizer.getVirtualItems();
	if (!groups || groups.length === 0) return null;
	return /* @__PURE__ */ jsx("div", {
		ref: scrollContainerRef,
		className: "overflow-auto h-full",
		children: /* @__PURE__ */ jsx("div", {
			className: "relative w-full",
			style: { height: virtualizer.getTotalSize() },
			children: virtualItems.map((virtualRow) => {
				const group = reversedGroups[virtualRow.index];
				if (!group) return null;
				return /* @__PURE__ */ jsx(TraceGroupItem, {
					groupId: group.id,
					groupName: group.name,
					groupStatus: group.status,
					groupStartTime: group.startTime,
					groupEndTime: group.endTime,
					totalSteps: group.metadata.totalSteps,
					activeSteps: group.metadata.activeSteps,
					isSelected: selectedGroupId === group.id,
					style: {
						position: "absolute",
						top: 0,
						left: 0,
						width: "100%",
						height: ROW_HEIGHT,
						transform: `translateY(${virtualRow.start}px)`
					}
				}, group.id);
			})
		})
	});
});
TracesGroups.displayName = "TracesGroups";

//#endregion
//#region src/components/observability-page.tsx
const ObservabilityPage = memo(() => {
	const $ = c(4);
	if ($[0] !== "eb4316941d86db32e402fdb6a9fa0b346d5506cfc5da93f56a7fbdbc084ca168") {
		for (let $i = 0; $i < 4; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "eb4316941d86db32e402fdb6a9fa0b346d5506cfc5da93f56a7fbdbc084ca168";
	}
	let t0;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t0 = /* @__PURE__ */ jsx(SearchBar, {});
		$[1] = t0;
	} else t0 = $[1];
	let t1;
	if ($[2] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = /* @__PURE__ */ jsx("div", {
			className: "w-[300px] border-r border-border overflow-auto h-full",
			"data-testid": "traces-container",
			children: /* @__PURE__ */ jsx(TracesGroups, {})
		});
		$[2] = t1;
	} else t1 = $[2];
	let t2;
	if ($[3] === Symbol.for("react.memo_cache_sentinel")) {
		t2 = /* @__PURE__ */ jsxs("div", {
			className: "grid grid-rows-[auto_1fr] h-full",
			children: [t0, /* @__PURE__ */ jsxs("div", {
				className: "grid grid-cols-[300px_1fr] overflow-hidden",
				children: [t1, /* @__PURE__ */ jsxs("div", {
					className: "overflow-auto",
					"data-testid": "trace-details",
					children: [/* @__PURE__ */ jsx(TraceTimeline, {}), /* @__PURE__ */ jsx(TraceEmptyState, {})]
				})]
			})]
		});
		$[3] = t2;
	} else t2 = $[3];
	return t2;
});
ObservabilityPage.displayName = "ObservabilityPage";

//#endregion
//#region src/components/observability-tab-label.tsx
const ObservabilityTabLabel = memo(() => {
	const $ = c(2);
	if ($[0] !== "12b48555066a60a0ff63053b795596dd6b1058a64ec587d2a2a72e7129e319e3") {
		for (let $i = 0; $i < 2; $i += 1) $[$i] = Symbol.for("react.memo_cache_sentinel");
		$[0] = "12b48555066a60a0ff63053b795596dd6b1058a64ec587d2a2a72e7129e319e3";
	}
	let t0;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t0 = /* @__PURE__ */ jsxs("div", {
			"data-testid": "observability-link",
			children: [/* @__PURE__ */ jsx(GanttChartSquare, { "aria-hidden": "true" }), /* @__PURE__ */ jsx("span", { children: "Tracing" })]
		});
		$[1] = t0;
	} else t0 = $[1];
	return t0;
});
ObservabilityTabLabel.displayName = "ObservabilityTabLabel";

//#endregion
export { ObservabilityPage, ObservabilityTabLabel };
