// @motiadev/workbench/dist/index.js
import { Handle, Position, Position as Position$1, useReactFlow, useUpdateNodeInternals } from "@xyflow/react";
import { c } from "react/compiler-runtime";
import { Button, Sidebar, cn, useThemeStore } from "@motiadev/ui";
import { CalendarClock, CircleOff, Link2, ScanSearch, Waypoints, X } from "lucide-react";
import { useEffect, useRef, useState } from "react";
import clsx from "clsx";
import { Fragment, jsx, jsxs } from "react/jsx-runtime";
import { cva } from "class-variance-authority";
import { Prism } from "react-syntax-highlighter";
import { oneDark, oneLight } from "react-syntax-highlighter/dist/esm/styles/prism";

//#region src/components/tutorial/engine/workbench-xpath.ts
const workbenchXPath = {
	sidebarContainer: "//div[@data-testid=\"sidebar-panel\"]",
	closePanelButton: "//div[@id=\"app-sidebar-container\"]//button[@data-testid=\"close-panel\"]",
	bottomPanel: "//div[@id=\"bottom-panel\"]",
	flows: {
		dropdownFlow: (flowId) => `//div[@data-testid="dropdown-${flowId}"]`,
		feature: (featureId) => `//div[@data-feature-id="${featureId}"]`,
		previewButton: (stepId) => `//button[@data-testid="open-code-preview-button-${stepId}"]`,
		node: (stepId) => `//div[@data-testid="node-${stepId}"]`
	},
	endpoints: {
		endpointsList: "//div[@data-testid=\"endpoints-list\"]",
		endpoint: (method, path) => `//div[@data-testid="endpoint-${method}-${path}"]`,
		callPanel: "//div[@data-testid=\"endpoint-details-panel\"]",
		specButton: "//button[@data-testid=\"endpoint-spec-button\"]",
		bodyTab: "//button[@data-testid=\"endpoint-body-tab\"]",
		headersTab: "//button[@data-testid=\"endpoint-headers-tab\"]",
		paramsTab: "//button[@data-testid=\"endpoint-params-tab\"]",
		callTab: "//button[@data-testid=\"endpoint-body-tab\"]",
		response: "//div[@data-testid=\"endpoint-response-container\"]",
		playButton: "//button[@data-testid=\"endpoint-play-button\"]"
	},
	tracing: {
		trace: (index) => `(//button[contains(@class, 'motia-trace-group')])[${index}]`,
		details: "//div[@data-testid=\"trace-details\"]",
		timeline: (index) => `(//div[@data-testid="trace-timeline-item"])[${index}]`
	},
	logs: {
		container: "//div[@data-testid=\"logs-container\"]",
		searchContainer: "//div[@data-testid=\"logs-search-container\"]",
		traceColumn: (index) => `(//button[starts-with(@data-testid, 'trace-filter')])[${index}]`,
		row: "//div[@data-testid=\"log-row\"]"
	},
	states: {
		container: "//div[@data-testid=\"states-container\"]",
		row: (index) => `(//tr[starts-with(@data-testid, 'item-')])[${index}]`
	},
	links: {
		flows: "//div[@data-testid=\"flows-dropdown-trigger\"]",
		endpoints: "//button[@data-testid=\"endpoints-link\"]",
		tracing: "//button[@data-testid=\"tracing-link\"]",
		logs: "//button[@data-testid=\"logs-link\"]",
		states: "//button[@data-testid=\"states-link\"]"
	}
};

//#endregion
//#region src/hooks/use-update-handle-positions.ts
const useHandlePositions = (data) => {
	const $ = c(21);
	const reactFlow = useReactFlow();
	const updateNodeInternals = useUpdateNodeInternals();
	const sourcePosition = data.nodeConfig?.sourceHandlePosition === "bottom" ? Position$1.Bottom : Position$1.Right;
	const targetPosition = data.nodeConfig?.targetHandlePosition === "top" ? Position$1.Top : Position$1.Left;
	let t0;
	if ($[0] !== data || $[1] !== reactFlow || $[2] !== updateNodeInternals) {
		t0 = (position) => {
			reactFlow.updateNode(data.id, { data: {
				...data,
				nodeConfig: {
					...data.nodeConfig,
					sourceHandlePosition: position
				}
			} });
			updateNodeInternals(data.id);
		};
		$[0] = data;
		$[1] = reactFlow;
		$[2] = updateNodeInternals;
		$[3] = t0;
	} else t0 = $[3];
	const updateSourcePosition = t0;
	let t1;
	if ($[4] !== data || $[5] !== reactFlow || $[6] !== updateNodeInternals) {
		t1 = (position_0) => {
			reactFlow.updateNode(data.id, { data: {
				...data,
				nodeConfig: {
					...data.nodeConfig,
					targetHandlePosition: position_0
				}
			} });
			updateNodeInternals(data.id);
		};
		$[4] = data;
		$[5] = reactFlow;
		$[6] = updateNodeInternals;
		$[7] = t1;
	} else t1 = $[7];
	const updateTargetPosition = t1;
	let t2;
	if ($[8] !== targetPosition || $[9] !== updateTargetPosition) {
		t2 = () => {
			updateTargetPosition(targetPosition === Position$1.Top ? Position$1.Left : Position$1.Top);
		};
		$[8] = targetPosition;
		$[9] = updateTargetPosition;
		$[10] = t2;
	} else t2 = $[10];
	const toggleTargetPosition = t2;
	let t3;
	if ($[11] !== sourcePosition || $[12] !== updateSourcePosition) {
		t3 = () => {
			updateSourcePosition(sourcePosition === Position$1.Bottom ? Position$1.Right : Position$1.Bottom);
		};
		$[11] = sourcePosition;
		$[12] = updateSourcePosition;
		$[13] = t3;
	} else t3 = $[13];
	const toggleSourcePosition = t3;
	let t4;
	if ($[14] !== sourcePosition || $[15] !== targetPosition || $[16] !== toggleSourcePosition || $[17] !== toggleTargetPosition || $[18] !== updateSourcePosition || $[19] !== updateTargetPosition) {
		t4 = {
			sourcePosition,
			targetPosition,
			updateSourcePosition,
			updateTargetPosition,
			toggleTargetPosition,
			toggleSourcePosition
		};
		$[14] = sourcePosition;
		$[15] = targetPosition;
		$[16] = toggleSourcePosition;
		$[17] = toggleTargetPosition;
		$[18] = updateSourcePosition;
		$[19] = updateTargetPosition;
		$[20] = t4;
	} else t4 = $[20];
	return t4;
};

//#endregion
//#region src/publicComponents/base-node/base-handle.tsx
const BaseHandle = (props) => {
	const $ = c(20);
	let isHidden;
	let onTogglePosition;
	let position;
	let rest;
	if ($[0] !== props) {
		({isHidden, position, onTogglePosition, ...rest} = props);
		$[0] = props;
		$[1] = isHidden;
		$[2] = onTogglePosition;
		$[3] = position;
		$[4] = rest;
	} else {
		isHidden = $[1];
		onTogglePosition = $[2];
		position = $[3];
		rest = $[4];
	}
	const isHorizontal = position === Position$1.Left || position === Position$1.Right;
	const t0 = position === Position$1.Top && "-top-[4px]";
	const t1 = position === Position$1.Bottom && "-bottom-[4px]";
	const t2 = position === Position$1.Left && "-left-[4px]";
	const t3 = position === Position$1.Right && "-right-[4px]";
	const t4 = isHorizontal ? "top-1/2 -mt-[2px]" : "left-1/2 -ml-[2px]";
	const t5 = isHidden && "hidden";
	let t6;
	if ($[5] !== t0 || $[6] !== t1 || $[7] !== t2 || $[8] !== t3 || $[9] !== t4 || $[10] !== t5) {
		t6 = clsx("absolute w-[6px] h-[6px]", t0, t1, t2, t3, t4, t5);
		$[5] = t0;
		$[6] = t1;
		$[7] = t2;
		$[8] = t3;
		$[9] = t4;
		$[10] = t5;
		$[11] = t6;
	} else t6 = $[11];
	let t7;
	if ($[12] === Symbol.for("react.memo_cache_sentinel")) {
		t7 = { background: "rgb(30,118,231)" };
		$[12] = t7;
	} else t7 = $[12];
	let t8;
	if ($[13] !== position || $[14] !== rest) {
		t8 = /* @__PURE__ */ jsx(Handle, {
			...rest,
			position,
			style: t7,
			className: "\n        bg-white/50\n        !static\n        !w-[6px]\n        !h-[6px]\n        !min-w-[6px]\n        !min-h-[6px]\n        !p-0\n        !border-none\n        !transform-none\n        !rounded-full\n        !outline-none\n        !shadow-none\n      "
		});
		$[13] = position;
		$[14] = rest;
		$[15] = t8;
	} else t8 = $[15];
	let t9;
	if ($[16] !== onTogglePosition || $[17] !== t6 || $[18] !== t8) {
		t9 = /* @__PURE__ */ jsx("div", {
			className: t6,
			onClick: onTogglePosition,
			children: t8
		});
		$[16] = onTogglePosition;
		$[17] = t6;
		$[18] = t8;
		$[19] = t9;
	} else t9 = $[19];
	return t9;
};

//#endregion
//#region src/publicComponents/base-node/node-header.tsx
const baseIcon = cva("rounded-md p-2", { variants: { variant: {
	event: "bg-[rgba(30,118,231,0.2)] text-[rgb(30,118,231)]",
	api: "dark:bg-[rgba(192,255,17,0.2)] dark:text-[rgb(192,255,17)] bg-[rgb(231,255,166)] text-[rgb(94,125,11)]",
	noop: "bg-[rgba(239,18,229,0.2)] text-[rgb(239,18,229)]",
	cron: "bg-[rgba(241,105,15,0.2)] text-[rgb(241,105,15)]"
} } });
const NodeIcon = (t0) => {
	const $ = c(8);
	const { variant } = t0;
	if (variant === "cron") {
		let t1$1;
		if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
			t1$1 = /* @__PURE__ */ jsx(CalendarClock, { className: "w-5 h-5" });
			$[0] = t1$1;
		} else t1$1 = $[0];
		return t1$1;
	} else if (variant === "api") {
		let t1$1;
		if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
			t1$1 = /* @__PURE__ */ jsx(Link2, { className: "w-5 h-5" });
			$[1] = t1$1;
		} else t1$1 = $[1];
		return t1$1;
	} else if (variant === "noop") {
		let t1$1;
		if ($[2] === Symbol.for("react.memo_cache_sentinel")) {
			t1$1 = /* @__PURE__ */ jsx(CircleOff, { className: "w-5 h-5" });
			$[2] = t1$1;
		} else t1$1 = $[2];
		return t1$1;
	} else if (variant === "event") {
		let t1$1;
		if ($[3] === Symbol.for("react.memo_cache_sentinel")) {
			t1$1 = /* @__PURE__ */ jsx(Waypoints, { className: "w-5 h-5" });
			$[3] = t1$1;
		} else t1$1 = $[3];
		return t1$1;
	}
	let t1;
	if ($[4] !== variant) {
		t1 = cn(baseIcon({ variant }));
		$[4] = variant;
		$[5] = t1;
	} else t1 = $[5];
	let t2;
	if ($[6] !== t1) {
		t2 = /* @__PURE__ */ jsx("div", {
			className: t1,
			children: "Icon"
		});
		$[6] = t1;
		$[7] = t2;
	} else t2 = $[7];
	return t2;
};
const NodeHeader = (t0) => {
	const $ = c(18);
	const { text, variant, children, className } = t0;
	let t1;
	if ($[0] !== className) {
		t1 = cn("flex items-center gap-2 p-2", className);
		$[0] = className;
		$[1] = t1;
	} else t1 = $[1];
	let t2;
	if ($[2] !== variant) {
		t2 = baseIcon({ variant });
		$[2] = variant;
		$[3] = t2;
	} else t2 = $[3];
	let t3;
	if ($[4] !== variant) {
		t3 = /* @__PURE__ */ jsx(NodeIcon, { variant });
		$[4] = variant;
		$[5] = t3;
	} else t3 = $[5];
	let t4;
	if ($[6] !== t2 || $[7] !== t3) {
		t4 = /* @__PURE__ */ jsx("div", {
			className: t2,
			children: t3
		});
		$[6] = t2;
		$[7] = t3;
		$[8] = t4;
	} else t4 = $[8];
	let t5;
	if ($[9] !== text) {
		t5 = /* @__PURE__ */ jsx("div", {
			className: "flex flex-col",
			children: /* @__PURE__ */ jsx("div", {
				className: "text-sm font-semibold leading-[1.25] tracking-[-0.25px]",
				children: text
			})
		});
		$[9] = text;
		$[10] = t5;
	} else t5 = $[10];
	let t6;
	if ($[11] !== children || $[12] !== t5) {
		t6 = /* @__PURE__ */ jsxs("div", {
			className: "flex flex-1 justify-between items-start gap-4",
			children: [t5, children]
		});
		$[11] = children;
		$[12] = t5;
		$[13] = t6;
	} else t6 = $[13];
	let t7;
	if ($[14] !== t1 || $[15] !== t4 || $[16] !== t6) {
		t7 = /* @__PURE__ */ jsxs("div", {
			className: t1,
			children: [t4, t6]
		});
		$[14] = t1;
		$[15] = t4;
		$[16] = t6;
		$[17] = t7;
	} else t7 = $[17];
	return t7;
};

//#endregion
//#region src/publicComponents/base-node/feature-card.tsx
const FeatureCard = (t0) => {
	const $ = c(16);
	const { feature, highlighted, onClick, onHover } = t0;
	const t1 = feature.id;
	const t2 = highlighted && "border-2 border-accent-1000 bg-accent-100";
	let t3;
	if ($[0] !== t2) {
		t3 = cn("p-4 rounded-lg bg-card shadow-sm cursor-pointer hover:bg-card/50 border-2 border-transparent", t2);
		$[0] = t2;
		$[1] = t3;
	} else t3 = $[1];
	let t4;
	if ($[2] !== feature.title) {
		t4 = /* @__PURE__ */ jsx("div", {
			className: "text-md font-semibold text-foreground leading-tight whitespace-nowrap mb-2",
			children: feature.title
		});
		$[2] = feature.title;
		$[3] = t4;
	} else t4 = $[3];
	let t5;
	if ($[4] !== feature.description) {
		t5 = /* @__PURE__ */ jsx("div", {
			className: "text-sm font-medium text-muted-foreground leading-tight",
			children: feature.description
		});
		$[4] = feature.description;
		$[5] = t5;
	} else t5 = $[5];
	let t6;
	if ($[6] !== feature.link) {
		t6 = feature.link && /* @__PURE__ */ jsx("div", {
			className: "text-sm font-medium text-muted-foreground leading-tight",
			children: /* @__PURE__ */ jsx("a", {
				href: feature.link,
				children: "Learn more"
			})
		});
		$[6] = feature.link;
		$[7] = t6;
	} else t6 = $[7];
	let t7;
	if ($[8] !== feature.id || $[9] !== onClick || $[10] !== onHover || $[11] !== t3 || $[12] !== t4 || $[13] !== t5 || $[14] !== t6) {
		t7 = /* @__PURE__ */ jsxs("div", {
			"data-feature-id": t1,
			className: t3,
			onClick,
			onMouseEnter: onHover,
			children: [
				t4,
				t5,
				t6
			]
		});
		$[8] = feature.id;
		$[9] = onClick;
		$[10] = onHover;
		$[11] = t3;
		$[12] = t4;
		$[13] = t5;
		$[14] = t6;
		$[15] = t7;
	} else t7 = $[15];
	return t7;
};

//#endregion
//#region src/publicComponents/base-node/language-indicator.tsx
const LanguageIndicator = (t0) => {
	const $ = c(13);
	const { language, className, size: t1, showLabel: t2 } = t0;
	const size = t1 === void 0 ? 18 : t1;
	const showLabel = t2 === void 0 ? false : t2;
	let t3;
	if ($[0] !== className || $[1] !== showLabel || $[2] !== size) {
		t3 = (language_0) => {
			if (language_0 === "typescript") return /* @__PURE__ */ jsxs(Fragment, { children: [showLabel ? /* @__PURE__ */ jsx(Fragment, { children: "TypeScript" }) : null, /* @__PURE__ */ jsx("svg", {
				xmlns: "http://www.w3.org/2000/svg",
				width: size,
				height: size,
				viewBox: "0 0 24 24",
				fill: "none",
				children: /* @__PURE__ */ jsx("path", {
					d: "M3 3V21H21V3H3ZM9 12H13.5V13.5H12V19.5H10.5V13.5H9V12ZM16.5 12H19.5V13.5H16.5V15H18C18.3975 15.0012 18.7783 15.1596 19.0593 15.4407C19.3404 15.7217 19.4988 16.1025 19.5 16.5V18C19.4988 18.3975 19.3404 18.7783 19.0593 19.0593C18.7783 19.3404 18.3975 19.4988 18 19.5H15V18H18V16.5H16.5C16.1025 16.4988 15.7217 16.3404 15.4407 16.0593C15.1596 15.7783 15.0012 15.3975 15 15V13.5C15.0012 13.1025 15.1596 12.7217 15.4407 12.4407C15.7217 12.1596 16.1025 12.0012 16.5 12Z",
					fill: "#0288D1"
				})
			})] });
			if (language_0 === "javascript") return /* @__PURE__ */ jsxs(Fragment, { children: [showLabel ? /* @__PURE__ */ jsx(Fragment, { children: "JavaScript" }) : null, /* @__PURE__ */ jsxs("svg", {
				width: size,
				height: size,
				viewBox: "0 0 20 20",
				fill: "none",
				xmlns: "http://www.w3.org/2000/svg",
				className,
				children: [/* @__PURE__ */ jsx("g", {
					clipPath: "url(#clip0_42_2938)",
					children: /* @__PURE__ */ jsx("path", {
						d: "M0 0H20V20H0V0ZM18.3617 15.23C18.2158 14.3175 17.6217 13.5508 15.8592 12.8358C15.2458 12.5483 14.5642 12.3483 14.3617 11.8858C14.2858 11.6108 14.2742 11.4608 14.3233 11.2983C14.4483 10.76 15.0858 10.5983 15.5858 10.7483C15.9108 10.8483 16.2108 11.0983 16.3992 11.4983C17.2608 10.935 17.2608 10.935 17.8617 10.5608C17.6367 10.2108 17.525 10.06 17.3733 9.91083C16.8483 9.32333 16.1492 9.02333 15.0117 9.04917L14.4242 9.12333C13.8608 9.26083 13.3242 9.56083 12.9992 9.96083C12.0492 11.0367 12.3233 12.9117 13.4733 13.6867C14.6108 14.5367 16.2742 14.7233 16.4867 15.5242C16.6867 16.4992 15.7617 16.8117 14.8483 16.6992C14.1725 16.5492 13.7983 16.2108 13.3858 15.5858L11.8608 16.4617C12.0358 16.8617 12.2358 17.0358 12.5358 17.3858C13.9858 18.8492 17.6108 18.7742 18.2617 16.5492C18.2858 16.4742 18.4617 15.9617 18.3233 15.1742L18.3617 15.23ZM10.8758 9.1925H9.0025C9.0025 10.8075 8.995 12.4125 8.995 14.03C8.995 15.0567 9.0475 15.9992 8.88 16.2892C8.605 16.8633 7.89667 16.79 7.575 16.6892C7.245 16.5258 7.0775 16.3008 6.88333 15.9767C6.83083 15.8892 6.79167 15.8133 6.7775 15.8133L5.25667 16.7508C5.51083 17.2758 5.88167 17.7275 6.36 18.015C7.0725 18.44 8.03 18.5775 9.0325 18.3525C9.685 18.1642 10.2475 17.7767 10.5417 17.1767C10.9667 16.4017 10.8767 15.4517 10.8725 14.3883C10.8825 12.6767 10.8725 10.9642 10.8725 9.23917L10.8758 9.1925Z",
						fill: "white"
					})
				}), /* @__PURE__ */ jsx("defs", { children: /* @__PURE__ */ jsx("clipPath", {
					id: "clip0_42_2938",
					children: /* @__PURE__ */ jsx("rect", {
						width: "20",
						height: "20",
						fill: "white"
					})
				}) })]
			})] });
			if (language_0 === "python") return /* @__PURE__ */ jsxs(Fragment, { children: [showLabel ? /* @__PURE__ */ jsx(Fragment, { children: "Python" }) : null, /* @__PURE__ */ jsxs("svg", {
				xmlns: "http://www.w3.org/2000/svg",
				width: size,
				height: size,
				viewBox: "0 0 24 24",
				fill: "none",
				children: [/* @__PURE__ */ jsx("path", {
					d: "M9.86 2C9.10148 2 8.37403 2.30132 7.83767 2.83767C7.30132 3.37403 7 4.10148 7 4.86V6.54H11.29C11.68 6.54 12 7.11 12 7.5H4.86C4.10148 7.5 3.37403 7.80132 2.83767 8.33767C2.30132 8.87403 2 9.60148 2 10.36V14.141C2 14.8995 2.30132 15.627 2.83767 16.1633C3.37403 16.6997 4.10148 17.001 4.86 17.001H6.04V14.321C6.03868 13.9459 6.11143 13.5742 6.25407 13.2273C6.3967 12.8803 6.60642 12.565 6.8712 12.2993C7.13598 12.0336 7.45061 11.8227 7.79704 11.6789C8.14347 11.535 8.51489 11.461 8.89 11.461H14.14C15.72 11.461 17 10.19 17 8.61V4.86C17 4.10148 16.6987 3.37403 16.1623 2.83767C15.626 2.30132 14.8985 2 14.14 2H9.86ZM9.14 3.61C9.54 3.61 9.86 3.73 9.86 4.32C9.86 4.91 9.54 5.211 9.14 5.211C8.75 5.211 8.43 4.911 8.43 4.321C8.43 3.731 8.75 3.61 9.14 3.61Z",
					fill: "#0288D1"
				}), /* @__PURE__ */ jsx("path", {
					d: "M17.959 7V9.68C17.9602 10.055 17.8873 10.4266 17.7446 10.7734C17.602 11.1202 17.3922 11.4355 17.1274 11.7011C16.8627 11.9667 16.5481 12.1774 16.2017 12.3212C15.8554 12.465 15.484 12.539 15.109 12.539H9.86C9.48489 12.5377 9.11321 12.6104 8.76628 12.7531C8.41935 12.8957 8.10399 13.1054 7.83828 13.3702C7.57257 13.635 7.36175 13.9496 7.2179 14.296C7.07405 14.6425 7 15.0139 7 15.389V19.139C7 19.8975 7.30132 20.625 7.83767 21.1613C8.37403 21.6977 9.10148 21.999 9.86 21.999H14.14C14.8983 21.999 15.6256 21.6978 16.162 21.1617C16.6983 20.6255 16.9997 19.8983 17 19.14V17.46H12.709C12.319 17.46 12 16.89 12 16.5H19.14C19.8985 16.5 20.626 16.1987 21.1623 15.6623C21.6987 15.126 22 14.3985 22 13.64V9.86C22 9.10148 21.6987 8.37403 21.1623 7.83767C20.626 7.30132 19.8985 7 19.14 7H17.959ZM8.32 11.513L8.316 11.517L8.354 11.513H8.32ZM14.86 18.789C15.25 18.789 15.57 19.089 15.57 19.679C15.57 19.7722 15.5516 19.8646 15.516 19.9507C15.4803 20.0368 15.428 20.1151 15.362 20.181C15.2961 20.247 15.2178 20.2993 15.1317 20.335C15.0456 20.3706 14.9532 20.389 14.86 20.389C14.46 20.389 14.14 20.269 14.14 19.679C14.14 19.089 14.46 18.789 14.86 18.789Z",
					fill: "#FDD835"
				})]
			})] });
			if (language_0 === "go") return /* @__PURE__ */ jsxs(Fragment, { children: [showLabel ? /* @__PURE__ */ jsx(Fragment, { children: "Go" }) : null, /* @__PURE__ */ jsx("svg", {
				viewBox: "0 0 128 128",
				width: size,
				height: size,
				className,
				children: /* @__PURE__ */ jsx("path", { d: "M108.7 64.4c-1.6-1.2-3.8-1.9-6-1.9-.1-3.6-.2-7.2-.2-10.7-.1-9.5-.1-19.3-2.6-28.6 6.5-2.5 6.2-7.6 5.9-9.2-.7-3.8-4-7.9-9-7.9-1.9 0-3.9.6-5.8 1.9C83.4 1.6 73 .5 62.5.5 53 1.3 46.4 2.8 41 5.4c-2.3 1.1-4.2 2.4-6 3.9-1.8-1.1-3.7-1.7-5.5-1.7-5.1 0-8.8 4.1-9.4 8.1-.6 4 1.6 7.3 6 8.9-2.2 8.9-1.5 18.1-.7 27 .3 3.7.6 7.6.7 11.4h-.5c-2.3 0-4.6.7-6.3 1.9-.7.5-2.1 2-2 3.3 0 .4.2.9.9 1.3.2 2.1 1.5 2.4 2.3 2.4 1.6 0 3.4-1.1 3.9-1.5.5-.3.9-.5 1.2-.6l.5-.2c-.1 2.2-.3 4.5-.5 7-.9 11.7-2 25 3.9 34.6 1.2 1.9 2.6 3.6 4.3 5.1l-.9.6c-2.7 1.9-6.7 4.7-4.3 8.5v.1c.4.4.6.5.9.5h.1c.3.5.9 1.6 2.4 1.6.6 0 1.3-.2 2-.6 1.1-.6 2.1-1.4 3-2.2 1.5-1.3 3-2.6 5-2.9l.6-.1.7.3c6.4 2.6 12.9 3.3 17.3 3.4h2.6c7.7 0 14.8-1.2 20.5-3.5 1.2-.5 2.4-1.1 3.5-1.7 1.2.5 2.3 1.7 3.4 3 1.4 1.6 2.8 3.2 4.5 3.2.8 0 1.6-.4 2.4-1.1 1.2-.7 1.8-1.7 1.8-2.9 0-2.6-2.9-5.7-5.2-7.2 2-2 3.6-4.2 4.9-6.7 5.9-11.8 5-26.2 4.2-39l.5.3c.5.4 2.3 1.5 3.9 1.5.8 0 2.1-.3 2.3-2.4.7-.4.8-.9.9-1.3 0-1.3-1.4-2.8-2.1-3.3zm-17.1-56c1.7-1.1 3.4-1.7 5.2-1.7 4.4 0 7.7 3.7 8.3 7.3.6 3.7-1.4 6.8-5.4 8.4 0-.2-.1-.3-.2-.5l-.6-2.1-.3-.8-.9-2.4c0-.1 0-.1-.1-.2 0 0 0-.1-.1-.1-.4-.7-.7-1.4-1.1-2l-.2-.3-1.2-1.8c-.1-.1-.2-.2-.3-.4-.4-.5-.8-1-1.3-1.5-.1-.2-.3-.3-.5-.5l-1.2-1.2c.1.1 0-.1-.1-.2-.2zM26.3 23.9c-4-1.5-6.1-4.5-5.5-8.1.6-3.7 4-7.5 8.7-7.5 1.7 0 3.3.5 4.9 1.4l-.1.1-.3.3c-.5.5-1 1-1.4 1.5l-.3.3c-.5.6-1 1.2-1.5 1.9l-.2.3c-.5.7-.9 1.3-1.3 2 0 0 0 .1-.1.1-.4.7-.8 1.5-1.1 2.3 0 .2-.1.4-.1.5-.3.8-.7 1.6-.9 2.4 0 .1 0 .1-.1.2l-.6 2-.1.3zm-.9 45.2c-.3.1-.8.3-1.4.7-1 .7-2.4 1.4-3.5 1.4-.6 0-1.4-.2-1.6-1.7.5-.2 1-.4 1.4-.9.1-.2.1-.4-.1-.5-.2-.1-.4-.1-.5.1-.3.3-.6.5-1.1.6h-.1c-.3-.2-.5-.4-.6-.7.1-.9 1.1-2.1 1.8-2.7 1.6-1.2 3.7-1.8 5.9-1.8h.6v2.9l-.1 1c-.1 1.4-.2 1.4-.7 1.6zm15.8 52l.1.1c-1.9.5-3.4 1.7-4.8 2.9-.9.8-1.8 1.6-2.9 2.1-.6.4-1.1.5-1.6.5-1 0-1.5-.8-1.7-1.2.1-.8.5-1.5 1-2.2.3-.4.6-.9.8-1.4.1-.2 0-.4-.2-.5-.2-.1-.4 0-.5.2-.2.4-.5.8-.7 1.3-.4.7-.9 1.4-1.1 2.2h-.1c-.1 0-.2-.1-.4-.3-2-3.2 1.4-5.5 4.2-7.5l.9-.6h.2l.2.1 1.2.9.4.3c.4.3.8.6 1.3.9l.5.3 1.3.8.6.3c.4.3.8.5 1.3.8zm52.3-5.3c2.1 1.4 5 4.4 5 6.7 0 .8-.4 1.5-1.1 2-.1-.3-.2-.6-.4-.9-.3-.9-.7-1.9-1.5-2.6-.1-.1-.4-.1-.5 0-.1.1-.1.4 0 .5.7.6 1 1.5 1.3 2.4l.4 1.1c-.6.5-1.2.8-1.8.8-1.4 0-2.6-1.5-3.9-3-1-1.1-2-2.3-3.1-3l.3-.2c.3-.1.5-.3.7-.5l1.3-.9.8-.6 1.4-1.1c.2-.2.4-.4.7-.6.2.1.3 0 .4-.1zm4.7-7.5c-1.2 2.5-3.4 5.7-7.1 8.5l-.2.1c-.3.3-.7.5-1.1.8l-1.1.7-.2.1-1.6.9-.7.4c-5.4 2.8-12.8 4.8-23.2 4.8h-2.6c-8.1-.3-14.3-2-18.9-4.2h-.1c-1.1-.5-2.1-1.1-3.1-1.7-4.6-2.9-7.1-6.1-8.3-8-5.8-9.4-4.8-22.5-3.8-34.1.2-3 .5-5.7.6-8.4l.1-1.8.1-.1v-.1l-.1-.5V63c-.1-4-.4-7.7-.7-11.5-.8-9-1.5-18.4.9-27.4.2-.8.4-1.6.7-2.3v-.1l.1-.3c.1-.4.3-.8.5-1.2.1-.3.2-.5.3-.7l.3-.8.4-.8.3-.6c.1-.3.3-.6.4-.8.1-.2.2-.4.3-.5.2-.4.5-.8.7-1.1.2-.3.4-.6.6-.8.1-.2.2-.3.4-.5l.6-.8c.1-.1.2-.2.3-.4l.9-1 .1-.1 1.2-1.2.2-.2c7-6 17-7.8 27.1-8.6 5.9 0 13.5.3 20 2.7 3.2 1.2 6 2.8 8.4 4.9.3.3.6.6.9.8l.9.9 1 1.2.6.7 1 1.4.5.7.9 1.5.4.8c.3.6.5 1.3.7 2 0 .1.1.2.1.4.2.6.4 1.2.6 1.7l.1.3c3 9.9 3.1 20.2 3.1 30.4 0 3.5 0 7.4.2 10.4v.6c0 .9.1 1.8.1 2.5v.1l.1 1.9c0 .6.1 1.3.1 2v.3c.9 12.6 1.9 27.1-4 38.8zm11.2-40c-.5-.2-.9-.3-1.2-.7-.1-.2-.3-.2-.5-.1s-.2.3-.1.5c.4.5.9.7 1.4.9-.2 1.5-1 1.7-1.6 1.7-1.1 0-2.5-.7-3.5-1.4-.4-.3-.7-.4-1-.5l-.2-2.4c0-.4 0-.8-.1-1.1 0-.6-.1-1.3-.1-2 2.1.1 4.1.7 5.6 1.8.7.5 1.8 1.8 1.7 2.6.1.3-.1.6-.4.7zM77.7 7.2c-3.8 0-7.5 1.6-9.9 4.1-2.2 2.4-3.3 5.4-3.1 8.8v.1C66 27.4 72 30 76.8 30c4.2 0 8.1-1.8 10.4-4.7 2-2.5 2.7-5.8 1.8-9.3-1.5-6.4-6.8-8.8-11.3-8.8zm8.9 17.5c-2.1 2.7-5.9 4.3-9.9 4.3-2.7 0-5.2-.7-7.2-2.1-2.2-1.6-3.6-3.9-4.1-6.9-.2-3.2.8-6 2.9-8.2 2.3-2.5 5.8-3.9 9.4-3.9 4.2 0 9.2 2.1 10.6 8.1.8 3.4.2 6.4-1.7 8.7zm-27-5.5c-.4-7.3-6.2-10.6-11.8-10.6-4.1 0-7.9 1.7-10.3 4.5-2.1 2.6-2.8 5.9-2.1 9.6 1.6 6.3 7 8.5 11.5 8.5 3.8 0 7.4-1.5 9.8-4 2.1-2.2 3.1-5 2.9-8zM47 30.5c-4.2 0-9.3-2.1-10.8-8-.7-3.5 0-6.6 2-9 2.2-2.7 5.8-4.3 9.7-4.3 7.7 0 10.8 5.4 11.1 10 .2 2.9-.7 5.5-2.6 7.5-2.3 2.4-5.8 3.8-9.4 3.8zm-5.4-13.9c-1.9 0-3.5 1.7-3.5 3.8 0 2.1 1.6 3.8 3.5 3.8s3.5-1.7 3.5-3.8c0-2.1-1.6-3.8-3.5-3.8zm1.6 5.7c-.5 0-.8-.4-.8-1 0-.5.4-1 .8-1 .5 0 .8.4.8 1 0 .5-.4 1-.8 1zm27.9-6.6c-1.9 0-3.5 1.7-3.5 3.8 0 2.1 1.6 3.8 3.5 3.8s3.5-1.7 3.5-3.8c0-2.1-1.5-3.8-3.5-3.8zm1.6 5.6c-.5 0-.8-.4-.8-1 0-.5.4-1 .8-1 .5 0 .8.4.8 1s-.3 1-.8 1zm-4.2 9c.1-.3.1-.7 0-1.1-1.1-4.1-10.4-3.5-10.4 1.3 0-.4-.1-.6-.1-.8 0 .3.1.6.1.8-1.5.2-2.8 1.2-3.6 2.7-.7 1.4-.8 2.9-.1 4.2.6 1.1 1.6 1.7 2.8 1.7-.4 2.1 0 4.6 1 6 .5.7 1.2 1.1 1.8 1.1.9 0 1.9-.7 2.8-1.8.6 1 1.5 1.7 2.7 1.9h.2c.2-.1.4-.2.6-.2.7-.2 1.4-.5 1.9-1.3v-.1c.3-1.4.1-2.9 0-4.3l-.1-1.2c.5.2 1 .3 1.5.3 1 0 1.8-.4 2.5-1.2.9-1 1.2-2 1-3-.1-2.3-2.4-4-4.6-5zm-5.9 13.2c-.9 1.1-1.7 1.7-2.4 1.7-.5 0-.9-.3-1.3-.7-.9-1.2-1.3-3.7-.9-5.6l.5-.1h.2l.6-.1h.1l1.2-.4.6-.2c.1 0 .1 0 .2-.1.2-.1.3-.1.5-.1.2-.1.4-.1.7-.2h.2l-.1 1.3c-.1 1.3-.2 2.9-.1 4.5zm5.2-3.3c.1 1.3.2 2.7 0 4-.3.6-.8.7-1.5 1l-.6.2c-1-.2-1.8-.8-2.2-1.8-.2-1.5-.1-3 .1-4.6l.1-1.5h.1c1.1 0 2.1.5 3.1.9l.7.3c.1.4.2.9.2 1.5zm4.1-2.7c-.6.7-1.3 1-2 1-.6 0-1.1-.2-1.7-.4-.3-.1-.6-.2-.9-.4-1.1-.4-2.2-.9-3.4-.9h-1.3l-.4.1c-.1 0-.2 0-.3.1-.2 0-.3.1-.5.2-.1 0-.2 0-.2.1l-.8.3c-.1.1-.3.1-.4.1l-.3.1-.3.1c-.2.1-.3.1-.5.1-.1 0-.2 0-.3.1l-.5.1h-.2l-.6.1c-1 0-1.7-.4-2.2-1.3-.6-1-.6-2.3.1-3.5.7-1.2 1.9-2.1 3.1-2.3 1.2 2.5 8.1 2.2 9.9-.2 2 .9 4.2 2.5 4.5 4.2.2.7-.1 1.5-.8 2.3zM58 29.4z" })
			})] });
			if (language_0 === "ruby") return /* @__PURE__ */ jsxs(Fragment, { children: [showLabel ? /* @__PURE__ */ jsx(Fragment, { children: "Ruby" }) : null, /* @__PURE__ */ jsxs("svg", {
				width: size,
				height: size,
				viewBox: "0 0 20 20",
				fill: "none",
				xmlns: "http://www.w3.org/2000/svg",
				className,
				children: [/* @__PURE__ */ jsx("g", {
					clipPath: "url(#clip0_42_2969)",
					children: /* @__PURE__ */ jsx("path", {
						d: "M16.7967 0.0691243C19.3242 0.506624 20.0408 2.23412 19.9875 4.04412L20 4.01829L18.8625 18.925L4.075 19.9383H4.08833C2.86083 19.8866 0.125 19.7741 0 15.9491L1.37083 13.4491L3.72 18.9375L4.13917 19.9141L6.47667 12.2941L6.45167 12.3L6.465 12.275L14.1775 14.7383L13.0142 10.2125L12.1892 6.96246L19.5392 6.48829L19.0267 6.06329L13.75 1.76162L16.7992 0.060791L16.7967 0.0691243ZM4.275 4.22746C7.2425 1.28329 11.0725 -0.456709 12.5433 1.02746C14.0117 2.50829 12.4558 6.11496 9.4825 9.05746C6.51333 12.0008 2.73 13.8358 1.2625 12.355C-0.209167 10.8741 1.3 7.17412 4.2725 4.22996L4.275 4.22746Z",
						fill: "white"
					})
				}), /* @__PURE__ */ jsx("defs", { children: /* @__PURE__ */ jsx("clipPath", {
					id: "clip0_42_2969",
					children: /* @__PURE__ */ jsx("rect", {
						width: "20",
						height: "20",
						fill: "white"
					})
				}) })]
			})] });
			if (language_0 === "php") return /* @__PURE__ */ jsxs(Fragment, { children: [showLabel ? /* @__PURE__ */ jsx(Fragment, { children: "PHP" }) : null, /* @__PURE__ */ jsx("svg", {
				viewBox: "0 0 128 128",
				width: size,
				height: size,
				className,
				children: /* @__PURE__ */ jsx("path", { d: "M64 30.332C28.654 30.332 0 45.407 0 64s28.654 33.668 64 33.668c35.345 0 64-15.075 64-33.668S99.346 30.332 64 30.332zm-5.982 9.81h7.293v.003l-1.745 8.968h6.496c4.087 0 6.908.714 8.458 2.139 1.553 1.427 2.017 2.017 3.737 3.737 5.946 1.719 2.209-2.209 2.209-5.928 0-8.137zm-51.051 51.372c0-3.757-1.072-5.686-3.214-5.791a6.03 6.03 0 00-2.495.409c-.646.231-1.082.461-1.311.692v8.968c1.371.86 2.588 1.26 3.649 1.197 2.247-.148 3.371-1.971 3.371-5.475zm2.643.157c0 1.909-.447 3.493-1.348 4.753-1.003 1.427-2.394 2.16-4.172 2.201-1.34.043-2.721-.378-4.142-1.258v8.151l-2.298-.82V107.14c.377-.462.862-.859 1.451-1.196 1.368-.798 3.031-1.207 4.987-1.228l.033.032c1.788-.022 3.166.712 4.134 2.201.902 1.366 1.355 3.117 1.355 5.257zm14.049 5.349c0 2.56-.257 4.333-.77 5.318-.516.986-1.497 1.773-2.945 2.359-1.174.463-2.444.714-3.808.757l-.38-1.448c1.386-.188 2.362-.378 2.928-.566 1.114-.377 1.878-.955 2.298-1.73.337-.631.503-1.835.503-3.618v-.599a11.809 11.809 0 01-4.941 1.068c-1.132 0-2.13-.354-2.99-1.068-.966-.777-1.449-1.764-1.449-2.958v-9.566l2.299-.787v9.63c0 1.028.332 1.82.996 2.376s1.524.822 2.578.803c1.054-.022 2.183-.431 3.382-1.228v-11.234h2.299v12.491zm8.973 1.479a9.457 9.457 0 01-.757.032c-1.3 0-2.314-.309-3.038-.93-.722-.622-1.084-1.479-1.084-2.573v-9.054h-1.574v-1.446h1.574v-3.84l2.296-.817v4.657h2.583v1.446h-2.583v8.991c0 .862.231 1.474.694 1.83.397.295 1.029.463 1.889.506v1.198zm13.917-.189h-2.298v-8.873c0-.902-.211-1.68-.631-2.329-.485-.734-1.159-1.102-2.024-1.102-1.054 0-2.372.556-3.954 1.668v10.636h-2.298V97.637l2.298-.725v9.659c1.469-1.068 3.073-1.604 4.816-1.604 1.218 0 2.203.41 2.958 1.228.757.817 1.134 1.836 1.134 3.053v9.597h-.001zm12.218-7.157c0-1.444-.274-2.636-.82-3.579-.649-1.149-1.657-1.756-3.021-1.818-2.52.146-3.778 1.951-3.778 5.412 0 1.587.262 2.912.79 3.976.674 1.356 1.685 2.024 3.033 2.002 2.531-.02 3.796-2.017 3.796-5.993zm2.518.015c0 2.055-.526 3.765-1.575 5.131-1.154 1.528-2.749 2.296-4.783 2.296-2.017 0-3.589-.768-4.723-2.296-1.028-1.366-1.542-3.076-1.542-5.131 0-1.932.556-3.556 1.668-4.879 1.174-1.403 2.718-2.107 4.627-2.107 1.909 0 3.463.704 4.66 2.107 1.111 1.323 1.668 2.947 1.668 4.879zm13.178 7.142h-2.299v-9.376c0-1.028-.31-1.831-.928-2.409-.619-.576-1.443-.855-2.472-.833-1.091.021-2.13.378-3.116 1.069v11.549h-2.299v-11.833c1.323-.963 2.54-1.592 3.652-1.886 1.049-.274 1.974-.41 2.771-.41.545 0 1.059.053 1.542.158.903.209 1.637.596 2.203 1.164.631.629.946 1.384.946 2.267v10.54z" })
			})] });
		};
		$[0] = className;
		$[1] = showLabel;
		$[2] = size;
		$[3] = t3;
	} else t3 = $[3];
	const renderIcon = t3;
	const t4 = !showLabel;
	let t5;
	if ($[4] !== showLabel || $[5] !== t4) {
		t5 = cn("text-muted-foreground", {
			"text-sm flex items-center justify-center flex-row gap-2": showLabel,
			"font-mono text-xs": t4
		});
		$[4] = showLabel;
		$[5] = t4;
		$[6] = t5;
	} else t5 = $[6];
	let t6;
	if ($[7] !== language || $[8] !== renderIcon) {
		t6 = renderIcon(language);
		$[7] = language;
		$[8] = renderIcon;
		$[9] = t6;
	} else t6 = $[9];
	let t7;
	if ($[10] !== t5 || $[11] !== t6) {
		t7 = /* @__PURE__ */ jsx("div", {
			className: t5,
			children: t6
		});
		$[10] = t5;
		$[11] = t6;
		$[12] = t7;
	} else t7 = $[12];
	return t7;
};

//#endregion
//#region src/publicComponents/base-node/code-display.tsx
const codeTagProps = { style: {
	fontFamily: "DM Mono, monospace",
	fontSize: "16px"
} };
const customStyle = {
	margin: 0,
	borderRadius: 0,
	padding: 0
};
const isHighlighted = (lines, lineNumber) => {
	return lines.some((line) => {
		const [start, end] = line.split("-").map((num) => parseInt(num, 10));
		if (end !== void 0) return lineNumber >= start && lineNumber <= end;
		return lineNumber == start;
	});
};
const getFirstLineNumber = (line) => {
	const [start] = line.split("-").map((num) => parseInt(num, 10));
	return start;
};
const CodeDisplay = (t0) => {
	const $ = c(21);
	const { code, language, features } = t0;
	const themeStyle = useThemeStore(_temp) === "dark" ? oneDark : oneLight;
	let t1;
	if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = [];
		$[0] = t1;
	} else t1 = $[0];
	const [highlightedLines, setHighlightedLines] = useState(t1);
	const [selectedFeature, setSelectedFeature] = useState(null);
	const ref = useRef(null);
	const handleFeatureClick = (feature) => {
		setSelectedFeature(feature);
		setHighlightedLines(feature.lines);
		const lineNumber = getFirstLineNumber(feature.lines[0]);
		const line = ref.current?.querySelector(`[data-line-number="${lineNumber}"]`);
		if (line) line.scrollIntoView({
			behavior: "smooth",
			block: "center"
		});
	};
	const t2 = "flex flex-col h-full overflow-hidden";
	let t3;
	let t4;
	if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
		t3 = /* @__PURE__ */ jsx("div", {
			className: "text-sm text-muted-foreground",
			children: "Read only"
		});
		t4 = /* @__PURE__ */ jsx("div", { className: "flex-1" });
		$[1] = t3;
		$[2] = t4;
	} else {
		t3 = $[1];
		t4 = $[2];
	}
	let t5;
	if ($[3] !== language) {
		t5 = /* @__PURE__ */ jsxs("div", {
			className: "flex items-center py-2 px-5 dark:bg-[#1e1e1e] gap-2 justify-center",
			children: [
				t3,
				t4,
				/* @__PURE__ */ jsx(LanguageIndicator, {
					language,
					className: "w-4 h-4",
					size: 16,
					showLabel: true
				})
			]
		});
		$[3] = language;
		$[4] = t5;
	} else t5 = $[4];
	const t6 = "flex flex-row h-[calc(100%-36px)]";
	const t7 = features && features.length > 0 && /* @__PURE__ */ jsx("div", {
		className: "flex flex-col gap-2 p-2 bg-card overflow-y-auto min-w-[200px] w-[300px]",
		children: features.map((feature_0, index) => /* @__PURE__ */ jsx(FeatureCard, {
			feature: feature_0,
			highlighted: selectedFeature === feature_0,
			onClick: () => handleFeatureClick(feature_0),
			onHover: () => handleFeatureClick(feature_0)
		}, index))
	});
	let t8;
	if ($[5] !== highlightedLines) {
		t8 = (lineNumber_0) => {
			if (isHighlighted(highlightedLines, lineNumber_0)) return {
				"data-line-number": lineNumber_0,
				style: {
					borderLeft: "2px solid var(--accent-1000)",
					backgroundColor: "rgb(from var(--accent-1000) r g b / 0.2)"
				}
			};
			return {
				"data-line-number": lineNumber_0,
				style: { borderLeft: "2px solid transparent" }
			};
		};
		$[5] = highlightedLines;
		$[6] = t8;
	} else t8 = $[6];
	let t9;
	if ($[7] !== code || $[8] !== language || $[9] !== t8 || $[10] !== themeStyle) {
		t9 = /* @__PURE__ */ jsx(Prism, {
			showLineNumbers: true,
			language,
			style: themeStyle,
			codeTagProps,
			customStyle,
			wrapLines: true,
			lineProps: t8,
			children: code
		});
		$[7] = code;
		$[8] = language;
		$[9] = t8;
		$[10] = themeStyle;
		$[11] = t9;
	} else t9 = $[11];
	let t10;
	if ($[12] !== ref || $[13] !== t9) {
		t10 = /* @__PURE__ */ jsx("div", {
			className: "overflow-y-auto",
			ref,
			children: t9
		});
		$[12] = ref;
		$[13] = t9;
		$[14] = t10;
	} else t10 = $[14];
	let t11;
	if ($[15] !== t10 || $[16] !== t7) {
		t11 = /* @__PURE__ */ jsxs("div", {
			className: t6,
			children: [t7, t10]
		});
		$[15] = t10;
		$[16] = t7;
		$[17] = t11;
	} else t11 = $[17];
	let t12;
	if ($[18] !== t11 || $[19] !== t5) {
		t12 = /* @__PURE__ */ jsxs("div", {
			className: t2,
			children: [t5, t11]
		});
		$[18] = t11;
		$[19] = t5;
		$[20] = t12;
	} else t12 = $[20];
	return t12;
};
function _temp(state) {
	return state.theme;
}

//#endregion
//#region src/publicComponents/base-node/node-sidebar.tsx
const NodeSidebar = (t0) => {
	const $ = c(13);
	const { content, title, subtitle, language, isOpen, onClose, features } = t0;
	if (!isOpen) return null;
	let t1;
	if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = /* @__PURE__ */ jsx(X, {});
		$[0] = t1;
	} else t1 = $[0];
	let t2;
	if ($[1] !== onClose) {
		t2 = [{
			icon: t1,
			onClick: onClose,
			label: "Close"
		}];
		$[1] = onClose;
		$[2] = t2;
	} else t2 = $[2];
	let t3;
	if ($[3] !== content || $[4] !== features || $[5] !== language) {
		t3 = /* @__PURE__ */ jsx(CodeDisplay, {
			code: content,
			language,
			features
		});
		$[3] = content;
		$[4] = features;
		$[5] = language;
		$[6] = t3;
	} else t3 = $[6];
	let t4;
	if ($[7] !== onClose || $[8] !== subtitle || $[9] !== t2 || $[10] !== t3 || $[11] !== title) {
		t4 = /* @__PURE__ */ jsx(Sidebar, {
			title,
			subtitle,
			initialWidth: 820,
			contentClassName: "p-0 h-full gap-0",
			onClose,
			actions: t2,
			children: t3
		});
		$[7] = onClose;
		$[8] = subtitle;
		$[9] = t2;
		$[10] = t3;
		$[11] = title;
		$[12] = t4;
	} else t4 = $[12];
	return t4;
};

//#endregion
//#region src/publicComponents/base-node/base-node.tsx
const BaseNode = (t0) => {
	const $ = c(56);
	const { title, variant, children, disableSourceHandle, disableTargetHandle, language, subtitle, data } = t0;
	const [isOpen, setIsOpen] = useState(false);
	const { sourcePosition, targetPosition, toggleTargetPosition, toggleSourcePosition } = useHandlePositions(data);
	const [content, setContent] = useState(null);
	let t1;
	if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
		t1 = [];
		$[0] = t1;
	} else t1 = $[0];
	const [features, setFeatures] = useState(t1);
	let t2;
	if ($[1] !== data.id) {
		t2 = async () => {
			const responseData = await (await fetch(`/__motia/step/${data.id}`)).json();
			setContent(responseData.content);
			setFeatures(responseData.features);
		};
		$[1] = data.id;
		$[2] = t2;
	} else t2 = $[2];
	const fetchContent = t2;
	let t3;
	let t4;
	if ($[3] !== data.id || $[4] !== fetchContent || $[5] !== isOpen) {
		t3 = () => {
			if (data.id && isOpen) fetchContent();
		};
		t4 = [
			data.id,
			isOpen,
			fetchContent
		];
		$[3] = data.id;
		$[4] = fetchContent;
		$[5] = isOpen;
		$[6] = t3;
		$[7] = t4;
	} else {
		t3 = $[6];
		t4 = $[7];
	}
	useEffect(t3, t4);
	let t5;
	if ($[8] !== isOpen) {
		t5 = cn("p-1 rounded-lg max-w-[350px]", { "bg-muted-foreground/20": isOpen });
		$[8] = isOpen;
		$[9] = t5;
	} else t5 = $[9];
	let t6;
	if ($[10] !== title) {
		t6 = title?.toLowerCase().replace(/ /g, "-");
		$[10] = title;
		$[11] = t6;
	} else t6 = $[11];
	const t7 = `node-${t6}`;
	let t8;
	if ($[12] !== title) {
		t8 = title?.toLowerCase();
		$[12] = title;
		$[13] = t8;
	} else t8 = $[13];
	const t9 = `open-code-preview-button-${t8}`;
	let t10;
	let t11;
	if ($[14] === Symbol.for("react.memo_cache_sentinel")) {
		t10 = () => setIsOpen(true);
		t11 = /* @__PURE__ */ jsx(ScanSearch, { className: "w-4 h-4" });
		$[14] = t10;
		$[15] = t11;
	} else {
		t10 = $[14];
		t11 = $[15];
	}
	let t12;
	if ($[16] !== t9) {
		t12 = /* @__PURE__ */ jsx("div", {
			className: "flex justify-end",
			children: /* @__PURE__ */ jsx(Button, {
				"data-testid": t9,
				variant: "ghost",
				className: "h-5 p-0.5",
				onClick: t10,
				children: t11
			})
		});
		$[16] = t9;
		$[17] = t12;
	} else t12 = $[17];
	let t13;
	if ($[18] !== t12 || $[19] !== title || $[20] !== variant) {
		t13 = /* @__PURE__ */ jsx(NodeHeader, {
			text: title,
			variant,
			className: "border-b-2 border-muted-foreground/10",
			children: t12
		});
		$[18] = t12;
		$[19] = title;
		$[20] = variant;
		$[21] = t13;
	} else t13 = $[21];
	let t14;
	if ($[22] !== subtitle) {
		t14 = subtitle && /* @__PURE__ */ jsx("div", {
			className: "py-4 px-6 text-sm text-muted-foreground",
			children: subtitle
		});
		$[22] = subtitle;
		$[23] = t14;
	} else t14 = $[23];
	let t15;
	if ($[24] !== children || $[25] !== variant) {
		t15 = children && /* @__PURE__ */ jsx("div", {
			className: "p-2",
			children: /* @__PURE__ */ jsx("div", {
				className: cn("space-y-3 p-4 text-sm text-muted-foreground", { "bg-card": variant !== "noop" }),
				children
			})
		});
		$[24] = children;
		$[25] = variant;
		$[26] = t15;
	} else t15 = $[26];
	let t16;
	if ($[27] !== disableTargetHandle || $[28] !== targetPosition || $[29] !== toggleTargetPosition) {
		t16 = !disableTargetHandle && /* @__PURE__ */ jsx(BaseHandle, {
			type: "target",
			position: targetPosition,
			onTogglePosition: toggleTargetPosition
		});
		$[27] = disableTargetHandle;
		$[28] = targetPosition;
		$[29] = toggleTargetPosition;
		$[30] = t16;
	} else t16 = $[30];
	let t17;
	if ($[31] !== disableSourceHandle || $[32] !== sourcePosition || $[33] !== toggleSourcePosition) {
		t17 = !disableSourceHandle && /* @__PURE__ */ jsx(BaseHandle, {
			type: "source",
			position: sourcePosition,
			onTogglePosition: toggleSourcePosition
		});
		$[31] = disableSourceHandle;
		$[32] = sourcePosition;
		$[33] = toggleSourcePosition;
		$[34] = t17;
	} else t17 = $[34];
	let t18;
	if ($[35] !== t13 || $[36] !== t14 || $[37] !== t15 || $[38] !== t16 || $[39] !== t17) {
		t18 = /* @__PURE__ */ jsxs("div", {
			className: "group relative",
			children: [
				t13,
				t14,
				t15,
				t16,
				t17
			]
		});
		$[35] = t13;
		$[36] = t14;
		$[37] = t15;
		$[38] = t16;
		$[39] = t17;
		$[40] = t18;
	} else t18 = $[40];
	let t19;
	if ($[41] !== t18 || $[42] !== t7) {
		t19 = /* @__PURE__ */ jsx("div", {
			className: "rounded-lg dark:bg-[#101010] bg-background border-1 border-muted-foreground/30 border-solid",
			"data-testid": t7,
			children: t18
		});
		$[41] = t18;
		$[42] = t7;
		$[43] = t19;
	} else t19 = $[43];
	let t20;
	if ($[44] !== content || $[45] !== features || $[46] !== isOpen || $[47] !== language || $[48] !== subtitle || $[49] !== title || $[50] !== variant) {
		t20 = content && /* @__PURE__ */ jsx(NodeSidebar, {
			features,
			content,
			title,
			subtitle,
			variant,
			language,
			isOpen,
			onClose: () => setIsOpen(false)
		});
		$[44] = content;
		$[45] = features;
		$[46] = isOpen;
		$[47] = language;
		$[48] = subtitle;
		$[49] = title;
		$[50] = variant;
		$[51] = t20;
	} else t20 = $[51];
	let t21;
	if ($[52] !== t19 || $[53] !== t20 || $[54] !== t5) {
		t21 = /* @__PURE__ */ jsxs("div", {
			className: t5,
			children: [t19, t20]
		});
		$[52] = t19;
		$[53] = t20;
		$[54] = t5;
		$[55] = t21;
	} else t21 = $[55];
	return t21;
};

//#endregion
//#region src/publicComponents/api-node.tsx
const ApiNode = (t0) => {
	const $ = c(8);
	const { data, children } = t0;
	const t1 = !data.emits?.length && !data.virtualEmits?.length;
	const t2 = !data.subscribes?.length && !data.virtualSubscribes?.length;
	let t3;
	if ($[0] !== data.webhookUrl) {
		t3 = data.webhookUrl && /* @__PURE__ */ jsx("div", {
			className: "flex gap-1 items-center text-muted-foreground",
			children: /* @__PURE__ */ jsx("div", {
				className: "font-mono",
				children: data.webhookUrl
			})
		});
		$[0] = data.webhookUrl;
		$[1] = t3;
	} else t3 = $[1];
	let t4;
	if ($[2] !== children || $[3] !== data || $[4] !== t1 || $[5] !== t2 || $[6] !== t3) {
		t4 = /* @__PURE__ */ jsxs(BaseNode, {
			data,
			variant: "api",
			title: data.name,
			language: data.language,
			subtitle: data.description,
			disableSourceHandle: t1,
			disableTargetHandle: t2,
			children: [t3, children]
		});
		$[2] = children;
		$[3] = data;
		$[4] = t1;
		$[5] = t2;
		$[6] = t3;
		$[7] = t4;
	} else t4 = $[7];
	return t4;
};

//#endregion
//#region src/publicComponents/event-node.tsx
const EventNode = (t0) => {
	const $ = c(5);
	const { data, children } = t0;
	const t1 = !data.emits?.length && !data.virtualEmits?.length;
	const t2 = !data.subscribes?.length && !data.virtualSubscribes?.length;
	let t3;
	if ($[0] !== children || $[1] !== data || $[2] !== t1 || $[3] !== t2) {
		t3 = /* @__PURE__ */ jsx(BaseNode, {
			data,
			variant: "event",
			title: data.name,
			subtitle: data.description,
			language: data.language,
			disableSourceHandle: t1,
			disableTargetHandle: t2,
			children
		});
		$[0] = children;
		$[1] = data;
		$[2] = t1;
		$[3] = t2;
		$[4] = t3;
	} else t3 = $[4];
	return t3;
};

//#endregion
//#region src/publicComponents/noop-node.tsx
const NoopNode = (t0) => {
	const $ = c(5);
	const { data, children } = t0;
	const t1 = !data.virtualEmits.length;
	const t2 = !data.subscribes?.length;
	let t3;
	if ($[0] !== children || $[1] !== data || $[2] !== t1 || $[3] !== t2) {
		t3 = /* @__PURE__ */ jsx(BaseNode, {
			data,
			variant: "noop",
			title: data.name,
			subtitle: data.description,
			disableSourceHandle: t1,
			disableTargetHandle: t2,
			children
		});
		$[0] = children;
		$[1] = data;
		$[2] = t1;
		$[3] = t2;
		$[4] = t3;
	} else t3 = $[4];
	return t3;
};

//#endregion
export { ApiNode, BaseHandle, BaseNode, EventNode, NoopNode, Position, workbenchXPath };
