// motia/dist/cloud/new-deployment/build.mjs
import { BuildError, BuildErrorType } from "../../utils/errors/build.error.mjs";
import { collectFlows, getStepFiles, getStreamFiles } from "../../generate-locked-data.mjs";
import { activatePythonVenv } from "../../utils/activate-python-env.mjs";
import { validatePythonEnvironment } from "../../utils/validate-python-environment.mjs";
import { Builder } from "../build/builder.mjs";
import { distDir, projectDir, stepsConfigPath } from "./constants.mjs";
import { NodeBuilder } from "../build/builders/node/index.mjs";
import { PythonBuilder } from "../build/builders/python/index.mjs";
import { LockedData, MemoryStreamAdapterManager, NoPrinter, isApiStep } from "@motiadev/core";
import fs from "fs";

//#region src/cloud/new-deployment/build.ts
const hasPythonSteps = (stepFiles) => {
	return stepFiles.some((file) => file.endsWith(".py"));
};
const build = async (listener) => {
	const builder = new Builder(projectDir, listener);
	const stepFiles = getStepFiles(projectDir);
	const streamFiles = getStreamFiles(projectDir);
	if (stepFiles.length === 0) throw new Error("Project contains no steps, please add some steps before building");
	builder.registerBuilder("node", new NodeBuilder(builder, listener));
	fs.rmSync(distDir, {
		recursive: true,
		force: true
	});
	fs.mkdirSync(distDir, { recursive: true });
	const lockedData = new LockedData(projectDir, new MemoryStreamAdapterManager(), new NoPrinter());
	const hasPython = hasPythonSteps([...stepFiles, ...streamFiles]);
	if (!(await validatePythonEnvironment({
		baseDir: projectDir,
		hasPythonFiles: hasPython
	})).success) throw new BuildError(BuildErrorType.COMPILATION, void 0, "");
	if (hasPython) {
		activatePythonVenv({ baseDir: projectDir });
		builder.registerBuilder("python", new PythonBuilder(builder, listener));
	}
	if ((await collectFlows(projectDir, lockedData).catch((err) => {
		const finalMessage = `${err.filePath ? `Build error in ${err.filePath}` : "Build error"}\nPlease check the logs above for details`;
		throw new BuildError(BuildErrorType.COMPILATION, err.filePath, finalMessage, err);
	})).length > 0) throw new Error("Project contains invalid steps, please fix them before building");
	await Promise.all(lockedData.activeSteps.map((step) => builder.buildStep(step)));
	await builder.buildApiSteps(lockedData.activeSteps.filter(isApiStep));
	const streams = lockedData.listStreams();
	for (const stream of streams) if (stream.config.baseConfig.storageType === "default") builder.registerStateStream(stream);
	else listener.onWarning(stream.filePath, "Custom streams are not supported yet in the cloud");
	const stepsFile = {
		steps: builder.stepsConfig,
		streams: builder.streamsConfig,
		routers: builder.routersConfig
	};
	fs.writeFileSync(stepsConfigPath, JSON.stringify(stepsFile, null, 2));
	return builder;
};

//#endregion
export { build };
